{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "dc50e92a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/joaquin/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return func(*args, **kwargs)\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-67-f72ccfea8cc3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     14\u001b[0m     \u001b[0;34m'annotation'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'geneChr'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'geneStart'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0;34m'geneEnd'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'geneLength'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'geneStrand'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'geneId'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'transcriptId'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'distanceToTSS'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m                     ], skiprows=1, usecols=['chrm','start','end','strand','context','mets', 'total','annotation']\n\u001b[0m\u001b[1;32m     17\u001b[0m                     ) #, usecols=['seqnames','start','end','width',\n\u001b[1;32m     18\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcompressedMethylomeDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    309\u001b[0m                     \u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstacklevel\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    310\u001b[0m                 )\n\u001b[0;32m--> 311\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    312\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    313\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[1;32m    584\u001b[0m     \u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwds_defaults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    486\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    487\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mparser\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 488\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    489\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    490\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, nrows)\u001b[0m\n\u001b[1;32m   1045\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnrows\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1046\u001b[0m         \u001b[0mnrows\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvalidate_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"nrows\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1047\u001b[0;31m         \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcol_dict\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1048\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1049\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mindex\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/python_parser.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, rows)\u001b[0m\n\u001b[1;32m    244\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrows\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    245\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 246\u001b[0;31m             \u001b[0mcontent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_lines\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    247\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mStopIteration\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    248\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_first_chunk\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/python_parser.py\u001b[0m in \u001b[0;36m_get_lines\u001b[0;34m(self, rows)\u001b[0m\n\u001b[1;32m   1053\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1054\u001b[0m                         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1055\u001b[0;31m                             \u001b[0mnew_row\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_next_iter_line\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrow_num\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpos\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mrows\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1056\u001b[0m                             \u001b[0mrows\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1057\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/python_parser.py\u001b[0m in \u001b[0;36m_next_iter_line\u001b[0;34m(self, row_num)\u001b[0m\n\u001b[1;32m    740\u001b[0m             \u001b[0;31m# assert for mypy, data is Iterator[str] or None, would error in next\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    741\u001b[0m             \u001b[0;32massert\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 742\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mnext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    743\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mcsv\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    744\u001b[0m             if (\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/python_parser.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m()\u001b[0m\n\u001b[1;32m    233\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m                 \u001b[0;32mfor\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 235\u001b[0;31m                     \u001b[0;32myield\u001b[0m \u001b[0mpat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    236\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    237\u001b[0m             \u001b[0mreader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "CORRESPONDENCE_METHYLOME_FILES_AND_1001_ID = '/home/joaquin/projects/gwasmet/data/commonData/correspondanceDoneRemoveNan.csv'\n",
    "METHYLOMES_FILES_FOLDER='/home/joaquin/projects/gwasmet/data/commonData/methylomechlpseek'\n",
    "idsDf = pd.read_csv(CORRESPONDENCE_METHYLOME_FILES_AND_1001_ID)\n",
    "allEcotypevalueDict = {}\n",
    "for index, methylome in idsDf.iterrows():\n",
    "\n",
    "    compressedMethylomePath = os.path.join(METHYLOMES_FILES_FOLDER, str(methylome.vcfsToUse)+'.bed')\n",
    "    compressedMethylomeDataFrame = pd.read_csv(compressedMethylomePath, sep='\\t', \n",
    "                    names=[\n",
    "    'seqnames','chrm','start','end','width',\n",
    "    'strand','V4','context','mets','total','V8',\n",
    "    'annotation','geneChr','geneStart',\n",
    "    'geneEnd','geneLength','geneStrand','geneId','transcriptId','distanceToTSS'\n",
    "                    ], skiprows=1, usecols=['chrm','start','end','strand','context','mets', 'total','annotation']\n",
    "                    ) #, usecols=['seqnames','start','end','width',\n",
    "    compressedMethylomeDataFrame['ratio'] = compressedMethylomeDataFrame['mets']/compressedMethylomeDataFrame['total']\n",
    "    CHHframe = compressedMethylomeDataFrame[\n",
    "    compressedMethylomeDataFrame.context.str.contains(r'C[A,T,C,M,W,Y,H][A,T,C,M,W,Y,H]')\n",
    "    ]\n",
    "    CHGframe = compressedMethylomeDataFrame[\n",
    "        compressedMethylomeDataFrame.context.str.contains(r'C[A,T,C,M,W,Y,H]G')\n",
    "        ]\n",
    "    CGframe = compressedMethylomeDataFrame[\n",
    "        compressedMethylomeDataFrame.context.str.contains(r'CG[A,T,G,C,N,M,R,W,S,Y,K,V,H,D,B]?')\n",
    "        ]\n",
    "    nonCGframe = compressedMethylomeDataFrame[\n",
    "        compressedMethylomeDataFrame.context.str.contains(r'C[A,T,C,M,W,Y,H][A,T,C,M,W,Y,H]|C[A,T,C,M,W,Y,H]G')\n",
    "        ]\n",
    "    print(compressedMethylomeDataFrame.head)\n",
    "\n",
    "#         compressedMethylomeDataFrame['ratio'] = compressedMethylomeDataFrame['mets']/compressedMethylomeDataFrame['total']\n",
    "#         CHHframe = compressedMethylomeDataFrame[\n",
    "#             compressedMethylomeDataFrame.context.str.contains(r'C[A,T,C,M,W,Y,H][A,T,C,M,W,Y,H]')\n",
    "#             ]\n",
    "#         CHGframe = compressedMethylomeDataFrame[\n",
    "#             compressedMethylomeDataFrame.context.str.contains(r'C[A,T,C,M,W,Y,H]G')\n",
    "#             ]\n",
    "#         CGframe = compressedMethylomeDataFrame[\n",
    "#             compressedMethylomeDataFrame.context.str.contains(r'CG[A,T,G,C,N,M,R,W,S,Y,K,V,H,D,B]?')\n",
    "#             ]\n",
    "#         nonCGframe = compressedMethylomeDataFrame[\n",
    "#             compressedMethylomeDataFrame.context.str.contains(r'C[A,T,C,M,W,Y,H][A,T,C,M,W,Y,H]|C[A,T,C,M,W,Y,H]G')\n",
    "#             ]\n",
    "#         checkIfNullsInExternalFiles(compressedMethylomeDataFrame)\n",
    "#         idgenomeProyect = str(methylome.vcfsToUse)\n",
    "#         allEcotypevalueDict[idgenomeProyect] = {}\n",
    "#         allEcotypevalueDict[idgenomeProyect]['CHH'] = float(CHHframe['ratio'].mean())\n",
    "#         allEcotypevalueDict[idgenomeProyect]['CHG'] = float(CHGframe['ratio'].mean())\n",
    "#         allEcotypevalueDict[idgenomeProyect]['CG'] = float(CGframe['ratio'].mean())\n",
    "#         allEcotypevalueDict[idgenomeProyect]['allMets'] = float(compressedMethylomeDataFrame['ratio'].mean())\n",
    "#         allEcotypevalueDict[idgenomeProyect]['nonCGMets'] = float(nonCGframe['ratio'].mean())\n",
    "\n",
    "#     with open(allEcotypePhenotypicValuesPath, 'w') as fp:\n",
    "#         json.dump(allEcotypevalueDict, fp,  indent=4)\n",
    "\n",
    "#     return allEcotypevalueDict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66b83a34",
   "metadata": {},
   "outputs": [],
   "source": [
    "usecols=['chrm','start','end','strand','context','mets', 'total','annotation']\n",
    "names=[\n",
    "    'chrm','start','end','width',\n",
    "    'strand','V4','context','mets','total','V8',\n",
    "    'annotation','geneChr','geneStart',\n",
    "    'geneEnd','geneLength','geneStrand','geneId','transcriptId','distanceToTSS'\n",
    "                    ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "ad29d3c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<bound method NDFrame.head of    fd  seqnames  start  end  width strand V4   V5  V6  V7  V8  \\\n",
      "0   4         1     34   34      1      *  -  CHG   4  11   1   \n",
      "1   5         1     38   38      1      *  +  CHH   1   2   0   \n",
      "2   6         1     39   39      1      *  +  CHH   1   2   0   \n",
      "3   7         1     45   45      1      *  +  CHH   1   2   0   \n",
      "4   8         1     46   46      1      *  +  CHH   1   2   0   \n",
      "5   9         1     47   47      1      *  +  CHH   2   2   1   \n",
      "\n",
      "          annotation  geneChr  geneStart  geneEnd  geneLength  geneStrand  \\\n",
      "0  Distal Intergenic        1       3631     5899        2269           1   \n",
      "1  Distal Intergenic        1       3631     5899        2269           1   \n",
      "2  Distal Intergenic        1       3631     5899        2269           1   \n",
      "3  Distal Intergenic        1       3631     5899        2269           1   \n",
      "4  Distal Intergenic        1       3631     5899        2269           1   \n",
      "5  Distal Intergenic        1       3631     5899        2269           1   \n",
      "\n",
      "           geneId transcriptId  distanceToTSS  \n",
      "0  gene:AT1G01010  AT1G01010.1          -3597  \n",
      "1  gene:AT1G01010  AT1G01010.1          -3593  \n",
      "2  gene:AT1G01010  AT1G01010.1          -3592  \n",
      "3  gene:AT1G01010  AT1G01010.1          -3586  \n",
      "4  gene:AT1G01010  AT1G01010.1          -3585  \n",
      "5  gene:AT1G01010  AT1G01010.1          -3584  >\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "compressedMethylomeDataFrame = pd.read_csv(\"/home/joaquin/projects/gwasmet/averquepasa.bed\",\n",
    "                                          \n",
    "                                           names=['fd','seqnames', 'start', 'end', 'width', 'strand', 'V4', 'V5', 'V6', 'V7',\n",
    "       'V8', 'annotation', 'geneChr', 'geneStart', 'geneEnd', 'geneLength',\n",
    "       'geneStrand', 'geneId', 'transcriptId', 'distanceToTSS'],\n",
    "                                           engine=\"python\",\n",
    "                                           sep=\",\", \n",
    "                                           skiprows=4) \n",
    "print(compressedMethylomeDataFrame.head)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "f2778bfe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fd</th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>V4</th>\n",
       "      <th>V5</th>\n",
       "      <th>V6</th>\n",
       "      <th>V7</th>\n",
       "      <th>V8</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneChr</th>\n",
       "      <th>geneStart</th>\n",
       "      <th>geneEnd</th>\n",
       "      <th>geneLength</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>geneId</th>\n",
       "      <th>transcriptId</th>\n",
       "      <th>distanceToTSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>-</td>\n",
       "      <td>CHG</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>1</td>\n",
       "      <td>3631</td>\n",
       "      <td>5899</td>\n",
       "      <td>2269</td>\n",
       "      <td>1</td>\n",
       "      <td>gene:AT1G01010</td>\n",
       "      <td>AT1G01010.1</td>\n",
       "      <td>-3597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>1</td>\n",
       "      <td>3631</td>\n",
       "      <td>5899</td>\n",
       "      <td>2269</td>\n",
       "      <td>1</td>\n",
       "      <td>gene:AT1G01010</td>\n",
       "      <td>AT1G01010.1</td>\n",
       "      <td>-3593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>1</td>\n",
       "      <td>3631</td>\n",
       "      <td>5899</td>\n",
       "      <td>2269</td>\n",
       "      <td>1</td>\n",
       "      <td>gene:AT1G01010</td>\n",
       "      <td>AT1G01010.1</td>\n",
       "      <td>-3592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>1</td>\n",
       "      <td>3631</td>\n",
       "      <td>5899</td>\n",
       "      <td>2269</td>\n",
       "      <td>1</td>\n",
       "      <td>gene:AT1G01010</td>\n",
       "      <td>AT1G01010.1</td>\n",
       "      <td>-3586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>1</td>\n",
       "      <td>3631</td>\n",
       "      <td>5899</td>\n",
       "      <td>2269</td>\n",
       "      <td>1</td>\n",
       "      <td>gene:AT1G01010</td>\n",
       "      <td>AT1G01010.1</td>\n",
       "      <td>-3585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>1</td>\n",
       "      <td>3631</td>\n",
       "      <td>5899</td>\n",
       "      <td>2269</td>\n",
       "      <td>1</td>\n",
       "      <td>gene:AT1G01010</td>\n",
       "      <td>AT1G01010.1</td>\n",
       "      <td>-3584</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fd  seqnames  start  end  width strand V4   V5  V6  V7  V8  \\\n",
       "0   4         1     34   34      1      *  -  CHG   4  11   1   \n",
       "1   5         1     38   38      1      *  +  CHH   1   2   0   \n",
       "2   6         1     39   39      1      *  +  CHH   1   2   0   \n",
       "3   7         1     45   45      1      *  +  CHH   1   2   0   \n",
       "4   8         1     46   46      1      *  +  CHH   1   2   0   \n",
       "5   9         1     47   47      1      *  +  CHH   2   2   1   \n",
       "\n",
       "          annotation  geneChr  geneStart  geneEnd  geneLength  geneStrand  \\\n",
       "0  Distal Intergenic        1       3631     5899        2269           1   \n",
       "1  Distal Intergenic        1       3631     5899        2269           1   \n",
       "2  Distal Intergenic        1       3631     5899        2269           1   \n",
       "3  Distal Intergenic        1       3631     5899        2269           1   \n",
       "4  Distal Intergenic        1       3631     5899        2269           1   \n",
       "5  Distal Intergenic        1       3631     5899        2269           1   \n",
       "\n",
       "           geneId transcriptId  distanceToTSS  \n",
       "0  gene:AT1G01010  AT1G01010.1          -3597  \n",
       "1  gene:AT1G01010  AT1G01010.1          -3593  \n",
       "2  gene:AT1G01010  AT1G01010.1          -3592  \n",
       "3  gene:AT1G01010  AT1G01010.1          -3586  \n",
       "4  gene:AT1G01010  AT1G01010.1          -3585  \n",
       "5  gene:AT1G01010  AT1G01010.1          -3584  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "compressedMethylomeDataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "a60023c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrm</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>context</th>\n",
       "      <th>mets</th>\n",
       "      <th>total</th>\n",
       "      <th>annotation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>*</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>*</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>*</td>\n",
       "      <td>CHG</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>34</td>\n",
       "      <td>*</td>\n",
       "      <td>CHG</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>*</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>*</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>*</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>*</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>*</td>\n",
       "      <td>CHH</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrm  start  end strand context  mets  total          annotation\n",
       "1     1     29   29      *     CHH     1      1   Distal Intergenic\n",
       "2     1     30   30      *     CHH     1      1   Distal Intergenic\n",
       "3     1     32   32      *     CHG     1      1   Distal Intergenic\n",
       "4     1     34   34      *     CHG     4     11   Distal Intergenic\n",
       "5     1     38   38      *     CHH     1      2   Distal Intergenic\n",
       "6     1     39   39      *     CHH     1      2   Distal Intergenic\n",
       "7     1     45   45      *     CHH     1      2   Distal Intergenic\n",
       "8     1     46   46      *     CHH     1      2   Distal Intergenic\n",
       "9     1     47   47      *     CHH     2      2   Distal Intergenic"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "compressedMethylomeDataFrame[['chrm','start','end','strand','context','mets', 'total','annotation']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "07061bfb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrom\tpos\tstrand\tmc_class\tmethylated_bases\ttotal_bases\tmethylation_call\n",
      "\n",
      "<bound method NDFrame.head of          context  mets  total\n",
      "0            CTA     1      1\n",
      "1            CCC     2      3\n",
      "2            CCT     2      3\n",
      "3            CTA     2      3\n",
      "4            CCG     2      3\n",
      "...          ...   ...    ...\n",
      "31282784     CTA     3      3\n",
      "31282785     CCT     2      3\n",
      "31282786     CCC     2      2\n",
      "31282787     CTA     1      1\n",
      "31282788     CCT     1      1\n",
      "\n",
      "[31282789 rows x 3 columns]>\n",
      "chrom\tpos\tstrand\tmc_class\tmethylated_bases\ttotal_bases\tmethylation_call\n",
      "\n",
      "<bound method NDFrame.head of          context  mets  total\n",
      "0            CCC     1      1\n",
      "1            CCT     0      1\n",
      "2            CTA     1      1\n",
      "3            CCC     1      1\n",
      "4            CCT     1      1\n",
      "...          ...   ...    ...\n",
      "29973428     CCT     0      2\n",
      "29973429     CCC     1      2\n",
      "29973430     CTA     0      2\n",
      "29973431     CCT     0      1\n",
      "29973432     CTA     1      1\n",
      "\n",
      "[29973433 rows x 3 columns]>\n",
      "chrom\tpos\tstrand\tmc_class\tmethylated_bases\ttotal_bases\tmethylation_call\n",
      "\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-34-c20fb86c0796>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     17\u001b[0m             compressedMethylomeDataFrame = pd.read_csv(compressedMethylomePath, sep='\\t', \n\u001b[1;32m     18\u001b[0m                     \u001b[0mnames\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'chr'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'position'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'strand'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'context'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'mets'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'total'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'cmyesornot'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 19\u001b[0;31m                     usecols=['context','mets', 'total'], skiprows=1)\n\u001b[0m\u001b[1;32m     20\u001b[0m             \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcompressedMethylomeDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    309\u001b[0m                     \u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstacklevel\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    310\u001b[0m                 )\n\u001b[0;32m--> 311\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    312\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    313\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[1;32m    584\u001b[0m     \u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwds_defaults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    486\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    487\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mparser\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 488\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mparser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    489\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    490\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, nrows)\u001b[0m\n\u001b[1;32m   1045\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnrows\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1046\u001b[0m         \u001b[0mnrows\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvalidate_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"nrows\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1047\u001b[0;31m         \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcol_dict\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1048\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1049\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mindex\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/io/parsers/c_parser_wrapper.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, nrows)\u001b[0m\n\u001b[1;32m    221\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlow_memory\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 223\u001b[0;31m                 \u001b[0mchunks\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_low_memory\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnrows\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    224\u001b[0m                 \u001b[0;31m# destructive to chunks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    225\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_concatenate_chunks\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchunks\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.read_low_memory\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._read_rows\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._convert_column_data\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._convert_tokens\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._convert_with_dtype\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/software/miniconda/envs/methylation/lib/python3.7/site-packages/pandas/core/dtypes/common.py\u001b[0m in \u001b[0;36mis_extension_array_dtype\u001b[0;34m(arr_or_dtype)\u001b[0m\n\u001b[1;32m   1418\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1419\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1420\u001b[0;31m \u001b[0;32mdef\u001b[0m \u001b[0mis_extension_array_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marr_or_dtype\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mbool\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1421\u001b[0m     \"\"\"\n\u001b[1;32m   1422\u001b[0m     \u001b[0mCheck\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0man\u001b[0m \u001b[0mobject\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0ma\u001b[0m \u001b[0mpandas\u001b[0m \u001b[0mextension\u001b[0m \u001b[0marray\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import gzip\n",
    "CORRESPONDENCE_METHYLOME_FILES_AND_1001_ID = '/home/joaquin/projects/gwasmet/data/commonData/correspondanceDoneRemoveNan.csv'\n",
    "METHYLOMES_FILES_FOLDER='/home/joaquin/projects/gwasmet/data/commonData/methylome'\n",
    "for index, methylome in idsDf.iterrows():\n",
    "\n",
    "    compressedMethylomePath = os.path.join(METHYLOMES_FILES_FOLDER, methylome.Name)\n",
    "    # in this case, we will take only the cols to make the calculation: mets and total. Some Experiments has initial col names, so\n",
    "    # we remove them with this if else statement\n",
    "\n",
    "    with gzip.open(compressedMethylomePath, 'rt') as f:\n",
    "        hola = next(f)\n",
    "        if 'methylated_bases' in hola:\n",
    "            print(hola)\n",
    "            compressedMethylomePath = os.path.join(METHYLOMES_FILES_FOLDER, methylome.Name)\n",
    "            compressedMethylomeDataFrame = pd.read_csv(compressedMethylomePath, sep='\\t', \n",
    "                    names=['chr', 'position', 'strand', 'context', 'mets', 'total', 'cmyesornot'],\n",
    "                    usecols=['context','mets', 'total'], skiprows=1)\n",
    "            print(compressedMethylomeDataFrame.head)\n",
    "            "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "b311b5e2c1aaa04d48066871c65b4e8ce80fb760bb38a0437727056581097150"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
