{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import os\n",
    "import shutil\n",
    "import subprocess\n",
    "import tempfile\n",
    "import glob\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "import re\n",
    "import pandas as pd\n",
    "import time\n",
    "\n",
    "intersect_peaks_file_path = '/home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15-round1_allpeaks.bed' #/home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15_Allpeaks.bed\n",
    " # '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/1ng', #/ZYMO71ngsorted.bam\n",
    "folders = [\n",
    "   '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/1ng',\n",
    "    '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/5ng', #/ZYMO75ngsorted.bam\n",
    "    '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/10ng', #/ZYMO710ngsorted.bam\n",
    "    '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/25ng', #/ZYMO725ngsorted.bam\n",
    "    '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/50ng', #/ZYMO750ngsorted.bam\n",
    "    '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/Input' #/ZYMO7Input.bam\n",
    "\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def performIntersect(folder, intersectFile):\n",
    "    \n",
    "    sortedBamFile = glob.glob(f'{folder}/*orted.bam')\n",
    "    \n",
    "    if len(sortedBamFile) != 1:\n",
    "        return print(folder, ' has a problem selecting File')\n",
    "    else:\n",
    "        sortedBamFile = sortedBamFile[0]\n",
    "    \n",
    "    outputFile = intersectFile.strip().split('/')[-1][:-4]+'_'+sortedBamFile.split('/')[-1][:-10]+'.bed'\n",
    "    outputFilePath = os.path.join(folder,outputFile)\n",
    "\n",
    "    print(outputFilePath)\n",
    "    print(outputFilePath)\n",
    "    \n",
    "    if os.path.isfile(outputFilePath):\n",
    "        os.remove(outputFilePath)\n",
    "    if not os.path.isfile(outputFilePath):\n",
    "        comandocall = 'samtools' + ' view -q1 -b ' + sortedBamFile + ' | ' + 'bedtools' + ' intersect -abam stdin -b ' + intersectFile + ' -bed -wb -f 0.5 ' + '> ' + outputFilePath \n",
    "        print(comandocall)\n",
    "        subprocess.call(\n",
    "            comandocall, shell=True\n",
    "        )\n",
    "    else:\n",
    "        print(outputFile, ' is already done')\n",
    "    \n",
    "    totalForBox = {}\n",
    "    with open(outputFilePath, 'r') as intersectOut:\n",
    "        intersectDf = pd.read_csv(\n",
    "            intersectOut, sep='\\t', usecols=[3, 12, 13, 14, 15],\n",
    "            names=['intersected', 'chr', 'start', 'end', 'boxname'],\n",
    "        )\n",
    "        for index, ip in intersectDf.iterrows():\n",
    "            intersectOcurrence = str(ip.intersected.split('/')[0])\n",
    "            box = ','.join([str(ip.chr), str(ip.start), str(ip.end), ip.boxname])\n",
    "            if box in totalForBox:\n",
    "                totalForBox[box].add(intersectOcurrence)\n",
    "            else:\n",
    "                totalForBox[box] = {intersectOcurrence}\n",
    "\n",
    "        for box in totalForBox:\n",
    "            boxlen = len(totalForBox[box])\n",
    "            totalForBox[box] = boxlen\n",
    "\n",
    "        with open(outputFilePath[:-4] + '_boxtotals.csv', 'w') as elcsv:\n",
    "            elcsv.write('chr,start,end,boxname,{}\\n'.format(sortedBamFile.split('/')[-1][:-10]))\n",
    "            for name, recount in totalForBox.items():\n",
    "                elcsv.write('{},{}\\n'.format(name, recount))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/1ng/ZYMO15-round1_allpeaks_ZYMO151ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/1ng/ZYMO15-round1_allpeaks_ZYMO151ng.bed\n",
      "samtools view -q1 -b /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/1ng/ZYMO151ngsorted.bam | bedtools intersect -abam stdin -b /home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15-round1_allpeaks.bed -bed -wb -f 0.5 > /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/1ng/ZYMO15-round1_allpeaks_ZYMO151ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/5ng/ZYMO15-round1_allpeaks_ZYMO155ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/5ng/ZYMO15-round1_allpeaks_ZYMO155ng.bed\n",
      "samtools view -q1 -b /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/5ng/ZYMO155ngsorted.bam | bedtools intersect -abam stdin -b /home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15-round1_allpeaks.bed -bed -wb -f 0.5 > /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/5ng/ZYMO15-round1_allpeaks_ZYMO155ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/10ng/ZYMO15-round1_allpeaks_ZYMO1510ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/10ng/ZYMO15-round1_allpeaks_ZYMO1510ng.bed\n",
      "samtools view -q1 -b /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/10ng/ZYMO1510ngsorted.bam | bedtools intersect -abam stdin -b /home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15-round1_allpeaks.bed -bed -wb -f 0.5 > /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/10ng/ZYMO15-round1_allpeaks_ZYMO1510ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/25ng/ZYMO15-round1_allpeaks_ZYMO1525ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/25ng/ZYMO15-round1_allpeaks_ZYMO1525ng.bed\n",
      "samtools view -q1 -b /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/25ng/ZYMO1525ngsorted.bam | bedtools intersect -abam stdin -b /home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15-round1_allpeaks.bed -bed -wb -f 0.5 > /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/25ng/ZYMO15-round1_allpeaks_ZYMO1525ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/50ng/ZYMO15-round1_allpeaks_ZYMO1550ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/50ng/ZYMO15-round1_allpeaks_ZYMO1550ng.bed\n",
      "samtools view -q1 -b /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/50ng/ZYMO1550ngsorted.bam | bedtools intersect -abam stdin -b /home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15-round1_allpeaks.bed -bed -wb -f 0.5 > /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/50ng/ZYMO15-round1_allpeaks_ZYMO1550ng.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/Input/ZYMO15-round1_allpeaks_ZYMO15Input.bed\n",
      "/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/Input/ZYMO15-round1_allpeaks_ZYMO15Input.bed\n",
      "samtools view -q1 -b /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/Input/ZYMO15Inputsorted.bam | bedtools intersect -abam stdin -b /home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/ZYMO15-round1_allpeaks.bed -bed -wb -f 0.5 > /home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15/Input/ZYMO15-round1_allpeaks_ZYMO15Input.bed\n"
     ]
    }
   ],
   "source": [
    "for fol in folders:\n",
    "    performIntersect(fol,intersect_peaks_file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def generateDfandNormaliceDataTPMs(folder,dataColName, totalReads):\n",
    "    csvFile = glob.glob(f'{folder}/ZYMO15-round1_allpeaks*.csv')\n",
    "    \n",
    "    if len(csvFile) != 1:\n",
    "        return print(folder, ' has a problem selecting File')\n",
    "    else:\n",
    "        csvFilePath = csvFile[0]\n",
    "\n",
    "    fileDf = pd.read_csv(csvFilePath,header=0, names=['chr','star','end','id',dataColName])\n",
    "\n",
    "    scalingFactor = totalReads/1000000\n",
    "    fileDf_counts = fileDf.copy()\n",
    "    fileDf[dataColName] = fileDf[dataColName].apply(lambda x: x/scalingFactor)\n",
    "\n",
    "    return fileDf_counts, fileDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_to_total_reads = '/home/joaquin/projects/methylation/peaks_and_alingment/concentraciones_summary.tsv'\n",
    "infosecund_df = pd.read_csv(path_to_total_reads,sep='\\t',names=['condition','reads','q','peaks'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['', 'home', 'joaquin', 'projects', 'methylation', 'data', 'data_concentraciones', 'ZYMO', '15', '1ng']\n",
      "ZYMO_15_1ng\n",
      "9108005\n",
      "['', 'home', 'joaquin', 'projects', 'methylation', 'data', 'data_concentraciones', 'ZYMO', '15', '5ng']\n",
      "ZYMO_15_5ng\n",
      "11313382\n",
      "['', 'home', 'joaquin', 'projects', 'methylation', 'data', 'data_concentraciones', 'ZYMO', '15', '10ng']\n",
      "ZYMO_15_10ng\n",
      "10277543\n",
      "['', 'home', 'joaquin', 'projects', 'methylation', 'data', 'data_concentraciones', 'ZYMO', '15', '25ng']\n",
      "ZYMO_15_25ng\n",
      "9319952\n",
      "['', 'home', 'joaquin', 'projects', 'methylation', 'data', 'data_concentraciones', 'ZYMO', '15', '50ng']\n",
      "ZYMO_15_50ng\n",
      "9756143\n",
      "['', 'home', 'joaquin', 'projects', 'methylation', 'data', 'data_concentraciones', 'ZYMO', '15', 'Input']\n",
      "ZYMO_15_Input\n",
      "8199725\n"
     ]
    }
   ],
   "source": [
    "main_dir = '/home/joaquin/projects/methylation/data/data_concentraciones/ZYMO/15'\n",
    "\n",
    "listofdfs_tpms = []\n",
    "listofdfs_counts = []\n",
    "\n",
    "for workingFolder in folders:\n",
    "    cachos = workingFolder.split('/')\n",
    "    print(cachos)\n",
    "    cachos[-3],cachos[-2],cachos[-1]\n",
    "    datacolname = '{}_{}_{}'.format(cachos[-3],cachos[-2],cachos[-1])\n",
    "    print(datacolname)\n",
    "    normalization_value = int(list(infosecund_df[infosecund_df['condition'] == datacolname]['reads'])[0])\n",
    "    print(normalization_value)\n",
    "    counts, tpms = generateDfandNormaliceDataTPMs(workingFolder,datacolname,normalization_value)\n",
    "    listofdfs_tpms.append(tpms)\n",
    "    listofdfs_counts.append(counts)\n",
    "    #             los valores que no estan en una de las replicas los completo con un 0\n",
    "df_merged_tpms = reduce(lambda  left,right: pd.merge(left,right,on=['chr','star','end','id'],\n",
    "                                how='outer'), listofdfs_tpms)\n",
    "\n",
    "df_merged_counts = reduce(lambda  left,right: pd.merge(left,right,on=['chr','star','end','id'],\n",
    "                                how='outer'), listofdfs_counts)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>star</th>\n",
       "      <th>end</th>\n",
       "      <th>id</th>\n",
       "      <th>ZYMO_15_1ng</th>\n",
       "      <th>ZYMO_15_5ng</th>\n",
       "      <th>ZYMO_15_10ng</th>\n",
       "      <th>ZYMO_15_25ng</th>\n",
       "      <th>ZYMO_15_50ng</th>\n",
       "      <th>ZYMO_15_Input</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1440</td>\n",
       "      <td>1641</td>\n",
       "      <td>5ngP06598</td>\n",
       "      <td>33.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>9954</td>\n",
       "      <td>10155</td>\n",
       "      <td>5ngP17889</td>\n",
       "      <td>18.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>10737</td>\n",
       "      <td>10938</td>\n",
       "      <td>1ngP06790</td>\n",
       "      <td>89.0</td>\n",
       "      <td>87.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11920</td>\n",
       "      <td>12121</td>\n",
       "      <td>1ngP08921</td>\n",
       "      <td>95.0</td>\n",
       "      <td>148.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>15423</td>\n",
       "      <td>15624</td>\n",
       "      <td>5ngP04555</td>\n",
       "      <td>59.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26124</th>\n",
       "      <td>5</td>\n",
       "      <td>25332780</td>\n",
       "      <td>25332981</td>\n",
       "      <td>10ngP08014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26125</th>\n",
       "      <td>5</td>\n",
       "      <td>26073373</td>\n",
       "      <td>26073574</td>\n",
       "      <td>10ngP06143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26126</th>\n",
       "      <td>1</td>\n",
       "      <td>14824452</td>\n",
       "      <td>14824653</td>\n",
       "      <td>50ngP03977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26127</th>\n",
       "      <td>3</td>\n",
       "      <td>14893747</td>\n",
       "      <td>14893948</td>\n",
       "      <td>10ngP07858</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26128</th>\n",
       "      <td>5</td>\n",
       "      <td>10555628</td>\n",
       "      <td>10555829</td>\n",
       "      <td>10ngP07810</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26129 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr      star       end          id  ZYMO_15_1ng  ZYMO_15_5ng  \\\n",
       "0        1      1440      1641   5ngP06598         33.0         96.0   \n",
       "1        1      9954     10155   5ngP17889         18.0         31.0   \n",
       "2        1     10737     10938   1ngP06790         89.0         87.0   \n",
       "3        1     11920     12121   1ngP08921         95.0        148.0   \n",
       "4        1     15423     15624   5ngP04555         59.0        170.0   \n",
       "...    ...       ...       ...         ...          ...          ...   \n",
       "26124    5  25332780  25332981  10ngP08014          NaN         22.0   \n",
       "26125    5  26073373  26073574  10ngP06143          NaN          4.0   \n",
       "26126    1  14824452  14824653  50ngP03977          NaN          NaN   \n",
       "26127    3  14893747  14893948  10ngP07858          NaN          NaN   \n",
       "26128    5  10555628  10555829  10ngP07810          NaN          NaN   \n",
       "\n",
       "       ZYMO_15_10ng  ZYMO_15_25ng  ZYMO_15_50ng  ZYMO_15_Input  \n",
       "0              71.0          51.0          71.0            8.0  \n",
       "1              21.0          17.0          31.0           26.0  \n",
       "2              49.0          40.0          53.0           12.0  \n",
       "3             101.0          74.0          66.0            9.0  \n",
       "4              90.0          79.0         103.0           11.0  \n",
       "...             ...           ...           ...            ...  \n",
       "26124          10.0           3.0           5.0            5.0  \n",
       "26125           2.0           1.0           8.0            4.0  \n",
       "26126           1.0           2.0           1.0            7.0  \n",
       "26127           1.0           NaN           1.0            4.0  \n",
       "26128           NaN           NaN           NaN            2.0  \n",
       "\n",
       "[26129 rows x 10 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merged_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>star</th>\n",
       "      <th>end</th>\n",
       "      <th>id</th>\n",
       "      <th>ZYMO_15_1ng</th>\n",
       "      <th>ZYMO_15_5ng</th>\n",
       "      <th>ZYMO_15_10ng</th>\n",
       "      <th>ZYMO_15_25ng</th>\n",
       "      <th>ZYMO_15_50ng</th>\n",
       "      <th>ZYMO_15_Input</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1440</td>\n",
       "      <td>1641</td>\n",
       "      <td>5ngP06598</td>\n",
       "      <td>3.623186</td>\n",
       "      <td>8.485526</td>\n",
       "      <td>6.908266</td>\n",
       "      <td>5.472131</td>\n",
       "      <td>7.277466</td>\n",
       "      <td>0.975642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>9954</td>\n",
       "      <td>10155</td>\n",
       "      <td>5ngP17889</td>\n",
       "      <td>1.976284</td>\n",
       "      <td>2.740118</td>\n",
       "      <td>2.043290</td>\n",
       "      <td>1.824044</td>\n",
       "      <td>3.177485</td>\n",
       "      <td>3.170838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>10737</td>\n",
       "      <td>10938</td>\n",
       "      <td>1ngP06790</td>\n",
       "      <td>9.771624</td>\n",
       "      <td>7.690008</td>\n",
       "      <td>4.767676</td>\n",
       "      <td>4.291868</td>\n",
       "      <td>5.432475</td>\n",
       "      <td>1.463464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11920</td>\n",
       "      <td>12121</td>\n",
       "      <td>1ngP08921</td>\n",
       "      <td>10.430385</td>\n",
       "      <td>13.081853</td>\n",
       "      <td>9.827252</td>\n",
       "      <td>7.939955</td>\n",
       "      <td>6.764968</td>\n",
       "      <td>1.097598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>15423</td>\n",
       "      <td>15624</td>\n",
       "      <td>5ngP04555</td>\n",
       "      <td>6.477818</td>\n",
       "      <td>15.026453</td>\n",
       "      <td>8.756957</td>\n",
       "      <td>8.476439</td>\n",
       "      <td>10.557451</td>\n",
       "      <td>1.341508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26124</th>\n",
       "      <td>5</td>\n",
       "      <td>25332780</td>\n",
       "      <td>25332981</td>\n",
       "      <td>10ngP08014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.944600</td>\n",
       "      <td>0.972995</td>\n",
       "      <td>0.321890</td>\n",
       "      <td>0.512498</td>\n",
       "      <td>0.609777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26125</th>\n",
       "      <td>5</td>\n",
       "      <td>26073373</td>\n",
       "      <td>26073574</td>\n",
       "      <td>10ngP06143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.353564</td>\n",
       "      <td>0.194599</td>\n",
       "      <td>0.107297</td>\n",
       "      <td>0.819996</td>\n",
       "      <td>0.487821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26126</th>\n",
       "      <td>1</td>\n",
       "      <td>14824452</td>\n",
       "      <td>14824653</td>\n",
       "      <td>50ngP03977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.097300</td>\n",
       "      <td>0.214593</td>\n",
       "      <td>0.102500</td>\n",
       "      <td>0.853687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26127</th>\n",
       "      <td>3</td>\n",
       "      <td>14893747</td>\n",
       "      <td>14893948</td>\n",
       "      <td>10ngP07858</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.097300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.102500</td>\n",
       "      <td>0.487821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26128</th>\n",
       "      <td>5</td>\n",
       "      <td>10555628</td>\n",
       "      <td>10555829</td>\n",
       "      <td>10ngP07810</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.243911</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26129 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr      star       end          id  ZYMO_15_1ng  ZYMO_15_5ng  \\\n",
       "0        1      1440      1641   5ngP06598     3.623186     8.485526   \n",
       "1        1      9954     10155   5ngP17889     1.976284     2.740118   \n",
       "2        1     10737     10938   1ngP06790     9.771624     7.690008   \n",
       "3        1     11920     12121   1ngP08921    10.430385    13.081853   \n",
       "4        1     15423     15624   5ngP04555     6.477818    15.026453   \n",
       "...    ...       ...       ...         ...          ...          ...   \n",
       "26124    5  25332780  25332981  10ngP08014          NaN     1.944600   \n",
       "26125    5  26073373  26073574  10ngP06143          NaN     0.353564   \n",
       "26126    1  14824452  14824653  50ngP03977          NaN          NaN   \n",
       "26127    3  14893747  14893948  10ngP07858          NaN          NaN   \n",
       "26128    5  10555628  10555829  10ngP07810          NaN          NaN   \n",
       "\n",
       "       ZYMO_15_10ng  ZYMO_15_25ng  ZYMO_15_50ng  ZYMO_15_Input  \n",
       "0          6.908266      5.472131      7.277466       0.975642  \n",
       "1          2.043290      1.824044      3.177485       3.170838  \n",
       "2          4.767676      4.291868      5.432475       1.463464  \n",
       "3          9.827252      7.939955      6.764968       1.097598  \n",
       "4          8.756957      8.476439     10.557451       1.341508  \n",
       "...             ...           ...           ...            ...  \n",
       "26124      0.972995      0.321890      0.512498       0.609777  \n",
       "26125      0.194599      0.107297      0.819996       0.487821  \n",
       "26126      0.097300      0.214593      0.102500       0.853687  \n",
       "26127      0.097300           NaN      0.102500       0.487821  \n",
       "26128           NaN           NaN           NaN       0.243911  \n",
       "\n",
       "[26129 rows x 10 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merged_tpms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_merged_tpms.to_excel(f'{main_dir}/ZYMO15-round1_allpeaks_ZYMO15_tpms.xlsx', index=False)\n",
    "df_merged_counts.to_excel(f'{main_dir}/ZYMO15-round1_allpeaks_ZYMO15_counts.xlsx', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "methylation",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
