{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv\n",
    "import gzip\n",
    "import time\n",
    "import os\n",
    "import pandas as pd\n",
    "import re\n",
    "from numpy import where as npwhere\n",
    "from functools import reduce\n",
    "\n",
    "def mapMethytilatedCitosine(metfile,intersection_file_path,output_file_path):\n",
    "    chrs_data = {}\n",
    "    with gzip.open(metfile, 'rt') as f:\n",
    "        for line in f:\n",
    "            chr_index = line.split('\\t')[0]\n",
    "            coordinate_index = int(line.split('\\t')[1])\n",
    "            try:\n",
    "                chrs_data[chr_index][coordinate_index] = line[:-5]\n",
    "            except:\n",
    "                chrs_data[chr_index] = {coordinate_index:line[:-5]}\n",
    "\n",
    "    with open(intersection_file_path) as f:\n",
    "        reader = csv.reader(f, delimiter='\\t')\n",
    "        data = list(reader)\n",
    "\n",
    "    with open(output_file_path, 'a+') as whateverBox:\n",
    "        for each_box in data:\n",
    "            box_areas = {}\n",
    "            box_start = int(each_box[1])\n",
    "            box_end = int(each_box[2])\n",
    "            chr_location = each_box[0]\n",
    "            sequence_identifier = each_box[3]\n",
    "            # tenemos que aplicar rango desde una posicion inicial porque es un archivo 0 based, si fuese 1 based el rango estarÃ­a mal y tendria que restarle\n",
    "            # uno al archivo al box start\n",
    "            box_areas['box'] = list(range(box_start, box_end))\n",
    "\n",
    "            for boxarea in box_areas:\n",
    "                for coordinate in box_areas[boxarea]:\n",
    "                    matchline = chrs_data[chr_location].get(coordinate, False)\n",
    "                    if matchline:\n",
    "                        whateverBox.write('{}\\t{}\\t{}\\n'.format(matchline, sequence_identifier, boxarea))\n",
    "                        \n",
    "\n",
    "def loadFileRemoveMtsAndLowMetReport(replicate):\n",
    "        bxdf = pd.read_csv(\n",
    "            replicate,\n",
    "            sep='\\t', \n",
    "            names=['chr', 'position', 'strand', 'mets', 'nomets', 'context', 'boxID', 'side'],\n",
    "            usecols=['chr', 'position', 'strand', 'boxID','mets', 'nomets', 'context']\n",
    "                )\n",
    "        bxdf.drop(bxdf[(bxdf.chr == 'Mt') | (bxdf.chr == 'Pt')].index, inplace=True)\n",
    "        bxdf = bxdf[bxdf['mets'] + bxdf['nomets'] >= 4]\n",
    "        return bxdf\n",
    "\n",
    "def calculateValueAsPercentageOfMetCs(replicatedf,replicanamedf):\n",
    "    replicatedf[replicanamedf] = replicatedf['mets'] / (replicatedf['nomets'] + replicatedf['mets'])\n",
    "    replicatedf = replicatedf.drop(columns=['mets','nomets'])\n",
    "    return replicatedf\n",
    "def calculateValueAsCualitative(replicatedf,replicanamedf):\n",
    "    \"\"\"only zero or one, meaning one that there is methylation\"\"\"\n",
    "    replicatedf[replicanamedf] = npwhere(replicatedf['mets']>replicatedf['nomets'], 1, 0)\n",
    "    replicatedf = replicatedf.drop(columns=['mets','nomets'])\n",
    "    return replicatedf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import shutil\n",
    "def movetofolderallthefiles(origin, destinationfolder):\n",
    "    shutil.copy2(origin, destinationfolder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976405_ros1-4_rep2_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976406_ros1-4_rep3_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976404_ros1-4_rep1_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976407_ros1-4_rep4_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976284_JA_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976282_JA_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976283_JA_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976402_Col0_rep3_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976401_Col0_rep2_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976403_Col0_rep4_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976400_Col0_rep1_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976281_WaterControl_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976280_WaterControl_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976279_WaterControl_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976411_nrpe1-11_rep4_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976408_nrpe1-11_rep1_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976410_nrpe1-11_rep3_bismark_CX_report.txtpeakMets.tsv\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976409_nrpe1-11_rep2_bismark_CX_report.txtpeakMets.tsv\n"
     ]
    }
   ],
   "source": [
    "working_folder = '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper'\n",
    "intersection_file = '/home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/allThePossiblePeaksnine.bed'\n",
    "basedmrfolder = '/home/joaquin/projects/methylation/data/DMRs/CX'\n",
    "experimentsClasification = {}\n",
    "for root, dirs, files in os.walk(working_folder):\n",
    "\n",
    "    for file in files:\n",
    "        if 'txt.gz' in file:\n",
    "            replicate, hour, condition =root.strip().split('/')[-3:]\n",
    "            experimentName = condition\n",
    "            experimentsClasification.setdefault(experimentName, [])\n",
    "            targetMetFilename = root+'/'+file\n",
    "            finalnamedestination = basedmrfolder+'/'+file\n",
    "            outputfile = targetMetFilename[:-3] + 'peakMets.tsv'\n",
    "            experimentsClasification[experimentName].append(outputfile)\n",
    "            print(outputfile)\n",
    "            # mapMethytilatedCitosine(targetMetFilename,intersection_file,outputfile)\n",
    "\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ros1': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976405_ros1-4_rep2_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976406_ros1-4_rep3_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976404_ros1-4_rep1_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976407_ros1-4_rep4_bismark_CX_report.txtpeakMets.tsv'],\n",
       " 'col0JA': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976284_JA_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976282_JA_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976283_JA_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv'],\n",
       " 'col0': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976402_Col0_rep3_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976401_Col0_rep2_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976403_Col0_rep4_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976400_Col0_rep1_bismark_CX_report.txtpeakMets.tsv'],\n",
       " 'col0W': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976281_WaterControl_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976280_WaterControl_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976279_WaterControl_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv'],\n",
       " 'nrpe1': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976411_nrpe1-11_rep4_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976408_nrpe1-11_rep1_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976410_nrpe1-11_rep3_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976409_nrpe1-11_rep2_bismark_CX_report.txtpeakMets.tsv']}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "experimentsClasification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'paperros1': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976405_ros1-4_rep2_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976406_ros1-4_rep3_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976404_ros1-4_rep1_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976407_ros1-4_rep4_bismark_CX_report.txtpeakMets.tsv'],\n",
       " 'papercol0JA': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976284_JA_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976282_JA_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976283_JA_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv'],\n",
       " 'papercol0': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976402_Col0_rep3_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976401_Col0_rep2_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976403_Col0_rep4_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976400_Col0_rep1_bismark_CX_report.txtpeakMets.tsv'],\n",
       " 'papercol0W': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976281_WaterControl_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976280_WaterControl_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976279_WaterControl_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv'],\n",
       " 'papernrpe1': ['/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976411_nrpe1-11_rep4_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976408_nrpe1-11_rep1_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976410_nrpe1-11_rep3_bismark_CX_report.txtpeakMets.tsv',\n",
       "  '/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976409_nrpe1-11_rep2_bismark_CX_report.txtpeakMets.tsv']}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "experimentsClasification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976405_ros1-4_rep2_bismark_CX_report.txtpeakMets.tsv\n",
      "ros1rep2\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976406_ros1-4_rep3_bismark_CX_report.txtpeakMets.tsv\n",
      "ros1rep3\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976404_ros1-4_rep1_bismark_CX_report.txtpeakMets.tsv\n",
      "ros1rep1\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/ros1/GSM5976407_ros1-4_rep4_bismark_CX_report.txtpeakMets.tsv\n",
      "ros1rep4\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976284_JA_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "col0JArep3\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976282_JA_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "col0JArep1\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0JA/GSM4976283_JA_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "col0JArep2\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976402_Col0_rep3_bismark_CX_report.txtpeakMets.tsv\n",
      "col0rep3\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976401_Col0_rep2_bismark_CX_report.txtpeakMets.tsv\n",
      "col0rep2\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976403_Col0_rep4_bismark_CX_report.txtpeakMets.tsv\n",
      "col0rep4\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0/GSM5976400_Col0_rep1_bismark_CX_report.txtpeakMets.tsv\n",
      "col0rep1\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976281_WaterControl_rep3_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "col0Wrep3\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976280_WaterControl_rep2_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "col0Wrep2\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/col0W/GSM4976279_WaterControl_rep1_bismark_genome_wide_cytosine_report.txtpeakMets.tsv\n",
      "col0Wrep1\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976411_nrpe1-11_rep4_bismark_CX_report.txtpeakMets.tsv\n",
      "nrpe1rep4\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976408_nrpe1-11_rep1_bismark_CX_report.txtpeakMets.tsv\n",
      "nrpe1rep1\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976410_nrpe1-11_rep3_bismark_CX_report.txtpeakMets.tsv\n",
      "nrpe1rep3\n",
      "/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/nrpe1/GSM5976409_nrpe1-11_rep2_bismark_CX_report.txtpeakMets.tsv\n",
      "nrpe1rep2\n"
     ]
    }
   ],
   "source": [
    "allthedata = []\n",
    "for experimentCondition in experimentsClasification:\n",
    "    listOfDfReplicatesPercentageOfMetCs = []\n",
    "    listOfDfReplicatesCualitative = []\n",
    "    listOfNamesReplicates = []\n",
    "    for replicatePath in experimentsClasification[experimentCondition]:\n",
    "        print(replicatePath)\n",
    "        replicaname =  experimentCondition+replicatePath.split('/')[-1].split('_')[2]\n",
    "        print(replicaname)\n",
    "        replicateDf = loadFileRemoveMtsAndLowMetReport(replicatePath)\n",
    "\n",
    "        replicateDfPercentageOfMetCs = calculateValueAsPercentageOfMetCs(replicateDf,replicaname)\n",
    "        listOfNamesReplicates.append(replicaname)\n",
    "        listOfDfReplicatesPercentageOfMetCs.append(replicateDfPercentageOfMetCs)\n",
    "\n",
    "        \n",
    "    df_merged = reduce(lambda  left,right: pd.merge(left,right,on=['chr','position', 'strand','context','boxID'],\n",
    "                                                how='outer'), listOfDfReplicatesPercentageOfMetCs).fillna(0)\n",
    "    df_merged[experimentCondition] = df_merged[listOfNamesReplicates].mean(axis=1)\n",
    "    df_merged = df_merged.drop(columns=listOfNamesReplicates)\n",
    "    allthedata.append(df_merged)\n",
    "finaldf = reduce(lambda  left,right: pd.merge(left,right,on=['chr','position', 'strand','context','boxID'],\n",
    "                                                how='outer'), allthedata).fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ros1</th>\n",
       "      <th>col0JA</th>\n",
       "      <th>col0</th>\n",
       "      <th>col0W</th>\n",
       "      <th>nrpe1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>boxID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1:10000619</th>\n",
       "      <td>0.001291</td>\n",
       "      <td>0.000782</td>\n",
       "      <td>0.003081</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:10001679</th>\n",
       "      <td>0.000791</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001319</td>\n",
       "      <td>0.005472</td>\n",
       "      <td>0.001718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:10002938</th>\n",
       "      <td>0.000627</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001425</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>0.002887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:1000436</th>\n",
       "      <td>0.002778</td>\n",
       "      <td>0.001058</td>\n",
       "      <td>0.001891</td>\n",
       "      <td>0.000919</td>\n",
       "      <td>0.000441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:10005021</th>\n",
       "      <td>0.002356</td>\n",
       "      <td>0.001488</td>\n",
       "      <td>0.002089</td>\n",
       "      <td>0.004249</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:9995993</th>\n",
       "      <td>0.232773</td>\n",
       "      <td>0.282835</td>\n",
       "      <td>0.267687</td>\n",
       "      <td>0.273808</td>\n",
       "      <td>0.275903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:9996636</th>\n",
       "      <td>0.329594</td>\n",
       "      <td>0.189585</td>\n",
       "      <td>0.226895</td>\n",
       "      <td>0.167944</td>\n",
       "      <td>0.276521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:9997107</th>\n",
       "      <td>0.231632</td>\n",
       "      <td>0.180334</td>\n",
       "      <td>0.247682</td>\n",
       "      <td>0.187670</td>\n",
       "      <td>0.224321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:9998635</th>\n",
       "      <td>0.212124</td>\n",
       "      <td>0.279672</td>\n",
       "      <td>0.253338</td>\n",
       "      <td>0.252742</td>\n",
       "      <td>0.233708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:9998667</th>\n",
       "      <td>0.228573</td>\n",
       "      <td>0.296566</td>\n",
       "      <td>0.267565</td>\n",
       "      <td>0.272042</td>\n",
       "      <td>0.244263</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59195 rows Ã 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ros1    col0JA      col0     col0W     nrpe1\n",
       "boxID                                                       \n",
       "1:10000619  0.001291  0.000782  0.003081  0.000000  0.002171\n",
       "1:10001679  0.000791  0.000000  0.001319  0.005472  0.001718\n",
       "1:10002938  0.000627  0.000000  0.001425  0.002875  0.002887\n",
       "1:1000436   0.002778  0.001058  0.001891  0.000919  0.000441\n",
       "1:10005021  0.002356  0.001488  0.002089  0.004249  0.000000\n",
       "...              ...       ...       ...       ...       ...\n",
       "5:9995993   0.232773  0.282835  0.267687  0.273808  0.275903\n",
       "5:9996636   0.329594  0.189585  0.226895  0.167944  0.276521\n",
       "5:9997107   0.231632  0.180334  0.247682  0.187670  0.224321\n",
       "5:9998635   0.212124  0.279672  0.253338  0.252742  0.233708\n",
       "5:9998667   0.228573  0.296566  0.267565  0.272042  0.244263\n",
       "\n",
       "[59195 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>position</th>\n",
       "      <th>strand</th>\n",
       "      <th>context</th>\n",
       "      <th>boxID</th>\n",
       "      <th>ros1finalreps_x</th>\n",
       "      <th>ros1finalreps_y</th>\n",
       "      <th>ros1finalreps_x</th>\n",
       "      <th>ros1finalreps_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1439</td>\n",
       "      <td>+</td>\n",
       "      <td>CG</td>\n",
       "      <td>1:1538</td>\n",
       "      <td>0.777778</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.714286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1472</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1:1538</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1475</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1:1538</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1477</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1:1538</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1496</td>\n",
       "      <td>+</td>\n",
       "      <td>CHH</td>\n",
       "      <td>1:1538</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3991172</th>\n",
       "      <td>5</td>\n",
       "      <td>26933655</td>\n",
       "      <td>-</td>\n",
       "      <td>CHH</td>\n",
       "      <td>5:26933620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3991173</th>\n",
       "      <td>5</td>\n",
       "      <td>26933658</td>\n",
       "      <td>-</td>\n",
       "      <td>CHH</td>\n",
       "      <td>5:26933620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3991174</th>\n",
       "      <td>5</td>\n",
       "      <td>26933661</td>\n",
       "      <td>-</td>\n",
       "      <td>CHH</td>\n",
       "      <td>5:26933620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3991175</th>\n",
       "      <td>5</td>\n",
       "      <td>26933669</td>\n",
       "      <td>-</td>\n",
       "      <td>CHH</td>\n",
       "      <td>5:26933620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3991176</th>\n",
       "      <td>5</td>\n",
       "      <td>26933680</td>\n",
       "      <td>-</td>\n",
       "      <td>CHH</td>\n",
       "      <td>5:26933620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3991177 rows Ã 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr  position strand context       boxID  ros1finalreps_x  \\\n",
       "0          1      1439      +      CG      1:1538         0.777778   \n",
       "1          1      1472      +     CHH      1:1538         0.000000   \n",
       "2          1      1475      +     CHH      1:1538         0.000000   \n",
       "3          1      1477      +     CHH      1:1538         0.000000   \n",
       "4          1      1496      +     CHH      1:1538         0.000000   \n",
       "...      ...       ...    ...     ...         ...              ...   \n",
       "3991172    5  26933655      -     CHH  5:26933620         0.000000   \n",
       "3991173    5  26933658      -     CHH  5:26933620         0.000000   \n",
       "3991174    5  26933661      -     CHH  5:26933620         0.000000   \n",
       "3991175    5  26933669      -     CHH  5:26933620         0.000000   \n",
       "3991176    5  26933680      -     CHH  5:26933620         0.000000   \n",
       "\n",
       "         ros1finalreps_y  ros1finalreps_x  ros1finalreps_y  \n",
       "0                    0.7              1.0         0.714286  \n",
       "1                    0.0              0.0         0.000000  \n",
       "2                    0.0              0.0         0.000000  \n",
       "3                    0.0              0.0         0.000000  \n",
       "4                    0.0              0.0         0.000000  \n",
       "...                  ...              ...              ...  \n",
       "3991172              0.0              0.0         0.000000  \n",
       "3991173              0.0              0.0         0.000000  \n",
       "3991174              0.0              0.0         0.000000  \n",
       "3991175              0.0              0.0         0.000000  \n",
       "3991176              0.0              0.0         0.000000  \n",
       "\n",
       "[3991177 rows x 9 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finaldf.to_csv('bisulfite.tsv',sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finaldf.to_csv('bisulfite.tsv',sep='\\t', index=False)\n",
    "CGdf = finaldf[(finaldf['context'] == 'CG')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "finaldf = finaldf[['chr','position','strand','context','boxID','1Mock','1JA','1ACC','6Mock','6JA','6ACC','24Mock','24JA','24ACC']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finaldf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "CGdf = finaldf[(finaldf['context'] == 'CG')]\n",
    "CHHdf = finaldf[(finaldf['context'] == 'CHH')]\n",
    "CHGdf = finaldf[(finaldf['context'] == 'CHG')]\n",
    "\n",
    "CGdf.to_csv('CG_BSMet.tsv',sep='\\t', index=False)\n",
    "\n",
    "CHGdf.to_csv('CHG_BSMet.tsv',sep='\\t', index=False)\n",
    "\n",
    "CHHdf.to_csv('CHH_BSMet.tsv',sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "CGdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "CGdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import math\n",
    "# CGdf = CGdf.drop(columns=['chr','position', 'strand','context','boxID'])\n",
    "# CGdf = CGdf.set_index()\n",
    "\n",
    "# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "CGdf = CGdf[['1Mock','6Mock','24Mock','1ACC','6ACC','24ACC','1JA','6JA','24JA']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.heatmap(CGdf, cmap=\"mako\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "fig, ax = plt.subplots(figsize=(10,20))         # Sample figsize in inches\n",
    "sns.heatmap(CGdf, cmap=\"tab10\", ax=ax)\n",
    "# sns.set_context(\"notebook\", font_scale=1.75, rc={\"lines.linewidth\": 2.5, 'font.family':'Helvetica'})\n",
    "# sns.heatmap(CGdf, cmap=\"tab10\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Sinmenosde1 = CGdf[(CGdf >= 0.01).all(1)]\n",
    "fig, ax = plt.subplots(figsize=(10,20)) # Sample figsize in inches\n",
    "sns.heatmap(Sinmenosde1, cmap=\"tab10\", ax=ax)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.set_context(\"paper\", font_scale=1.75, rc={\"lines.linewidth\": 2.5, 'font.family':'Helvetica'})\n",
    "fig, ax = plt.subplots(figsize=(10,30))         # Sample figsize in inches\n",
    "sns.heatmap(Sinmenosde1, cmap=\"mako\", ax=ax)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Sinmenosde1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for col in ['1Mock','6Mock','24Mock','1ACC','6ACC','24ACC','1JA','6JA','24JA']:\n",
    "    print(col, CGdf[col].mean()*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHHdf = finaldf[(finaldf['context'] == 'CHH')]\n",
    "for col in ['1Mock','6Mock','24Mock','1ACC','6ACC','24ACC','1JA','6JA','24JA']:\n",
    "    print(col, CHHdf[col].mean()*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHGdf = finaldf[(finaldf['context'] == 'CHG')]\n",
    "for col in ['1Mock','6Mock','24Mock','1ACC','6ACC','24ACC','1JA','6JA','24JA']:\n",
    "    print(col, CHGdf[col].mean()*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mergeResultsMets(folder):\n",
    "    with cd('/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2/reports/'):\n",
    "        file_names = os.listdir()\n",
    "        for rep in ['rep1', 'rep2']:\n",
    "            for side in ['.motif.', 'motifSide']:\n",
    "                for metcontex in ['CG', 'CHG', 'CHH']:\n",
    "                    for box in ['G_box', 'GAC_box', 'PBE_box', 'GCC_box', 'TG_box']:\n",
    "                        boxfiles = []\n",
    "                        for file in file_names:\n",
    "                            # print(file_names)\n",
    "                            # print('hello')\n",
    "                            if rep in file and side in file and metcontex in file and file.startswith(box):\n",
    "                                # print('hello')\n",
    "                                if '_merged_intersect.csv' not in file:\n",
    "                                    boxfiles.append(file)\n",
    "                        # print(boxfiles)\n",
    "                        \n",
    "                        if box + rep + side + metcontex + '_merged_intersect.csv' in file_names:\n",
    "                            experiment1df = pd.read_csv(box + rep + side + metcontex + '_merged_intersect.csv')\n",
    "                            experiment2df = pd.read_csv(boxfiles[0])\n",
    "                            mergedDf = experiment1df.merge(experiment2df, on=['chr', 'position', 'strand','boxID'], how='outer')\n",
    "                            mergedDf.to_csv(box + rep + side + metcontex + '_merged_intersect.csv', index=False)\n",
    "                            os.remove(boxfiles[0])\n",
    "                        elif len(boxfiles) == 2:\n",
    "                            # print('hello')\n",
    "                            experiment1df = pd.read_csv(boxfiles[0])\n",
    "                            experiment2df = pd.read_csv(boxfiles[1])\n",
    "                            # print(experiment1df)\n",
    "                            mergedDf = experiment1df.merge(experiment2df, on=['chr', 'position', 'strand','boxID'], how='outer')\n",
    "                            mergedDf.to_csv(box + rep + side + metcontex + '_merged_intersect.csv', index=False)\n",
    "                            os.remove(boxfiles[0])\n",
    "                            os.remove(boxfiles[1])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "peaksfile = '/home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/allThePossiblePeaksnine.bed'\n",
    "\n",
    "peaks = pd.read_csv(peaksfile, sep='\\t', names=['chrm','start','end','boxID'])\n",
    "def getpositions(x):\n",
    "    chrm, center = x.strip().split(':')\n",
    "    return chrm, int(center)-100, int(center)+101\n",
    "df  = finaldf[(finaldf['context'] != 'CHG')]\n",
    "metadata = df.iloc[:,4:5]\n",
    "values = df.iloc[:,5:]\n",
    "# values = values.loc[(values!=0).any(axis=1)]\n",
    "# merged = metadata.merge(values, left_index=True, right_index=True, how='inner')\n",
    "# meandf = merged.groupby('boxID', as_index=False).mean()\n",
    "# peaks = peaks.merge(meandf, on='boxID', how='outer')\n",
    "# peaks =peaks.fillna(0)\n",
    "\n",
    "# # peaks.to_csv('C/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/G_BSMet_mean.tsv',sep='\\t', index=False)\n",
    "\n",
    "# # peaks.to_csv('/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/pape/rCHG_BSMet_mean.tsv',sep='\\t', index=False)\n",
    "\n",
    "# peaks.to_csv('/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/non-CG_BSMetmean.tsv',sep='\\t', index=False)\n",
    "# # meandf['chr'], meandf['start'], meandf['end'] =  zip(*meandf['boxID'].map(lambda x: getpositions(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ros1</th>\n",
       "      <th>col0JA</th>\n",
       "      <th>col0</th>\n",
       "      <th>col0W</th>\n",
       "      <th>nrpe1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.798016</td>\n",
       "      <td>0.047619</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.035714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.027778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.027778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.027778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4256523</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4256524</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4256525</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4256526</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4256527</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3733031 rows Ã 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             ros1    col0JA  col0  col0W     nrpe1\n",
       "0        0.798016  0.047619   0.0    0.0  0.035714\n",
       "1        0.000000  0.000000   0.0    0.0  0.027778\n",
       "2        0.000000  0.000000   0.0    0.0  0.027778\n",
       "3        0.000000  0.000000   0.0    0.0  0.027778\n",
       "4        0.000000  0.000000   0.0    0.0  0.000000\n",
       "...           ...       ...   ...    ...       ...\n",
       "4256523  0.000000  0.000000   0.0    0.0  0.000000\n",
       "4256524  0.000000  0.000000   0.0    0.0  0.000000\n",
       "4256525  0.000000  0.000000   0.0    0.0  0.000000\n",
       "4256526  0.000000  0.000000   0.0    0.0  0.000000\n",
       "4256527  0.000000  0.000000   0.0    0.0  0.000000\n",
       "\n",
       "[3733031 rows x 5 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "peaksfile = '/home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/allThePossiblePeaksnine.bed'\n",
    "\n",
    "peaks = pd.read_csv(peaksfile, sep='\\t', names=['chrm','start','end','boxID'])\n",
    "def getpositions(x):\n",
    "    chrm, center = x.strip().split(':')\n",
    "    return chrm, int(center)-100, int(center)+101\n",
    "df  = finaldf[(finaldf['context'] != 'CG')]\n",
    "# metadata = df.iloc[:,4:5]\n",
    "# values = df.iloc[:,5:]\n",
    "values = df.set_index(['chr','position','strand','context','boxID'])\n",
    "values = values.loc[(values!=0).any(axis=1)]\n",
    "# merged = metadata.merge(values, left_index=True, right_index=True, how='inner')\n",
    "meandf = values.groupby('boxID').mean()\n",
    "peaks = peaks.merge(meandf, on='boxID', how='outer')\n",
    "peaks =peaks.fillna(0)\n",
    "\n",
    "# peaks.to_csv('/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/CG_BSMet_mean.tsv',sep='\\t', index=False)\n",
    "\n",
    "# peaks.to_csv('/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/CHG_BSMet_mean.tsv',sep='\\t', index=False)\n",
    "\n",
    "peaks.to_csv('/home/joaquin/projects/methylation/data/bisulfite_rep1_rep2_rep3/finalreps/paper/non-CG_BSMetmean.tsv',sep='\\t', index=False)\n",
    "# meandf['chr'], meandf['start'], meandf['end'] =  zip(*meandf['boxID'].map(lambda x: getpositions(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrm</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>boxID</th>\n",
       "      <th>ros1</th>\n",
       "      <th>col0JA</th>\n",
       "      <th>col0</th>\n",
       "      <th>col0W</th>\n",
       "      <th>nrpe1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1438</td>\n",
       "      <td>1639</td>\n",
       "      <td>1:1538</td>\n",
       "      <td>0.276505</td>\n",
       "      <td>0.007937</td>\n",
       "      <td>0.005208</td>\n",
       "      <td>0.006944</td>\n",
       "      <td>0.024702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>10141</td>\n",
       "      <td>10342</td>\n",
       "      <td>1:10241</td>\n",
       "      <td>0.035714</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.041667</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>10610</td>\n",
       "      <td>10811</td>\n",
       "      <td>1:10710</td>\n",
       "      <td>0.050000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>10632</td>\n",
       "      <td>10833</td>\n",
       "      <td>1:10732</td>\n",
       "      <td>0.050000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>10738</td>\n",
       "      <td>10939</td>\n",
       "      <td>1:10838</td>\n",
       "      <td>0.050000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59745</th>\n",
       "      <td>5</td>\n",
       "      <td>26974230</td>\n",
       "      <td>26974431</td>\n",
       "      <td>5:26974330</td>\n",
       "      <td>0.860217</td>\n",
       "      <td>0.480849</td>\n",
       "      <td>0.330296</td>\n",
       "      <td>0.582856</td>\n",
       "      <td>0.519475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59746</th>\n",
       "      <td>5</td>\n",
       "      <td>26974256</td>\n",
       "      <td>26974457</td>\n",
       "      <td>5:26974356</td>\n",
       "      <td>0.860217</td>\n",
       "      <td>0.480849</td>\n",
       "      <td>0.330296</td>\n",
       "      <td>0.582856</td>\n",
       "      <td>0.519475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59747</th>\n",
       "      <td>5</td>\n",
       "      <td>26974303</td>\n",
       "      <td>26974504</td>\n",
       "      <td>5:26974403</td>\n",
       "      <td>0.778080</td>\n",
       "      <td>0.266035</td>\n",
       "      <td>0.218155</td>\n",
       "      <td>0.289394</td>\n",
       "      <td>0.018717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59748</th>\n",
       "      <td>5</td>\n",
       "      <td>26974337</td>\n",
       "      <td>26974538</td>\n",
       "      <td>5:26974437</td>\n",
       "      <td>0.607143</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0.233631</td>\n",
       "      <td>0.060049</td>\n",
       "      <td>0.010417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59749</th>\n",
       "      <td>5</td>\n",
       "      <td>26974754</td>\n",
       "      <td>26974955</td>\n",
       "      <td>5:26974854</td>\n",
       "      <td>0.903990</td>\n",
       "      <td>0.710707</td>\n",
       "      <td>0.651515</td>\n",
       "      <td>0.675527</td>\n",
       "      <td>0.652236</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59750 rows Ã 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrm     start       end       boxID      ros1    col0JA      col0  \\\n",
       "0         1      1438      1639      1:1538  0.276505  0.007937  0.005208   \n",
       "1         1     10141     10342     1:10241  0.035714  0.000000  0.000000   \n",
       "2         1     10610     10811     1:10710  0.050000  0.000000  0.000000   \n",
       "3         1     10632     10833     1:10732  0.050000  0.000000  0.000000   \n",
       "4         1     10738     10939     1:10838  0.050000  0.000000  0.000000   \n",
       "...     ...       ...       ...         ...       ...       ...       ...   \n",
       "59745     5  26974230  26974431  5:26974330  0.860217  0.480849  0.330296   \n",
       "59746     5  26974256  26974457  5:26974356  0.860217  0.480849  0.330296   \n",
       "59747     5  26974303  26974504  5:26974403  0.778080  0.266035  0.218155   \n",
       "59748     5  26974337  26974538  5:26974437  0.607143  0.015152  0.233631   \n",
       "59749     5  26974754  26974955  5:26974854  0.903990  0.710707  0.651515   \n",
       "\n",
       "          col0W     nrpe1  \n",
       "0      0.006944  0.024702  \n",
       "1      0.041667  0.000000  \n",
       "2      0.000000  0.000000  \n",
       "3      0.000000  0.000000  \n",
       "4      0.000000  0.000000  \n",
       "...         ...       ...  \n",
       "59745  0.582856  0.519475  \n",
       "59746  0.582856  0.519475  \n",
       "59747  0.289394  0.018717  \n",
       "59748  0.060049  0.010417  \n",
       "59749  0.675527  0.652236  \n",
       "\n",
       "[59750 rows x 9 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "peaksfile = '/home/joaquin/projects/methylation/data/commonData/arabidopsisThaliana/intersect/allThePossiblePeaksnine.bed'\n",
    "\n",
    "peaks = pd.read_csv(peaksfile, sep='\\t', names=['chrm','start','end','boxID'])\n",
    "def getpositions(x):\n",
    "    chrm, center = x.strip().split(':')\n",
    "    return chrm, int(center)-100, int(center)+101\n",
    "df  = pd.read_csv('CG_BSMet.tsv', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = df.iloc[:,4:5]\n",
    "values = df.iloc[:,5:]\n",
    "values = values.loc[(values!=0).any(axis=1)]\n",
    "merged = metadata.merge(values, left_index=True, right_index=True, how='inner')\n",
    "meandf = merged.groupby('boxID', as_index=False).mean()\n",
    "peaks = peaks.merge(meandf, on='boxID', how='outer')\n",
    "peaks =peaks.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "with pd.option_context('display.max_rows', None, 'display.max_columns', None):  # more options can be specified also\n",
    "    print(null_data)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "b311b5e2c1aaa04d48066871c65b4e8ce80fb760bb38a0437727056581097150"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
