pigz 2.6
Analysis complete for INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  
Archive:  INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX9_BKDL210038288-1A-9_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  
pigz 2.6
Analysis complete for INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  
Archive:  INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/03 16:31:20

Loading data...
    Loading reads from: CS142-CDFvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/FVechrs/ --s 150000000 --expt CS142-CDFvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDF/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 10101714	HitCounts: 2.3237999E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 10101714	HitCounts: 1.7398383E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 83 reads.


	IP: 17398383	Ctrl: 15832687	 IP/Ctrl: 1.10

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 0.14

The genome is segmented into 201014 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/201014	53.3s
1000	/201014	1.2m
10000	/201014	3.4m
100000	/201014	4.4m
201014	/201014	4.4m

Scaling condition , IP/Control = 0.47

---------------------------
Events discovered 
Significant:	23851
Insignificant:	334
Filtered:	1269

Total 17433 homotypic events (within 501bp of other events).

Finish binding event prediction: 4.4m

Refine read distribution from 2000 binding events. 


============================ Round 1 ============================

Running with 10 threads ...

100	/201014	1.1s
1000	/201014	2.1s
10000	/201014	7.4s
100000	/201014	1.2m
200000	/201014	2.2m
201014	/201014	3.6m
---------------------------
Events discovered 
Significant:	32586
Insignificant:	113
Filtered:	28

Total 6536 homotypic events (within 545bp of other events).

Finish binding event prediction: 3.6m

Refine read distribution from 2000 binding events. 

Loading genome sequences ...
Done, 2.2s
GC content=0.41

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.31. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=10
k=6+1, gapped kmer=19, base kmer=28
k=6+2, gapped kmer=37, base kmer=43
k=6+3, gapped kmer=41, base kmer=34
k=6+4, gapped kmer=37, base kmer=23

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 10

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTTTTT/AAAAAG	303	311	209	-9.0    	0	407.9	3.0	1223.7	8
TACTTT/AAAGTA	132	133	61	-8.4    	6	244.6	2.0	489.3	3

Align and cluster 24 k-mers with seed CTTTTT,   	#0
Align and cluster 18 k-mers with seed TACTTT,   	#1

Merging redundant motifs ...
118ms: Remove motif #1.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= TACTTT 	2.21, 144, kAUC=17.9	PWM= ACTTT	7.21, 387, pAUC=7.7
#1	k=6	KSM= CTTTTT 	2.70, 107, kAUC=15.7	PWM= CTTTTT	6.92, 496, pAUC=7.2

After merging:
#0	k=6	KSM= TACTTT 	2.21, 144, kAUC=17.9	PWM= ACTTT	7.21, 387, pAUC=7.7



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=10
k=7+1, gapped kmer=22, base kmer=31
k=7+2, gapped kmer=27, base kmer=28
k=7+3, gapped kmer=44, base kmer=24
k=7+4, gapped kmer=23, base kmer=14

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 10

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTCTTTT/AAAAGAA	156	160	97	-5.5    	0	206.3	4.0	825.2	4
ATAAAGG/CCTTTAT	39	40	6	-7.9    	8	79.0	2.0	158.0	2
CAACTTT/AAAGTTG	42	43	12	-5.6    	7	71.8	2.0	143.7	1

Align and cluster 24 k-mers with seed TTCTTTT,   	#0
Align and cluster 8 k-mers with seed ATAAAGG,   	#1
Align and cluster 15 k-mers with seed CAACTTT,   	#2

Merging redundant motifs ...
120ms: Remove motif #1.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= CAACTTT 	2.67, 36, kAUC=19.7	PWM= CTACTTT	7.13, 431, pAUC=13.4
#1	k=7	KSM= TTCTTTT 	1.97, 110, kAUC=16.5	PWM= TTCTTTT	8.15, 311, pAUC=8.0
#2	k=7	KSM= ATAAAGG 	3.31, 40, kAUC=10.0	PWM= ATAAAGG	8.54, 74, pAUC=9.8

After merging:
#0	k=7	KSM= CAACTTT 	2.67, 36, kAUC=19.7	PWM= CTACTTT	7.13, 431, pAUC=13.4
#2	k=7	KSM= ATAAAGG 	3.31, 40, kAUC=10.0	PWM= ATAAAGG	8.54, 74, pAUC=9.8



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=9
k=8+1, gapped kmer=16, base kmer=22
k=8+2, gapped kmer=31, base kmer=27
k=8+3, gapped kmer=22, base kmer=17
k=8+4, gapped kmer=26, base kmer=15

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 9

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTCTTT/AAAGAAAA	111	115	53	-7.2    	1	145.1	5.0	725.3	4
AAAAGTGA/TCACTTTT	30	31	7	-5.0    	4	72.2	2.0	144.3	2
ACTTTTCT/AGAAAAGT	25	27	3	-6.5    	5	42.8	2.0	85.7	1
TTTCTTTT/AAAAGAAA	113	117	60	-6.0

Align and cluster 33 k-mers with seed TTTTCTTT,   	#0
Align and cluster 13 k-mers with seed AAAAGTGA,   	#1
Align and cluster 14 k-mers with seed ACTTTTCT,   	#2
Align and cluster 31 k-mers with seed TTTCTTTT,   	#3

Merging redundant motifs ...
96ms: Remove motif #1.
251ms: Remove motif #2.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= TTTTCTTT 	1.94, 111, kAUC=24.8	PWM= TTTTCTTT	9.18, 260, pAUC=10.4
#1	k=8	KSM= TTTCTTTT 	2.18, 92, kAUC=22.8	PWM= TTTCTTTT	8.65, 272, pAUC=9.4
#2	k=8	KSM= ACTTTTCT 	2.07, 116, kAUC=16.6	PWM= TCTTTTCT	8.70, 292, pAUC=11.7
#3	k=8	KSM= AAAAGTGA 	2.49, 36, kAUC=11.9	PWM= AAAAGTGA	9.74, 89, pAUC=12.5

After merging:
#0	k=8	KSM= TTTTCTTT 	2.60, 77, kAUC=28.2	PWM= TTTTCTTT	8.46, 286, pAUC=10.4
#3	k=8	KSM= AAAAGTGA 	2.49, 36, kAUC=11.9	PWM= AAAAGTGA	9.74, 89, pAUC=12.5



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=5
k=9+1, gapped kmer=14, base kmer=18
k=9+2, gapped kmer=22, base kmer=20
k=9+3, gapped kmer=32, base kmer=15
k=9+4, gapped kmer=60, base kmer=17

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 5

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTCTTTT/AAAAGAAAA	84	87	33	-7.4    	0	111.9	3.0	335.8	5

Align and cluster 37 k-mers with seed TTTTCTTTT,   	#0

------------------------- k = 9 ----------------------------

#0	k=9	KSM= TTTTCTTTT 	3.79, 30, kAUC=17.1	PWM= TTTTCTTTT	10.91, 158, pAUC=11.9



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=5
k=10+1, gapped kmer=8, base kmer=12
k=10+2, gapped kmer=34, base kmer=18
k=10+3, gapped kmer=30, base kmer=17
k=10+4, gapped kmer=30, base kmer=14

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 5

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTTCTTTT/AAAAGAAAAA	61	64	23	-6.0    	0	80.0	2.0	160.0	5

Align and cluster 54 k-mers with seed TTTTTCTTTT,   	#0

------------------------- k = 10 ----------------------------

#0	k=10	KSM= TTTTTCTTTT 	2.30, 44, kAUC=15.3	PWM= TTTTTCTTTT	11.64, 146, pAUC=13.0



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=6
k=11+1, gapped kmer=12, base kmer=13
k=11+2, gapped kmer=29, base kmer=17
k=11+3, gapped kmer=24, base kmer=14
k=11+4, gapped kmer=5, base kmer=8

------------------------- k = 11 ----------------------------

Total number of k-mers: n = 6

Top 5 cluster center k-mers:
# k-mer/r.c.           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTTTTTCTTTT/AAAAGAAAAAG	27	29	4	-6.3    	0	38.8	4.0	155.3	4

Align and cluster 52 k-mers with seed CTTTTTCTTTT,   	#0

------------------------- k = 11 ----------------------------

#0	k=11	KSM= CTTTTTCTTTT 	2.78, 44, kAUC=15.3	PWM= CTTTTTCTTTT	12.84, 134, pAUC=13.4



----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=5
k=12+1, gapped kmer=9, base kmer=13
k=12+2, gapped kmer=26, base kmer=14
k=12+3, gapped kmer=13, base kmer=9
k=12+4, gapped kmer=1, base kmer=3

------------------------- k = 12 ----------------------------

Total number of k-mers: n = 5

Top 5 cluster center k-mers:
# k-mer/r.c.             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTCTTTTTCTTT/AAAGAAAAAGAA	17	19	2	-5.0    	0	29.6	4.0	118.3	5

Align and cluster 42 k-mers with seed TTCTTTTTCTTT,   	#0

------------------------- k = 12 ----------------------------

#0	k=12	KSM= TTCTTTTTCTTT 	1.89, 34, kAUC=12.7	PWM= TTCTTTTTCTTT	13.77, 114, pAUC=13.2



----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=2
k=13+1, gapped kmer=9, base kmer=9
k=13+2, gapped kmer=11, base kmer=9
k=13+3, gapped kmer=2, base kmer=4
k=13+4, gapped kmer=0, base kmer=0

------------------------- k = 13 ----------------------------

Total number of k-mers: n = 2

Top 5 cluster center k-mers:
# k-mer/r.c.               	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAAAAGAAA/TTTCTTTTTTTTT	16	16	1	-5.2    	0	21.2	3.0	63.5	2

Align and cluster 16 k-mers with seed AAAAAAAAAGAAA,   	#0

------------------------- k = 13 ----------------------------

#0	k=13	KSM= AAAAAAAAAGAAA 	1.09, 49, kAUC=12.1	PWM= AAAAAAAAAGAAA	13.64, 124, pAUC=13.7



----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=2
k=14+1, gapped kmer=6, base kmer=7
k=14+2, gapped kmer=2, base kmer=3
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

------------------------- k = 14 ----------------------------

Total number of k-mers: n = 2

Top 5 cluster center k-mers:
# k-mer/r.c.                 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAAAAGAAAA/TTTTCTTTTTTTTT	14	14	0	-5.2    	0	18.3	1.0	18.3	2

Align and cluster 8 k-mers with seed AAAAAAAAAGAAAA,   	#0

------------------------- k = 14 ----------------------------

#0	k=14	KSM= AAAAAAAAAGAAAA 	1.10, 45, kAUC=11.5	PWM= AAAAAAAAAGAAAA	15.50, 97, pAUC=13.7



----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=1, base kmer=2
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= TTTTCTTT 	2.60, 77, kAUC=28.2	PWM= TTTTCTTT	8.46, 286, pAUC=10.4
#1	k=7	KSM= CAACTTT 	2.67, 36, kAUC=19.7	PWM= CTACTTT	7.13, 431, pAUC=13.4
#2	k=6	KSM= TACTTT 	2.21, 144, kAUC=17.9	PWM= ACTTT	7.21, 387, pAUC=7.7
#3	k=9	KSM= TTTTCTTTT 	3.79, 30, kAUC=17.1	PWM= TTTTCTTTT	10.91, 158, pAUC=11.9
#4	k=10	KSM= TTTTTCTTTT 	2.30, 44, kAUC=15.3	PWM= TTTTTCTTTT	11.64, 146, pAUC=13.0
#5	k=11	KSM= CTTTTTCTTTT 	2.78, 44, kAUC=15.3	PWM= CTTTTTCTTTT	12.84, 134, pAUC=13.4
#6	k=12	KSM= TTCTTTTTCTTT 	1.89, 34, kAUC=12.7	PWM= TTCTTTTTCTTT	13.77, 114, pAUC=13.2
#7	k=13	KSM= AAAAAAAAAGAAA 	1.09, 49, kAUC=12.1	PWM= AAAAAAAAAGAAA	13.64, 124, pAUC=13.7
#8	k=8	KSM= AAAAGTGA 	2.49, 36, kAUC=11.9	PWM= AAAAGTGA	9.74, 89, pAUC=12.5
#9	k=14	KSM= AAAAAAAAAGAAAA 	1.10, 45, kAUC=11.5	PWM= AAAAAAAAAGAAAA	15.50, 97, pAUC=13.7
#10	k=7	KSM= ATAAAGG 	3.31, 40, kAUC=10.0	PWM= ATAAAGG	8.54, 74, pAUC=9.8

116ms: Remove motif #1.
122ms: PWM TTTTCTTTT has been removed.
189ms: Remove motif #4.
258ms: Remove motif #5.
327ms: Remove motif #6.
398ms: Remove motif #7.
470ms: Remove motif #9.
474ms: PWM TCACTTTT has been removed.

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= TTTTCTTT 	2.60, 77, kAUC=28.2	PWM= TTTTCTTT	8.46, 286, pAUC=10.4
#1	k=7	KSM= CAACTTT 	2.67, 36, kAUC=19.7	PWM= CTACTTT	7.13, 431, pAUC=13.4
#2	k=6	KSM= TACTTT 	2.21, 144, kAUC=17.9	PWM= ACTTT	7.21, 387, pAUC=7.7
#3	k=9	KSM= TTTTCTTTT 	3.79, 30, kAUC=17.1	PWM= TTTTCTTTT	10.91, 158, pAUC=11.9
#4	k=10	KSM= TTTTTCTTTT 	2.30, 44, kAUC=15.3	PWM= TTTTTCTTTT	11.64, 146, pAUC=13.0
#5	k=11	KSM= CTTTTTCTTTT 	2.78, 44, kAUC=15.3	PWM= CTTTTTCTTTT	12.84, 134, pAUC=13.4
#6	k=12	KSM= TTCTTTTTCTTT 	1.89, 34, kAUC=12.7	PWM= TTCTTTTTCTTT	13.77, 114, pAUC=13.2
#7	k=13	KSM= AAAAAAAAAGAAA 	1.09, 49, kAUC=12.1	PWM= AAAAAAAAAGAAA	13.64, 124, pAUC=13.7
#8	k=8	KSM= AAAAGTGA 	2.49, 36, kAUC=11.9	PWM= AAAAGTGA	9.74, 89, pAUC=12.5
#9	k=14	KSM= AAAAAAAAAGAAAA 	1.10, 45, kAUC=11.5	PWM= AAAAAAAAAGAAAA	15.50, 97, pAUC=13.7
#10	k=7	KSM= ATAAAGG 	3.31, 40, kAUC=10.0	PWM= ATAAAGG	8.54, 74, pAUC=9.8

------------------------- GEMout_1, k = ALL (after merging) -------------------------
#0	k=8	KSM= TTTTCTTT 	2.60, 77, kAUC=28.2	PWM= TTTTCTTT	8.46, 286, pAUC=10.4
#2	k=6	KSM= TACTTT 	2.21, 144, kAUC=17.9	PWM= ACTTT	7.21, 387, pAUC=7.7
#10	k=7	KSM= ATAAAGG 	3.31, 40, kAUC=10.0	PWM= ATAAAGG	8.54, 74, pAUC=9.8

510ms: Finalizing 3 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_1 final motifs ----------------------------


GEMout_1 motif #0

KSM top k-mer: TTTTCTTT/AAAGAAAA, total 21 k-mers
KSM threshold: 2.60, 		hit=77+/4-, kAUC=28.2

    |
TTTTCTTT
    TC  
        
        
PWM threshold: 8.46/12.23, 	hit=286+/251-, pAUC=10.4
--------------------------------------------------------------

GEMout_1 motif #1

KSM top k-mer: TACTTT/AAAGTA, total 14 k-mers
KSM threshold: 2.21, 		hit=144+/37-, kAUC=17.9

    |
ACTTT
     
     
     
PWM threshold: 7.21/9.06, 	hit=387+/266-, pAUC=7.7
--------------------------------------------------------------

GEMout_1 motif #2

KSM top k-mer: ATAAAGG/CCTTTAT, total 1 k-mers
KSM threshold: 3.31, 		hit=40+/6-, kAUC=10.0

|
ATAAAGG
       
       
       
PWM threshold: 8.54/12.46, 	hit=74+/43-, pAUC=9.8
--------------------------------------------------------------



============================ Round 2 ============================
Running EM with motif positional prior ...

Running with 10 threads ...

100	/201014	26.1s
1000	/201014	27.1s
10000	/201014	33.1s
100000	/201014	1.9m
200000	/201014	3.2m
201014	/201014	4.5m
---------------------------
Events discovered 
Significant:	34499
Insignificant:	161
Filtered:	8

Total 10094 homotypic events (within 545bp of other events).

Finish binding event prediction: 4.5m

Refine read distribution from 2000 binding events. 

Percentage of significant events with a motif match, divided in 10 bins (~3450 events per bin):
46 45 45 44 41 43 41 42 42 44 

Percentage of filtered events with a motif match, divided in 8 bins (~1 events per bin):
0 0 0 0 0 0 0 0 

Percentage of insignificant events with a motif match, divided in 9 bins (~17 events per bin):
47 41 52 41 47 11 29 23 29 62 

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.32. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=16
k=6+1, gapped kmer=20, base kmer=34
k=6+2, gapped kmer=48, base kmer=46
k=6+3, gapped kmer=61, base kmer=42
k=6+4, gapped kmer=47, base kmer=31

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 16

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTTTTT/AAAAAG	298	304	211	-7.7    	0	411.5	4.0	1646.0	12
TCACTT/AAGTGA	82	82	36	-5.7    	14	182.2	2.0	364.5	1
TACTTT/AAAGTA	114	114	53	-7.1    	11	235.9	1.0	235.9	2
CCATCT/AGATGG	17	16	1	-5.2    	15	52.6	2.0	105.2	1

Align and cluster 31 k-mers with seed CTTTTT,   	#0
Align and cluster 14 k-mers with seed TCACTT,   	#1
Align and cluster 26 k-mers with seed TACTTT,   	#2
Align and cluster 11 k-mers with seed CCATCT,   	#3

Merging redundant motifs ...
275ms: Remove motif #1.
396ms: Remove motif #2.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= CTTTTT 	2.06, 227, kAUC=17.7	PWM= CTTTTT	6.73, 519, pAUC=7.1
#1	k=6	KSM= TACTTT 	2.37, 173, kAUC=17.0	PWM= TACTTT	6.48, 510, pAUC=10.7
#2	k=6	KSM= TCACTT 	1.94, 152, kAUC=16.5	PWM= TCACTT	6.64, 242, pAUC=10.6
#3	k=6	KSM= CCATCT 	2.42, 16, kAUC=7.5	PWM= CCATCT	5.53, 36, pAUC=8.4

After merging:
#0	k=6	KSM= CTTTTT 	1.96, 194, kAUC=18.4	PWM= CTTTTT	7.03, 512, pAUC=7.1
#3	k=6	KSM= CCATCT 	2.42, 16, kAUC=7.5	PWM= CCATCT	5.53, 36, pAUC=8.4



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=13
k=7+1, gapped kmer=26, base kmer=34
k=7+2, gapped kmer=35, base kmer=39
k=7+3, gapped kmer=54, base kmer=35
k=7+4, gapped kmer=49, base kmer=29

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 13

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TCTTTTT/AAAAAGA	156	160	95	-5.8    	3	255.2	4.0	1020.7	9
TTCACTT/AAGTGAA	63	64	22	-6.3    	7	134.8	2.0	269.6	1
CAACTTT/AAAGTTG	38	39	11	-5.1    	11	79.0	2.0	158.0	1
TTTTCTT/AAGAAAA	172	176	84	-10.6

Align and cluster 44 k-mers with seed TCTTTTT,   	#0
Align and cluster 23 k-mers with seed TTCACTT,   	#1
Align and cluster 9 k-mers with seed CAACTTT,   	#2
Align and cluster 57 k-mers with seed TTTTCTT,   	#3

Merging redundant motifs ...
115ms: Remove motif #2.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= TTTTCTT 	2.40, 248, kAUC=24.8	PWM= TTTTCTT	8.28, 288, pAUC=10.2
#1	k=7	KSM= TTCACTT 	2.07, 227, kAUC=20.4	PWM= TTCACTT	7.82, 124, pAUC=12.5
#2	k=7	KSM= TCTTTTT 	2.50, 224, kAUC=20.4	PWM= TCTTTTT	7.90, 375, pAUC=8.2
#3	k=7	KSM= CAACTTT 	2.45, 39, kAUC=8.9	PWM= CAACTTT	7.68, 67, pAUC=8.9

After merging:
#0	k=7	KSM= TTTTCTT 	2.14, 179, kAUC=25.1	PWM= TTTTCTT	7.56, 353, pAUC=10.2
#1	k=7	KSM= TTCACTT 	2.07, 227, kAUC=20.4	PWM= TTCACTT	7.82, 124, pAUC=12.5
#3	k=7	KSM= CAACTTT 	2.45, 39, kAUC=8.9	PWM= CAACTTT	7.68, 67, pAUC=8.9



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=12
k=8+1, gapped kmer=24, base kmer=31
k=8+2, gapped kmer=35, base kmer=33
k=8+3, gapped kmer=57, base kmer=32
k=8+4, gapped kmer=47, base kmer=21

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 12

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTCTTT/AAAGAAAA	133	137	52	-11.6    	0	182.1	5.0	910.4	5
AAAAGTGA/TCACTTTT	46	47	10	-7.4    	8	94.2	2.0	188.5	1
ACTTTTCT/AGAAAAGT	42	42	10	-6.2    	9	82.5	2.0	165.0	1
AGCTACTA/TAGTAGCT	14	15	0	-5.5    	11	15.0	5.0	75.0	1

Align and cluster 72 k-mers with seed TTTTCTTT,   	#0
Align and cluster 20 k-mers with seed AAAAGTGA,   	#1
Align and cluster 26 k-mers with seed ACTTTTCT,   	#2
Align and cluster 1 k-mers with seed AGCTACTA,   	#3

Merging redundant motifs ...
95ms: Remove motif #1.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= TTTTCTTT 	4.02, 58, kAUC=26.8	PWM= TTTTCTTT	8.92, 255, pAUC=12.8
#1	k=8	KSM= ACTTTTCT 	2.88, 48, kAUC=24.3	PWM= TCTTTTCT	9.33, 155, pAUC=13.4
#2	k=8	KSM= AAAAGTGA 	3.29, 35, kAUC=16.9	PWM= AAAAGTGA	9.45, 127, pAUC=15.0
#3	k=8	KSM= AGCTACTA 	2.46, 15, kAUC=7.2	PWM= AGCTACTA	9.49, 17, pAUC=5.4

After merging:
#0	k=8	KSM= TTTTCTTT 	2.75, 65, kAUC=27.7	PWM= TTTTCTTT	9.23, 268, pAUC=12.8
#2	k=8	KSM= AAAAGTGA 	3.29, 35, kAUC=16.9	PWM= AAAAGTGA	9.45, 127, pAUC=15.0
#3	k=8	KSM= AGCTACTA 	2.46, 15, kAUC=7.2	PWM= AGCTACTA	9.49, 17, pAUC=5.4



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=15
k=9+1, gapped kmer=22, base kmer=27
k=9+2, gapped kmer=38, base kmer=29
k=9+3, gapped kmer=57, base kmer=26
k=9+4, gapped kmer=59, base kmer=20

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 15

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTCTTTT/AAAAGAAAA	98	101	29	-11.8    	0	151.1	5.0	755.4	13
AAGCTACTA/TAGTAGCTT	14	15	0	-5.5    	14	15.0	4.0	60.0	1

Align and cluster 78 k-mers with seed TTTTCTTTT,   	#0
Align and cluster 2 k-mers with seed AAGCTACTA,   	#1

Merging redundant motifs ...

------------------------- k = 9 ----------------------------

#0	k=9	KSM= TTTTCTTTT 	3.24, 76, kAUC=28.8	PWM= TTTTCTTTT	10.23, 261, pAUC=15.8
#1	k=9	KSM= AAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAGCTACTA	9.26, 23, pAUC=6.9

After merging:
#0	k=9	KSM= TTTTCTTTT 	3.24, 76, kAUC=28.8	PWM= TTTTCTTTT	10.23, 261, pAUC=15.8
#1	k=9	KSM= AAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAGCTACTA	9.26, 23, pAUC=6.9



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=16
k=10+1, gapped kmer=21, base kmer=25
k=10+2, gapped kmer=36, base kmer=25
k=10+3, gapped kmer=27, base kmer=23
k=10+4, gapped kmer=51, base kmer=15

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 16

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTTCTTTT/AAAAGAAAAA	71	74	21	-8.8    	0	120.7	5.0	603.7	15
AAAGCTACTA/TAGTAGCTTT	14	15	0	-5.5    	15	15.0	4.0	60.0	1

Align and cluster 83 k-mers with seed TTTTTCTTTT,   	#0
Align and cluster 9 k-mers with seed AAAGCTACTA,   	#1

Merging redundant motifs ...

------------------------- k = 10 ----------------------------

#0	k=10	KSM= TTTTTCTTTT 	3.13, 38, kAUC=22.2	PWM= TTTTTCTTTT	11.55, 170, pAUC=15.4
#1	k=10	KSM= AAAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAAGCAACTA	7.86, 109, pAUC=11.5

After merging:
#0	k=10	KSM= TTTTTCTTTT 	3.13, 38, kAUC=22.2	PWM= TTTTTCTTTT	11.55, 170, pAUC=15.4
#1	k=10	KSM= AAAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAAGCAACTA	7.86, 109, pAUC=11.5



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=14
k=11+1, gapped kmer=15, base kmer=22
k=11+2, gapped kmer=22, base kmer=20
k=11+3, gapped kmer=28, base kmer=17
k=11+4, gapped kmer=22, base kmer=8

------------------------- k = 11 ----------------------------

Total number of k-mers: n = 14

Top 5 cluster center k-mers:
# k-mer/r.c.           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTTCTTTTT/AAAAAGAAAAA	42	43	13	-5.2    	0	66.9	5.0	334.4	13

Align and cluster 63 k-mers with seed TTTTTCTTTTT,   	#0

------------------------- k = 11 ----------------------------

#0	k=11	KSM= TTTTTCTTTTT 	2.71, 50, kAUC=17.2	PWM= TTTTTCTTTTT	12.35, 188, pAUC=18.0



----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=9
k=12+1, gapped kmer=10, base kmer=16
k=12+2, gapped kmer=17, base kmer=15
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=6, base kmer=5

------------------------- k = 12 ----------------------------

Total number of k-mers: n = 9

Top 5 cluster center k-mers:
# k-mer/r.c.             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TCTTTTTCTTTT/AAAAGAAAAAGA	28	29	2	-8.0    	0	43.4	4.0	173.7	9

Align and cluster 31 k-mers with seed TCTTTTTCTTTT,   	#0

------------------------- k = 12 ----------------------------

#0	k=12	KSM= TCTTTTTCTTTT 	3.11, 22, kAUC=14.9	PWM= TCTTTTTCTTTT	12.55, 158, pAUC=16.5



----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=6
k=13+1, gapped kmer=8, base kmer=11
k=13+2, gapped kmer=6, base kmer=7
k=13+3, gapped kmer=3, base kmer=4
k=13+4, gapped kmer=3, base kmer=3

------------------------- k = 13 ----------------------------

Total number of k-mers: n = 6

Top 5 cluster center k-mers:
# k-mer/r.c.               	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTCTTTTTCTTTT/AAAAGAAAAAGAA	21	22	2	-5.9    	0	35.1	4.0	140.4	6

Align and cluster 17 k-mers with seed TTCTTTTTCTTTT,   	#0

------------------------- k = 13 ----------------------------

#0	k=13	KSM= TTCTTTTTCTTTT 	1.93, 43, kAUC=13.2	PWM= TTCTTTTTCTTTT	13.20, 156, pAUC=15.7



----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=2
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

------------------------- k = 14 ----------------------------

Total number of k-mers: n = 2

Top 5 cluster center k-mers:
# k-mer/r.c.                 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAAAAAGAAAA/TTTTCTTTTTCTTT	16	17	1	-5.5    	0	23.0	2.0	45.9	2

Align and cluster 2 k-mers with seed AAAGAAAAAGAAAA,   	#0

------------------------- k = 14 ----------------------------

#0	k=14	KSM= AAAGAAAAAGAAAA 	2.52, 17, kAUC=7.4	PWM= AAAGAAAAAGAAAA	13.22, 135, pAUC=15.2



----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=1, base kmer=2
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=9	KSM= TTTTCTTTT 	3.24, 76, kAUC=28.8	PWM= TTTTCTTTT	10.23, 261, pAUC=15.8
#1	k=8	KSM= TTTTCTTT 	2.75, 65, kAUC=27.7	PWM= TTTTCTTT	9.23, 268, pAUC=12.8
#2	k=7	KSM= TTTTCTT 	2.14, 179, kAUC=25.1	PWM= TTTTCTT	7.56, 353, pAUC=10.2
#3	k=10	KSM= TTTTTCTTTT 	3.13, 38, kAUC=22.2	PWM= TTTTTCTTTT	11.55, 170, pAUC=15.4
#4	k=7	KSM= TTCACTT 	2.07, 227, kAUC=20.4	PWM= TTCACTT	7.82, 124, pAUC=12.5
#5	k=6	KSM= CTTTTT 	1.96, 194, kAUC=18.4	PWM= CTTTTT	7.03, 512, pAUC=7.1
#6	k=11	KSM= TTTTTCTTTTT 	2.71, 50, kAUC=17.2	PWM= TTTTTCTTTTT	12.35, 188, pAUC=18.0
#7	k=8	KSM= AAAAGTGA 	3.29, 35, kAUC=16.9	PWM= AAAAGTGA	9.45, 127, pAUC=15.0
#8	k=12	KSM= TCTTTTTCTTTT 	3.11, 22, kAUC=14.9	PWM= TCTTTTTCTTTT	12.55, 158, pAUC=16.5
#9	k=13	KSM= TTCTTTTTCTTTT 	1.93, 43, kAUC=13.2	PWM= TTCTTTTTCTTTT	13.20, 156, pAUC=15.7
#10	k=7	KSM= CAACTTT 	2.45, 39, kAUC=8.9	PWM= CAACTTT	7.68, 67, pAUC=8.9
#11	k=6	KSM= CCATCT 	2.42, 16, kAUC=7.5	PWM= CCATCT	5.53, 36, pAUC=8.4
#12	k=14	KSM= AAAGAAAAAGAAAA 	2.52, 17, kAUC=7.4	PWM= AAAGAAAAAGAAAA	13.22, 135, pAUC=15.2
#13	k=10	KSM= AAAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAAGCAACTA	7.86, 109, pAUC=11.5
#14	k=9	KSM= AAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAGCTACTA	9.26, 23, pAUC=6.9
#15	k=8	KSM= AGCTACTA 	2.46, 15, kAUC=7.2	PWM= AGCTACTA	9.49, 17, pAUC=5.4

158ms: Remove motif #1.
367ms: Remove motif #2.
377ms: PWM TTTTTCTTTT has been removed.
507ms: Remove motif #5.
579ms: Remove motif #6.
704ms: Remove motif #8.
808ms: Remove motif #9.
913ms: Remove motif #12.
982ms: Remove motif #7.
1.0s: Remove motif #14.
1.1s: PWM AGCTACTA has been removed.
1.2s: Remove motif #4.

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=9	KSM= TTTTCTTTT 	3.24, 76, kAUC=28.8	PWM= TTTTCTTTT	10.23, 261, pAUC=15.8
#1	k=8	KSM= TTTTCTTT 	2.75, 65, kAUC=27.7	PWM= TTTTCTTT	9.23, 268, pAUC=12.8
#2	k=7	KSM= TTTTCTT 	2.14, 179, kAUC=25.1	PWM= TTTTCTT	7.56, 353, pAUC=10.2
#3	k=10	KSM= TTTTTCTTTT 	3.13, 38, kAUC=22.2	PWM= TTTTTCTTTT	11.55, 170, pAUC=15.4
#4	k=7	KSM= TTCACTT 	2.07, 227, kAUC=20.4	PWM= TTCACTT	7.82, 124, pAUC=12.5
#5	k=6	KSM= CTTTTT 	1.96, 194, kAUC=18.4	PWM= CTTTTT	7.03, 512, pAUC=7.1
#6	k=11	KSM= TTTTTCTTTTT 	2.71, 50, kAUC=17.2	PWM= TTTTTCTTTTT	12.35, 188, pAUC=18.0
#7	k=8	KSM= AAAAGTGA 	3.29, 35, kAUC=16.9	PWM= AAAAGTGA	9.45, 127, pAUC=15.0
#8	k=12	KSM= TCTTTTTCTTTT 	3.11, 22, kAUC=14.9	PWM= TCTTTTTCTTTT	12.55, 158, pAUC=16.5
#9	k=13	KSM= TTCTTTTTCTTTT 	1.93, 43, kAUC=13.2	PWM= TTCTTTTTCTTTT	13.20, 156, pAUC=15.7
#10	k=7	KSM= CAACTTT 	2.45, 39, kAUC=8.9	PWM= CAACTTT	7.68, 67, pAUC=8.9
#11	k=6	KSM= CCATCT 	2.42, 16, kAUC=7.5	PWM= CCATCT	5.53, 36, pAUC=8.4
#12	k=14	KSM= AAAGAAAAAGAAAA 	2.52, 17, kAUC=7.4	PWM= AAAGAAAAAGAAAA	13.22, 135, pAUC=15.2
#13	k=10	KSM= AAAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAAGCAACTA	7.86, 109, pAUC=11.5
#14	k=9	KSM= AAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAGCTACTA	9.26, 23, pAUC=6.9
#15	k=8	KSM= AGCTACTA 	2.46, 15, kAUC=7.2	PWM= AGCTACTA	9.49, 17, pAUC=5.4

------------------------- GEMout_2, k = ALL (after merging) -------------------------
#0	k=9	KSM= TTTTCTTTT 	2.97, 73, kAUC=29.9	PWM= TTTTCTTTT	10.19, 271, pAUC=16.5
#10	k=7	KSM= CAACTTT 	2.45, 39, kAUC=8.9	PWM= CAACTTT	7.68, 67, pAUC=8.9
#11	k=6	KSM= CCATCT 	2.42, 16, kAUC=7.5	PWM= CCATCT	5.53, 36, pAUC=8.4
#13	k=10	KSM= AAAGCTACTA 	2.46, 15, kAUC=7.2	PWM= AAAGCAACTA	7.86, 109, pAUC=11.5

1.2s: Finalizing 4 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_2 final motifs ----------------------------


GEMout_2 motif #0

KSM top k-mer: TTTTCTTTT/AAAAGAAAA, total 53 k-mers
KSM threshold: 2.97, 		hit=73+/4-, kAUC=29.9

        |
TTTTCTTTT
  C      
         
         
PWM threshold: 10.19/14.81, 	hit=271+/112-, pAUC=16.5
--------------------------------------------------------------

GEMout_2 motif #1

KSM top k-mer: CAACTTT/AAAGTTG, total 1 k-mers
KSM threshold: 2.45, 		hit=39+/11-, kAUC=8.9

           |
CAACTTT
       
       
       
PWM threshold: 7.68/12.63, 	hit=67+/42-, pAUC=8.9
--------------------------------------------------------------

GEMout_2 motif #2

KSM top k-mer: CCATCT/AGATGG, total 1 k-mers
KSM threshold: 2.42, 		hit=16+/1-, kAUC=7.5

         |
CCATCT
     C
      
      
PWM threshold: 5.53/11.05, 	hit=36+/13-, pAUC=8.4
--------------------------------------------------------------

GEMout_2 motif #3

KSM top k-mer: AAAGCTACTA/TAGTAGCTTT, total 1 k-mers
KSM threshold: 2.46, 		hit=15+/0-, kAUC=7.2

 |
AAAGCAACTA
     T  A 
          
          
PWM threshold: 7.86/15.32, 	hit=109+/53-, pAUC=11.5
--------------------------------------------------------------



Finished! GEM analysis results are printed to:
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDF/vescaAlpina/GEMout/GEMout.GEM_events.txt
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDF/vescaAlpina/GEMout/GEMout.results.htm
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDF/vescaAlpina/GEMout/GEMout_outputs (folder with all other files)

----------------------------------

End time: 2021/12/03 16:47:15

Total running time: 15.9m

pigz 2.6
Analysis complete for INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  
Archive:  INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX6_BKDL210038288-1A-6_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/03 17:40:03

Loading data...
    Loading reads from: CS144-MYBvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/FVechrs/ --s 150000000 --expt CS144-MYBvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYB/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 26286361	HitCounts: 3.6501077E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 26286361	HitCounts: 2.9880566E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 132 reads.


	IP: 29880566	Ctrl: 15832687	 IP/Ctrl: 1.89

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 2.01

The genome is segmented into 224014 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/224014	1.3s
1000	/224014	35.3s
10000	/224014	47.3s
100000	/224014	49.3s
224014	/224014	49.3s

Scaling condition , IP/Control = 1.78

---------------------------
Events discovered 
Significant:	155
Insignificant:	77
Filtered:	11

Total 89 homotypic events (within 501bp of other events).

Finish binding event prediction: 51.6s


============================ Round 1 ============================

Running with 10 threads ...

100	/224014	1.2s
1000	/224014	2.2s
10000	/224014	3.2s
100000	/224014	9.2s
200000	/224014	13.2s
224014	/224014	19.2s
---------------------------
Events discovered 
Significant:	178
Insignificant:	94
Filtered:	14

Total 91 homotypic events (within 501bp of other events).

Finish binding event prediction: 21.0s

Loading genome sequences ...
Done, 2.0s
GC content=0.41

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.57. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 272+/272- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=223
k=6+1, gapped kmer=328, base kmer=406
k=6+2, gapped kmer=762, base kmer=583
k=6+3, gapped kmer=1322, base kmer=669
k=6+4, gapped kmer=1838, base kmer=687

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 223

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTGCCG/CGGCAG	84	92	14	-18.8    	0	144.2	6.0	865.2	26
GGGGCT/AGCCCC	65	74	38	-4.4    	11	122.3	4.0	489.0	15
TCGACA/TGTCGA	84	90	5	-25.7    	1	131.8	3.0	395.4	27
GCAGGG/CCCTGC	68	78	46	-3.5    	6	125.8	2.0	251.7	18
GGCTCG/CGAGCC	71	81	31	-7.7    	3	117.3	2.0	234.7	18

Align and cluster 132 k-mers with seed CTGCCG,   	#0
Align and cluster 81 k-mers with seed GGGGCT,   	#1
Align and cluster 162 k-mers with seed TCGACA,   	#2
Align and cluster 108 k-mers with seed GCAGGG,   	#3
Align and cluster 110 k-mers with seed GGCTCG,   	#4

Merging redundant motifs ...
44ms: Remove motif #2.
86ms: Remove motif #3.
166ms: Remove motif #4.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= GGGGCT 	1.30, 165, kAUC=53.8	PWM= GGGGGG	6.25, 125, pAUC=0.7
#1	k=6	KSM= GCAGGG 	1.63, 135, kAUC=51.8	PWM= GGCAGG	7.34, 116, pAUC=6.7
#2	k=6	KSM= GGCTCG 	1.78, 124, kAUC=43.0	PWM= GGCTCG	8.00, 90, pAUC=11.5
#3	k=6	KSM= TCGACA 	5.66, 85, kAUC=35.1	PWM= TCGGCA	7.35, 129, pAUC=28.8
#4	k=6	KSM= CTGCCG 	3.26, 62, kAUC=33.4	PWM= CTGCCG	7.87, 112, pAUC=26.2

After merging:
#1	k=6	KSM= GCAGGG 	1.35, 154, kAUC=58.7	PWM= GGCAGG	7.57, 130, pAUC=6.7
#0	k=6	KSM= GGGGCT 	1.30, 165, kAUC=53.8	PWM= GGGGGG	6.25, 125, pAUC=0.7



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=516
k=7+1, gapped kmer=284, base kmer=397
k=7+2, gapped kmer=506, base kmer=530
k=7+3, gapped kmer=716, base kmer=619
k=7+4, gapped kmer=1039, base kmer=675

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 516

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GCAGGGC/GCCCTGC	66	75	22	-9.5    	1	116.8	7.0	817.8	15
TGCCGAG/CTCGGCA	64	72	7	-17.4    	13	106.0	4.0	423.9	23
TGCTACT/AGTAGCA	76	72	12	-13.7    	14	93.3	4.0	373.4	21
CGGCAGG/CCTGCCG	64	73	7	-17.7    	7	103.2	2.0	206.5	25
GGCTCGG/CCGAGCC	63	72	19	-10.0    	12	98.6	2.0	197.2	18
GGGCTCG/CGAGCCC	67	76	21	-10.2

Align and cluster 115 k-mers with seed GCAGGGC,   	#0
Align and cluster 151 k-mers with seed TGCCGAG,   	#1
Align and cluster 207 k-mers with seed TGCTACT,   	#2
Align and cluster 154 k-mers with seed CGGCAGG,   	#3
Align and cluster 124 k-mers with seed GGCTCGG,   	#4

Merging redundant motifs ...
82ms: Remove motif #2.
226ms: Remove motif #3.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= CGGCAGG 	2.29, 100, kAUC=52.6	PWM= CGGCAGG	9.03, 95, pAUC=24.8
#1	k=7	KSM= TGCTACT 	1.96, 76, kAUC=49.6	PWM= TGCTACT	8.46, 92, pAUC=23.3
#2	k=7	KSM= TGCCGAG 	1.88, 98, kAUC=49.5	PWM= TGCCGAG	8.91, 123, pAUC=27.8
#3	k=7	KSM= GCAGGGC 	3.11, 51, kAUC=41.3	PWM= GCAGGGC	8.81, 90, pAUC=14.5
#4	k=7	KSM= GGCTCGG 	2.55, 90, kAUC=37.5	PWM= GGCTCGA	8.29, 87, pAUC=21.7

After merging:
#0	k=7	KSM= CGGCAGG 	2.29, 100, kAUC=52.6	PWM= CGGCAGG	9.03, 95, pAUC=24.8
#1	k=7	KSM= TGCTACT 	1.96, 76, kAUC=49.6	PWM= TGCTACT	8.46, 92, pAUC=23.3
#4	k=7	KSM= GGCTCGG 	2.55, 90, kAUC=37.5	PWM= GGCTCGA	8.29, 87, pAUC=21.7



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=623
k=8+1, gapped kmer=257, base kmer=363
k=8+2, gapped kmer=319, base kmer=409
k=8+3, gapped kmer=338, base kmer=456
k=8+4, gapped kmer=548, base kmer=562

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 623

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GCTCGGCA/TGCCGAGC	62	70	4	-19.6    	12	91.7	8.0	733.2	15
CGGCAGGG/CCCTGCCG	64	73	5	-19.6    	2	90.8	5.0	454.1	14
CAGGGCTC/GAGCCCTG	63	71	13	-12.8    	9	87.2	4.0	348.6	10
TGCTACTG/CAGTAGCA	62	58	9	-11.5    	19	73.8	4.0	295.4	17
AGGGGGCT/AGCCCCCT	64	73	10	-15.4    	0	80.1	3.0	240.4	10

Align and cluster 214 k-mers with seed GCTCGGCA,   	#0
Align and cluster 244 k-mers with seed CGGCAGGG,   	#1
Align and cluster 171 k-mers with seed CAGGGCTC,   	#2
Align and cluster 306 k-mers with seed TGCTACTG,   	#3
Align and cluster 153 k-mers with seed AGGGGGCT,   	#4

Merging redundant motifs ...
48ms: Remove motif #2.
106ms: Remove motif #3.
174ms: Remove motif #4.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= CGGCAGGG 	1.10, 138, kAUC=51.3	PWM= CGACAGGG	9.75, 110, pAUC=32.3
#1	k=8	KSM= TGCTACTG 	1.10, 102, kAUC=41.3	PWM= TGCTACTG	9.70, 89, pAUC=25.4
#2	k=8	KSM= GCTCGGCA 	1.96, 88, kAUC=40.9	PWM= GCTCGACA	7.75, 108, pAUC=32.7
#3	k=8	KSM= CAGGGCTC 	7.03, 44, kAUC=30.0	PWM= CAGGGCTC	8.66, 73, pAUC=17.6
#4	k=8	KSM= AGGGGGCT 	1.31, 84, kAUC=26.3	PWM= AGGGGGCT	9.49, 75, pAUC=19.2

After merging:
#0	k=8	KSM= CGGCAGGG 	1.10, 140, kAUC=51.5	PWM= CGACAGGG	8.48, 125, pAUC=36.6
#1	k=8	KSM= TGCTACTG 	1.10, 102, kAUC=41.3	PWM= TGCTACTG	9.70, 89, pAUC=25.4



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=675
k=9+1, gapped kmer=249, base kmer=353
k=9+2, gapped kmer=246, base kmer=347
k=9+3, gapped kmer=202, base kmer=328
k=9+4, gapped kmer=293, base kmer=391

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 675

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GCAGGGCTC/GAGCCCTGC	61	69	8	-15.6    	12	104.4	8.0	835.3	26
TCGGCAGGG/CCCTGCCGA	64	73	4	-20.6    	3	97.4	6.0	584.5	33
GGCTCGGCA/TGCCGAGCC	62	70	2	-22.0    	11	96.1	5.0	480.6	35
TGCTACTGC/GCAGTAGCA	47	45	3	-12.2    	26	68.7	4.0	275.0	48
GTGTTGTTG/CAACAACAC	36	34	5	-7.2    	35	63.9	3.0	191.6	63
AGGGGGCTC/GAGCCCCCT	64	73	10	-15.4

Align and cluster 122 k-mers with seed GCAGGGCTC,   	#0
Align and cluster 168 k-mers with seed TCGGCAGGG,   	#1
Align and cluster 181 k-mers with seed GGCTCGGCA,   	#2
Align and cluster 251 k-mers with seed TGCTACTGC,   	#3
Align and cluster 259 k-mers with seed GTGTTGTTG,   	#4

Merging redundant motifs ...
46ms: Remove motif #3.
150ms: Remove motif #4.

------------------------- k = 9 ----------------------------

#0	k=9	KSM= TCGGCAGGG 	1.00, 115, kAUC=39.1	PWM= TCGACAGGG	10.26, 98, pAUC=33.4
#1	k=9	KSM= TGCTACTGC 	1.99, 62, kAUC=35.3	PWM= TGCTACTGC	10.89, 87, pAUC=29.9
#2	k=9	KSM= GTGTTGTTG 	1.07, 86, kAUC=34.1	PWM= GTGTTGTTG	9.99, 65, pAUC=18.0
#3	k=9	KSM= GGCTCGGCA 	5.13, 68, kAUC=27.7	PWM= GGCTCGACA	10.90, 77, pAUC=29.3
#4	k=9	KSM= GCAGGGCTC 	4.30, 72, kAUC=25.3	PWM= GCAGGGCTC	13.45, 71, pAUC=19.7

After merging:
#0	k=9	KSM= TCGGCAGGG 	1.00, 115, kAUC=39.1	PWM= TCGACAGGG	10.26, 98, pAUC=33.4
#1	k=9	KSM= TGCTACTGC 	1.99, 62, kAUC=35.3	PWM= TGCTACTGC	10.89, 87, pAUC=29.9
#2	k=9	KSM= GTGTTGTTG 	1.07, 86, kAUC=34.1	PWM= GTGTTGTTG	9.99, 65, pAUC=18.0



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=687
k=10+1, gapped kmer=249, base kmer=350
k=10+2, gapped kmer=211, base kmer=309
k=10+3, gapped kmer=135, base kmer=247
k=10+4, gapped kmer=177, base kmer=284

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 687

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GGCTCGGCAG/CTGCCGAGCC	62	70	2	-22.0    	7	90.1	9.0	811.0	28
TCGGCAGGGC/GCCCTGCCGA	62	70	2	-22.0    	9	88.5	5.0	442.7	27
GTGCTGCTAC/GTAGCAGCAC	37	35	2	-10.1    	29	55.5	5.0	277.5	33
AGGGGGCTCG/CGAGCCCCCT	64	73	2	-23.1    	0	77.8	3.0	233.5	17
TGTTGTTGTG/CACAACAACA	36	34	3	-8.7    	31	49.5	4.0	197.9	43

Align and cluster 257 k-mers with seed GGCTCGGCAG,   	#0
Align and cluster 226 k-mers with seed TCGGCAGGGC,   	#1
Align and cluster 291 k-mers with seed GTGCTGCTAC,   	#2
Align and cluster 161 k-mers with seed AGGGGGCTCG,   	#3
Align and cluster 338 k-mers with seed TGTTGTTGTG,   	#4

Merging redundant motifs ...
56ms: Remove motif #1.
59ms: PWM AGGGGGCTCG has been removed.

------------------------- k = 10 ----------------------------

#0	k=10	KSM= TCGGCAGGGC 	1.00, 111, kAUC=37.7	PWM= TCGGCAGGGG	12.90, 90, pAUC=30.1
#1	k=10	KSM= GGCTCGGCAG 	1.09, 107, kAUC=37.7	PWM= GGCTCGGCAG	10.99, 88, pAUC=28.2
#2	k=10	KSM= GTGCTGCTAC 	1.29, 72, kAUC=34.6	PWM= GTGCTGCTAC	11.53, 71, pAUC=26.4
#3	k=10	KSM= TGTTGTTGTG 	1.21, 76, kAUC=32.9	PWM= TGTTGTTGTG	10.53, 50, pAUC=16.9
#4	k=10	KSM= AGGGGGCTCG 	1.42, 68, kAUC=26.1	PWM= AGGGGGCTCG	21.56, 73, pAUC=22.7

After merging:
#0	k=10	KSM= TCGGCAGGGC 	1.16, 112, kAUC=38.8	PWM= TCGGCAGGGG	12.77, 90, pAUC=29.7
#2	k=10	KSM= GTGCTGCTAC 	1.29, 72, kAUC=34.6	PWM= GTGCTGCTAC	11.53, 71, pAUC=26.4
#3	k=10	KSM= TGTTGTTGTG 	1.21, 76, kAUC=32.9	PWM= TGTTGTTGTG	10.53, 50, pAUC=16.9



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=692
k=11+1, gapped kmer=237, base kmer=339
k=11+2, gapped kmer=186, base kmer=298
k=11+3, gapped kmer=103, base kmer=196
k=11+4, gapped kmer=113, base kmer=198

------------------------- k = 11 ----------------------------

Total number of k-mers: n = 692

Top 5 cluster center k-mers:
# k-mer/r.c.           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GGCTCGGCAGG/CCTGCCGAGCC	62	70	2	-22.0    	3	82.8	10.0	828.3	26
GTGCTGCTACT/AGTAGCAGCAC	36	34	1	-11.0    	30	50.8	6.0	305.0	25
AGGGGGCTCGG/CCGAGCCCCCT	62	71	2	-22.4    	1	75.8	4.0	303.3	17
TGTTGTTGTGT/ACACAACAACA	34	32	0	-11.0    	33	45.6	6.0	273.6	33
CGGCAGGGCTC/GAGCCCTGCCG	61	69	2	-21.7    	5	72.4	2.0	144.8	17
CAGGGGGCTCG/CGAGCCCCCTG	64	73	2	-23.1

Align and cluster 225 k-mers with seed GGCTCGGCAGG,   	#0
Align and cluster 224 k-mers with seed GTGCTGCTACT,   	#1
Align and cluster 122 k-mers with seed AGGGGGCTCGG,   	#2
Align and cluster 243 k-mers with seed TGTTGTTGTGT,   	#3
Align and cluster 157 k-mers with seed CGGCAGGGCTC,   	#4

Merging redundant motifs ...
53ms: Remove motif #3.
56ms: PWM CGGCAGGGCTC has been removed.

------------------------- k = 11 ----------------------------

#0	k=11	KSM= TGTTGTTGTGT 	0.94, 80, kAUC=31.3	PWM= TGTTGTTGTGT	11.22, 48, pAUC=18.0
#1	k=11	KSM= GTGCTGCTACT 	1.92, 58, kAUC=31.1	PWM= GTGCTGCTACT	13.70, 58, pAUC=27.8
#2	k=11	KSM= GGCTCGGCAGG 	7.08, 76, kAUC=26.8	PWM= GGCTCGGCAGG	15.40, 76, pAUC=23.6
#3	k=11	KSM= AGGGGGCTCGG 	1.21, 74, kAUC=25.8	PWM= AGGGGGCTCGG	12.09, 73, pAUC=22.5
#4	k=11	KSM= CGGCAGGGCTC 	1.31, 76, kAUC=25.4	PWM= CGGCAGGGCTC	13.56, 71, pAUC=22.3

After merging:
#0	k=11	KSM= TGTTGTTGTGT 	0.94, 80, kAUC=31.3	PWM= TGTTGTTGTGT	11.22, 48, pAUC=18.0
#1	k=11	KSM= GTGCTGCTACT 	1.92, 58, kAUC=31.1	PWM= GTGCTGCTACT	13.70, 58, pAUC=27.8
#2	k=11	KSM= GGCTCGGCAGG 	7.08, 76, kAUC=26.8	PWM= GGCTCGGCAGG	15.40, 76, pAUC=23.6



----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=674
k=12+1, gapped kmer=223, base kmer=319
k=12+2, gapped kmer=181, base kmer=288
k=12+3, gapped kmer=77, base kmer=161
k=12+4, gapped kmer=66, base kmer=143

------------------------- k = 12 ----------------------------

Total number of k-mers: n = 674

Top 5 cluster center k-mers:
# k-mer/r.c.             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GGGGGCTCGGCA/TGCCGAGCCCCC	60	68	2	-21.3    	4	84.9	11.0	933.9	32
TGTGCTGCTACT/AGTAGCAGCACA	34	32	1	-10.4    	32	50.9	7.0	356.4	41
CGACAGGGGGCT/AGCCCCCTGTCG	44	50	0	-17.2    	25	72.1	4.0	288.4	22
TGTTGTGTGCTG/CAGCACACAACA	24	23	5	-4.2    	40	50.8	4.0	203.0	32
TGCTACTGCGTG/CACGCAGTAGCA	23	22	1	-7.1    	43	43.7	4.0	175.0	38
AGGGGGCTCGGC/GCCGAGCCCCCT	62	71	1	-24.0

Align and cluster 202 k-mers with seed GGGGGCTCGGCA,   	#0
Align and cluster 230 k-mers with seed TGTGCTGCTACT,   	#1
Align and cluster 169 k-mers with seed CGACAGGGGGCT,   	#2
Align and cluster 227 k-mers with seed TGTTGTGTGCTG,   	#3
Align and cluster 256 k-mers with seed TGCTACTGCGTG,   	#4

Merging redundant motifs ...
64ms: Remove motif #1.
105ms: Remove motif #3.

------------------------- k = 12 ----------------------------

#0	k=12	KSM= TGCTACTGCGTG 	1.09, 82, kAUC=35.2	PWM= TGCTACTGCGGG	12.13, 87, pAUC=32.6
#1	k=12	KSM= TGTGCTGCTACT 	1.21, 70, kAUC=33.8	PWM= TGTGCTGCTACT	14.10, 70, pAUC=30.1
#2	k=12	KSM= GGGGGCTCGGCA 	6.10, 69, kAUC=26.0	PWM= GGGGGCTCGGCA	17.08, 74, pAUC=23.3
#3	k=12	KSM= CGACAGGGGGCT 	5.61, 73, kAUC=25.7	PWM= CGGCAGGGGGCT	23.01, 73, pAUC=23.5
#4	k=12	KSM= TGTTGTGTGCTG 	1.10, 52, kAUC=21.9	PWM= TGTTGTGTGCTG	15.29, 49, pAUC=15.3

After merging:
#0	k=12	KSM= TGCTACTGCGTG 	1.10, 82, kAUC=35.5	PWM= TGCTACTGCGTG	11.91, 88, pAUC=32.2
#2	k=12	KSM= GGGGGCTCGGCA 	1.00, 76, kAUC=26.2	PWM= GGGGGCTCGGCA	17.08, 74, pAUC=23.2
#4	k=12	KSM= TGTTGTGTGCTG 	1.10, 52, kAUC=21.9	PWM= TGTTGTGTGCTG	15.29, 49, pAUC=15.3



----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=658
k=13+1, gapped kmer=216, base kmer=314
k=13+2, gapped kmer=173, base kmer=284
k=13+3, gapped kmer=62, base kmer=137
k=13+4, gapped kmer=53, base kmer=130

------------------------- k = 13 ----------------------------

Total number of k-mers: n = 658

Top 5 cluster center k-mers:
# k-mer/r.c.               	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GGGGGCTCGGCAG/CTGCCGAGCCCCC	59	67	2	-21.0    	4	77.7	12.0	931.8	30
GCTCGGCAGGGCT/AGCCCTGCCGAGC	59	66	1	-22.2    	5	75.8	8.0	606.2	30
TGTGCTGCTACTG/CAGTAGCAGCACA	28	26	1	-8.4    	35	44.7	8.0	357.9	32
TCGACAGGGGGCT/AGCCCCCTGTCGA	44	50	0	-17.2    	27	68.9	5.0	344.6	22
TGTTGTGTGCTGC/GCAGCACACAACA	23	22	1	-7.1    	40	44.0	4.0	176.0	19
CAGGGGGCTCGGC/GCCGAGCCCCCTG	62	71	1	-24.0

Align and cluster 181 k-mers with seed GGGGGCTCGGCAG,   	#0
Align and cluster 164 k-mers with seed GCTCGGCAGGGCT,   	#1
Align and cluster 193 k-mers with seed TGTGCTGCTACTG,   	#2
Align and cluster 143 k-mers with seed TCGACAGGGGGCT,   	#3
Align and cluster 144 k-mers with seed TGTTGTGTGCTGC,   	#4

Merging redundant motifs ...
14ms: PWM TGTTGTGTGCTGC has been removed.
18ms: PWM TCGGCAGGGGGCT has been removed.
22ms: PWM GGGGGCTCGGCAG has been removed.

------------------------- k = 13 ----------------------------

#0	k=13	KSM= TGTGCTGCTACTG 	1.09, 73, kAUC=33.1	PWM= TGTGCTGCTACTG	11.30, 79, pAUC=31.2
#1	k=13	KSM= GCTCGGCAGGGCT 	1.31, 76, kAUC=27.6	PWM= GCTCGGCAGGGCT	18.47, 73, pAUC=23.6
#2	k=13	KSM= TCGACAGGGGGCT 	9.31, 73, kAUC=26.8	PWM= TCGGCAGGGGGCT	24.87, 73, pAUC=23.6
#3	k=13	KSM= GGGGGCTCGGCAG 	6.95, 73, kAUC=26.1	PWM= GGGGGCTCGGCAG	19.06, 73, pAUC=22.7
#4	k=13	KSM= TGTTGTGTGCTGC 	1.10, 51, kAUC=23.5	PWM= TGTTGTGTGCTGC	14.91, 48, pAUC=17.2

After merging:
#0	k=13	KSM= TGTGCTGCTACTG 	1.09, 73, kAUC=33.1	PWM= TGTGCTGCTACTG	11.30, 79, pAUC=31.2
#1	k=13	KSM= GCTCGGCAGGGCT 	1.31, 76, kAUC=27.6	PWM= GCTCGGCAGGGCT	18.47, 73, pAUC=23.6



----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=649
k=14+1, gapped kmer=206, base kmer=302
k=14+2, gapped kmer=165, base kmer=274
k=14+3, gapped kmer=58, base kmer=135
k=14+4, gapped kmer=40, base kmer=98

------------------------- k = 14 ----------------------------

Total number of k-mers: n = 649

Top 5 cluster center k-mers:
# k-mer/r.c.                 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GGCTCGGCAGGGCT/AGCCCTGCCGAGCC	58	65	0	-22.6    	5	72.1	13.0	937.4	29
CGGCAGGGGGCTCG/CGAGCCCCCTGCCG	54	61	0	-21.1    	8	66.7	6.0	400.4	18
GTGTGCTGCTACTG/CAGTAGCAGCACAC	24	23	0	-8.1    	37	37.8	8.0	302.1	28
GTTGTTGTGTGCTG/CAGCACACAACAAC	21	20	0	-7.1    	40	31.3	6.0	187.8	27
AGGGGGCTCGGCAG/CTGCCGAGCCCCCT	58	65	0	-22.6    	3	70.3	2.0	140.6	25
CAGGGGGCTCGGCA/TGCCGAGCCCCCTG	59	67	0	-23.3

Align and cluster 207 k-mers with seed GGCTCGGCAGGGCT,   	#0
Align and cluster 164 k-mers with seed CGGCAGGGGGCTCG,   	#1
Align and cluster 207 k-mers with seed GTGTGCTGCTACTG,   	#2
Align and cluster 209 k-mers with seed GTTGTTGTGTGCTG,   	#3
Align and cluster 213 k-mers with seed AGGGGGCTCGGCAG,   	#4

Merging redundant motifs ...
14ms: PWM CGACAGGGGGCTCG has been removed.
18ms: PWM AGGGGGCTCGGCAG has been removed.
61ms: Remove motif #4.

------------------------- k = 14 ----------------------------

#0	k=14	KSM= GGCTCGGCAGGGCT 	1.10, 76, kAUC=27.6	PWM= GGCTCGGCAGGGCT	20.05, 73, pAUC=24.0
#1	k=14	KSM= CGGCAGGGGGCTCG 	3.14, 73, kAUC=26.6	PWM= CGACAGGGGGCTCG	28.93, 73, pAUC=23.6
#2	k=14	KSM= AGGGGGCTCGGCAG 	2.60, 73, kAUC=26.4	PWM= AGGGGGCTCGGCAG	20.77, 73, pAUC=23.7
#3	k=14	KSM= GTTGTTGTGTGCTG 	1.00, 51, kAUC=22.8	PWM= GTTGTTGTGTGCTG	18.39, 48, pAUC=19.2
#4	k=14	KSM= GTGTGCTGCTACTG 	1.00, 48, kAUC=22.4	PWM= GTGTGCTGCTACTG	16.68, 52, pAUC=22.3

After merging:
#0	k=14	KSM= GGCTCGGCAGGGCT 	1.10, 76, kAUC=27.6	PWM= GGCTCGGCAGGGCT	20.05, 73, pAUC=24.0
#3	k=14	KSM= GTTGTTGTGTGCTG 	1.00, 51, kAUC=22.8	PWM= GTTGTTGTGTGCTG	18.39, 48, pAUC=19.2



----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=627
k=15+1, gapped kmer=192, base kmer=288
k=15+2, gapped kmer=156, base kmer=259
k=15+3, gapped kmer=49, base kmer=127
k=15+4, gapped kmer=34, base kmer=81

------------------------- k = 15 ----------------------------

Total number of k-mers: n = 627

Top 5 cluster center k-mers:
# k-mer/r.c.                   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AGGGGGCTCGGCAGG/CCTGCCGAGCCCCCT	58	65	0	-22.6    	0	72.6	13.0	943.3	42
GCTCGGCAGGGCTCG/CGAGCCCTGCCGAGC	55	62	0	-21.5    	6	72.2	8.0	577.3	37
TGTGTGCTGCTACTG/CAGTAGCAGCACACA	22	21	0	-7.5    	37	35.5	8.0	284.0	33
TGTTGTTGTGTGCTG/CAGCACACAACAACA	20	19	0	-6.8    	42	31.1	7.0	217.9	38
TCGACACGGCAGTAG/CTACTGCCGTGTCGA	9	9	0	-3.7    	75	26.7	5.0	133.3	25

Align and cluster 201 k-mers with seed AGGGGGCTCGGCAGG,   	#0
Align and cluster 173 k-mers with seed GCTCGGCAGGGCTCG,   	#1
Align and cluster 158 k-mers with seed TGTGTGCTGCTACTG,   	#2
Align and cluster 175 k-mers with seed TGTTGTTGTGTGCTG,   	#3
Align and cluster 136 k-mers with seed TCGACACGGCAGTAG,   	#4

Merging redundant motifs ...
54ms: Remove motif #1.
93ms: Remove motif #3.

------------------------- k = 15 ----------------------------

#0	k=15	KSM= GCTCGGCAGGGCTCG 	1.10, 73, kAUC=27.4	PWM= GCTCGGCAGGGCTCG	21.84, 71, pAUC=23.7
#1	k=15	KSM= AGGGGGCTCGGCAGG 	4.87, 73, kAUC=27.1	PWM= AGGGGGCTCGGCAGG	23.03, 73, pAUC=23.9
#2	k=15	KSM= TGTTGTTGTGTGCTG 	1.00, 51, kAUC=23.9	PWM= TGTTGTTGTGTGCTG	17.83, 52, pAUC=19.7
#3	k=15	KSM= TGTGTGCTGCTACTG 	1.00, 47, kAUC=22.5	PWM= TGTGTGCTGCTACTG	18.29, 52, pAUC=21.0
#4	k=15	KSM= TCGACACGGCAGTAG 	1.00, 28, kAUC=15.1	PWM= TCGACACGGCAGTAG	17.27, 32, pAUC=20.0

After merging:
#0	k=15	KSM= GCTCGGCAGGGCTCG 	1.10, 73, kAUC=27.4	PWM= GCTCGGCAGGGCTCG	21.84, 71, pAUC=23.7
#2	k=15	KSM= TGTTGTTGTGTGCTG 	1.00, 51, kAUC=23.9	PWM= TGTTGTTGTGTGCTG	17.83, 52, pAUC=19.7
#4	k=15	KSM= TCGACACGGCAGTAG 	1.00, 28, kAUC=15.1	PWM= TCGACACGGCAGTAG	17.27, 32, pAUC=20.0



----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=601
k=16+1, gapped kmer=177, base kmer=263
k=16+2, gapped kmer=138, base kmer=238
k=16+3, gapped kmer=46, base kmer=115
k=16+4, gapped kmer=22, base kmer=56

------------------------- k = 16 ----------------------------

Total number of k-mers: n = 601

Top 5 cluster center k-mers:
# k-mer/r.c.                     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GGGCTCGGCAGGGCTC/GAGCCCTGCCGAGCCC	57	64	0	-22.2    	1	69.7	14.0	976.4	41
CTCGGCAGGGGGCTCG/CGAGCCCCCTGCCGAG	52	59	0	-20.4    	6	65.6	7.0	459.4	24
TGTGCTGCTACTGCGT/ACGCAGTAGCAGCACA	16	15	0	-5.5    	48	29.5	9.0	265.5	34
TGTTGTTGTGTGCTGC/GCAGCACACAACAACA	19	18	0	-6.5    	40	25.8	7.0	180.6	27
CAGGGGGCTCGGCAGG/CCTGCCGAGCCCCCTG	58	65	0	-22.6    	0	68.9	2.0	137.8	35

Align and cluster 202 k-mers with seed GGGCTCGGCAGGGCTC,   	#0
Align and cluster 190 k-mers with seed CTCGGCAGGGGGCTCG,   	#1
Align and cluster 189 k-mers with seed TGTGCTGCTACTGCGT,   	#2
Align and cluster 178 k-mers with seed TGTTGTTGTGTGCTGC,   	#3
Align and cluster 211 k-mers with seed CAGGGGGCTCGGCAGG,   	#4

Merging redundant motifs ...
64ms: Remove motif #4.
68ms: PWM CTCGGCAGGGGGCTCG has been removed.
72ms: PWM GGGCTCGGCAGGGCTC has been removed.

------------------------- k = 16 ----------------------------

#0	k=16	KSM= TGTGCTGCTACTGCGT 	1.00, 75, kAUC=32.7	PWM= TGTGCTGCTACTGCGT	18.64, 80, pAUC=32.3
#1	k=16	KSM= CAGGGGGCTCGGCAGG 	1.31, 73, kAUC=27.4	PWM= CAGGGGGCTCGGCAGG	25.29, 73, pAUC=23.9
#2	k=16	KSM= CTCGGCAGGGGGCTCG 	1.10, 73, kAUC=27.4	PWM= CTCGGCAGGGGGCTCG	31.75, 73, pAUC=23.6
#3	k=16	KSM= GGGCTCGGCAGGGCTC 	2.76, 73, kAUC=27.1	PWM= GGGCTCGGCAGGGCTC	24.52, 73, pAUC=23.5
#4	k=16	KSM= TGTTGTTGTGTGCTGC 	1.00, 50, kAUC=23.5	PWM= TGTTGTTGTGTGCTGC	19.36, 52, pAUC=20.0

After merging:
#0	k=16	KSM= TGTGCTGCTACTGCGT 	1.00, 75, kAUC=32.7	PWM= TGTGCTGCTACTGCGT	18.64, 80, pAUC=32.3
#1	k=16	KSM= CAGGGGGCTCGGCAGG 	1.31, 73, kAUC=27.4	PWM= CAGGGGGCTCGGCAGG	25.29, 73, pAUC=23.9



----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=572
k=17+1, gapped kmer=163, base kmer=247
k=17+2, gapped kmer=123, base kmer=216
k=17+3, gapped kmer=45, base kmer=110
k=17+4, gapped kmer=13, base kmer=34

------------------------- k = 17 ----------------------------

Total number of k-mers: n = 572

Top 5 cluster center k-mers:
# k-mer/r.c.                       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GGGGGCTCGGCAGGGCT/AGCCCTGCCGAGCCCCC	56	63	0	-21.8    	3	69.2	15.0	1037.9	42
CGGCAGGGGGCTCGGCA/TGCCGAGCCCCCTGCCG	47	53	0	-18.2    	7	62.2	6.0	373.2	24
GTGTGCTGCTACTGCGT/ACGCAGTAGCAGCACAC	15	14	0	-5.2    	47	24.5	9.0	220.8	30
TACTGCGTGTGTTGTTG/CAACAACACACGCAGTA	7	7	0	-3.0    	90	13.5	7.0	94.4	21
GTTGTTGTGTGCTGCTA/TAGCAGCACACAACAAC	17	16	0	-5.9    	41	22.3	4.0	89.1	20
AGGGGGCTCGGCAGGGC/GCCCTGCCGAGCCCCCT	56	63	0	-21.8

Align and cluster 191 k-mers with seed GGGGGCTCGGCAGGGCT,   	#0
Align and cluster 169 k-mers with seed CGGCAGGGGGCTCGGCA,   	#1
Align and cluster 145 k-mers with seed GTGTGCTGCTACTGCGT,   	#2
Align and cluster 76 k-mers with seed TACTGCGTGTGTTGTTG,   	#3
Align and cluster 102 k-mers with seed GTTGTTGTGTGCTGCTA,   	#4

Merging redundant motifs ...
17ms: PWM GGGGGCTCGGCAGGGCT has been removed.
63ms: Remove motif #3.

------------------------- k = 17 ----------------------------

#0	k=17	KSM= CGGCAGGGGGCTCGGCA 	1.10, 73, kAUC=27.4	PWM= CGGCAGGGGGCTCGGCA	27.23, 73, pAUC=24.4
#1	k=17	KSM= GGGGGCTCGGCAGGGCT 	2.53, 73, kAUC=27.4	PWM= GGGGGCTCGGCAGGGCT	24.85, 70, pAUC=23.3
#2	k=17	KSM= GTTGTTGTGTGCTGCTA 	1.00, 45, kAUC=21.8	PWM= GTTGTTGTGTGCTGCTA	20.40, 50, pAUC=20.5
#3	k=17	KSM= GTGTGCTGCTACTGCGT 	1.00, 45, kAUC=21.8	PWM= GTGTGCTGCTACTGCGT	18.10, 53, pAUC=25.5
#4	k=17	KSM= TACTGCGTGTGTTGTTG 	1.00, 38, kAUC=19.3	PWM= TACTGCGTGTGTTGTTG	19.68, 36, pAUC=16.5

After merging:
#0	k=17	KSM= CGGCAGGGGGCTCGGCA 	1.10, 73, kAUC=27.4	PWM= CGGCAGGGGGCTCGGCA	27.23, 73, pAUC=24.4
#2	k=17	KSM= GTTGTTGTGTGCTGCTA 	1.00, 45, kAUC=21.8	PWM= GTTGTTGTGTGCTGCTA	20.40, 50, pAUC=20.5
#4	k=17	KSM= TACTGCGTGTGTTGTTG 	1.00, 38, kAUC=19.3	PWM= TACTGCGTGTGTTGTTG	19.68, 36, pAUC=16.5



----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=546
k=18+1, gapped kmer=153, base kmer=230
k=18+2, gapped kmer=108, base kmer=191
k=18+3, gapped kmer=41, base kmer=104
k=18+4, gapped kmer=10, base kmer=25

------------------------- k = 18 ----------------------------

Total number of k-mers: n = 546

Top 5 cluster center k-mers:
# k-mer/r.c.                         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AGGGGGCTCGGCAGGGCT/AGCCCTGCCGAGCCCCCT	56	63	0	-21.8    	0	68.7	16.0	1099.8	50
CGAGCCCCCTGCCGAGCC/GGCTCGGCAGGGGGCTCG	47	53	0	-18.2    	4	62.2	7.0	435.4	36
GTGTGCTGCTACTGCGTG/CACGCAGTAGCAGCACAC	14	13	0	-4.9    	46	23.4	9.0	210.3	32
CCTGCCGAGCCCTGCCGA/TCGGCAGGGCTCGGCAGG	44	49	0	-16.8    	13	59.0	2.0	118.0	28
TGTCGACCGGAGACGTGC/GCACGTCTCCGGTCGACA	6	5	0	-2.4    	205	16.6	5.0	82.9	18

Align and cluster 209 k-mers with seed AGGGGGCTCGGCAGGGCT,   	#0
Align and cluster 208 k-mers with seed CGAGCCCCCTGCCGAGCC,   	#1
Align and cluster 148 k-mers with seed GTGTGCTGCTACTGCGTG,   	#2
Align and cluster 194 k-mers with seed CCTGCCGAGCCCTGCCGA,   	#3
Align and cluster 112 k-mers with seed TGTCGACCGGAGACGTGC,   	#4

Merging redundant motifs ...
16ms: PWM TCGGCAGGGCTCGGCAGG has been removed.
67ms: Remove motif #2.

------------------------- k = 18 ----------------------------

#0	k=18	KSM= AGGGGGCTCGGCAGGGCT 	1.21, 73, kAUC=27.4	PWM= AGGGGGCTCGGCAGGGCT	22.15, 71, pAUC=23.4
#1	k=18	KSM= CCTGCCGAGCCCTGCCGA 	1.10, 73, kAUC=27.4	PWM= CCTGCCGAGCCCTGCCGA	27.50, 71, pAUC=23.2
#2	k=18	KSM= CGAGCCCCCTGCCGAGCC 	1.21, 73, kAUC=27.1	PWM= CGAGCCCCCTGCCGAGCC	36.73, 73, pAUC=23.5
#3	k=18	KSM= GTGTGCTGCTACTGCGTG 	1.00, 42, kAUC=20.7	PWM= GTGTGCTGCTACTGCGTG	21.57, 52, pAUC=22.8
#4	k=18	KSM= TGTCGACCGGAGACGTGC 	1.00, 22, kAUC=13.0	PWM= TGTCGACCGGAGACGTGC	18.58, 31, pAUC=20.4

After merging:
#0	k=18	KSM= AGGGGGCTCGGCAGGGCT 	1.21, 73, kAUC=27.4	PWM= AGGGGGCTCGGCAGGGCT	22.15, 71, pAUC=23.4
#3	k=18	KSM= GTGTGCTGCTACTGCGTG 	1.00, 42, kAUC=20.7	PWM= GTGTGCTGCTACTGCGTG	21.57, 52, pAUC=22.8
#4	k=18	KSM= TGTCGACCGGAGACGTGC 	1.00, 22, kAUC=13.0	PWM= TGTCGACCGGAGACGTGC	18.58, 31, pAUC=20.4



----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=519
k=19+1, gapped kmer=145, base kmer=215
k=19+2, gapped kmer=94, base kmer=169
k=19+3, gapped kmer=34, base kmer=87
k=19+4, gapped kmer=11, base kmer=22

------------------------- k = 19 ----------------------------

Total number of k-mers: n = 519

Top 5 cluster center k-mers:
# k-mer/r.c.                           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CAGGGGGCTCGGCAGGGCT/AGCCCTGCCGAGCCCCCTG	56	63	0	-21.8    	0	66.9	16.0	1070.1	43
CGAGCCCCCTGCCGAGCCC/GGGCTCGGCAGGGGGCTCG	47	53	0	-18.2    	3	62.2	7.0	435.4	39
TTGTTGTGTGCTGCTACTG/CAGTAGCAGCACACAACAA	15	14	0	-5.2    	45	20.1	10.0	201.5	21
GTGTGTTGTTGTGTGCTGC/GCAGCACACAACAACACAC	12	11	0	-4.3    	53	18.8	8.0	150.1	29
CTCGGCAGGGCTCGGCAGG/CCTGCCGAGCCCTGCCGAG	42	47	0	-16.1    	12	57.8	2.0	115.5	29

Align and cluster 219 k-mers with seed CAGGGGGCTCGGCAGGGCT,   	#0
Align and cluster 207 k-mers with seed CGAGCCCCCTGCCGAGCCC,   	#1
Align and cluster 74 k-mers with seed TTGTTGTGTGCTGCTACTG,   	#2
Align and cluster 94 k-mers with seed GTGTGTTGTTGTGTGCTGC,   	#3
Align and cluster 203 k-mers with seed CTCGGCAGGGCTCGGCAGG,   	#4

Merging redundant motifs ...
93ms: Remove motif #1.
179ms: Remove motif #2.
226ms: Remove motif #4.

------------------------- k = 19 ----------------------------

#0	k=19	KSM= CTCGGCAGGGCTCGGCAGG 	1.10, 73, kAUC=27.4	PWM= CTCGGCAGGGCTCGGCAGG	30.25, 70, pAUC=23.8
#1	k=19	KSM= CGAGCCCCCTGCCGAGCCC 	1.10, 73, kAUC=27.4	PWM= CGAGCCCCCTGCCGAGCCC	38.99, 73, pAUC=23.5
#2	k=19	KSM= CAGGGGGCTCGGCAGGGCT 	1.00, 73, kAUC=27.4	PWM= CAGGGGGCTCGGCAGGGCT	25.83, 71, pAUC=23.5
#3	k=19	KSM= GTGTGTTGTTGTGTGCTGC 	1.00, 41, kAUC=20.4	PWM= GTGTGTTGTTGTGTGCTGC	20.79, 41, pAUC=18.2
#4	k=19	KSM= TTGTTGTGTGCTGCTACTG 	1.00, 37, kAUC=19.0	PWM= TTGTTGTGTGCTGCTACTG	23.43, 51, pAUC=20.8

After merging:
#0	k=19	KSM= CTCGGCAGGGCTCGGCAGG 	1.10, 73, kAUC=27.4	PWM= CTCGGCAGGGCTCGGCAGG	30.25, 70, pAUC=23.8
#3	k=19	KSM= GTGTGTTGTTGTGTGCTGC 	1.00, 41, kAUC=20.4	PWM= GTGTGTTGTTGTGTGCTGC	20.79, 41, pAUC=18.2



----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=491
k=20+1, gapped kmer=135, base kmer=201
k=20+2, gapped kmer=84, base kmer=161
k=20+3, gapped kmer=35, base kmer=81
k=20+4, gapped kmer=10, base kmer=21

------------------------- k = 20 ----------------------------

Total number of k-mers: n = 491

Top 5 cluster center k-mers:
# k-mer/r.c.                             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CAGGGGGCTCGGCAGGGCTC/GAGCCCTGCCGAGCCCCCTG	55	62	0	-21.5    	0	66.3	17.0	1127.9	46
CTCGGCAGGGGGCTCGGCAG/CTGCCGAGCCCCCTGCCGAG	44	49	0	-16.8    	3	59.8	8.0	478.5	35
TTGTTGTGTGCTGCTACTGC/GCAGTAGCAGCACACAACAA	12	11	0	-4.3    	53	17.2	10.0	172.3	17
TCGACACGGCAGTAGCACGT/ACGTGCTACTGCCGTGTCGA	7	7	0	-3.0    	75	12.1	8.0	97.0	22
TGTGATGTTGGTGTGTGTTG/CAACACACACCAACATCACA	5	5	0	-2.4    	180	11.4	6.0	68.4	20

Align and cluster 244 k-mers with seed CAGGGGGCTCGGCAGGGCTC,   	#0
Align and cluster 217 k-mers with seed CTCGGCAGGGGGCTCGGCAG,   	#1
Align and cluster 74 k-mers with seed TTGTTGTGTGCTGCTACTGC,   	#2
Align and cluster 88 k-mers with seed TCGACACGGCAGTAGCACGT,   	#3
Align and cluster 97 k-mers with seed TGTGATGTTGGTGTGTGTTG,   	#4

Merging redundant motifs ...
17ms: PWM CAGGGGGCTCGGCAGGGCTC has been removed.
60ms: Remove motif #3.

------------------------- k = 20 ----------------------------

#0	k=20	KSM= CTCGGCAGGGGGCTCGGCAG 	1.42, 73, kAUC=27.4	PWM= CTCGGCAGGGGGCTCGGCAG	28.70, 73, pAUC=24.0
#1	k=20	KSM= CAGGGGGCTCGGCAGGGCTC 	1.00, 73, kAUC=27.4	PWM= CAGGGGGCTCGGCAGGGCTC	28.05, 71, pAUC=23.4
#2	k=20	KSM= TGTGATGTTGGTGTGTGTTG 	1.00, 39, kAUC=19.7	PWM= TGTGTTGTTGTTGTGTGTTG	18.29, 50, pAUC=19.3
#3	k=20	KSM= TTGTTGTGTGCTGCTACTGC 	1.00, 35, kAUC=17.9	PWM= TTGTTGTGTGCTGCTACTGC	25.23, 51, pAUC=21.0
#4	k=20	KSM= TCGACACGGCAGTAGCACGT 	1.00, 22, kAUC=13.0	PWM= TCGACACGGCAGTAGCACGT	25.15, 32, pAUC=19.6

After merging:
#0	k=20	KSM= CTCGGCAGGGGGCTCGGCAG 	1.42, 73, kAUC=27.4	PWM= CTCGGCAGGGGGCTCGGCAG	28.70, 73, pAUC=24.0
#2	k=20	KSM= TGTGATGTTGGTGTGTGTTG 	1.00, 39, kAUC=19.7	PWM= TGTGTTGTTGTTGTGTGTTG	18.29, 50, pAUC=19.3
#4	k=20	KSM= TCGACACGGCAGTAGCACGT 	1.00, 22, kAUC=13.0	PWM= TCGACACGGCAGTAGCACGT	25.15, 32, pAUC=19.6



------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=6	KSM= GCAGGG 	1.35, 154, kAUC=58.7	PWM= GGCAGG	7.57, 130, pAUC=6.7
#1	k=6	KSM= GGGGCT 	1.30, 165, kAUC=53.8	PWM= GGGGGG	6.25, 125, pAUC=0.7
#2	k=7	KSM= CGGCAGG 	2.29, 100, kAUC=52.6	PWM= CGGCAGG	9.03, 95, pAUC=24.8
#3	k=8	KSM= CGGCAGGG 	1.10, 140, kAUC=51.5	PWM= CGACAGGG	8.48, 125, pAUC=36.6
#4	k=7	KSM= TGCTACT 	1.96, 76, kAUC=49.6	PWM= TGCTACT	8.46, 92, pAUC=23.3
#5	k=8	KSM= TGCTACTG 	1.10, 102, kAUC=41.3	PWM= TGCTACTG	9.70, 89, pAUC=25.4
#6	k=9	KSM= TCGGCAGGG 	1.00, 115, kAUC=39.1	PWM= TCGACAGGG	10.26, 98, pAUC=33.4
#7	k=10	KSM= TCGGCAGGGC 	1.16, 112, kAUC=38.8	PWM= TCGGCAGGGG	12.77, 90, pAUC=29.7
#8	k=7	KSM= GGCTCGG 	2.55, 90, kAUC=37.5	PWM= GGCTCGA	8.29, 87, pAUC=21.7
#9	k=12	KSM= TGCTACTGCGTG 	1.10, 82, kAUC=35.5	PWM= TGCTACTGCGTG	11.91, 88, pAUC=32.2
#10	k=9	KSM= TGCTACTGC 	1.99, 62, kAUC=35.3	PWM= TGCTACTGC	10.89, 87, pAUC=29.9
#11	k=10	KSM= GTGCTGCTAC 	1.29, 72, kAUC=34.6	PWM= GTGCTGCTAC	11.53, 71, pAUC=26.4
#12	k=9	KSM= GTGTTGTTG 	1.07, 86, kAUC=34.1	PWM= GTGTTGTTG	9.99, 65, pAUC=18.0
#13	k=13	KSM= TGTGCTGCTACTG 	1.09, 73, kAUC=33.1	PWM= TGTGCTGCTACTG	11.30, 79, pAUC=31.2
#14	k=10	KSM= TGTTGTTGTG 	1.21, 76, kAUC=32.9	PWM= TGTTGTTGTG	10.53, 50, pAUC=16.9
#15	k=16	KSM= TGTGCTGCTACTGCGT 	1.00, 75, kAUC=32.7	PWM= TGTGCTGCTACTGCGT	18.64, 80, pAUC=32.3
#16	k=11	KSM= TGTTGTTGTGT 	0.94, 80, kAUC=31.3	PWM= TGTTGTTGTGT	11.22, 48, pAUC=18.0
#17	k=11	KSM= GTGCTGCTACT 	1.92, 58, kAUC=31.1	PWM= GTGCTGCTACT	13.70, 58, pAUC=27.8
#18	k=14	KSM= GGCTCGGCAGGGCT 	1.10, 76, kAUC=27.6	PWM= GGCTCGGCAGGGCT	20.05, 73, pAUC=24.0
#19	k=13	KSM= GCTCGGCAGGGCT 	1.31, 76, kAUC=27.6	PWM= GCTCGGCAGGGCT	18.47, 73, pAUC=23.6
#20	k=18	KSM= AGGGGGCTCGGCAGGGCT 	1.21, 73, kAUC=27.4	PWM= AGGGGGCTCGGCAGGGCT	22.15, 71, pAUC=23.4
#21	k=17	KSM= CGGCAGGGGGCTCGGCA 	1.10, 73, kAUC=27.4	PWM= CGGCAGGGGGCTCGGCA	27.23, 73, pAUC=24.4
#22	k=20	KSM= CTCGGCAGGGGGCTCGGCAG 	1.42, 73, kAUC=27.4	PWM= CTCGGCAGGGGGCTCGGCAG	28.70, 73, pAUC=24.0
#23	k=16	KSM= CAGGGGGCTCGGCAGG 	1.31, 73, kAUC=27.4	PWM= CAGGGGGCTCGGCAGG	25.29, 73, pAUC=23.9
#24	k=15	KSM= GCTCGGCAGGGCTCG 	1.10, 73, kAUC=27.4	PWM= GCTCGGCAGGGCTCG	21.84, 71, pAUC=23.7
#25	k=19	KSM= CTCGGCAGGGCTCGGCAGG 	1.10, 73, kAUC=27.4	PWM= CTCGGCAGGGCTCGGCAGG	30.25, 70, pAUC=23.8
#26	k=11	KSM= GGCTCGGCAGG 	7.08, 76, kAUC=26.8	PWM= GGCTCGGCAGG	15.40, 76, pAUC=23.6
#27	k=12	KSM= GGGGGCTCGGCA 	1.00, 76, kAUC=26.2	PWM= GGGGGCTCGGCA	17.08, 74, pAUC=23.2
#28	k=15	KSM= TGTTGTTGTGTGCTG 	1.00, 51, kAUC=23.9	PWM= TGTTGTTGTGTGCTG	17.83, 52, pAUC=19.7
#29	k=14	KSM= GTTGTTGTGTGCTG 	1.00, 51, kAUC=22.8	PWM= GTTGTTGTGTGCTG	18.39, 48, pAUC=19.2
#30	k=12	KSM= TGTTGTGTGCTG 	1.10, 52, kAUC=21.9	PWM= TGTTGTGTGCTG	15.29, 49, pAUC=15.3
#31	k=17	KSM= GTTGTTGTGTGCTGCTA 	1.00, 45, kAUC=21.8	PWM= GTTGTTGTGTGCTGCTA	20.40, 50, pAUC=20.5
#32	k=18	KSM= GTGTGCTGCTACTGCGTG 	1.00, 42, kAUC=20.7	PWM= GTGTGCTGCTACTGCGTG	21.57, 52, pAUC=22.8
#33	k=19	KSM= GTGTGTTGTTGTGTGCTGC 	1.00, 41, kAUC=20.4	PWM= GTGTGTTGTTGTGTGCTGC	20.79, 41, pAUC=18.2
#34	k=20	KSM= TGTGATGTTGGTGTGTGTTG 	1.00, 39, kAUC=19.7	PWM= TGTGTTGTTGTTGTGTGTTG	18.29, 50, pAUC=19.3
#35	k=17	KSM= TACTGCGTGTGTTGTTG 	1.00, 38, kAUC=19.3	PWM= TACTGCGTGTGTTGTTG	19.68, 36, pAUC=16.5
#36	k=15	KSM= TCGACACGGCAGTAG 	1.00, 28, kAUC=15.1	PWM= TCGACACGGCAGTAG	17.27, 32, pAUC=20.0
#37	k=18	KSM= TGTCGACCGGAGACGTGC 	1.00, 22, kAUC=13.0	PWM= TGTCGACCGGAGACGTGC	18.58, 31, pAUC=20.4
#38	k=20	KSM= TCGACACGGCAGTAGCACGT 	1.00, 22, kAUC=13.0	PWM= TCGACACGGCAGTAGCACGT	25.15, 32, pAUC=19.6

142ms: Remove motif #2.
208ms: Remove motif #3.
285ms: Remove motif #6.
357ms: Remove motif #7.
359ms: PWM GGCTCGGCAGGGCT has been removed.
362ms: PWM GCTCGGCAGGGCT has been removed.
365ms: PWM AGGGGGCTCGGCAGGGCT has been removed.
367ms: PWM GCTCGGCAGGGCTCG has been removed.
370ms: PWM CTCGGCAGGGCTCGGCAGG has been removed.
472ms: Remove motif #26.
570ms: Remove motif #36.
645ms: Remove motif #38.
688ms: Remove motif #8.
691ms: PWM CGGCAGGGGGCTCGGCA has been removed.
695ms: PWM CTCGGCAGGGGGCTCGGCAG has been removed.
698ms: PWM CAGGGGGCTCGGCAGG has been removed.
701ms: PWM GGGGGCTCGGCA has been removed.
770ms: Remove motif #5.
819ms: Remove motif #9.
869ms: Remove motif #10.
937ms: Remove motif #11.
1.0s: Remove motif #13.
1.1s: Remove motif #15.
1.2s: Remove motif #17.
1.2s: Remove motif #32.
1.3s: Remove motif #34.
1.3s: Remove motif #14.
1.3s: PWM TGTTGTTGTGT has been removed.
1.4s: Remove motif #28.
1.5s: Remove motif #30.
1.6s: Remove motif #33.
1.6s: PWM TACTGCGTGTGTTGTTG has been removed.
1.6s: Remove motif #31.

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=6	KSM= GCAGGG 	1.35, 154, kAUC=58.7	PWM= GGCAGG	7.57, 130, pAUC=6.7
#1	k=6	KSM= GGGGCT 	1.30, 165, kAUC=53.8	PWM= GGGGGG	6.25, 125, pAUC=0.7
#2	k=7	KSM= CGGCAGG 	2.29, 100, kAUC=52.6	PWM= CGGCAGG	9.03, 95, pAUC=24.8
#3	k=8	KSM= CGGCAGGG 	1.10, 140, kAUC=51.5	PWM= CGACAGGG	8.48, 125, pAUC=36.6
#4	k=7	KSM= TGCTACT 	1.96, 76, kAUC=49.6	PWM= TGCTACT	8.46, 92, pAUC=23.3
#5	k=8	KSM= TGCTACTG 	1.10, 102, kAUC=41.3	PWM= TGCTACTG	9.70, 89, pAUC=25.4
#6	k=9	KSM= TCGGCAGGG 	1.00, 115, kAUC=39.1	PWM= TCGACAGGG	10.26, 98, pAUC=33.4
#7	k=10	KSM= TCGGCAGGGC 	1.16, 112, kAUC=38.8	PWM= TCGGCAGGGG	12.77, 90, pAUC=29.7
#8	k=7	KSM= GGCTCGG 	2.55, 90, kAUC=37.5	PWM= GGCTCGA	8.29, 87, pAUC=21.7
#9	k=12	KSM= TGCTACTGCGTG 	1.10, 82, kAUC=35.5	PWM= TGCTACTGCGTG	11.91, 88, pAUC=32.2
#10	k=9	KSM= TGCTACTGC 	1.99, 62, kAUC=35.3	PWM= TGCTACTGC	10.89, 87, pAUC=29.9
#11	k=10	KSM= GTGCTGCTAC 	1.29, 72, kAUC=34.6	PWM= GTGCTGCTAC	11.53, 71, pAUC=26.4
#12	k=9	KSM= GTGTTGTTG 	1.07, 86, kAUC=34.1	PWM= GTGTTGTTG	9.99, 65, pAUC=18.0
#13	k=13	KSM= TGTGCTGCTACTG 	1.09, 73, kAUC=33.1	PWM= TGTGCTGCTACTG	11.30, 79, pAUC=31.2
#14	k=10	KSM= TGTTGTTGTG 	1.21, 76, kAUC=32.9	PWM= TGTTGTTGTG	10.53, 50, pAUC=16.9
#15	k=16	KSM= TGTGCTGCTACTGCGT 	1.00, 75, kAUC=32.7	PWM= TGTGCTGCTACTGCGT	18.64, 80, pAUC=32.3
#16	k=11	KSM= TGTTGTTGTGT 	0.94, 80, kAUC=31.3	PWM= TGTTGTTGTGT	11.22, 48, pAUC=18.0
#17	k=11	KSM= GTGCTGCTACT 	1.92, 58, kAUC=31.1	PWM= GTGCTGCTACT	13.70, 58, pAUC=27.8
#18	k=14	KSM= GGCTCGGCAGGGCT 	1.10, 76, kAUC=27.6	PWM= GGCTCGGCAGGGCT	20.05, 73, pAUC=24.0
#19	k=13	KSM= GCTCGGCAGGGCT 	1.31, 76, kAUC=27.6	PWM= GCTCGGCAGGGCT	18.47, 73, pAUC=23.6
#20	k=18	KSM= AGGGGGCTCGGCAGGGCT 	1.21, 73, kAUC=27.4	PWM= AGGGGGCTCGGCAGGGCT	22.15, 71, pAUC=23.4
#21	k=17	KSM= CGGCAGGGGGCTCGGCA 	1.10, 73, kAUC=27.4	PWM= CGGCAGGGGGCTCGGCA	27.23, 73, pAUC=24.4
#22	k=20	KSM= CTCGGCAGGGGGCTCGGCAG 	1.42, 73, kAUC=27.4	PWM= CTCGGCAGGGGGCTCGGCAG	28.70, 73, pAUC=24.0
#23	k=16	KSM= CAGGGGGCTCGGCAGG 	1.31, 73, kAUC=27.4	PWM= CAGGGGGCTCGGCAGG	25.29, 73, pAUC=23.9
#24	k=15	KSM= GCTCGGCAGGGCTCG 	1.10, 73, kAUC=27.4	PWM= GCTCGGCAGGGCTCG	21.84, 71, pAUC=23.7
#25	k=19	KSM= CTCGGCAGGGCTCGGCAGG 	1.10, 73, kAUC=27.4	PWM= CTCGGCAGGGCTCGGCAGG	30.25, 70, pAUC=23.8
#26	k=11	KSM= GGCTCGGCAGG 	7.08, 76, kAUC=26.8	PWM= GGCTCGGCAGG	15.40, 76, pAUC=23.6
#27	k=12	KSM= GGGGGCTCGGCA 	1.00, 76, kAUC=26.2	PWM= GGGGGCTCGGCA	17.08, 74, pAUC=23.2
#28	k=15	KSM= TGTTGTTGTGTGCTG 	1.00, 51, kAUC=23.9	PWM= TGTTGTTGTGTGCTG	17.83, 52, pAUC=19.7
#29	k=14	KSM= GTTGTTGTGTGCTG 	1.00, 51, kAUC=22.8	PWM= GTTGTTGTGTGCTG	18.39, 48, pAUC=19.2
#30	k=12	KSM= TGTTGTGTGCTG 	1.10, 52, kAUC=21.9	PWM= TGTTGTGTGCTG	15.29, 49, pAUC=15.3
#31	k=17	KSM= GTTGTTGTGTGCTGCTA 	1.00, 45, kAUC=21.8	PWM= GTTGTTGTGTGCTGCTA	20.40, 50, pAUC=20.5
#32	k=18	KSM= GTGTGCTGCTACTGCGTG 	1.00, 42, kAUC=20.7	PWM= GTGTGCTGCTACTGCGTG	21.57, 52, pAUC=22.8
#33	k=19	KSM= GTGTGTTGTTGTGTGCTGC 	1.00, 41, kAUC=20.4	PWM= GTGTGTTGTTGTGTGCTGC	20.79, 41, pAUC=18.2
#34	k=20	KSM= TGTGATGTTGGTGTGTGTTG 	1.00, 39, kAUC=19.7	PWM= TGTGTTGTTGTTGTGTGTTG	18.29, 50, pAUC=19.3
#35	k=17	KSM= TACTGCGTGTGTTGTTG 	1.00, 38, kAUC=19.3	PWM= TACTGCGTGTGTTGTTG	19.68, 36, pAUC=16.5
#36	k=15	KSM= TCGACACGGCAGTAG 	1.00, 28, kAUC=15.1	PWM= TCGACACGGCAGTAG	17.27, 32, pAUC=20.0
#37	k=18	KSM= TGTCGACCGGAGACGTGC 	1.00, 22, kAUC=13.0	PWM= TGTCGACCGGAGACGTGC	18.58, 31, pAUC=20.4
#38	k=20	KSM= TCGACACGGCAGTAGCACGT 	1.00, 22, kAUC=13.0	PWM= TCGACACGGCAGTAGCACGT	25.15, 32, pAUC=19.6

------------------------- GEMout_1, k = ALL (after merging) -------------------------
#0	k=6	KSM= GCAGGG 	1.35, 154, kAUC=58.7	PWM= GGCAGG	7.57, 130, pAUC=6.7
#1	k=6	KSM= GGGGCT 	1.30, 165, kAUC=53.8	PWM= GGGGGG	6.25, 125, pAUC=0.7
#4	k=7	KSM= TGCTACT 	1.84, 86, kAUC=52.0	PWM= TGCTACT	7.60, 88, pAUC=23.8
#12	k=9	KSM= GTGTTGTTG 	1.07, 85, kAUC=35.7	PWM= GTGTTGTTG	10.06, 61, pAUC=17.1
#29	k=14	KSM= GTTGTTGTGTGCTG 	1.00, 51, kAUC=22.8	PWM= GTTGTTGTGTGCTG	18.39, 48, pAUC=19.2
#37	k=18	KSM= TGTCGACCGGAGACGTGC 	1.00, 22, kAUC=13.0	PWM= TGTCGACCGGAGACGTGC	18.58, 31, pAUC=20.4

1.7s: Finalizing 6 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_1 final motifs ----------------------------


GEMout_1 motif #0

KSM top k-mer: GCAGGG/CCCTGC, total 61 k-mers
KSM threshold: 1.35, 		hit=154+/3-, kAUC=58.7

   |
GGCAGG
     T
      
      
PWM threshold: 7.57/11.02, 	hit=130+/87-, pAUC=6.7
--------------------------------------------------------------

GEMout_1 motif #1

KSM top k-mer: GGGGCT/AGCCCC, total 34 k-mers
KSM threshold: 1.30, 		hit=165+/7-, kAUC=53.8

    |
GGGGGG
    CC
    TT
      
PWM threshold: 6.25/8.93, 	hit=125+/135-, pAUC=0.7
--------------------------------------------------------------

GEMout_1 motif #2

KSM top k-mer: TGCTACT/AGTAGCA, total 155 k-mers
KSM threshold: 1.84, 		hit=86+/0-, kAUC=52.0

  |
TGCTACT
       
       
       
PWM threshold: 7.60/11.02, 	hit=88+/40-, pAUC=23.8
--------------------------------------------------------------

GEMout_1 motif #3

KSM top k-mer: GTGTTGTTG/CAACAACAC, total 144 k-mers
KSM threshold: 1.07, 		hit=85+/0-, kAUC=35.7

       |
GTGTTGTTG
         
         
         
PWM threshold: 10.06/16.07, 	hit=61+/21-, pAUC=17.1
--------------------------------------------------------------

GEMout_1 motif #4

KSM top k-mer: GTTGTTGTGTGCTG/CAGCACACAACAAC, total 150 k-mers
KSM threshold: 1.00, 		hit=51+/0-, kAUC=22.8

         |
GTTGTTGTGTGCTG
              
              
              
PWM threshold: 18.39/26.66, 	hit=48+/3-, pAUC=19.2
--------------------------------------------------------------

GEMout_1 motif #5

KSM top k-mer: TGTCGACCGGAGACGTGC/GCACGTCTCCGGTCGACA, total 40 k-mers
KSM threshold: 1.00, 		hit=22+/0-, kAUC=13.0

             |
TGTCGACCGGAGACGTGC
       GA         
                  
                  
PWM threshold: 18.58/33.49, 	hit=31+/0-, pAUC=20.4
--------------------------------------------------------------



============================ Round 2 ============================
Running EM with motif positional prior ...

Running with 10 threads ...

100	/224014	20.1s
1000	/224014	21.1s
10000	/224014	22.1s
100000	/224014	35.1s
200000	/224014	47.1s
224014	/224014	2.9m
---------------------------
Events discovered 
Significant:	953
Insignificant:	241
Filtered:	4

Total 476 homotypic events (within 501bp of other events).

Finish binding event prediction: 3.0m

Refine read distribution from 468 binding events. 

Percentage of significant events with a motif match, divided in 10 bins (~96 events per bin):
96 97 93 95 100 97 96 97 95 97 

Percentage of filtered events with a motif match, divided in 4 bins (~1 events per bin):
0 0 0 0 

Percentage of insignificant events with a motif match, divided in 10 bins (~25 events per bin):
96 100 100 84 96 92 96 100 96 93 

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.49. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=302
k=6+1, gapped kmer=709, base kmer=748
k=6+2, gapped kmer=2976, base kmer=1434
k=6+3, gapped kmer=5428, base kmer=1552
k=6+4, gapped kmer=7607, base kmer=1542

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 302

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTAC/GTAGCA	207	214	41	-36.8    	0	309.3	5.0	1546.4	24
TTGTTG/CAACAA	150	153	51	-15.5    	5	286.1	4.0	1144.4	40
ACATGC/GCATGT	108	109	36	-11.0    	13	240.6	3.0	721.7	34
CTACTG/CAGTAG	155	162	36	-24.1    	3	252.0	2.0	504.0	16
CTGCGT/ACGCAG	100	104	17	-18.4    	15	228.9	2.0	457.9	23

Align and cluster 238 k-mers with seed TGCTAC,   	#0
Align and cluster 311 k-mers with seed TTGTTG,   	#1
Align and cluster 258 k-mers with seed ACATGC,   	#2
Align and cluster 208 k-mers with seed CTACTG,   	#3
Align and cluster 194 k-mers with seed CTGCGT,   	#4

Merging redundant motifs ...
173ms: Remove motif #3.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= ACATGC 	1.71, 261, kAUC=55.3	PWM= ACATGC	6.63, 355, pAUC=14.1
#1	k=6	KSM= TGCTAC 	2.47, 372, kAUC=46.8	PWM= TGCTAC	6.48, 307, pAUC=23.2
#2	k=6	KSM= CTGCGT 	2.80, 253, kAUC=40.9	PWM= CTGCGT	6.76, 231, pAUC=21.0
#3	k=6	KSM= CTACTG 	2.41, 218, kAUC=39.4	PWM= CTACTG	6.48, 284, pAUC=19.5
#4	k=6	KSM= TTGTTG 	2.40, 319, kAUC=37.0	PWM= TTGTTG	6.78, 236, pAUC=14.4

After merging:
#0	k=6	KSM= ACATGC 	1.71, 261, kAUC=55.3	PWM= ACATGC	6.63, 355, pAUC=14.1
#1	k=6	KSM= TGCTAC 	2.49, 373, kAUC=47.5	PWM= TGCTAC	6.51, 306, pAUC=23.2
#2	k=6	KSM= CTGCGT 	2.80, 253, kAUC=40.9	PWM= CTGCGT	6.76, 231, pAUC=21.0
#4	k=6	KSM= TTGTTG 	2.40, 319, kAUC=37.0	PWM= TTGTTG	6.78, 236, pAUC=14.4



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=873
k=7+1, gapped kmer=931, base kmer=1067
k=7+2, gapped kmer=2064, base kmer=1445
k=7+3, gapped kmer=3222, base kmer=1519
k=7+4, gapped kmer=4757, base kmer=1536

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 873

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTACT/AGTAGCA	161	169	23	-33.8    	0	257.1	6.0	1542.4	32
GTTGTTG/CAACAAC	145	148	21	-28.6    	3	239.3	5.0	1196.6	53
ACTGCGT/ACGCAGT	94	98	5	-26.1    	9	185.7	4.0	742.9	33
GATAGTT/AACTATC	60	61	2	-17.9    	29	124.7	3.0	374.2	31
TACATGC/GCATGTA	72	73	11	-13.6    	24	176.1	2.0	352.3	44

Align and cluster 373 k-mers with seed TGCTACT,   	#0
Align and cluster 527 k-mers with seed GTTGTTG,   	#1
Align and cluster 297 k-mers with seed ACTGCGT,   	#2
Align and cluster 279 k-mers with seed GATAGTT,   	#3
Align and cluster 328 k-mers with seed TACATGC,   	#4

Merging redundant motifs ...

------------------------- k = 7 ----------------------------

#0	k=7	KSM= TGCTACT 	2.88, 187, kAUC=54.5	PWM= TGCTACT	8.11, 245, pAUC=26.4
#1	k=7	KSM= ACTGCGT 	1.31, 276, kAUC=48.7	PWM= ACTGCGT	8.11, 186, pAUC=21.6
#2	k=7	KSM= GTTGTTG 	3.23, 228, kAUC=48.3	PWM= GTTGTTG	6.79, 217, pAUC=25.4
#3	k=7	KSM= TACATGC 	2.67, 159, kAUC=38.0	PWM= TACATGC	8.08, 167, pAUC=17.5
#4	k=7	KSM= GATAGTT 	2.35, 141, kAUC=37.6	PWM= GATAGTT	7.63, 107, pAUC=17.3

After merging:
#0	k=7	KSM= TGCTACT 	2.88, 187, kAUC=54.5	PWM= TGCTACT	8.11, 245, pAUC=26.4
#1	k=7	KSM= ACTGCGT 	1.31, 276, kAUC=48.7	PWM= ACTGCGT	8.11, 186, pAUC=21.6
#2	k=7	KSM= GTTGTTG 	3.23, 228, kAUC=48.3	PWM= GTTGTTG	6.79, 217, pAUC=25.4
#3	k=7	KSM= TACATGC 	2.67, 159, kAUC=38.0	PWM= TACATGC	8.08, 167, pAUC=17.5
#4	k=7	KSM= GATAGTT 	2.35, 141, kAUC=37.6	PWM= GATAGTT	7.63, 107, pAUC=17.3



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=1439
k=8+1, gapped kmer=719, base kmer=915
k=8+2, gapped kmer=1247, base kmer=1153
k=8+3, gapped kmer=1856, base kmer=1373
k=8+4, gapped kmer=2614, base kmer=1458

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 1439

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTACTG/CAGTAGCA	144	152	18	-32.0    	0	206.3	7.0	1444.1	32
TACTGCGT/ACGCAGTA	76	80	1	-25.6    	12	141.7	3.0	425.1	21
CTGCGTGT/ACACGCAG	61	65	4	-16.8    	22	131.5	3.0	394.5	23
GTTACATG/CATGTAAC	62	63	2	-18.5    	25	114.7	3.0	344.2	29
GCTGCTAC/GTAGCAGC	94	99	8	-23.5    	5	163.5	2.0	327.0	28

Align and cluster 717 k-mers with seed TGCTACTG,   	#0
Align and cluster 578 k-mers with seed TACTGCGT,   	#1
Align and cluster 557 k-mers with seed CTGCGTGT,   	#2
Align and cluster 556 k-mers with seed GTTACATG,   	#3
Align and cluster 578 k-mers with seed GCTGCTAC,   	#4

Merging redundant motifs ...
109ms: Remove motif #1.
187ms: Remove motif #4.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= GCTGCTAC 	2.24, 225, kAUC=46.3	PWM= GCTGCTAC	9.26, 179, pAUC=24.1
#1	k=8	KSM= TGCTACTG 	2.60, 147, kAUC=45.6	PWM= TGCTACTG	9.56, 184, pAUC=21.7
#2	k=8	KSM= TACTGCGT 	2.68, 150, kAUC=44.6	PWM= TACTGCGT	9.03, 177, pAUC=24.4
#3	k=8	KSM= GTTACATG 	2.62, 115, kAUC=41.5	PWM= GTTACGTG	8.20, 214, pAUC=16.2
#4	k=8	KSM= CTGCGTGT 	2.79, 118, kAUC=38.3	PWM= CTGCGTGT	9.72, 162, pAUC=23.0

After merging:
#0	k=8	KSM= GCTGCTAC 	1.48, 261, kAUC=52.0	PWM= GCTGCTAC	9.44, 228, pAUC=29.3
#2	k=8	KSM= TACTGCGT 	2.68, 150, kAUC=44.6	PWM= TACTGCGT	9.03, 177, pAUC=24.4
#3	k=8	KSM= GTTACATG 	2.62, 115, kAUC=41.5	PWM= GTTACGTG	8.20, 214, pAUC=16.2



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=1552
k=9+1, gapped kmer=629, base kmer=842
k=9+2, gapped kmer=912, base kmer=925
k=9+3, gapped kmer=1180, base kmer=1094
k=9+4, gapped kmer=1591, base kmer=1238

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 1552

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTACTGC/GCAGTAGCA	132	139	4	-41.2    	0	228.3	7.0	1598.2	89
TACTGCGTG/CACGCAGTA	72	77	0	-25.3    	10	152.2	4.0	609.0	60
TACATGCAT/ATGCATGTA	55	56	4	-14.1    	26	125.0	3.0	375.0	84
TCGACACGG/CCGTGTCGA	35	38	0	-12.6    	48	90.2	4.0	360.7	55
ACACGGCAG/CTGCCGTGT	41	44	0	-14.6    	37	113.0	3.0	338.9	52

Align and cluster 636 k-mers with seed TGCTACTGC,   	#0
Align and cluster 424 k-mers with seed TACTGCGTG,   	#1
Align and cluster 487 k-mers with seed TACATGCAT,   	#2
Align and cluster 297 k-mers with seed TCGACACGG,   	#3
Align and cluster 311 k-mers with seed ACACGGCAG,   	#4

Merging redundant motifs ...
92ms: Remove motif #1.
228ms: Remove motif #2.

------------------------- k = 9 ----------------------------

#0	k=9	KSM= TACTGCGTG 	1.51, 217, kAUC=45.5	PWM= TACTGCCTG	8.87, 201, pAUC=28.2
#1	k=9	KSM= ACACGGCAG 	1.41, 217, kAUC=43.7	PWM= CCCCCGCAG	7.95, 229, pAUC=21.1
#2	k=9	KSM= TGCTACTGC 	1.72, 169, kAUC=37.4	PWM= TGCTACTGC	10.43, 176, pAUC=26.6
#3	k=9	KSM= TACATGCAT 	1.44, 123, kAUC=28.9	PWM= TACATGCAT	8.30, 96, pAUC=13.8
#4	k=9	KSM= TCGACACGG 	1.00, 88, kAUC=18.2	PWM= TCGACACGG	9.72, 73, pAUC=13.1

After merging:
#0	k=9	KSM= TACTGCGTG 	1.41, 232, kAUC=48.1	PWM= TACTGCGTG	9.26, 223, pAUC=31.0
#3	k=9	KSM= TACATGCAT 	1.44, 123, kAUC=28.9	PWM= TACATGCAT	8.30, 96, pAUC=13.8
#4	k=9	KSM= TCGACACGG 	1.00, 88, kAUC=18.2	PWM= TCGACACGG	9.72, 73, pAUC=13.1



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=1542
k=10+1, gapped kmer=553, base kmer=751
k=10+2, gapped kmer=784, base kmer=782
k=10+3, gapped kmer=860, base kmer=872
k=10+4, gapped kmer=980, base kmer=946

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 1542

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTGCTACT/AGTAGCAGCA	91	96	5	-25.5    	0	156.8	8.0	1254.1	67
TGTTGTTGTG/CACAACAACA	69	72	10	-13.9    	8	130.9	5.0	654.5	83
GATAGTTACA/TGTAACTATC	56	57	0	-18.7    	17	103.5	5.0	517.6	46
CTACTGCGTG/CACGCAGTAG	69	73	0	-24.0    	7	125.0	4.0	500.0	49
AGTTACATGC/GCATGTAACT	48	49	0	-16.2    	30	98.0	4.0	392.0	44

Align and cluster 716 k-mers with seed TGCTGCTACT,   	#0
Align and cluster 921 k-mers with seed TGTTGTTGTG,   	#1
Align and cluster 502 k-mers with seed GATAGTTACA,   	#2
Align and cluster 648 k-mers with seed CTACTGCGTG,   	#3
Align and cluster 533 k-mers with seed AGTTACATGC,   	#4

Merging redundant motifs ...
139ms: Remove motif #2.
240ms: Remove motif #4.

------------------------- k = 10 ----------------------------

#0	k=10	KSM= AGTTACATGC 	1.82, 142, kAUC=35.3	PWM= AGTTACATGC	9.48, 135, pAUC=23.3
#1	k=10	KSM= TGCTGCTACT 	1.32, 187, kAUC=35.1	PWM= TGCTGCTACT	11.79, 187, pAUC=26.4
#2	k=10	KSM= GATAGTTACA 	2.42, 122, kAUC=31.0	PWM= GATAGTTACG	8.55, 75, pAUC=13.4
#3	k=10	KSM= TGTTGTTGTG 	2.04, 137, kAUC=30.2	PWM= TGTTGTTGTG	12.30, 123, pAUC=16.4
#4	k=10	KSM= CTACTGCGTG 	1.10, 138, kAUC=25.2	PWM= CTACTGCGTG	11.44, 121, pAUC=20.0

After merging:
#0	k=10	KSM= AGTTACATGC 	2.78, 110, kAUC=35.8	PWM= AGATACATGC	8.78, 139, pAUC=23.0
#1	k=10	KSM= TGCTGCTACT 	1.32, 187, kAUC=35.1	PWM= TGCTGCTACT	11.79, 187, pAUC=26.4
#3	k=10	KSM= TGTTGTTGTG 	2.04, 137, kAUC=30.2	PWM= TGTTGTTGTG	12.30, 123, pAUC=16.4



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=1539
k=11+1, gapped kmer=525, base kmer=717
k=11+2, gapped kmer=749, base kmer=736
k=11+3, gapped kmer=655, base kmer=716
k=11+4, gapped kmer=644, base kmer=759

------------------------- k = 11 ----------------------------

Total number of k-mers: n = 1539

Top 5 cluster center k-mers:
# k-mer/r.c.           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTGCTACTG/CAGTAGCAGCA	83	88	3	-25.2    	0	126.9	9.0	1142.1	48
GATAGTTACAT/ATGTAACTATC	56	57	0	-18.7    	10	79.2	6.0	475.1	39
TGTTGTTGTGT/ACACAACAACA	64	67	6	-15.5    	6	116.9	4.0	467.6	54
TACATGCATAG/CTATGCATGTA	48	49	0	-16.2    	21	71.7	5.0	358.6	36
GCTACTGCGTG/CACGCAGTAGC	67	71	0	-23.3    	5	108.9	2.0	217.8	43

Align and cluster 619 k-mers with seed TGCTGCTACTG,   	#0
Align and cluster 358 k-mers with seed GATAGTTACAT,   	#1
Align and cluster 683 k-mers with seed TGTTGTTGTGT,   	#2
Align and cluster 442 k-mers with seed TACATGCATAG,   	#3
Align and cluster 535 k-mers with seed GCTACTGCGTG,   	#4

Merging redundant motifs ...
118ms: Remove motif #1.

------------------------- k = 11 ----------------------------

#0	k=11	KSM= TGCTGCTACTG 	1.78, 152, kAUC=28.9	PWM= TGCTGCTACTG	13.58, 135, pAUC=25.0
#1	k=11	KSM= GCTACTGCGTG 	1.10, 126, kAUC=22.3	PWM= GCTACTGCGTG	12.64, 121, pAUC=21.6
#2	k=11	KSM= TGTTGTTGTGT 	1.47, 109, kAUC=22.2	PWM= TGTTGTTGTGT	13.24, 109, pAUC=15.6
#3	k=11	KSM= GATAGTTACAT 	1.00, 107, kAUC=21.7	PWM= GATAGTTACAT	9.93, 75, pAUC=15.5
#4	k=11	KSM= TACATGCATAG 	1.00, 96, kAUC=19.6	PWM= TACATGCATAG	10.28, 91, pAUC=16.5

After merging:
#0	k=11	KSM= TGCTGCTACTG 	1.78, 152, kAUC=28.9	PWM= TGCTGCTACTG	13.58, 135, pAUC=25.0
#2	k=11	KSM= TGTTGTTGTGT 	1.47, 109, kAUC=22.2	PWM= TGTTGTTGTGT	13.24, 109, pAUC=15.6
#3	k=11	KSM= GATAGTTACAT 	1.00, 107, kAUC=21.7	PWM= GATAGTTACAT	9.93, 75, pAUC=15.5
#4	k=11	KSM= TACATGCATAG 	1.00, 96, kAUC=19.6	PWM= TACATGCATAG	10.28, 91, pAUC=16.5



----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=1533
k=12+1, gapped kmer=510, base kmer=706
k=12+2, gapped kmer=706, base kmer=697
k=12+3, gapped kmer=524, base kmer=632
k=12+4, gapped kmer=463, base kmer=585

------------------------- k = 12 ----------------------------

Total number of k-mers: n = 1533

Top 5 cluster center k-mers:
# k-mer/r.c.             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTGCTACTGC/GCAGTAGCAGCA	80	85	1	-27.3    	0	145.2	10.0	1451.9	86
TGCATAGTTACA/TGTAACTATGCA	47	48	0	-15.8    	15	79.6	6.0	477.6	60
TGTTGTTGTGTG/CACACAACAACA	62	65	2	-19.1    	4	118.2	4.0	472.9	95
ACATGCATAGTT/AACTATGCATGT	48	49	0	-16.2    	10	76.7	6.0	460.1	61
TAGTTACATGCA/TGCATGTAACTA	45	46	0	-15.2    	20	90.7	5.0	453.7	61

Align and cluster 661 k-mers with seed TGCTGCTACTGC,   	#0
Align and cluster 488 k-mers with seed TGCATAGTTACA,   	#1
Align and cluster 816 k-mers with seed TGTTGTTGTGTG,   	#2
Align and cluster 531 k-mers with seed ACATGCATAGTT,   	#3
Align and cluster 538 k-mers with seed TAGTTACATGCA,   	#4

Merging redundant motifs ...
352ms: Remove motif #3.
470ms: Remove motif #4.
634ms: Remove motif #2.

------------------------- k = 12 ----------------------------

#0	k=12	KSM= TGCTGCTACTGC 	1.31, 146, kAUC=26.1	PWM= TGCTGCTACTGC	12.94, 132, pAUC=24.8
#1	k=12	KSM= TAGTTACATGCA 	1.00, 111, kAUC=22.2	PWM= TAGTTACATGCA	10.57, 91, pAUC=19.2
#2	k=12	KSM= TGTTGTTGTGTG 	3.39, 101, kAUC=22.0	PWM= TGTTGTTGTGTG	15.33, 109, pAUC=16.7
#3	k=12	KSM= ACATGCATAGTT 	1.00, 73, kAUC=16.1	PWM= ACATGCATAGTT	11.80, 72, pAUC=15.8
#4	k=12	KSM= TGCATAGTTACA 	1.10, 72, kAUC=15.8	PWM= TGCATAGTTACA	14.41, 72, pAUC=14.1

After merging:
#1	k=12	KSM= TAGTTACATGCA 	1.31, 215, kAUC=40.7	PWM= TAGTTACATGCA	10.15, 170, pAUC=25.9
#0	k=12	KSM= TGCTGCTACTGC 	1.31, 146, kAUC=26.1	PWM= TGCTGCTACTGC	12.94, 132, pAUC=24.8



----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=1518
k=13+1, gapped kmer=494, base kmer=676
k=13+2, gapped kmer=661, base kmer=685
k=13+3, gapped kmer=467, base kmer=575
k=13+4, gapped kmer=349, base kmer=465

------------------------- k = 13 ----------------------------

Total number of k-mers: n = 1518

Top 5 cluster center k-mers:
# k-mer/r.c.               	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GTGCTGCTACTGC/GCAGTAGCAGCAC	73	78	0	-25.6    	0	120.4	10.0	1204.5	69
ATAGTTACATGCA/TGCATGTAACTAT	44	45	0	-14.9    	14	74.4	6.0	446.4	49
TGTTGTTGTGTGC/GCACACAACAACA	42	45	0	-14.9    	15	82.7	5.0	413.5	69
TCGACACGGCAGT/ACTGCCGTGTCGA	28	30	0	-10.1    	34	40.2	6.0	241.5	27
TCGGTGAACACTG/CAGTGTTCACCGA	23	23	0	-7.9    	87	33.9	6.0	203.5	8

Align and cluster 566 k-mers with seed GTGCTGCTACTGC,   	#0
Align and cluster 442 k-mers with seed ATAGTTACATGCA,   	#1
Align and cluster 663 k-mers with seed TGTTGTTGTGTGC,   	#2
Align and cluster 158 k-mers with seed TCGACACGGCAGT,   	#3
Align and cluster 82 k-mers with seed TCGGTGAACACTG,   	#4

Merging redundant motifs ...

------------------------- k = 13 ----------------------------

#0	k=13	KSM= GTGCTGCTACTGC 	1.31, 184, kAUC=34.0	PWM= GTGCTGCTACTGC	12.15, 194, pAUC=30.8
#1	k=13	KSM= TGTTGTTGTGTGC 	1.82, 115, kAUC=22.2	PWM= TGTTGTTGTGTGC	16.54, 106, pAUC=17.0
#2	k=13	KSM= ATAGTTACATGCA 	1.00, 88, kAUC=18.8	PWM= ATAGTTACATGCA	12.62, 72, pAUC=16.4
#3	k=13	KSM= TCGACACGGCAGT 	1.00, 51, kAUC=12.6	PWM= TCGACACGGCAGT	15.05, 55, pAUC=13.2
#4	k=13	KSM= TCGGTGAACACTG 	1.00, 24, kAUC=8.8	PWM= TCGGTGAACACTG	20.93, 25, pAUC=9.8

After merging:
#0	k=13	KSM= GTGCTGCTACTGC 	1.31, 184, kAUC=34.0	PWM= GTGCTGCTACTGC	12.15, 194, pAUC=30.8
#1	k=13	KSM= TGTTGTTGTGTGC 	1.82, 115, kAUC=22.2	PWM= TGTTGTTGTGTGC	16.54, 106, pAUC=17.0
#2	k=13	KSM= ATAGTTACATGCA 	1.00, 88, kAUC=18.8	PWM= ATAGTTACATGCA	12.62, 72, pAUC=16.4
#3	k=13	KSM= TCGACACGGCAGT 	1.00, 51, kAUC=12.6	PWM= TCGACACGGCAGT	15.05, 55, pAUC=13.2
#4	k=13	KSM= TCGGTGAACACTG 	1.00, 24, kAUC=8.8	PWM= TCGGTGAACACTG	20.93, 25, pAUC=9.8



----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=1476
k=14+1, gapped kmer=466, base kmer=647
k=14+2, gapped kmer=613, base kmer=641
k=14+3, gapped kmer=404, base kmer=510
k=14+4, gapped kmer=243, base kmer=401

------------------------- k = 14 ----------------------------

Total number of k-mers: n = 1476

Top 5 cluster center k-mers:
# k-mer/r.c.                 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGTGCTGCTACTGC/GCAGTAGCAGCACA	69	73	0	-24.0    	0	105.3	11.0	1158.7	56
TACATGCATAGTTA/TAACTATGCATGTA	47	48	0	-15.8    	6	62.4	7.0	436.5	47
ATAGTTACATGCAT/ATGCATGTAACTAT	44	45	0	-14.9    	10	62.2	7.0	435.5	43
TGTTGTGTGCTGCT/AGCAGCACACAACA	39	41	3	-10.5    	16	69.6	4.0	278.2	65
ACACGGCAGTAGCA/TGCTACTGCCGTGT	27	29	0	-9.8    	26	66.2	4.0	264.7	34

Align and cluster 606 k-mers with seed TGTGCTGCTACTGC,   	#0
Align and cluster 487 k-mers with seed TACATGCATAGTTA,   	#1
Align and cluster 555 k-mers with seed ATAGTTACATGCAT,   	#2
Align and cluster 808 k-mers with seed TGTTGTGTGCTGCT,   	#3
Align and cluster 466 k-mers with seed ACACGGCAGTAGCA,   	#4

Merging redundant motifs ...
150ms: Remove motif #1.
159ms: PWM TACATGCATAGTTA has been removed.

------------------------- k = 14 ----------------------------

#0	k=14	KSM= ACACGGCAGTAGCA 	1.10, 183, kAUC=32.1	PWM= AACACGCAGTAGCA	12.03, 172, pAUC=27.0
#1	k=14	KSM= TGTGCTGCTACTGC 	1.00, 125, kAUC=22.9	PWM= TGTGCTGCTACTGC	16.35, 124, pAUC=21.6
#2	k=14	KSM= TGTTGTGTGCTGCT 	1.20, 120, kAUC=22.2	PWM= TGTTGTGTGCTGCT	15.93, 120, pAUC=17.8
#3	k=14	KSM= ATAGTTACATGCAT 	1.00, 73, kAUC=16.4	PWM= ATAGTTACATGCAT	15.54, 73, pAUC=17.5
#4	k=14	KSM= TACATGCATAGTTA 	1.10, 71, kAUC=16.1	PWM= TACATGCATAGTTA	15.56, 69, pAUC=15.1

After merging:
#0	k=14	KSM= ACACGGCAGTAGCA 	1.10, 183, kAUC=32.1	PWM= AACACGCAGTAGCA	12.03, 172, pAUC=27.0
#2	k=14	KSM= TGTTGTGTGCTGCT 	1.20, 120, kAUC=22.2	PWM= TGTTGTGTGCTGCT	15.93, 120, pAUC=17.8
#3	k=14	KSM= ATAGTTACATGCAT 	1.00, 73, kAUC=16.4	PWM= ATAGTTACATGCAT	15.54, 73, pAUC=17.5



----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=1418
k=15+1, gapped kmer=450, base kmer=618
k=15+2, gapped kmer=555, base kmer=595
k=15+3, gapped kmer=328, base kmer=450
k=15+4, gapped kmer=194, base kmer=334

------------------------- k = 15 ----------------------------

Total number of k-mers: n = 1418

Top 5 cluster center k-mers:
# k-mer/r.c.                   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGCTGCTACTGCGTG/CACGCAGTAGCAGCA	54	58	0	-19.1    	0	95.4	12.0	1145.1	71
TACATGCATAGTTAC/GTAACTATGCATGTA	44	45	0	-14.9    	7	70.4	9.0	633.2	65
ATAGTTACATGCATA/TATGCATGTAACTAT	41	42	0	-13.9    	9	61.1	8.0	489.1	62
TGTTGTTGTGTGCTG/CAGCACACAACAACA	35	37	0	-12.3    	17	66.4	5.0	331.8	91
GCCCGTGGGATGTTG/CAACATCCCACGGGC	16	16	0	-5.8    	203	40.8	6.0	244.8	15

Align and cluster 453 k-mers with seed TGCTGCTACTGCGTG,   	#0
Align and cluster 392 k-mers with seed TACATGCATAGTTAC,   	#1
Align and cluster 427 k-mers with seed ATAGTTACATGCATA,   	#2
Align and cluster 680 k-mers with seed TGTTGTTGTGTGCTG,   	#3
Align and cluster 157 k-mers with seed GCCCGTGGGATGTTG,   	#4

Merging redundant motifs ...
35ms: PWM TACATGCATAGTTAC has been removed.

------------------------- k = 15 ----------------------------

#0	k=15	KSM= TGCTGCTACTGCGTG 	1.00, 125, kAUC=23.1	PWM= TGCTGCTACTGCGTG	18.08, 124, pAUC=21.1
#1	k=15	KSM= TGTTGTTGTGTGCTG 	1.10, 124, kAUC=23.0	PWM= TGTTGTTGTGTGCTG	16.81, 112, pAUC=18.0
#2	k=15	KSM= ATAGTTACATGCATA 	1.00, 72, kAUC=16.2	PWM= ATAGTTACATGCATA	14.94, 72, pAUC=15.9
#3	k=15	KSM= TACATGCATAGTTAC 	1.10, 71, kAUC=16.1	PWM= TACATGCATAGTTAC	18.37, 69, pAUC=16.3
#4	k=15	KSM= GCCCGTGGGATGTTG 	1.00, 17, kAUC=7.9	PWM= GCCCGTGGGATGTTG	26.00, 17, pAUC=7.0

After merging:
#0	k=15	KSM= TGCTGCTACTGCGTG 	1.00, 125, kAUC=23.1	PWM= TGCTGCTACTGCGTG	18.08, 124, pAUC=21.1
#1	k=15	KSM= TGTTGTTGTGTGCTG 	1.10, 124, kAUC=23.0	PWM= TGTTGTTGTGTGCTG	16.81, 112, pAUC=18.0
#2	k=15	KSM= ATAGTTACATGCATA 	1.00, 72, kAUC=16.2	PWM= ATAGTTACATGCATA	14.94, 72, pAUC=15.9
#4	k=15	KSM= GCCCGTGGGATGTTG 	1.00, 17, kAUC=7.9	PWM= GCCCGTGGGATGTTG	26.00, 17, pAUC=7.0



----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=1367
k=16+1, gapped kmer=425, base kmer=583
k=16+2, gapped kmer=468, base kmer=537
k=16+3, gapped kmer=274, base kmer=393
k=16+4, gapped kmer=170, base kmer=298

------------------------- k = 16 ----------------------------

Total number of k-mers: n = 1367

Top 5 cluster center k-mers:
# k-mer/r.c.                     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GTGCTGCTACTGCGTG/CACGCAGTAGCAGCAC	49	53	0	-17.4    	0	86.8	13.0	1127.8	66
TACATGCATAGTTACA/TGTAACTATGCATGTA	44	45	0	-14.9    	3	62.2	10.0	622.1	58
ATAGTTACATGCATAG/CTATGCATGTAACTAT	37	38	0	-12.6    	10	56.5	9.0	508.5	52
TGTTGTTGTGTGCTGC/GCAGCACACAACAACA	34	36	0	-12.0    	15	64.6	5.0	323.1	88
TGTGTGTTGTTGTGTG/CACACAACAACACACA	18	19	0	-6.7    	90	47.7	6.0	286.5	69

Align and cluster 513 k-mers with seed GTGCTGCTACTGCGTG,   	#0
Align and cluster 482 k-mers with seed TACATGCATAGTTACA,   	#1
Align and cluster 458 k-mers with seed ATAGTTACATGCATAG,   	#2
Align and cluster 698 k-mers with seed TGTTGTTGTGTGCTGC,   	#3
Align and cluster 558 k-mers with seed TGTGTGTTGTTGTGTG,   	#4

Merging redundant motifs ...
127ms: Remove motif #2.
137ms: PWM ATAGTTACATGCATAG has been removed.

------------------------- k = 16 ----------------------------

#0	k=16	KSM= GTGCTGCTACTGCGTG 	1.00, 125, kAUC=23.3	PWM= GTGCTGCTACTGCGTG	16.52, 125, pAUC=22.3
#1	k=16	KSM= TGTTGTTGTGTGCTGC 	1.00, 125, kAUC=23.2	PWM= TGTTGTTGTGTGCTGC	17.96, 115, pAUC=18.6
#2	k=16	KSM= TGTGTGTTGTTGTGTG 	1.10, 122, kAUC=23.1	PWM= TGTGTGTTGTTGTGTG	14.59, 88, pAUC=12.3
#3	k=16	KSM= TACATGCATAGTTACA 	1.00, 71, kAUC=16.1	PWM= TACATGCATAGTTACA	15.25, 71, pAUC=15.6
#4	k=16	KSM= ATAGTTACATGCATAG 	1.00, 68, kAUC=15.6	PWM= ATAGTTACATGCATAG	17.60, 61, pAUC=16.5

After merging:
#0	k=16	KSM= GTGCTGCTACTGCGTG 	1.00, 125, kAUC=23.3	PWM= GTGCTGCTACTGCGTG	16.52, 125, pAUC=22.3
#1	k=16	KSM= TGTTGTTGTGTGCTGC 	1.00, 125, kAUC=23.2	PWM= TGTTGTTGTGTGCTGC	17.96, 115, pAUC=18.6
#3	k=16	KSM= TACATGCATAGTTACA 	1.00, 71, kAUC=16.1	PWM= TACATGCATAGTTACA	15.25, 71, pAUC=15.6



----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=1299
k=17+1, gapped kmer=384, base kmer=540
k=17+2, gapped kmer=394, base kmer=483
k=17+3, gapped kmer=246, base kmer=351
k=17+4, gapped kmer=144, base kmer=271

------------------------- k = 17 ----------------------------

Total number of k-mers: n = 1299

Top 5 cluster center k-mers:
# k-mer/r.c.                       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGTGCTGCTACTGCGTG/CACGCAGTAGCAGCACA	46	49	0	-16.2    	0	75.4	13.0	980.1	63
ATAGTTACATGCATAGT/ACTATGCATGTAACTAT	37	38	0	-12.6    	4	55.8	10.0	558.2	46
TGTTGTTGTGTGCTGCT/AGCAGCACACAACAACA	31	33	0	-11.1    	14	60.5	5.0	302.6	88
GAATATTATCGGTGAAC/GTTCACCGATAATATTC	23	23	0	-7.9    	32	29.9	9.0	269.5	6
ACGAAGATGACCCGTTC/GAACGGGTCATCTTCGT	23	23	0	-7.9    	23	24.5	9.0	220.1	6

Align and cluster 422 k-mers with seed TGTGCTGCTACTGCGTG,   	#0
Align and cluster 366 k-mers with seed ATAGTTACATGCATAGT,   	#1
Align and cluster 601 k-mers with seed TGTTGTTGTGTGCTGCT,   	#2
Align and cluster 42 k-mers with seed GAATATTATCGGTGAAC,   	#3
Align and cluster 44 k-mers with seed ACGAAGATGACCCGTTC,   	#4

Merging redundant motifs ...
133ms: Remove motif #1.

------------------------- k = 17 ----------------------------

#0	k=17	KSM= TGTGCTGCTACTGCGTG 	1.00, 124, kAUC=23.4	PWM= TGTGCTGCTACTGCGTG	18.71, 123, pAUC=21.8
#1	k=17	KSM= TGTTGTTGTGTGCTGCT 	1.00, 123, kAUC=23.3	PWM= TGTTGTTGTGTGCTGCT	16.04, 119, pAUC=19.1
#2	k=17	KSM= ATAGTTACATGCATAGT 	1.10, 63, kAUC=14.8	PWM= ATAGTTACATGCATAGT	18.06, 62, pAUC=16.3
#3	k=17	KSM= ACGAAGATGACCCGTTC 	1.00, 25, kAUC=9.0	PWM= ACGAAGATGACCCGTTC	28.79, 25, pAUC=12.5
#4	k=17	KSM= GAATATTATCGGTGAAC 	1.10, 25, kAUC=9.0	PWM= GAATATTATCGGTGAAC	27.20, 25, pAUC=11.9

After merging:
#0	k=17	KSM= TGTGCTGCTACTGCGTG 	1.00, 124, kAUC=23.4	PWM= TGTGCTGCTACTGCGTG	18.71, 123, pAUC=21.8
#2	k=17	KSM= ATAGTTACATGCATAGT 	1.10, 63, kAUC=14.8	PWM= ATAGTTACATGCATAGT	18.06, 62, pAUC=16.3
#3	k=17	KSM= ACGAAGATGACCCGTTC 	1.00, 25, kAUC=9.0	PWM= ACGAAGATGACCCGTTC	28.79, 25, pAUC=12.5
#4	k=17	KSM= GAATATTATCGGTGAAC 	1.10, 25, kAUC=9.0	PWM= GAATATTATCGGTGAAC	27.20, 25, pAUC=11.9



----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=1230
k=18+1, gapped kmer=340, base kmer=486
k=18+2, gapped kmer=342, base kmer=432
k=18+3, gapped kmer=219, base kmer=321
k=18+4, gapped kmer=128, base kmer=254

------------------------- k = 18 ----------------------------

Total number of k-mers: n = 1230

Top 5 cluster center k-mers:
# k-mer/r.c.                         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGTGTGCTGCTACTGCGT/ACGCAGTAGCAGCACACA	32	35	0	-11.7    	7	64.5	14.0	903.5	81
ATAGTTACATGCATAGTT/AACTATGCATGTAACTAT	37	38	0	-12.6    	0	56.5	11.0	621.5	64
TGTTGTTGTGTGCTGCTA/TAGCAGCACACAACAACA	30	32	0	-10.8    	11	60.1	6.0	360.8	105
GAATATTATCGGTGAACA/TGTTCACCGATAATATTC	23	23	0	-7.9    	28	36.2	9.0	325.9	9
AAGTTCTTACGAAGATGA/TCATCTTCGTAAGAACTT	22	22	0	-7.6    	40	25.7	9.0	231.2	8

Align and cluster 460 k-mers with seed TGTGTGCTGCTACTGCGT,   	#0
Align and cluster 386 k-mers with seed ATAGTTACATGCATAGTT,   	#1
Align and cluster 560 k-mers with seed TGTTGTTGTGTGCTGCTA,   	#2
Align and cluster 65 k-mers with seed GAATATTATCGGTGAACA,   	#3
Align and cluster 72 k-mers with seed AAGTTCTTACGAAGATGA,   	#4

Merging redundant motifs ...
187ms: Remove motif #1.

------------------------- k = 18 ----------------------------

#0	k=18	KSM= TGTGTGCTGCTACTGCGT 	1.00, 120, kAUC=23.0	PWM= TGTGTGCTGCTACTGCGT	20.72, 119, pAUC=20.9
#1	k=18	KSM= TGTTGTTGTGTGCTGCTA 	1.00, 120, kAUC=22.8	PWM= TGTTGTTGTGTGCTGCTA	18.36, 119, pAUC=19.6
#2	k=18	KSM= ATAGTTACATGCATAGTT 	1.00, 68, kAUC=15.6	PWM= ATAGTTACATGCATAGTT	18.81, 60, pAUC=16.8
#3	k=18	KSM= GAATATTATCGGTGAACA 	1.00, 25, kAUC=9.0	PWM= GAATATTATCGGTGAACA	28.90, 25, pAUC=12.8
#4	k=18	KSM= AAGTTCTTACGAAGATGA 	1.20, 25, kAUC=9.0	PWM= AAGTTCTTACGAAGATGA	26.05, 25, pAUC=11.8

After merging:
#0	k=18	KSM= TGTGTGCTGCTACTGCGT 	1.00, 120, kAUC=23.0	PWM= TGTGTGCTGCTACTGCGT	20.72, 119, pAUC=20.9
#2	k=18	KSM= ATAGTTACATGCATAGTT 	1.00, 68, kAUC=15.6	PWM= ATAGTTACATGCATAGTT	18.81, 60, pAUC=16.8
#3	k=18	KSM= GAATATTATCGGTGAACA 	1.00, 25, kAUC=9.0	PWM= GAATATTATCGGTGAACA	28.90, 25, pAUC=12.8
#4	k=18	KSM= AAGTTCTTACGAAGATGA 	1.20, 25, kAUC=9.0	PWM= AAGTTCTTACGAAGATGA	26.05, 25, pAUC=11.8



----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=1153
k=19+1, gapped kmer=304, base kmer=441
k=19+2, gapped kmer=299, base kmer=393
k=19+3, gapped kmer=193, base kmer=280
k=19+4, gapped kmer=118, base kmer=226

------------------------- k = 19 ----------------------------

Total number of k-mers: n = 1153

Top 5 cluster center k-mers:
# k-mer/r.c.                           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGTTGTGTGCTGCTACTGC/GCAGTAGCAGCACACAACA	34	36	0	-12.0    	1	59.7	15.0	895.5	76
ATAGTTACATGCATAGTTA/TAACTATGCATGTAACTAT	36	37	0	-12.3    	0	54.1	11.0	594.7	56
ATATTATCGGTGAACACTG/CAGTGTTCACCGATAATAT	23	23	0	-7.9    	18	29.9	9.0	269.5	8
ACGAAGATGACCCGTTCGT/ACGAACGGGTCATCTTCGT	23	23	0	-7.9    	16	24.5	10.0	244.5	8
GTGAACACTGCTACAAGTT/AACTTGTAGCAGTGTTCAC	24	24	0	-8.3    	12	24.5	9.0	220.5	10

Align and cluster 404 k-mers with seed TGTTGTGTGCTGCTACTGC,   	#0
Align and cluster 364 k-mers with seed ATAGTTACATGCATAGTTA,   	#1
Align and cluster 30 k-mers with seed ATATTATCGGTGAACACTG,   	#2
Align and cluster 31 k-mers with seed ACGAAGATGACCCGTTCGT,   	#3
Align and cluster 50 k-mers with seed GTGAACACTGCTACAAGTT,   	#4

Merging redundant motifs ...
42ms: PWM ATATTATCGGTGAACACTG has been removed.

------------------------- k = 19 ----------------------------

#0	k=19	KSM= TGTTGTGTGCTGCTACTGC 	1.00, 115, kAUC=22.3	PWM= TGTTGTGTGCTGCTACTGC	21.10, 116, pAUC=20.4
#1	k=19	KSM= ATAGTTACATGCATAGTTA 	1.00, 60, kAUC=14.3	PWM= ATAGTTACATGCATAGTTA	20.70, 60, pAUC=17.1
#2	k=19	KSM= GTGAACACTGCTACAAGTT 	1.00, 25, kAUC=9.0	PWM= GTGAACACTGCTACAAGTT	31.75, 25, pAUC=7.1
#3	k=19	KSM= ATATTATCGGTGAACACTG 	3.32, 23, kAUC=8.6	PWM= ATATTATCGGTGAACACTG	31.12, 25, pAUC=10.7
#4	k=19	KSM= ACGAAGATGACCCGTTCGT 	1.00, 23, kAUC=8.6	PWM= ACGAAGATGACCCGTTCGT	32.71, 25, pAUC=10.0

After merging:
#0	k=19	KSM= TGTTGTGTGCTGCTACTGC 	1.00, 115, kAUC=22.3	PWM= TGTTGTGTGCTGCTACTGC	21.10, 116, pAUC=20.4
#1	k=19	KSM= ATAGTTACATGCATAGTTA 	1.00, 60, kAUC=14.3	PWM= ATAGTTACATGCATAGTTA	20.70, 60, pAUC=17.1
#2	k=19	KSM= GTGAACACTGCTACAAGTT 	1.00, 25, kAUC=9.0	PWM= GTGAACACTGCTACAAGTT	31.75, 25, pAUC=7.1
#4	k=19	KSM= ACGAAGATGACCCGTTCGT 	1.00, 23, kAUC=8.6	PWM= ACGAAGATGACCCGTTCGT	32.71, 25, pAUC=10.0



----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=1092
k=20+1, gapped kmer=273, base kmer=402
k=20+2, gapped kmer=265, base kmer=352
k=20+3, gapped kmer=173, base kmer=251
k=20+4, gapped kmer=103, base kmer=200

------------------------- k = 20 ----------------------------

Total number of k-mers: n = 1092

Top 5 cluster center k-mers:
# k-mer/r.c.                             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TGTTGTTGTGTGCTGCTACT/AGTAGCAGCACACAACAACA	30	32	0	-10.8    	2	53.7	16.0	858.7	83
ATAGTTACATGCATAGTTAC/GTAACTATGCATGTAACTAT	34	35	0	-11.7    	0	45.8	13.0	595.7	53
TTGTGTGCTGCTACTGCGTG/CACGCAGTAGCAGCACACAA	27	29	0	-9.8    	4	50.8	6.0	304.8	62
TACGAAGATGACCCGTTCGT/ACGAACGGGTCATCTTCGTA	23	23	0	-7.9    	21	23.0	10.0	230.0	7
CGGTGAACACTGCTACAAGT/ACTTGTAGCAGTGTTCACCG	23	23	0	-7.9    	16	24.0	9.0	215.8	9

Align and cluster 410 k-mers with seed TGTTGTTGTGTGCTGCTACT,   	#0
Align and cluster 427 k-mers with seed ATAGTTACATGCATAGTTAC,   	#1
Align and cluster 361 k-mers with seed TTGTGTGCTGCTACTGCGTG,   	#2
Align and cluster 39 k-mers with seed TACGAAGATGACCCGTTCGT,   	#3
Align and cluster 88 k-mers with seed CGGTGAACACTGCTACAAGT,   	#4

Merging redundant motifs ...
132ms: Remove motif #1.

------------------------- k = 20 ----------------------------

#0	k=20	KSM= TTGTGTGCTGCTACTGCGTG 	1.00, 111, kAUC=21.8	PWM= TTGTGTGCTGCTACTGCGTG	23.64, 118, pAUC=20.4
#1	k=20	KSM= TGTTGTTGTGTGCTGCTACT 	1.00, 110, kAUC=21.6	PWM= TGTTGTTGTGTGCTGCTACT	24.93, 112, pAUC=19.8
#2	k=20	KSM= ATAGTTACATGCATAGTTAC 	1.00, 58, kAUC=14.0	PWM= ATAGTTACATGCATAGTTAC	22.26, 60, pAUC=17.0
#3	k=20	KSM= CGGTGAACACTGCTACAAGT 	1.00, 25, kAUC=9.0	PWM= CGGTGAACACTGCTACAAGT	34.40, 25, pAUC=7.1
#4	k=20	KSM= TACGAAGATGACCCGTTCGT 	1.00, 24, kAUC=8.8	PWM= TACGAAGATGACCCGTTCGT	34.41, 25, pAUC=8.7

After merging:
#0	k=20	KSM= TTGTGTGCTGCTACTGCGTG 	1.00, 111, kAUC=21.8	PWM= TTGTGTGCTGCTACTGCGTG	22.16, 119, pAUC=20.4
#2	k=20	KSM= ATAGTTACATGCATAGTTAC 	1.00, 58, kAUC=14.0	PWM= ATAGTTACATGCATAGTTAC	22.26, 60, pAUC=17.0
#3	k=20	KSM= CGGTGAACACTGCTACAAGT 	1.00, 25, kAUC=9.0	PWM= CGGTGAACACTGCTACAAGT	34.40, 25, pAUC=7.1
#4	k=20	KSM= TACGAAGATGACCCGTTCGT 	1.00, 24, kAUC=8.8	PWM= TACGAAGATGACCCGTTCGT	34.41, 25, pAUC=8.7



------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=6	KSM= ACATGC 	1.71, 261, kAUC=55.3	PWM= ACATGC	6.63, 355, pAUC=14.1
#1	k=7	KSM= TGCTACT 	2.88, 187, kAUC=54.5	PWM= TGCTACT	8.11, 245, pAUC=26.4
#2	k=8	KSM= GCTGCTAC 	1.48, 261, kAUC=52.0	PWM= GCTGCTAC	9.44, 228, pAUC=29.3
#3	k=7	KSM= ACTGCGT 	1.31, 276, kAUC=48.7	PWM= ACTGCGT	8.11, 186, pAUC=21.6
#4	k=7	KSM= GTTGTTG 	3.23, 228, kAUC=48.3	PWM= GTTGTTG	6.79, 217, pAUC=25.4
#5	k=9	KSM= TACTGCGTG 	1.41, 232, kAUC=48.1	PWM= TACTGCGTG	9.26, 223, pAUC=31.0
#6	k=6	KSM= TGCTAC 	2.49, 373, kAUC=47.5	PWM= TGCTAC	6.51, 306, pAUC=23.2
#7	k=8	KSM= TACTGCGT 	2.68, 150, kAUC=44.6	PWM= TACTGCGT	9.03, 177, pAUC=24.4
#8	k=8	KSM= GTTACATG 	2.62, 115, kAUC=41.5	PWM= GTTACGTG	8.20, 214, pAUC=16.2
#9	k=6	KSM= CTGCGT 	2.80, 253, kAUC=40.9	PWM= CTGCGT	6.76, 231, pAUC=21.0
#10	k=12	KSM= TAGTTACATGCA 	1.31, 215, kAUC=40.7	PWM= TAGTTACATGCA	10.15, 170, pAUC=25.9
#11	k=7	KSM= TACATGC 	2.67, 159, kAUC=38.0	PWM= TACATGC	8.08, 167, pAUC=17.5
#12	k=7	KSM= GATAGTT 	2.35, 141, kAUC=37.6	PWM= GATAGTT	7.63, 107, pAUC=17.3
#13	k=6	KSM= TTGTTG 	2.40, 319, kAUC=37.0	PWM= TTGTTG	6.78, 236, pAUC=14.4
#14	k=10	KSM= AGTTACATGC 	2.78, 110, kAUC=35.8	PWM= AGATACATGC	8.78, 139, pAUC=23.0
#15	k=10	KSM= TGCTGCTACT 	1.32, 187, kAUC=35.1	PWM= TGCTGCTACT	11.79, 187, pAUC=26.4
#16	k=13	KSM= GTGCTGCTACTGC 	1.31, 184, kAUC=34.0	PWM= GTGCTGCTACTGC	12.15, 194, pAUC=30.8
#17	k=14	KSM= ACACGGCAGTAGCA 	1.10, 183, kAUC=32.1	PWM= AACACGCAGTAGCA	12.03, 172, pAUC=27.0
#18	k=10	KSM= TGTTGTTGTG 	2.04, 137, kAUC=30.2	PWM= TGTTGTTGTG	12.30, 123, pAUC=16.4
#19	k=11	KSM= TGCTGCTACTG 	1.78, 152, kAUC=28.9	PWM= TGCTGCTACTG	13.58, 135, pAUC=25.0
#20	k=9	KSM= TACATGCAT 	1.44, 123, kAUC=28.9	PWM= TACATGCAT	8.30, 96, pAUC=13.8
#21	k=12	KSM= TGCTGCTACTGC 	1.31, 146, kAUC=26.1	PWM= TGCTGCTACTGC	12.94, 132, pAUC=24.8
#22	k=17	KSM= TGTGCTGCTACTGCGTG 	1.00, 124, kAUC=23.4	PWM= TGTGCTGCTACTGCGTG	18.71, 123, pAUC=21.8
#23	k=16	KSM= GTGCTGCTACTGCGTG 	1.00, 125, kAUC=23.3	PWM= GTGCTGCTACTGCGTG	16.52, 125, pAUC=22.3
#24	k=16	KSM= TGTTGTTGTGTGCTGC 	1.00, 125, kAUC=23.2	PWM= TGTTGTTGTGTGCTGC	17.96, 115, pAUC=18.6
#25	k=15	KSM= TGCTGCTACTGCGTG 	1.00, 125, kAUC=23.1	PWM= TGCTGCTACTGCGTG	18.08, 124, pAUC=21.1
#26	k=18	KSM= TGTGTGCTGCTACTGCGT 	1.00, 120, kAUC=23.0	PWM= TGTGTGCTGCTACTGCGT	20.72, 119, pAUC=20.9
#27	k=15	KSM= TGTTGTTGTGTGCTG 	1.10, 124, kAUC=23.0	PWM= TGTTGTTGTGTGCTG	16.81, 112, pAUC=18.0
#28	k=19	KSM= TGTTGTGTGCTGCTACTGC 	1.00, 115, kAUC=22.3	PWM= TGTTGTGTGCTGCTACTGC	21.10, 116, pAUC=20.4
#29	k=11	KSM= TGTTGTTGTGT 	1.47, 109, kAUC=22.2	PWM= TGTTGTTGTGT	13.24, 109, pAUC=15.6
#30	k=14	KSM= TGTTGTGTGCTGCT 	1.20, 120, kAUC=22.2	PWM= TGTTGTGTGCTGCT	15.93, 120, pAUC=17.8
#31	k=13	KSM= TGTTGTTGTGTGC 	1.82, 115, kAUC=22.2	PWM= TGTTGTTGTGTGC	16.54, 106, pAUC=17.0
#32	k=20	KSM= TTGTGTGCTGCTACTGCGTG 	1.00, 111, kAUC=21.8	PWM= TTGTGTGCTGCTACTGCGTG	22.16, 119, pAUC=20.4
#33	k=11	KSM= GATAGTTACAT 	1.00, 107, kAUC=21.7	PWM= GATAGTTACAT	9.93, 75, pAUC=15.5
#34	k=11	KSM= TACATGCATAG 	1.00, 96, kAUC=19.6	PWM= TACATGCATAG	10.28, 91, pAUC=16.5
#35	k=13	KSM= ATAGTTACATGCA 	1.00, 88, kAUC=18.8	PWM= ATAGTTACATGCA	12.62, 72, pAUC=16.4
#36	k=9	KSM= TCGACACGG 	1.00, 88, kAUC=18.2	PWM= TCGACACGG	9.72, 73, pAUC=13.1
#37	k=14	KSM= ATAGTTACATGCAT 	1.00, 73, kAUC=16.4	PWM= ATAGTTACATGCAT	15.54, 73, pAUC=17.5
#38	k=15	KSM= ATAGTTACATGCATA 	1.00, 72, kAUC=16.2	PWM= ATAGTTACATGCATA	14.94, 72, pAUC=15.9
#39	k=16	KSM= TACATGCATAGTTACA 	1.00, 71, kAUC=16.1	PWM= TACATGCATAGTTACA	15.25, 71, pAUC=15.6
#40	k=18	KSM= ATAGTTACATGCATAGTT 	1.00, 68, kAUC=15.6	PWM= ATAGTTACATGCATAGTT	18.81, 60, pAUC=16.8
#41	k=17	KSM= ATAGTTACATGCATAGT 	1.10, 63, kAUC=14.8	PWM= ATAGTTACATGCATAGT	18.06, 62, pAUC=16.3
#42	k=19	KSM= ATAGTTACATGCATAGTTA 	1.00, 60, kAUC=14.3	PWM= ATAGTTACATGCATAGTTA	20.70, 60, pAUC=17.1
#43	k=20	KSM= ATAGTTACATGCATAGTTAC 	1.00, 58, kAUC=14.0	PWM= ATAGTTACATGCATAGTTAC	22.26, 60, pAUC=17.0
#44	k=13	KSM= TCGACACGGCAGT 	1.00, 51, kAUC=12.6	PWM= TCGACACGGCAGT	15.05, 55, pAUC=13.2
#45	k=18	KSM= GAATATTATCGGTGAACA 	1.00, 25, kAUC=9.0	PWM= GAATATTATCGGTGAACA	28.90, 25, pAUC=12.8
#46	k=17	KSM= ACGAAGATGACCCGTTC 	1.00, 25, kAUC=9.0	PWM= ACGAAGATGACCCGTTC	28.79, 25, pAUC=12.5
#47	k=17	KSM= GAATATTATCGGTGAAC 	1.10, 25, kAUC=9.0	PWM= GAATATTATCGGTGAAC	27.20, 25, pAUC=11.9
#48	k=18	KSM= AAGTTCTTACGAAGATGA 	1.20, 25, kAUC=9.0	PWM= AAGTTCTTACGAAGATGA	26.05, 25, pAUC=11.8
#49	k=20	KSM= CGGTGAACACTGCTACAAGT 	1.00, 25, kAUC=9.0	PWM= CGGTGAACACTGCTACAAGT	34.40, 25, pAUC=7.1
#50	k=19	KSM= GTGAACACTGCTACAAGTT 	1.00, 25, kAUC=9.0	PWM= GTGAACACTGCTACAAGTT	31.75, 25, pAUC=7.1
#51	k=13	KSM= TCGGTGAACACTG 	1.00, 24, kAUC=8.8	PWM= TCGGTGAACACTG	20.93, 25, pAUC=9.8
#52	k=20	KSM= TACGAAGATGACCCGTTCGT 	1.00, 24, kAUC=8.8	PWM= TACGAAGATGACCCGTTCGT	34.41, 25, pAUC=8.7
#53	k=19	KSM= ACGAAGATGACCCGTTCGT 	1.00, 23, kAUC=8.6	PWM= ACGAAGATGACCCGTTCGT	32.71, 25, pAUC=10.0
#54	k=15	KSM= GCCCGTGGGATGTTG 	1.00, 17, kAUC=7.9	PWM= GCCCGTGGGATGTTG	26.00, 17, pAUC=7.0

588ms: Remove motif #7.
747ms: Remove motif #8.
861ms: Remove motif #10.
1.0s: Remove motif #11.
1.3s: Remove motif #14.
1.4s: Remove motif #17.
1.6s: Remove motif #20.
1.7s: Remove motif #34.
1.8s: PWM ATAGTTACATGCA has been removed.
1.9s: Remove motif #36.
1.9s: PWM ATAGTTACATGCAT has been removed.
1.9s: PWM ATAGTTACATGCATA has been removed.
1.9s: PWM ATAGTTACATGCATAGTT has been removed.
1.9s: PWM ATAGTTACATGCATAGT has been removed.
1.9s: PWM ATAGTTACATGCATAGTTA has been removed.
1.9s: PWM ATAGTTACATGCATAGTTAC has been removed.
2.1s: Remove motif #44.
2.2s: Remove motif #2.
2.3s: Remove motif #6.
2.4s: Remove motif #15.
2.5s: Remove motif #16.
2.7s: Remove motif #19.
2.8s: Remove motif #21.
2.9s: Remove motif #22.
3.1s: Remove motif #23.
3.2s: Remove motif #25.
3.3s: Remove motif #26.
3.4s: Remove motif #32.
3.5s: PWM CGGTGAACACTGCTACAAGT has been removed.
3.5s: PWM GTGAACACTGCTACAAGTT has been removed.
3.6s: Remove motif #5.
3.7s: Remove motif #9.
3.8s: Remove motif #13.
3.9s: Remove motif #18.
4.0s: Remove motif #27.
4.1s: Remove motif #29.
4.2s: Remove motif #31.
4.3s: Remove motif #33.
4.4s: Remove motif #39.
4.5s: Remove motif #28.
4.7s: Remove motif #30.
4.7s: PWM GAATATTATCGGTGAAC has been removed.
4.7s: PWM TCGGTGAACACTG has been removed.
4.7s: PWM AAGTTCTTACGAAGATGA has been removed.
4.7s: PWM TACGAAGATGACCCGTTCGT has been removed.
4.7s: PWM ACGAAGATGACCCGTTCGT has been removed.

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=6	KSM= ACATGC 	1.71, 261, kAUC=55.3	PWM= ACATGC	6.63, 355, pAUC=14.1
#1	k=7	KSM= TGCTACT 	2.88, 187, kAUC=54.5	PWM= TGCTACT	8.11, 245, pAUC=26.4
#2	k=8	KSM= GCTGCTAC 	1.48, 261, kAUC=52.0	PWM= GCTGCTAC	9.44, 228, pAUC=29.3
#3	k=7	KSM= ACTGCGT 	1.31, 276, kAUC=48.7	PWM= ACTGCGT	8.11, 186, pAUC=21.6
#4	k=7	KSM= GTTGTTG 	3.23, 228, kAUC=48.3	PWM= GTTGTTG	6.79, 217, pAUC=25.4
#5	k=9	KSM= TACTGCGTG 	1.41, 232, kAUC=48.1	PWM= TACTGCGTG	9.26, 223, pAUC=31.0
#6	k=6	KSM= TGCTAC 	2.49, 373, kAUC=47.5	PWM= TGCTAC	6.51, 306, pAUC=23.2
#7	k=8	KSM= TACTGCGT 	2.68, 150, kAUC=44.6	PWM= TACTGCGT	9.03, 177, pAUC=24.4
#8	k=8	KSM= GTTACATG 	2.62, 115, kAUC=41.5	PWM= GTTACGTG	8.20, 214, pAUC=16.2
#9	k=6	KSM= CTGCGT 	2.80, 253, kAUC=40.9	PWM= CTGCGT	6.76, 231, pAUC=21.0
#10	k=12	KSM= TAGTTACATGCA 	1.31, 215, kAUC=40.7	PWM= TAGTTACATGCA	10.15, 170, pAUC=25.9
#11	k=7	KSM= TACATGC 	2.67, 159, kAUC=38.0	PWM= TACATGC	8.08, 167, pAUC=17.5
#12	k=7	KSM= GATAGTT 	2.35, 141, kAUC=37.6	PWM= GATAGTT	7.63, 107, pAUC=17.3
#13	k=6	KSM= TTGTTG 	2.40, 319, kAUC=37.0	PWM= TTGTTG	6.78, 236, pAUC=14.4
#14	k=10	KSM= AGTTACATGC 	2.78, 110, kAUC=35.8	PWM= AGATACATGC	8.78, 139, pAUC=23.0
#15	k=10	KSM= TGCTGCTACT 	1.32, 187, kAUC=35.1	PWM= TGCTGCTACT	11.79, 187, pAUC=26.4
#16	k=13	KSM= GTGCTGCTACTGC 	1.31, 184, kAUC=34.0	PWM= GTGCTGCTACTGC	12.15, 194, pAUC=30.8
#17	k=14	KSM= ACACGGCAGTAGCA 	1.10, 183, kAUC=32.1	PWM= AACACGCAGTAGCA	12.03, 172, pAUC=27.0
#18	k=10	KSM= TGTTGTTGTG 	2.04, 137, kAUC=30.2	PWM= TGTTGTTGTG	12.30, 123, pAUC=16.4
#19	k=11	KSM= TGCTGCTACTG 	1.78, 152, kAUC=28.9	PWM= TGCTGCTACTG	13.58, 135, pAUC=25.0
#20	k=9	KSM= TACATGCAT 	1.44, 123, kAUC=28.9	PWM= TACATGCAT	8.30, 96, pAUC=13.8
#21	k=12	KSM= TGCTGCTACTGC 	1.31, 146, kAUC=26.1	PWM= TGCTGCTACTGC	12.94, 132, pAUC=24.8
#22	k=17	KSM= TGTGCTGCTACTGCGTG 	1.00, 124, kAUC=23.4	PWM= TGTGCTGCTACTGCGTG	18.71, 123, pAUC=21.8
#23	k=16	KSM= GTGCTGCTACTGCGTG 	1.00, 125, kAUC=23.3	PWM= GTGCTGCTACTGCGTG	16.52, 125, pAUC=22.3
#24	k=16	KSM= TGTTGTTGTGTGCTGC 	1.00, 125, kAUC=23.2	PWM= TGTTGTTGTGTGCTGC	17.96, 115, pAUC=18.6
#25	k=15	KSM= TGCTGCTACTGCGTG 	1.00, 125, kAUC=23.1	PWM= TGCTGCTACTGCGTG	18.08, 124, pAUC=21.1
#26	k=18	KSM= TGTGTGCTGCTACTGCGT 	1.00, 120, kAUC=23.0	PWM= TGTGTGCTGCTACTGCGT	20.72, 119, pAUC=20.9
#27	k=15	KSM= TGTTGTTGTGTGCTG 	1.10, 124, kAUC=23.0	PWM= TGTTGTTGTGTGCTG	16.81, 112, pAUC=18.0
#28	k=19	KSM= TGTTGTGTGCTGCTACTGC 	1.00, 115, kAUC=22.3	PWM= TGTTGTGTGCTGCTACTGC	21.10, 116, pAUC=20.4
#29	k=11	KSM= TGTTGTTGTGT 	1.47, 109, kAUC=22.2	PWM= TGTTGTTGTGT	13.24, 109, pAUC=15.6
#30	k=14	KSM= TGTTGTGTGCTGCT 	1.20, 120, kAUC=22.2	PWM= TGTTGTGTGCTGCT	15.93, 120, pAUC=17.8
#31	k=13	KSM= TGTTGTTGTGTGC 	1.82, 115, kAUC=22.2	PWM= TGTTGTTGTGTGC	16.54, 106, pAUC=17.0
#32	k=20	KSM= TTGTGTGCTGCTACTGCGTG 	1.00, 111, kAUC=21.8	PWM= TTGTGTGCTGCTACTGCGTG	22.16, 119, pAUC=20.4
#33	k=11	KSM= GATAGTTACAT 	1.00, 107, kAUC=21.7	PWM= GATAGTTACAT	9.93, 75, pAUC=15.5
#34	k=11	KSM= TACATGCATAG 	1.00, 96, kAUC=19.6	PWM= TACATGCATAG	10.28, 91, pAUC=16.5
#35	k=13	KSM= ATAGTTACATGCA 	1.00, 88, kAUC=18.8	PWM= ATAGTTACATGCA	12.62, 72, pAUC=16.4
#36	k=9	KSM= TCGACACGG 	1.00, 88, kAUC=18.2	PWM= TCGACACGG	9.72, 73, pAUC=13.1
#37	k=14	KSM= ATAGTTACATGCAT 	1.00, 73, kAUC=16.4	PWM= ATAGTTACATGCAT	15.54, 73, pAUC=17.5
#38	k=15	KSM= ATAGTTACATGCATA 	1.00, 72, kAUC=16.2	PWM= ATAGTTACATGCATA	14.94, 72, pAUC=15.9
#39	k=16	KSM= TACATGCATAGTTACA 	1.00, 71, kAUC=16.1	PWM= TACATGCATAGTTACA	15.25, 71, pAUC=15.6
#40	k=18	KSM= ATAGTTACATGCATAGTT 	1.00, 68, kAUC=15.6	PWM= ATAGTTACATGCATAGTT	18.81, 60, pAUC=16.8
#41	k=17	KSM= ATAGTTACATGCATAGT 	1.10, 63, kAUC=14.8	PWM= ATAGTTACATGCATAGT	18.06, 62, pAUC=16.3
#42	k=19	KSM= ATAGTTACATGCATAGTTA 	1.00, 60, kAUC=14.3	PWM= ATAGTTACATGCATAGTTA	20.70, 60, pAUC=17.1
#43	k=20	KSM= ATAGTTACATGCATAGTTAC 	1.00, 58, kAUC=14.0	PWM= ATAGTTACATGCATAGTTAC	22.26, 60, pAUC=17.0
#44	k=13	KSM= TCGACACGGCAGT 	1.00, 51, kAUC=12.6	PWM= TCGACACGGCAGT	15.05, 55, pAUC=13.2
#45	k=18	KSM= GAATATTATCGGTGAACA 	1.00, 25, kAUC=9.0	PWM= GAATATTATCGGTGAACA	28.90, 25, pAUC=12.8
#46	k=17	KSM= ACGAAGATGACCCGTTC 	1.00, 25, kAUC=9.0	PWM= ACGAAGATGACCCGTTC	28.79, 25, pAUC=12.5
#47	k=17	KSM= GAATATTATCGGTGAAC 	1.10, 25, kAUC=9.0	PWM= GAATATTATCGGTGAAC	27.20, 25, pAUC=11.9
#48	k=18	KSM= AAGTTCTTACGAAGATGA 	1.20, 25, kAUC=9.0	PWM= AAGTTCTTACGAAGATGA	26.05, 25, pAUC=11.8
#49	k=20	KSM= CGGTGAACACTGCTACAAGT 	1.00, 25, kAUC=9.0	PWM= CGGTGAACACTGCTACAAGT	34.40, 25, pAUC=7.1
#50	k=19	KSM= GTGAACACTGCTACAAGTT 	1.00, 25, kAUC=9.0	PWM= GTGAACACTGCTACAAGTT	31.75, 25, pAUC=7.1
#51	k=13	KSM= TCGGTGAACACTG 	1.00, 24, kAUC=8.8	PWM= TCGGTGAACACTG	20.93, 25, pAUC=9.8
#52	k=20	KSM= TACGAAGATGACCCGTTCGT 	1.00, 24, kAUC=8.8	PWM= TACGAAGATGACCCGTTCGT	34.41, 25, pAUC=8.7
#53	k=19	KSM= ACGAAGATGACCCGTTCGT 	1.00, 23, kAUC=8.6	PWM= ACGAAGATGACCCGTTCGT	32.71, 25, pAUC=10.0
#54	k=15	KSM= GCCCGTGGGATGTTG 	1.00, 17, kAUC=7.9	PWM= GCCCGTGGGATGTTG	26.00, 17, pAUC=7.0

------------------------- GEMout_2, k = ALL (after merging) -------------------------
#3	k=7	KSM= ACTGCGT 	3.23, 189, kAUC=60.1	PWM= ACTGCCT	8.30, 274, pAUC=29.0
#0	k=6	KSM= ACATGC 	1.31, 399, kAUC=58.5	PWM= ACATGC	6.12, 330, pAUC=14.1
#1	k=7	KSM= TGCTACT 	2.12, 264, kAUC=56.5	PWM= TGCTACT	8.74, 235, pAUC=26.0
#4	k=7	KSM= GTTGTTG 	1.47, 279, kAUC=52.1	PWM= GTTGTTG	9.38, 191, pAUC=24.2
#12	k=7	KSM= GATAGTT 	2.40, 131, kAUC=37.8	PWM= GATAGTT	7.65, 107, pAUC=17.1
#24	k=16	KSM= TGTTGTTGTGTGCTGC 	1.00, 125, kAUC=23.2	PWM= TGTTGTTGTGTGCTGC	18.10, 119, pAUC=18.7
#45	k=18	KSM= GAATATTATCGGTGAACA 	1.00, 25, kAUC=9.0	PWM= GAATATTATCGGTGAACA	28.90, 25, pAUC=12.8
#46	k=17	KSM= ACGAAGATGACCCGTTC 	1.00, 25, kAUC=9.0	PWM= ACGAAGATGACCCGTTC	28.79, 25, pAUC=12.5
#54	k=15	KSM= GCCCGTGGGATGTTG 	1.00, 17, kAUC=7.9	PWM= GCCCGTGGGATGTTG	26.00, 17, pAUC=7.0

5.0s: Finalizing 9 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_2 final motifs ----------------------------


GEMout_2 motif #0

KSM top k-mer: ACTGCGT/ACGCAGT, total 164 k-mers
KSM threshold: 3.23, 		hit=189+/0-, kAUC=60.1

    |
ACTGCCT
     G 
       
       
PWM threshold: 8.30/11.94, 	hit=274+/65-, pAUC=29.0
--------------------------------------------------------------

GEMout_2 motif #1

KSM top k-mer: ACATGC/GCATGT, total 145 k-mers
KSM threshold: 1.31, 		hit=399+/35-, kAUC=58.5

      |
ACATGC
   C  
      
      
PWM threshold: 6.12/9.06, 	hit=330+/224-, pAUC=14.1
--------------------------------------------------------------

GEMout_2 motif #2

KSM top k-mer: TGCTACT/AGTAGCA, total 228 k-mers
KSM threshold: 2.12, 		hit=264+/4-, kAUC=56.5

        |
TGCTACT
       
       
       
PWM threshold: 8.74/12.78, 	hit=235+/80-, pAUC=26.0
--------------------------------------------------------------

GEMout_2 motif #3

KSM top k-mer: GTTGTTG/CAACAAC, total 275 k-mers
KSM threshold: 1.47, 		hit=279+/12-, kAUC=52.1

       |
GTTGTTG
       
       
       
PWM threshold: 9.38/13.42, 	hit=191+/34-, pAUC=24.2
--------------------------------------------------------------

GEMout_2 motif #4

KSM top k-mer: GATAGTT/AACTATC, total 118 k-mers
KSM threshold: 2.40, 		hit=131+/0-, kAUC=37.8

                |
GATAGTT
       
       
       
PWM threshold: 7.65/12.49, 	hit=107+/20-, pAUC=17.1
--------------------------------------------------------------

GEMout_2 motif #5

KSM top k-mer: TGTTGTTGTGTGCTGC/GCAGCACACAACAACA, total 562 k-mers
KSM threshold: 1.00, 		hit=125+/0-, kAUC=23.2

                     |
TGTTGTTGTGTGCTGC
                
                
                
PWM threshold: 18.10/30.23, 	hit=119+/8-, pAUC=18.7
--------------------------------------------------------------

GEMout_2 motif #6

KSM top k-mer: GAATATTATCGGTGAACA/TGTTCACCGATAATATTC, total 11 k-mers
KSM threshold: 1.00, 		hit=25+/0-, kAUC=9.0

                           |
GAATATTATCGGTGAACA
                  
                  
                  
PWM threshold: 28.90/33.61, 	hit=25+/0-, pAUC=12.8
--------------------------------------------------------------

GEMout_2 motif #7

KSM top k-mer: ACGAAGATGACCCGTTC/GAACGGGTCATCTTCGT, total 10 k-mers
KSM threshold: 1.00, 		hit=25+/0-, kAUC=9.0

ACGAAGATGACCCGTTC
PWM threshold: 28.79/33.49, 	hit=25+/0-, pAUC=12.5
--------------------------------------------------------------

GEMout_2 motif #8

KSM top k-mer: GCCCGTGGGATGTTG/CAACATCCCACGGGC, total 8 k-mers
KSM threshold: 1.00, 		hit=17+/0-, kAUC=7.9

 |
GCCCGTGGGATGTTG
               
               
               
PWM threshold: 26.00/30.23, 	hit=17+/0-, pAUC=7.0
--------------------------------------------------------------



Finished! GEM analysis results are printed to:
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYB/vescaAlpina/GEMout/GEMout.GEM_events.txt
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYB/vescaAlpina/GEMout/GEMout.results.htm
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYB/vescaAlpina/GEMout/GEMout_outputs (folder with all other files)

----------------------------------

End time: 2021/12/03 17:49:14

Total running time: 9.2m

pigz 2.6
Analysis complete for INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  
Archive:  INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX7_BKDL210038288-1A-7_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/03 18:27:23

Loading data...
    Loading reads from: CS149-HAT9vescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/FVechrs/ --s 150000000 --expt CS149-HAT9vescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 15058283	HitCounts: 2.603525E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 15058283	HitCounts: 2.2961594E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 105 reads.


	IP: 22961594	Ctrl: 15832687	 IP/Ctrl: 1.45

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 0.23

The genome is segmented into 226406 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/226406	42.3s
1000	/226406	1.0m
10000	/226406	4.8m
100000	/226406	5.8m
200000	/226406	5.8m
226406	/226406	5.8m

Scaling condition , IP/Control = 0.77

---------------------------
Events discovered 
Significant:	24358
Insignificant:	238
Filtered:	1022

Total 16037 homotypic events (within 501bp of other events).

Finish binding event prediction: 5.8m

Refine read distribution from 2000 binding events. 


============================ Round 1 ============================

Running with 10 threads ...

100	/226406	4.1s
1000	/226406	28.1s
10000	/226406	39.1s
100000	/226406	2.5m
200000	/226406	4.3m
226406	/226406	5.3m
---------------------------
Events discovered 
Significant:	36737
Insignificant:	68
Filtered:	25

Total 7169 homotypic events (within 545bp of other events).

Finish binding event prediction: 5.3m

Refine read distribution from 2000 binding events. 

Loading genome sequences ...
Done, 2.1s
GC content=0.41

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.32. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=11
k=6+1, gapped kmer=15, base kmer=31
k=6+2, gapped kmer=29, base kmer=51
k=6+3, gapped kmer=22, base kmer=37
k=6+4, gapped kmer=4, base kmer=9

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 11

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATCAT/ATGATT	229	236	77	-26.4    	0	345.2	3.0	1035.7	7
GATTAC/GTAATC	49	49	13	-6.5    	10	127.0	2.0	253.9	1
TAATGA/TCATTA	118	122	67	-5.4    	6	238.4	1.0	238.4	3

Align and cluster 32 k-mers with seed AATCAT,   	#0
Align and cluster 18 k-mers with seed GATTAC,   	#1
Align and cluster 32 k-mers with seed TAATGA,   	#2

Merging redundant motifs ...
201ms: Remove motif #1.
328ms: Remove motif #2.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= AATCAT 	3.09, 223, kAUC=27.1	PWM= AATCAT	6.74, 458, pAUC=14.9
#1	k=6	KSM= TAATGA 	1.96, 245, kAUC=26.7	PWM= TAATCA	5.20, 473, pAUC=10.9
#2	k=6	KSM= GATTAC 	3.15, 74, kAUC=20.0	PWM= ATTA	4.09, 571, pAUC=6.1

After merging:
#0	k=6	KSM= AATCAT 	3.09, 207, kAUC=27.8	PWM= AATCAT	6.74, 458, pAUC=14.9



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=23
k=7+1, gapped kmer=29, base kmer=49
k=7+2, gapped kmer=20, base kmer=42
k=7+3, gapped kmer=2, base kmer=4
k=7+4, gapped kmer=0, base kmer=0

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 23

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TAATCAT/ATGATTA	102	104	27	-13.3    	2	220.4	3.0	661.1	17
AATGATT/AATCATT	126	132	30	-18.9

Align and cluster 53 k-mers with seed TAATCAT,   	#0
Align and cluster 41 k-mers with seed AATGATT,   	#1

Merging redundant motifs ...
121ms: Remove motif #1.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= TAATCAT 	2.43, 160, kAUC=32.2	PWM= CAATCAT	7.08, 432, pAUC=19.7
#1	k=7	KSM= AATGATT 	3.96, 93, kAUC=31.1	PWM= AATGATT	8.12, 352, pAUC=20.7

After merging:
#0	k=7	KSM= TAATCAT 	3.88, 82, kAUC=33.1	PWM= CAATCAT	7.41, 411, pAUC=19.7



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=20
k=8+1, gapped kmer=20, base kmer=34
k=8+2, gapped kmer=5, base kmer=6
k=8+3, gapped kmer=0, base kmer=0
k=8+4, gapped kmer=0, base kmer=0

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 20

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATGATTG/CAATCATT	53	57	7	-11.9    	0	120.6	4.0	482.2	12
AACAATCA/TGATTGTT	22	23	3	-5.4    	17	30.7	2.0	61.4	1
CTAATGAT/ATCATTAG	22	24	1	-7.6    	16	56.7	1.0	56.7	1

Align and cluster 43 k-mers with seed AATGATTG,   	#0
Align and cluster 34 k-mers with seed AACAATCA,   	#1
Align and cluster 37 k-mers with seed CTAATGAT,   	#2

Merging redundant motifs ...
79ms: Remove motif #1.
142ms: Remove motif #2.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= AATGATTG 	4.44, 50, kAUC=32.9	PWM= AATGATTG	7.97, 241, pAUC=24.5
#1	k=8	KSM= CTAATGAT 	3.63, 43, kAUC=28.3	PWM= CTAATGAT	8.43, 226, pAUC=17.3
#2	k=8	KSM= AACAATCA 	2.33, 107, kAUC=26.9	PWM= AACAATCA	8.83, 165, pAUC=15.8

After merging:
#0	k=8	KSM= AATGATTG 	3.82, 61, kAUC=35.0	PWM= AATGATTG	8.85, 238, pAUC=21.5



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=7
k=9+1, gapped kmer=2, base kmer=4
k=9+2, gapped kmer=0, base kmer=0
k=9+3, gapped kmer=0, base kmer=0
k=9+4, gapped kmer=0, base kmer=0

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 7

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATAATTGA/TCAATTATT	25	26	2	-7.1    	0	46.5	4.0	185.8	6

Align and cluster 8 k-mers with seed AATAATTGA,   	#0

------------------------- k = 9 ----------------------------

#0	k=9	KSM= AATAATTGA 	1.31, 69, kAUC=15.2	PWM= AATGATTGA	8.41, 196, pAUC=21.7



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=0
k=10+1, gapped kmer=0, base kmer=0
k=10+2, gapped kmer=0, base kmer=0
k=10+3, gapped kmer=0, base kmer=0
k=10+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=0
k=11+1, gapped kmer=0, base kmer=0
k=11+2, gapped kmer=0, base kmer=0
k=11+3, gapped kmer=0, base kmer=0
k=11+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=0
k=12+1, gapped kmer=0, base kmer=0
k=12+2, gapped kmer=0, base kmer=0
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=0
k=13+1, gapped kmer=0, base kmer=0
k=13+2, gapped kmer=0, base kmer=0
k=13+3, gapped kmer=0, base kmer=0
k=13+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=0
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=0, base kmer=0
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= AATGATTG 	3.82, 61, kAUC=35.0	PWM= AATGATTG	8.85, 238, pAUC=21.5
#1	k=7	KSM= TAATCAT 	3.88, 82, kAUC=33.1	PWM= CAATCAT	7.41, 411, pAUC=19.7
#2	k=6	KSM= AATCAT 	3.09, 207, kAUC=27.8	PWM= AATCAT	6.74, 458, pAUC=14.9
#3	k=9	KSM= AATAATTGA 	1.31, 69, kAUC=15.2	PWM= AATGATTGA	8.41, 196, pAUC=21.7

85ms: Remove motif #1.
157ms: Remove motif #2.
228ms: Remove motif #3.

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= AATGATTG 	3.82, 61, kAUC=35.0	PWM= AATGATTG	8.85, 238, pAUC=21.5
#1	k=7	KSM= TAATCAT 	3.88, 82, kAUC=33.1	PWM= CAATCAT	7.41, 411, pAUC=19.7
#2	k=6	KSM= AATCAT 	3.09, 207, kAUC=27.8	PWM= AATCAT	6.74, 458, pAUC=14.9
#3	k=9	KSM= AATAATTGA 	1.31, 69, kAUC=15.2	PWM= AATGATTGA	8.41, 196, pAUC=21.7

------------------------- GEMout_1, k = ALL (after merging) -------------------------
#0	k=8	KSM= AATGATTG 	4.08, 58, kAUC=37.1	PWM= AATGATTG	8.48, 309, pAUC=22.3

294ms: Finalizing 1 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_1 final motifs ----------------------------


GEMout_1 motif #0

KSM top k-mer: AATGATTG/CAATCATT, total 43 k-mers
KSM threshold: 4.08, 		hit=58+/0-, kAUC=37.1

      |
AATGATTG
       A
        
        
PWM threshold: 8.48/12.17, 	hit=309+/87-, pAUC=22.3
--------------------------------------------------------------



============================ Round 2 ============================
Running EM with motif positional prior ...

Running with 10 threads ...

100	/226406	1.2s
1000	/226406	31.2s
10000	/226406	39.2s
100000	/226406	2.2m
200000	/226406	4.0m
226406	/226406	5.0m
---------------------------
Events discovered 
Significant:	38850
Insignificant:	63
Filtered:	9

Total 11030 homotypic events (within 545bp of other events).

Finish binding event prediction: 5.0m

Refine read distribution from 2000 binding events. 

Percentage of significant events with a motif match, divided in 10 bins (~3885 events per bin):
39 43 41 38 39 38 38 37 37 38 

Percentage of filtered events with a motif match, divided in 9 bins (~1 events per bin):
0 0 0 0 0 0 0 0 0 

Percentage of insignificant events with a motif match, divided in 9 bins (~7 events per bin):
42 14 57 14 14 0 14 14 14 

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.32. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=14
k=6+1, gapped kmer=23, base kmer=37
k=6+2, gapped kmer=39, base kmer=65
k=6+3, gapped kmer=26, base kmer=58
k=6+4, gapped kmer=7, base kmer=14

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 14

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATCAT/ATGATT	302	307	103	-36.5    	0	411.7	5.0	2058.3	8
CTAATC/GATTAG	57	58	22	-5.2    	12	145.1	2.0	290.2	1
TAATTG/CAATTA	122	122	43	-11.4    	7	250.4	1.0	250.4	4
TTACCC/GGGTAA	17	17	1	-5.5    	13	17.0	3.0	51.0	1

Align and cluster 43 k-mers with seed AATCAT,   	#0
Align and cluster 31 k-mers with seed CTAATC,   	#1
Align and cluster 33 k-mers with seed TAATTG,   	#2
Align and cluster 5 k-mers with seed TTACCC,   	#3

Merging redundant motifs ...
181ms: Remove motif #1.
296ms: Remove motif #2.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= TAATTG 	2.66, 264, kAUC=32.8	PWM= TGATTG	6.50, 502, pAUC=13.8
#1	k=6	KSM= AATCAT 	1.97, 321, kAUC=29.0	PWM= AATCAT	7.15, 484, pAUC=14.7
#2	k=6	KSM= CTAATC 	3.10, 210, kAUC=24.8	PWM= CTAATC	6.94, 312, pAUC=11.3
#3	k=6	KSM= TTACCC 	2.42, 17, kAUC=7.5	PWM= TTACCC	7.85, 47, pAUC=7.7

After merging:
#0	k=6	KSM= TAATTG 	2.66, 264, kAUC=32.8	PWM= TGATTG	6.50, 502, pAUC=13.8
#3	k=6	KSM= TTACCC 	2.42, 17, kAUC=7.5	PWM= TTACCC	7.85, 47, pAUC=7.7



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=23
k=7+1, gapped kmer=42, base kmer=59
k=7+2, gapped kmer=33, base kmer=59
k=7+3, gapped kmer=2, base kmer=4
k=7+4, gapped kmer=0, base kmer=0

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 23

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATGATT/AATCATT	166	171	33	-28.2    	0	268.9	5.0	1344.7	9
TCAATCA/TGATTGA	73	77	32	-5.9    	8	169.2	2.0	338.5	8

Align and cluster 47 k-mers with seed AATGATT,   	#0
Align and cluster 29 k-mers with seed TCAATCA,   	#1

Merging redundant motifs ...
98ms: Remove motif #1.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= AATGATT 	2.80, 252, kAUC=37.9	PWM= AATGATT	7.39, 413, pAUC=22.7
#1	k=7	KSM= TCAATCA 	2.54, 182, kAUC=34.8	PWM= TCAATCA	7.64, 327, pAUC=12.3

After merging:
#0	k=7	KSM= AATGATT 	1.59, 343, kAUC=38.0	PWM= AATGATT	7.50, 451, pAUC=22.7



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=25
k=8+1, gapped kmer=33, base kmer=51
k=8+2, gapped kmer=5, base kmer=12
k=8+3, gapped kmer=0, base kmer=0
k=8+4, gapped kmer=0, base kmer=0

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 25

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATGATTG/CAATCATT	84	87	10	-18.2    	0	171.7	4.0	686.9	13
ATAATAAT/ATTATTAT	31	31	7	-5.0    	17	85.2	2.0	170.3	3
AACAATCA/TGATTGTT	31	33	5	-6.7    	15	64.5	1.0	64.5	1

Align and cluster 57 k-mers with seed AATGATTG,   	#0
Align and cluster 53 k-mers with seed ATAATAAT,   	#1
Align and cluster 46 k-mers with seed AACAATCA,   	#2

Merging redundant motifs ...
94ms: Remove motif #1.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= AACAATCA 	3.16, 147, kAUC=43.2	PWM= ATAATC	5.84, 519, pAUC=9.2
#1	k=8	KSM= AATGATTG 	4.80, 108, kAUC=40.1	PWM= AATGATTG	9.25, 276, pAUC=30.4
#2	k=8	KSM= ATAATAAT 	2.78, 108, kAUC=34.4	PWM= TAATAA	5.68, 411, pAUC=8.9

After merging:
#0	k=8	KSM= AACAATCA 	3.16, 147, kAUC=43.2	PWM= ATAATC	5.84, 519, pAUC=9.2
#2	k=8	KSM= ATAATAAT 	2.78, 108, kAUC=34.4	PWM= TAATAA	5.68, 411, pAUC=8.9



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=12
k=9+1, gapped kmer=7, base kmer=13
k=9+2, gapped kmer=0, base kmer=0
k=9+3, gapped kmer=0, base kmer=0
k=9+4, gapped kmer=0, base kmer=0

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 12

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TCAATCATT/AATGATTGA	33	34	4	-7.7    	0	76.2	4.0	305.0	11

Align and cluster 19 k-mers with seed TCAATCATT,   	#0

------------------------- k = 9 ----------------------------

#0	k=9	KSM= TCAATCATT 	2.89, 83, kAUC=29.6	PWM= CAATCAT	8.32, 230, pAUC=22.0



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=1
k=10+1, gapped kmer=0, base kmer=0
k=10+2, gapped kmer=0, base kmer=0
k=10+3, gapped kmer=0, base kmer=0
k=10+4, gapped kmer=0, base kmer=0

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 1

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AACAATCATT/AATGATTGTT	15	16	1	-5.2    	0	16.0	0.0	0.0	1

Align and cluster 1 k-mers with seed AACAATCATT,   	#0

------------------------- k = 10 ----------------------------

#0	k=10	KSM= AACAATCATT 	2.33, 16, kAUC=7.2	PWM= AACAATCATT	9.27, 268, pAUC=27.4



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=0
k=11+1, gapped kmer=0, base kmer=0
k=11+2, gapped kmer=0, base kmer=0
k=11+3, gapped kmer=0, base kmer=0
k=11+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=0
k=12+1, gapped kmer=0, base kmer=0
k=12+2, gapped kmer=0, base kmer=0
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=0
k=13+1, gapped kmer=0, base kmer=0
k=13+2, gapped kmer=0, base kmer=0
k=13+3, gapped kmer=0, base kmer=0
k=13+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=0
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=0, base kmer=0
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=8	KSM= AACAATCA 	3.16, 147, kAUC=43.2	PWM= ATAATC	5.84, 519, pAUC=9.2
#1	k=7	KSM= AATGATT 	1.59, 343, kAUC=38.0	PWM= AATGATT	7.50, 451, pAUC=22.7
#2	k=8	KSM= ATAATAAT 	2.78, 108, kAUC=34.4	PWM= TAATAA	5.68, 411, pAUC=8.9
#3	k=6	KSM= TAATTG 	2.66, 264, kAUC=32.8	PWM= TGATTG	6.50, 502, pAUC=13.8
#4	k=9	KSM= TCAATCATT 	2.89, 83, kAUC=29.6	PWM= CAATCAT	8.32, 230, pAUC=22.0
#5	k=6	KSM= TTACCC 	2.42, 17, kAUC=7.5	PWM= TTACCC	7.85, 47, pAUC=7.7
#6	k=10	KSM= AACAATCATT 	2.33, 16, kAUC=7.2	PWM= AACAATCATT	9.27, 268, pAUC=27.4

86ms: Remove motif #3.
167ms: Remove motif #6.
247ms: Remove motif #4.

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=8	KSM= AACAATCA 	3.16, 147, kAUC=43.2	PWM= ATAATC	5.84, 519, pAUC=9.2
#1	k=7	KSM= AATGATT 	1.59, 343, kAUC=38.0	PWM= AATGATT	7.50, 451, pAUC=22.7
#2	k=8	KSM= ATAATAAT 	2.78, 108, kAUC=34.4	PWM= TAATAA	5.68, 411, pAUC=8.9
#3	k=6	KSM= TAATTG 	2.66, 264, kAUC=32.8	PWM= TGATTG	6.50, 502, pAUC=13.8
#4	k=9	KSM= TCAATCATT 	2.89, 83, kAUC=29.6	PWM= CAATCAT	8.32, 230, pAUC=22.0
#5	k=6	KSM= TTACCC 	2.42, 17, kAUC=7.5	PWM= TTACCC	7.85, 47, pAUC=7.7
#6	k=10	KSM= AACAATCATT 	2.33, 16, kAUC=7.2	PWM= AACAATCATT	9.27, 268, pAUC=27.4

------------------------- GEMout_2, k = ALL (after merging) -------------------------
#0	k=8	KSM= AACAATCA 	3.16, 147, kAUC=43.2	PWM= ATAATC	5.84, 519, pAUC=9.2
#1	k=7	KSM= AATGATT 	1.59, 343, kAUC=38.0	PWM= AATGATT	7.50, 451, pAUC=22.7
#2	k=8	KSM= ATAATAAT 	2.78, 108, kAUC=34.4	PWM= TAATAA	5.68, 411, pAUC=8.9
#5	k=6	KSM= TTACCC 	2.42, 17, kAUC=7.5	PWM= TTACCC	7.85, 47, pAUC=7.7

282ms: Finalizing 4 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_2 final motifs ----------------------------


GEMout_2 motif #0

KSM top k-mer: AACAATCA/TGATTGTT, total 40 k-mers
KSM threshold: 3.16, 		hit=147+/4-, kAUC=43.2

    |
ATAATC
 CC  T
      
      
PWM threshold: 5.84/8.57, 	hit=519+/384-, pAUC=9.2
--------------------------------------------------------------

GEMout_2 motif #1

KSM top k-mer: AATGATT/AATCATT, total 45 k-mers
KSM threshold: 1.59, 		hit=343+/49-, kAUC=38.0

      |
AATGATT
       
       
       
PWM threshold: 7.50/11.45, 	hit=451+/192-, pAUC=22.7
--------------------------------------------------------------

GEMout_2 motif #2

KSM top k-mer: ATAATAAT/ATTATTAT, total 22 k-mers
KSM threshold: 2.78, 		hit=108+/3-, kAUC=34.4

        |
TAATAA
   AT 
      
      
PWM threshold: 5.68/8.38, 	hit=411+/347-, pAUC=8.9
--------------------------------------------------------------

GEMout_2 motif #3

KSM top k-mer: TTACCC/GGGTAA, total 1 k-mers
KSM threshold: 2.42, 		hit=17+/1-, kAUC=7.5

TTACCC
 G
PWM threshold: 7.85/10.67, 	hit=47+/32-, pAUC=7.7
--------------------------------------------------------------



Finished! GEM analysis results are printed to:
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9/vescaAlpina/GEMout/GEMout.GEM_events.txt
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9/vescaAlpina/GEMout/GEMout.results.htm
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9/vescaAlpina/GEMout/GEMout_outputs (folder with all other files)

----------------------------------

End time: 2021/12/03 18:47:21

Total running time: 20.0m

pigz 2.6
Archive:  INDEX5_BKDL210038288-1A-5_HG7TTCCX2_L8_2_val_2_fastqc.zip
pigz 2.6
Checking MD5 from JM11-INDEX42
Checking Fastaq lenghts from JM11-INDEX42
Doing Trim galore in CS142-CDFb/vescaAlpina from JM11-INDEX42
pigz 2.6
Analysis complete for INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  
Archive:  INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/04 15:10:05

Loading data...
    Loading reads from: CS142-CDFbvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/Fvechrs/ --s 150000000 --expt CS142-CDFbvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDFb/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 34042355	HitCounts: 5.3025792E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 34042355	HitCounts: 4.7276613E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 198 reads.


	IP: 47276613	Ctrl: 15832687	 IP/Ctrl: 2.99

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 1.00

The genome is segmented into 221325 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/221325	1.8m
1000	/221325	2.2m
10000	/221325	4.8m
100000	/221325	7.5m
221325	/221325	7.5m

Scaling condition , IP/Control = 2.92

---------------------------
Events discovered 
Significant:	6844
Insignificant:	467
Filtered:	319

Total 815 homotypic events (within 501bp of other events).

Finish binding event prediction: 7.6m

Refine read distribution from 2000 binding events. 


============================ Round 1 ============================

Running with 10 threads ...

100	/221325	1.2s
1000	/221325	5.2s
10000	/221325	21.2s
100000	/221325	3.3m
200000	/221325	6.3m
221325	/221325	7.6m
---------------------------
Events discovered 
Significant:	45017
Insignificant:	968
Filtered:	79

Total 9666 homotypic events (within 545bp of other events).

Finish binding event prediction: 7.7m

Refine read distribution from 2000 binding events. 

Loading genome sequences ...
pigz 2.6
Analysis complete for INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  
Archive:  INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/04 16:50:17

Loading data...
    Loading reads from: CS144-MYBbvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/Fvechrs/ --s 150000000 --expt CS144-MYBbvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYBb/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 38627642	HitCounts: 5.270847E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 38627642	HitCounts: 4.4444446E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 187 reads.


	IP: 44444446	Ctrl: 15832687	 IP/Ctrl: 2.81

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 2.53

The genome is segmented into 221604 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/221604	38.4s
1000	/221604	2.0m
10000	/221604	2.7m
100000	/221604	3.6m
221604	/221604	3.6m

Scaling condition , IP/Control = 2.27

---------------------------
Events discovered 
Significant:	90
Insignificant:	54
Filtered:	2

Total 57 homotypic events (within 501bp of other events).

Finish binding event prediction: 3.6m


============================ Round 1 ============================

Running with 10 threads ...

100	/221604	1.2s
1000	/221604	2.2s
10000	/221604	5.2s
100000	/221604	48.2s
200000	/221604	1.5m
221604	/221604	2.1m
---------------------------
Events discovered 
Significant:	129
Insignificant:	55
Filtered:	5

Total 83 homotypic events (within 501bp of other events).

Finish binding event prediction: 2.1m

Loading genome sequences ...
pigz 2.6
Analysis complete for INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  
Archive:  INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/04 18:03:07

Loading data...
    Loading reads from: CS149-HAT9bvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/Fvechrs/ --s 150000000 --expt CS149-HAT9bvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9b/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 21523559	HitCounts: 3.9209356E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 21523559	HitCounts: 3.537534E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 153 reads.


	IP: 35375340	Ctrl: 15832687	 IP/Ctrl: 2.23

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 0.21

The genome is segmented into 204058 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/204058	1.1m
1000	/204058	1.5m
10000	/204058	4.8m
100000	/204058	7.3m
204058	/204058	7.3m

Scaling condition , IP/Control = 1.10

---------------------------
Events discovered 
Significant:	33104
Insignificant:	372
Filtered:	2123

Total 20258 homotypic events (within 501bp of other events).

Finish binding event prediction: 7.3m

Refine read distribution from 2000 binding events. 


============================ Round 1 ============================

Running with 10 threads ...

100	/204058	1.1s
1000	/204058	2.1s
10000	/204058	19.1s
100000	/204058	3.0m
200000	/204058	6.3m
204058	/204058	6.9m
---------------------------
Events discovered 
Significant:	47767
Insignificant:	89
Filtered:	35

Total 11346 homotypic events (within 545bp of other events).

Finish binding event prediction: 6.9m

Refine read distribution from 2000 binding events. 

Loading genome sequences ...
Checking MD5 from JM11-INDEX42
Checking Fastaq lenghts from JM11-INDEX42
Doing Trim galore in CS142-CDFb/vescaAlpina from JM11-INDEX42
Trim galore finished,checking results...
Doing Bowtie2 in CS142-CDFb/vescaAlpina from JM11-INDEX42
Bowtie2 summary: b''
Bowtie2 errors if any: b'Time loading reference: 00:00:00\nTime loading forward index: 00:00:00\nTime loading mirror index: 00:00:00\nMultiseed full-index search: 00:22:43\n28744913 reads; of these:\n  28744913 (100.00%) were paired; of these:\n    3442465 (11.98%) aligned concordantly 0 times\n    18270039 (63.56%) aligned concordantly exactly 1 time\n    7032409 (24.46%) aligned concordantly >1 times\n    ----\n    3442465 pairs aligned concordantly 0 times; of these:\n      472687 (13.73%) aligned discordantly 1 time\n    ----\n    2969778 pairs aligned 0 times concordantly or discordantly; of these:\n      5939556 mates make up the pairs; of these:\n        4464034 (75.16%) aligned 0 times\n        788931 (13.28%) aligned exactly 1 time\n        686591 (11.56%) aligned >1 times\n92.24% overall alignment rate\nTime searching: 00:22:43\nOverall time: 00:22:43\n'
Checking MD5 from JM11-INDEX44
Checking Fastaq lenghts from JM11-INDEX44
Doing Trim galore in CS144-MYBb/vescaAlpina from JM11-INDEX44
Trim galore finished,checking results...
Doing Bowtie2 in CS144-MYBb/vescaAlpina from JM11-INDEX44
Bowtie2 summary: b''
Bowtie2 errors if any: b'Time loading reference: 00:00:00\nTime loading forward index: 00:00:01\nTime loading mirror index: 00:00:00\nMultiseed full-index search: 00:24:06\n31197576 reads; of these:\n  31197576 (100.00%) were paired; of these:\n    5446315 (17.46%) aligned concordantly 0 times\n    18322695 (58.73%) aligned concordantly exactly 1 time\n    7428566 (23.81%) aligned concordantly >1 times\n    ----\n    5446315 pairs aligned concordantly 0 times; of these:\n      173005 (3.18%) aligned discordantly 1 time\n    ----\n    5273310 pairs aligned 0 times concordantly or discordantly; of these:\n      10546620 mates make up the pairs; of these:\n        9686682 (91.85%) aligned 0 times\n        518867 (4.92%) aligned exactly 1 time\n        341071 (3.23%) aligned >1 times\n84.48% overall alignment rate\nTime searching: 00:24:07\nOverall time: 00:24:07\n'
Checking MD5 from JM11-INDEX49
Checking Fastaq lenghts from JM11-INDEX49
Doing Trim galore in CS149-HAT9b/vescaAlpina from JM11-INDEX49
Trim galore finished,checking results...
Doing Bowtie2 in CS149-HAT9b/vescaAlpina from JM11-INDEX49
Bowtie2 summary: b''
Bowtie2 errors if any: b'Time loading reference: 00:00:00\nTime loading forward index: 00:00:00\nTime loading mirror index: 00:00:00\nMultiseed full-index search: 00:23:21\n20525632 reads; of these:\n  20525632 (100.00%) were paired; of these:\n    1803672 (8.79%) aligned concordantly 0 times\n    12422009 (60.52%) aligned concordantly exactly 1 time\n    6299951 (30.69%) aligned concordantly >1 times\n    ----\n    1803672 pairs aligned concordantly 0 times; of these:\n      348944 (19.35%) aligned discordantly 1 time\n    ----\n    1454728 pairs aligned 0 times concordantly or discordantly; of these:\n      2909456 mates make up the pairs; of these:\n        1841908 (63.31%) aligned 0 times\n        444419 (15.27%) aligned exactly 1 time\n        623129 (21.42%) aligned >1 times\n95.51% overall alignment rate\nTime searching: 00:23:21\nOverall time: 00:23:21\n'
pigz 2.6
Analysis complete for INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  
Archive:  INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX42_BKDL210038288-1A-42_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/04 20:26:41

Loading data...
    Loading reads from: CS142-CDFbvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/FVechrs/ --s 150000000 --expt CS142-CDFbvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDFb/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 34042355	HitCounts: 5.3025792E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 34042355	HitCounts: 4.7276613E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 198 reads.


	IP: 47276613	Ctrl: 15832687	 IP/Ctrl: 2.99

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 1.00

The genome is segmented into 221325 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/221325	1.4m
1000	/221325	1.7m
10000	/221325	3.9m
100000	/221325	6.6m
221325	/221325	6.6m

Scaling condition , IP/Control = 2.92

---------------------------
Events discovered 
Significant:	6844
Insignificant:	467
Filtered:	319

Total 815 homotypic events (within 501bp of other events).

Finish binding event prediction: 6.6m

Refine read distribution from 2000 binding events. 


============================ Round 1 ============================

Running with 10 threads ...

100	/221325	2.2s
1000	/221325	6.2s
10000	/221325	23.2s
100000	/221325	3.7m
200000	/221325	7.2m
221325	/221325	9.3m
---------------------------
Events discovered 
Significant:	45017
Insignificant:	968
Filtered:	79

Total 9666 homotypic events (within 545bp of other events).

Finish binding event prediction: 9.3m

Refine read distribution from 2000 binding events. 

Loading genome sequences ...
Done, 3.0s
GC content=0.41

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.34. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=10
k=6+1, gapped kmer=19, base kmer=27
k=6+2, gapped kmer=29, base kmer=33
k=6+3, gapped kmer=16, base kmer=23
k=6+4, gapped kmer=15, base kmer=14

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 10

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTTTTT/AAAAAG	208	208	141	-5.1    	0	316.0	3.0	947.9	7
ACTTTA/TAAAGT	105	106	45	-7.6    	6	210.7	1.0	210.7	3

Align and cluster 31 k-mers with seed CTTTTT,   	#0
Align and cluster 17 k-mers with seed ACTTTA,   	#1

Merging redundant motifs ...
164ms: Remove motif #1.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= CTTTTT 	2.41, 123, kAUC=18.8	PWM= CTTTTT	6.68, 459, pAUC=7.4
#1	k=6	KSM= ACTTTA 	2.14, 167, kAUC=15.3	PWM= ACTTTA	6.76, 376, pAUC=11.4

After merging:
#0	k=6	KSM= CTTTTT 	2.33, 116, kAUC=20.0	PWM= CTTTTT	6.44, 508, pAUC=7.4



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=10
k=7+1, gapped kmer=18, base kmer=27
k=7+2, gapped kmer=11, base kmer=16
k=7+3, gapped kmer=12, base kmer=17
k=7+4, gapped kmer=12, base kmer=8

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 10

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTTTTTC/GAAAAAG	77	77	31	-6.1    	0	116.1	5.0	580.4	4
AAAGGTA/TACCTTT	33	33	7	-5.5    	6	45.2	2.0	90.5	1

Align and cluster 19 k-mers with seed CTTTTTC,   	#0
Align and cluster 13 k-mers with seed AAAGGTA,   	#1

Merging redundant motifs ...

------------------------- k = 7 ----------------------------

#0	k=7	KSM= CTTTTTC 	2.05, 85, kAUC=20.1	PWM= CTTTTTC	9.99, 170, pAUC=13.1
#1	k=7	KSM= AAAGGTA 	2.03, 43, kAUC=14.3	PWM= AAGGT	8.09, 158, pAUC=7.0

After merging:
#0	k=7	KSM= CTTTTTC 	2.05, 85, kAUC=20.1	PWM= CTTTTTC	9.99, 170, pAUC=13.1
#1	k=7	KSM= AAAGGTA 	2.03, 43, kAUC=14.3	PWM= AAGGT	8.09, 158, pAUC=7.0



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=5
k=8+1, gapped kmer=11, base kmer=16
k=8+2, gapped kmer=5, base kmer=9
k=8+3, gapped kmer=6, base kmer=8
k=8+4, gapped kmer=1, base kmer=3

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 5

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CCTTTTTC/GAAAAAGG	19	19	2	-5.0    	0	24.7	4.0	98.6	2
AACTAAAG/CTTTAGTT	16	16	1	-5.2    	3	16.0	3.0	48.0	1
AAAAGGTA/TACCTTTT	17	17	0	-6.1    	1	21.0	2.0	42.0	1

Align and cluster 9 k-mers with seed CCTTTTTC,   	#0
Align and cluster 5 k-mers with seed AACTAAAG,   	#1
Align and cluster 12 k-mers with seed AAAAGGTA,   	#2

Merging redundant motifs ...

------------------------- k = 8 ----------------------------

#0	k=8	KSM= AAAAGGTA 	2.14, 62, kAUC=20.9	PWM= AAAAGG	11.55, 152, pAUC=12.8
#1	k=8	KSM= CCTTTTTC 	2.24, 42, kAUC=14.9	PWM= CCTTTTTC	9.04, 139, pAUC=14.4
#2	k=8	KSM= AACTAAAG 	2.42, 16, kAUC=7.4	PWM= AACTAAAG	9.58, 45, pAUC=7.2

After merging:
#0	k=8	KSM= AAAAGGTA 	2.14, 62, kAUC=20.9	PWM= AAAAGG	11.55, 152, pAUC=12.8
#1	k=8	KSM= CCTTTTTC 	2.24, 42, kAUC=14.9	PWM= CCTTTTTC	9.04, 139, pAUC=14.4
#2	k=8	KSM= AACTAAAG 	2.42, 16, kAUC=7.4	PWM= AACTAAAG	9.58, 45, pAUC=7.2



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=3
k=9+1, gapped kmer=5, base kmer=8
k=9+2, gapped kmer=3, base kmer=5
k=9+3, gapped kmer=0, base kmer=0
k=9+4, gapped kmer=0, base kmer=0

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 3

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTCTTTTTC/GAAAAAGAA	27	27	5	-5.1    	1	32.9	4.0	131.5	2
CTTTTTCTT/AAGAAAAAG	31	31	6	-5.6

Align and cluster 6 k-mers with seed TTCTTTTTC,   	#0
Align and cluster 7 k-mers with seed CTTTTTCTT,   	#1

Merging redundant motifs ...
90ms: Remove motif #1.

------------------------- k = 9 ----------------------------

#0	k=9	KSM= CTTTTTCTT 	1.93, 38, kAUC=14.3	PWM= CTTTTTCTT	9.85, 116, pAUC=13.5
#1	k=9	KSM= TTCTTTTTC 	1.93, 26, kAUC=11.7	PWM= TTCTTTTTC	9.48, 113, pAUC=13.3

After merging:
#0	k=9	KSM= CTTTTTCTT 	1.56, 46, kAUC=14.4	PWM= CTTTTTCTT	9.73, 130, pAUC=12.4



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=0
k=10+1, gapped kmer=1, base kmer=2
k=10+2, gapped kmer=2, base kmer=3
k=10+3, gapped kmer=0, base kmer=0
k=10+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=0
k=11+1, gapped kmer=0, base kmer=0
k=11+2, gapped kmer=0, base kmer=0
k=11+3, gapped kmer=0, base kmer=0
k=11+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=0
k=12+1, gapped kmer=1, base kmer=2
k=12+2, gapped kmer=0, base kmer=0
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=0
k=13+1, gapped kmer=0, base kmer=0
k=13+2, gapped kmer=0, base kmer=0
k=13+3, gapped kmer=0, base kmer=0
k=13+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=0
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=0, base kmer=0
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= AAAAGGTA 	2.14, 62, kAUC=20.9	PWM= AAAAGG	11.55, 152, pAUC=12.8
#1	k=7	KSM= CTTTTTC 	2.05, 85, kAUC=20.1	PWM= CTTTTTC	9.99, 170, pAUC=13.1
#2	k=6	KSM= CTTTTT 	2.33, 116, kAUC=20.0	PWM= CTTTTT	6.44, 508, pAUC=7.4
#3	k=8	KSM= CCTTTTTC 	2.24, 42, kAUC=14.9	PWM= CCTTTTTC	9.04, 139, pAUC=14.4
#4	k=9	KSM= CTTTTTCTT 	1.56, 46, kAUC=14.4	PWM= CTTTTTCTT	9.73, 130, pAUC=12.4
#5	k=7	KSM= AAAGGTA 	2.03, 43, kAUC=14.3	PWM= AAGGT	8.09, 158, pAUC=7.0
#6	k=8	KSM= AACTAAAG 	2.42, 16, kAUC=7.4	PWM= AACTAAAG	9.58, 45, pAUC=7.2

161ms: Remove motif #2.
233ms: Remove motif #3.
327ms: Remove motif #4.
486ms: Remove motif #1.

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= AAAAGGTA 	2.14, 62, kAUC=20.9	PWM= AAAAGG	11.55, 152, pAUC=12.8
#1	k=7	KSM= CTTTTTC 	2.05, 85, kAUC=20.1	PWM= CTTTTTC	9.99, 170, pAUC=13.1
#2	k=6	KSM= CTTTTT 	2.33, 116, kAUC=20.0	PWM= CTTTTT	6.44, 508, pAUC=7.4
#3	k=8	KSM= CCTTTTTC 	2.24, 42, kAUC=14.9	PWM= CCTTTTTC	9.04, 139, pAUC=14.4
#4	k=9	KSM= CTTTTTCTT 	1.56, 46, kAUC=14.4	PWM= CTTTTTCTT	9.73, 130, pAUC=12.4
#5	k=7	KSM= AAAGGTA 	2.03, 43, kAUC=14.3	PWM= AAGGT	8.09, 158, pAUC=7.0
#6	k=8	KSM= AACTAAAG 	2.42, 16, kAUC=7.4	PWM= AACTAAAG	9.58, 45, pAUC=7.2

------------------------- GEMout_1, k = ALL (after merging) -------------------------
#0	k=8	KSM= AAAAGGTA 	2.05, 71, kAUC=23.0	PWM= AAAAGGG	7.22, 415, pAUC=11.9
#5	k=7	KSM= AAAGGTA 	2.03, 43, kAUC=14.3	PWM= AAGGT	8.09, 158, pAUC=7.0
#6	k=8	KSM= AACTAAAG 	2.42, 16, kAUC=7.4	PWM= AACTAAAG	9.58, 45, pAUC=7.2

594ms: Finalizing 3 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_1 final motifs ----------------------------


GEMout_1 motif #0

KSM top k-mer: AAAAGGTA/TACCTTTT, total 11 k-mers
KSM threshold: 2.05, 		hit=71+/3-, kAUC=23.0

       |
AAAAGGG
     AA
       
       
PWM threshold: 7.22/10.65, 	hit=415+/255-, pAUC=11.9
--------------------------------------------------------------

GEMout_1 motif #1

KSM top k-mer: AAAGGTA/TACCTTT, total 4 k-mers
KSM threshold: 2.03, 		hit=43+/3-, kAUC=14.3

           |
AAGGT
 G   
     
     
PWM threshold: 8.09/9.44, 	hit=158+/136-, pAUC=7.0
--------------------------------------------------------------

GEMout_1 motif #2

KSM top k-mer: AACTAAAG/CTTTAGTT, total 1 k-mers
KSM threshold: 2.42, 		hit=16+/1-, kAUC=7.4

     |
AACTAAAG
        
        
        
PWM threshold: 9.58/13.87, 	hit=45+/28-, pAUC=7.2
--------------------------------------------------------------



============================ Round 2 ============================
Running EM with motif positional prior ...

Running with 10 threads ...

100	/221325	24.2s
1000	/221325	29.3s
10000	/221325	46.3s
100000	/221325	3.8m
200000	/221325	7.2m
221325	/221325	8.9m
---------------------------
Events discovered 
Significant:	55177
Insignificant:	976
Filtered:	80

Total 12533 homotypic events (within 545bp of other events).

Finish binding event prediction: 9.0m

Refine read distribution from 2000 binding events. 

Percentage of significant events with a motif match, divided in 10 bins (~5518 events per bin):
49 50 48 47 46 46 46 47 46 48 

Percentage of filtered events with a motif match, divided in 10 bins (~8 events per bin):
0 0 0 0 0 0 0 0 0 0 

Percentage of insignificant events with a motif match, divided in 10 bins (~98 events per bin):
50 50 57 61 58 55 58 55 35 45 

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.34. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=9
k=6+1, gapped kmer=20, base kmer=30
k=6+2, gapped kmer=45, base kmer=49
k=6+3, gapped kmer=31, base kmer=37
k=6+4, gapped kmer=31, base kmer=32

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 9

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAG/CTTTTT	255	255	162	-8.3    	0	349.1	3.0	1047.4	6
AAAGGA/TCCTTT	115	115	56	-6.5    	6	202.1	2.0	404.1	1
TTTTCC/GGAAAA	132	132	73	-5.7    	4	182.4	2.0	364.8	2

Align and cluster 32 k-mers with seed AAAAAG,   	#0
Align and cluster 28 k-mers with seed AAAGGA,   	#1
Align and cluster 22 k-mers with seed TTTTCC,   	#2

Merging redundant motifs ...
118ms: Remove motif #2.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= TTTTCC 	2.98, 176, kAUC=25.3	PWM= TTTTCC	6.14, 375, pAUC=9.0
#1	k=6	KSM= AAAGGA 	1.48, 254, kAUC=24.9	PWM= AAAGGG	6.79, 455, pAUC=8.4
#2	k=6	KSM= AAAAAG 	2.48, 110, kAUC=22.5	PWM= AAAAAG	6.70, 464, pAUC=7.9

After merging:
#0	k=6	KSM= TTTTCC 	2.98, 176, kAUC=25.3	PWM= TTTTCC	6.14, 375, pAUC=9.0
#1	k=6	KSM= AAAGGA 	1.52, 251, kAUC=25.0	PWM= AAAGGG	6.55, 525, pAUC=8.4



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=16
k=7+1, gapped kmer=32, base kmer=41
k=7+2, gapped kmer=34, base kmer=40
k=7+3, gapped kmer=35, base kmer=32
k=7+4, gapped kmer=17, base kmer=15

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 16

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAGAA/TTCTTTT	114	114	59	-5.8    	2	182.8	4.0	731.0	6
CTTTATC/GATAAAG	28	28	3	-6.8    	13	73.9	1.0	73.9	2
GCACAAA/TTTGTGC	14	14	0	-5.2    	15	14.0	3.0	42.0	1
AAGAAAA/TTTTCTT	116	116	63	-5.3

Align and cluster 38 k-mers with seed AAAAGAA,   	#0
Align and cluster 19 k-mers with seed CTTTATC,   	#1
Align and cluster 5 k-mers with seed GCACAAA,   	#2
Align and cluster 31 k-mers with seed AAGAAAA,   	#3

Merging redundant motifs ...
114ms: Remove motif #1.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= AAGAAAA 	3.74, 90, kAUC=27.8	PWM= AAGGAAA	7.43, 303, pAUC=13.8
#1	k=7	KSM= AAAAGAA 	4.55, 64, kAUC=27.0	PWM= AAAAGGA	8.00, 292, pAUC=14.0
#2	k=7	KSM= CTTTATC 	2.92, 68, kAUC=25.9	PWM= CTTTTTC	7.37, 259, pAUC=15.2
#3	k=7	KSM= GCACAAA 	2.46, 14, kAUC=7.2	PWM= GCACAAA	8.67, 39, pAUC=8.0

After merging:
#0	k=7	KSM= AAGAAAA 	3.74, 90, kAUC=27.8	PWM= AAGGAAA	7.43, 303, pAUC=13.8
#2	k=7	KSM= CTTTATC 	2.92, 68, kAUC=25.9	PWM= CTTTTTC	7.37, 259, pAUC=15.2
#3	k=7	KSM= GCACAAA 	2.46, 14, kAUC=7.2	PWM= GCACAAA	8.67, 39, pAUC=8.0



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=12
k=8+1, gapped kmer=19, base kmer=27
k=8+2, gapped kmer=25, base kmer=28
k=8+3, gapped kmer=15, base kmer=16
k=8+4, gapped kmer=11, base kmer=10

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 12

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTCTTT/AAAGAAAA	76	76	33	-5.4    	0	128.7	4.0	514.9	6
AAAAGTTA/TAACTTTT	39	39	8	-6.5    	7	65.8	2.0	131.6	1
GAAAAAGG/CCTTTTTC	39	39	5	-8.3    	8	65.5	1.0	65.5	2

Align and cluster 50 k-mers with seed TTTTCTTT,   	#0
Align and cluster 18 k-mers with seed AAAAGTTA,   	#1
Align and cluster 16 k-mers with seed GAAAAAGG,   	#2

Merging redundant motifs ...

------------------------- k = 8 ----------------------------

#0	k=8	KSM= TTTTCTTT 	2.99, 93, kAUC=29.4	PWM= TTTCCTTT	8.69, 179, pAUC=17.8
#1	k=8	KSM= AAAAGTTA 	1.62, 179, kAUC=27.2	PWM= AAAAGTGA	9.08, 211, pAUC=15.2
#2	k=8	KSM= GAAAAAGG 	3.24, 48, kAUC=22.1	PWM= GAAAAAGG	9.26, 238, pAUC=18.5

After merging:
#0	k=8	KSM= TTTTCTTT 	2.99, 93, kAUC=29.4	PWM= TTTCCTTT	8.69, 179, pAUC=17.8
#1	k=8	KSM= AAAAGTTA 	1.62, 179, kAUC=27.2	PWM= AAAAGTGA	9.08, 211, pAUC=15.2
#2	k=8	KSM= GAAAAAGG 	3.24, 48, kAUC=22.1	PWM= GAAAAAGG	9.26, 238, pAUC=18.5



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=12
k=9+1, gapped kmer=18, base kmer=23
k=9+2, gapped kmer=7, base kmer=10
k=9+3, gapped kmer=8, base kmer=10
k=9+4, gapped kmer=6, base kmer=4

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 12

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAGAAAA/TTTTCTTTT	53	53	15	-6.6    	0	103.7	5.0	518.6	9
TGAAAAAGG/CCTTTTTCA	14	14	0	-5.2    	11	35.3	2.0	70.6	1

Align and cluster 36 k-mers with seed AAAAGAAAA,   	#0
Align and cluster 11 k-mers with seed TGAAAAAGG,   	#1

Merging redundant motifs ...

------------------------- k = 9 ----------------------------

#0	k=9	KSM= AAAAGAAAA 	2.48, 90, kAUC=24.1	PWM= AAAAGGAAA	9.50, 168, pAUC=19.4
#1	k=9	KSM= TGAAAAAGG 	2.35, 50, kAUC=19.5	PWM= AGAAAAAGG	9.12, 174, pAUC=18.9

After merging:
#0	k=9	KSM= AAAAGAAAA 	2.48, 90, kAUC=24.1	PWM= AAAAGGAAA	9.50, 168, pAUC=19.4
#1	k=9	KSM= TGAAAAAGG 	2.35, 50, kAUC=19.5	PWM= AGAAAAAGG	9.12, 174, pAUC=18.9



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=8
k=10+1, gapped kmer=5, base kmer=8
k=10+2, gapped kmer=5, base kmer=6
k=10+3, gapped kmer=0, base kmer=0
k=10+4, gapped kmer=0, base kmer=0

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 8

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTCTTTTT/AAAAAGAAAA	36	36	9	-5.3    	0	57.5	3.0	172.6	8

Align and cluster 18 k-mers with seed TTTTCTTTTT,   	#0

------------------------- k = 10 ----------------------------

#0	k=10	KSM= TTTTCTTTTT 	1.96, 53, kAUC=17.4	PWM= TTTTCCTTTT	11.29, 160, pAUC=16.5



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=4
k=11+1, gapped kmer=3, base kmer=6
k=11+2, gapped kmer=0, base kmer=0
k=11+3, gapped kmer=0, base kmer=0
k=11+4, gapped kmer=0, base kmer=0

------------------------- k = 11 ----------------------------

Total number of k-mers: n = 4

Top 5 cluster center k-mers:
# k-mer/r.c.           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAGGAAAAA/TTTTTCCTTTT	19	19	2	-5.0    	3	30.8	4.0	123.3	3
AAGAAAAAGAA/TTCTTTTTCTT	21	21	2	-5.6

Align and cluster 6 k-mers with seed AAAAGGAAAAA,   	#0
Align and cluster 6 k-mers with seed AAGAAAAAGAA,   	#1

Merging redundant motifs ...
116ms: Remove motif #1.

------------------------- k = 11 ----------------------------

#0	k=11	KSM= AAAAGGAAAAA 	1.31, 60, kAUC=13.4	PWM= AAAAGGAAAAA	10.92, 152, pAUC=15.7
#1	k=11	KSM= AAGAAAAAGAA 	1.31, 43, kAUC=11.1	PWM= AAGAAAAAGAA	10.44, 133, pAUC=15.4

After merging:
#0	k=11	KSM= AAAAGGAAAAA 	1.31, 60, kAUC=13.4	PWM= AAAAGGAAAAA	10.92, 152, pAUC=15.7



----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=1
k=12+1, gapped kmer=0, base kmer=0
k=12+2, gapped kmer=0, base kmer=0
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=0, base kmer=0

------------------------- k = 12 ----------------------------

Total number of k-mers: n = 1

Top 5 cluster center k-mers:
# k-mer/r.c.             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTTTTCTTTTTC/GAAAAAGAAAAA	14	14	0	-5.2    	0	14.0	0.0	0.0	1

Align and cluster 1 k-mers with seed TTTTTCTTTTTC,   	#0

------------------------- k = 12 ----------------------------

#0	k=12	KSM= TTTTTCTTTTTC 	2.35, 14, kAUC=7.2	PWM= TTTTTCTTTTTC	10.95, 151, pAUC=17.1



----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=0
k=13+1, gapped kmer=0, base kmer=0
k=13+2, gapped kmer=0, base kmer=0
k=13+3, gapped kmer=0, base kmer=0
k=13+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=0
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=0, base kmer=0
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=8	KSM= TTTTCTTT 	2.99, 93, kAUC=29.4	PWM= TTTCCTTT	8.69, 179, pAUC=17.8
#1	k=7	KSM= AAGAAAA 	3.74, 90, kAUC=27.8	PWM= AAGGAAA	7.43, 303, pAUC=13.8
#2	k=8	KSM= AAAAGTTA 	1.62, 179, kAUC=27.2	PWM= AAAAGTGA	9.08, 211, pAUC=15.2
#3	k=7	KSM= CTTTATC 	2.92, 68, kAUC=25.9	PWM= CTTTTTC	7.37, 259, pAUC=15.2
#4	k=6	KSM= TTTTCC 	2.98, 176, kAUC=25.3	PWM= TTTTCC	6.14, 375, pAUC=9.0
#5	k=6	KSM= AAAGGA 	1.52, 251, kAUC=25.0	PWM= AAAGGG	6.55, 525, pAUC=8.4
#6	k=9	KSM= AAAAGAAAA 	2.48, 90, kAUC=24.1	PWM= AAAAGGAAA	9.50, 168, pAUC=19.4
#7	k=8	KSM= GAAAAAGG 	3.24, 48, kAUC=22.1	PWM= GAAAAAGG	9.26, 238, pAUC=18.5
#8	k=9	KSM= TGAAAAAGG 	2.35, 50, kAUC=19.5	PWM= AGAAAAAGG	9.12, 174, pAUC=18.9
#9	k=10	KSM= TTTTCTTTTT 	1.96, 53, kAUC=17.4	PWM= TTTTCCTTTT	11.29, 160, pAUC=16.5
#10	k=11	KSM= AAAAGGAAAAA 	1.31, 60, kAUC=13.4	PWM= AAAAGGAAAAA	10.92, 152, pAUC=15.7
#11	k=12	KSM= TTTTTCTTTTTC 	2.35, 14, kAUC=7.2	PWM= TTTTTCTTTTTC	10.95, 151, pAUC=17.1
#12	k=7	KSM= GCACAAA 	2.46, 14, kAUC=7.2	PWM= GCACAAA	8.67, 39, pAUC=8.0

110ms: Remove motif #1.
291ms: Remove motif #4.
391ms: Remove motif #5.
396ms: PWM TTTCCTTTT has been removed.
468ms: Remove motif #9.
541ms: Remove motif #10.
661ms: Remove motif #11.
756ms: Remove motif #7.
851ms: Remove motif #8.
1.0s: Remove motif #2.
1.1s: Remove motif #3.

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=8	KSM= TTTTCTTT 	2.99, 93, kAUC=29.4	PWM= TTTCCTTT	8.69, 179, pAUC=17.8
#1	k=7	KSM= AAGAAAA 	3.74, 90, kAUC=27.8	PWM= AAGGAAA	7.43, 303, pAUC=13.8
#2	k=8	KSM= AAAAGTTA 	1.62, 179, kAUC=27.2	PWM= AAAAGTGA	9.08, 211, pAUC=15.2
#3	k=7	KSM= CTTTATC 	2.92, 68, kAUC=25.9	PWM= CTTTTTC	7.37, 259, pAUC=15.2
#4	k=6	KSM= TTTTCC 	2.98, 176, kAUC=25.3	PWM= TTTTCC	6.14, 375, pAUC=9.0
#5	k=6	KSM= AAAGGA 	1.52, 251, kAUC=25.0	PWM= AAAGGG	6.55, 525, pAUC=8.4
#6	k=9	KSM= AAAAGAAAA 	2.48, 90, kAUC=24.1	PWM= AAAAGGAAA	9.50, 168, pAUC=19.4
#7	k=8	KSM= GAAAAAGG 	3.24, 48, kAUC=22.1	PWM= GAAAAAGG	9.26, 238, pAUC=18.5
#8	k=9	KSM= TGAAAAAGG 	2.35, 50, kAUC=19.5	PWM= AGAAAAAGG	9.12, 174, pAUC=18.9
#9	k=10	KSM= TTTTCTTTTT 	1.96, 53, kAUC=17.4	PWM= TTTTCCTTTT	11.29, 160, pAUC=16.5
#10	k=11	KSM= AAAAGGAAAAA 	1.31, 60, kAUC=13.4	PWM= AAAAGGAAAAA	10.92, 152, pAUC=15.7
#11	k=12	KSM= TTTTTCTTTTTC 	2.35, 14, kAUC=7.2	PWM= TTTTTCTTTTTC	10.95, 151, pAUC=17.1
#12	k=7	KSM= GCACAAA 	2.46, 14, kAUC=7.2	PWM= GCACAAA	8.67, 39, pAUC=8.0

------------------------- GEMout_2, k = ALL (after merging) -------------------------
#0	k=8	KSM= TTTTCTTT 	2.60, 146, kAUC=32.1	PWM= TTCCTTTT	8.01, 381, pAUC=15.6
#12	k=7	KSM= GCACAAA 	2.46, 14, kAUC=7.2	PWM= GCACAAA	8.67, 39, pAUC=8.0

1.1s: Finalizing 2 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_2 final motifs ----------------------------


GEMout_2 motif #0

KSM top k-mer: TTTTCTTT/AAAGAAAA, total 45 k-mers
KSM threshold: 2.60, 		hit=146+/12-, kAUC=32.1

  |
TTCCTTTT
  A     
  T     
        
PWM threshold: 8.01/11.58, 	hit=381+/181-, pAUC=15.6
--------------------------------------------------------------

GEMout_2 motif #1

KSM top k-mer: GCACAAA/TTTGTGC, total 1 k-mers
KSM threshold: 2.46, 		hit=14+/0-, kAUC=7.2

          |
GCACAAA
       
       
       
PWM threshold: 8.67/12.65, 	hit=39+/19-, pAUC=8.0
--------------------------------------------------------------



Finished! GEM analysis results are printed to:
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDFb/vescaAlpina/GEMout/GEMout.GEM_events.txt
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDFb/vescaAlpina/GEMout/GEMout.results.htm
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS142-CDFb/vescaAlpina/GEMout/GEMout_outputs (folder with all other files)

----------------------------------

End time: 2021/12/04 20:56:53

Total running time: 30.2m

pigz 2.6
Analysis complete for INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  
Archive:  INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX44_BKDL210038288-1A-44_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/04 22:14:27

Loading data...
    Loading reads from: CS144-MYBbvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/FVechrs/ --s 150000000 --expt CS144-MYBbvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYBb/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 38627642	HitCounts: 5.270847E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 38627642	HitCounts: 4.4444446E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 187 reads.


	IP: 44444446	Ctrl: 15832687	 IP/Ctrl: 2.81

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 2.53

The genome is segmented into 221604 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/221604	46.4s
1000	/221604	1.8m
10000	/221604	2.4m
100000	/221604	3.2m
200000	/221604	3.2m
221604	/221604	3.2m

Scaling condition , IP/Control = 2.25

---------------------------
Events discovered 
Significant:	90
Insignificant:	54
Filtered:	2

Total 57 homotypic events (within 501bp of other events).

Finish binding event prediction: 3.3m


============================ Round 1 ============================

Running with 10 threads ...

100	/221604	1.2s
1000	/221604	2.2s
10000	/221604	5.2s
100000	/221604	51.2s
200000	/221604	1.6m
221604	/221604	2.1m
---------------------------
Events discovered 
Significant:	131
Insignificant:	55
Filtered:	5

Total 83 homotypic events (within 501bp of other events).

Finish binding event prediction: 2.1m

Loading genome sequences ...
Done, 2.2s
GC content=0.41

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.32. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 186+/186- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=157
k=6+1, gapped kmer=119, base kmer=162
k=6+2, gapped kmer=417, base kmer=267
k=6+3, gapped kmer=249, base kmer=133
k=6+4, gapped kmer=61, base kmer=56

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 157

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAA/TTCTTT	37	37	25	-1.5    	0	75.7	5.0	378.6	17
TTTATC/GATAAA	31	32	15	-2.6    	5	72.2	2.0	144.4	18
ATTCAT/ATGAAT	32	33	12	-3.7    	2	71.5	2.0	143.0	20
CAAATA/TATTTG	22	22	6	-3.5    	16	53.5	2.0	107.0	14
CTTGTT/AACAAG	16	16	6	-2.2    	29	44.2	2.0	88.4	10

Align and cluster 136 k-mers with seed AAAGAA,   	#0
Align and cluster 112 k-mers with seed TTTATC,   	#1
Align and cluster 123 k-mers with seed ATTCAT,   	#2
Align and cluster 109 k-mers with seed CAAATA,   	#3
Align and cluster 83 k-mers with seed CTTGTT,   	#4

Merging redundant motifs ...
53ms: Remove motif #3.
86ms: Remove motif #4.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= CAAATA 	1.21, 75, kAUC=37.8	PWM= CAAATA	5.72, 71, pAUC=12.4
#1	k=6	KSM= AAAGAA 	1.55, 60, kAUC=34.7	PWM= AAAGAA	5.83, 112, pAUC=7.4
#2	k=6	KSM= TTTATC 	1.21, 94, kAUC=33.3	PWM= TTTATC	5.93, 114, pAUC=10.1
#3	k=6	KSM= ATTCAT 	1.37, 74, kAUC=31.1	PWM= ATTCAT	6.19, 118, pAUC=12.5
#4	k=6	KSM= CTTGTT 	1.43, 53, kAUC=29.9	PWM= CTTGTT	6.93, 60, pAUC=9.8

After merging:
#0	k=6	KSM= CAAATA 	1.21, 75, kAUC=37.8	PWM= CAAATA	5.72, 71, pAUC=12.4
#1	k=6	KSM= AAAGAA 	1.54, 65, kAUC=36.3	PWM= AAAGAA	5.98, 148, pAUC=7.4
#2	k=6	KSM= TTTATC 	1.21, 94, kAUC=33.3	PWM= TTTATC	5.93, 114, pAUC=10.1



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=284
k=7+1, gapped kmer=88, base kmer=131
k=7+2, gapped kmer=102, base kmer=89
k=7+3, gapped kmer=32, base kmer=37
k=7+4, gapped kmer=14, base kmer=19

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 284

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAAA/TTTCTTT	19	19	11	-1.4    	0	48.1	6.0	288.9	28
AAAATTG/CAATTTT	12	12	5	-1.7    	12	28.6	3.0	85.7	13
ATCAAAT/ATTTGAT	9	9	1	-3.1    	30	30.4	2.0	60.9	23
TTATTCA/TGAATAA	9	10	4	-1.6    	25	29.8	2.0	59.7	13
AGATCTA/TAGATCT	7	7	1	-2.4    	53	22.1	2.0	44.3	9

Align and cluster 135 k-mers with seed AAAGAAA,   	#0
Align and cluster 70 k-mers with seed AAAATTG,   	#1
Align and cluster 112 k-mers with seed ATCAAAT,   	#2
Align and cluster 109 k-mers with seed TTATTCA,   	#3
Align and cluster 70 k-mers with seed AGATCTA,   	#4

Merging redundant motifs ...

------------------------- k = 7 ----------------------------

#0	k=7	KSM= AAAGAAA 	1.49, 49, kAUC=40.3	PWM= AAAGAAA	6.82, 79, pAUC=8.1
#1	k=7	KSM= TTATTCA 	1.32, 56, kAUC=38.6	PWM= TTATTCA	7.51, 41, pAUC=11.3
#2	k=7	KSM= ATCAAAT 	1.39, 46, kAUC=37.6	PWM= ATCAAAT	7.13, 67, pAUC=11.1
#3	k=7	KSM= AAAATTG 	1.32, 36, kAUC=27.2	PWM= AAATT	5.96, 144, pAUC=5.0
#4	k=7	KSM= AGATCTA 	1.00, 19, kAUC=12.3	PWM= AGATCTA	6.94, 23, pAUC=10.8

After merging:
#0	k=7	KSM= AAAGAAA 	1.49, 49, kAUC=40.3	PWM= AAAGAAA	6.82, 79, pAUC=8.1
#1	k=7	KSM= TTATTCA 	1.32, 56, kAUC=38.6	PWM= TTATTCA	7.51, 41, pAUC=11.3
#2	k=7	KSM= ATCAAAT 	1.39, 46, kAUC=37.6	PWM= ATCAAAT	7.13, 67, pAUC=11.1
#3	k=7	KSM= AAAATTG 	1.32, 36, kAUC=27.2	PWM= AAATT	5.96, 144, pAUC=5.0
#4	k=7	KSM= AGATCTA 	1.00, 19, kAUC=12.3	PWM= AGATCTA	6.94, 23, pAUC=10.8



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=292
k=8+1, gapped kmer=9, base kmer=14
k=8+2, gapped kmer=12, base kmer=17
k=8+3, gapped kmer=9, base kmer=13
k=8+4, gapped kmer=2, base kmer=4

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 292

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
ATTTTTTT/AAAAAAAT	15	15	7	-1.6    	0	34.2	7.0	239.4	22
TTTCAATT/AATTGAAA	7	7	1	-2.4    	7	22.4	2.0	44.9	20
AGAAATGA/TCATTTCT	6	6	2	-1.5    	15	16.4	2.0	32.9	10
GATTCATT/AATGAATC	5	5	1	-1.8    	47	16.4	2.0	32.9	12
TTTATCTC/GAGATAAA	4	4	1	-1.5    	126	14.6	2.0	29.1	9

Align and cluster 214 k-mers with seed ATTTTTTT,   	#0
Align and cluster 182 k-mers with seed TTTCAATT,   	#1
Align and cluster 149 k-mers with seed AGAAATGA,   	#2
Align and cluster 150 k-mers with seed GATTCATT,   	#3
Align and cluster 162 k-mers with seed TTTATCTC,   	#4

Merging redundant motifs ...
29ms: Remove motif #1.
62ms: Remove motif #2.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= ATTTTTTT 	1.21, 120, kAUC=67.6	PWM= TTTTTT	5.70, 121, pAUC=5.1
#1	k=8	KSM= AGAAATGA 	1.21, 89, kAUC=55.4	PWM= AAAAAAGA	7.09, 76, pAUC=7.9
#2	k=8	KSM= TTTATCTC 	1.43, 40, kAUC=53.0	PWM= TTTTTTTT	6.90, 70, pAUC=4.9
#3	k=8	KSM= TTTCAATT 	1.54, 30, kAUC=51.6	PWM= ATCAAA	5.43, 112, pAUC=8.1
#4	k=8	KSM= GATTCATT 	1.31, 31, kAUC=37.4	PWM= ATTCAT	6.88, 72, pAUC=13.2

After merging:
#0	k=8	KSM= ATTTTTTT 	1.32, 90, kAUC=67.9	PWM= TTTTTT	5.92, 121, pAUC=5.1
#3	k=8	KSM= TTTCAATT 	1.54, 30, kAUC=51.6	PWM= ATCAAA	5.43, 112, pAUC=8.1
#4	k=8	KSM= GATTCATT 	1.31, 31, kAUC=37.4	PWM= ATTCAT	6.88, 72, pAUC=13.2



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=148
k=9+1, gapped kmer=2, base kmer=3
k=9+2, gapped kmer=2, base kmer=4
k=9+3, gapped kmer=1, base kmer=2
k=9+4, gapped kmer=2, base kmer=4

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 148

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATAAAAAA/TTTTTTATT	8	8	3	-1.5    	0	28.1	7.0	197.0	49
ATGAATTGA/TCAATTCAT	4	3	0	-1.8    	59	12.2	3.0	36.7	12
AATTCTTGG/CCAAGAATT	3	3	0	-1.8    	45	10.8	3.0	32.4	5
ATATAATTT/AAATTATAT	4	4	0	-2.1    	18	15.1	2.0	30.2	19
AATGCAGTT/AACTGCATT	3	3	0	-1.8    	43	7.5	3.0	22.6	1

Align and cluster 84 k-mers with seed AATAAAAAA,   	#0
Align and cluster 41 k-mers with seed ATGAATTGA,   	#1
Align and cluster 45 k-mers with seed AATTCTTGG,   	#2
Align and cluster 64 k-mers with seed ATATAATTT,   	#3
Align and cluster 28 k-mers with seed AATGCAGTT,   	#4

Merging redundant motifs ...

------------------------- k = 9 ----------------------------

#0	k=9	KSM= AATAAAAAA 	1.32, 30, kAUC=30.4	PWM= AATAAAAAA	9.80, 36, pAUC=11.3
#1	k=9	KSM= ATGAATTGA 	1.11, 49, kAUC=29.1	PWM= AAGAATTGA	6.90, 58, pAUC=12.6
#2	k=9	KSM= AATTCTTGG 	1.11, 24, kAUC=17.8	PWM= ATTCTTG	7.50, 39, pAUC=8.6
#3	k=9	KSM= ATATAATTT 	1.11, 19, kAUC=15.8	PWM= ATATAATTT	8.99, 19, pAUC=10.1
#4	k=9	KSM= AATGCAGTT 	1.32, 3, kAUC=6.5	PWM= AATGCAGTT	9.39, 5, pAUC=3.2

After merging:
#0	k=9	KSM= AATAAAAAA 	1.32, 30, kAUC=30.4	PWM= AATAAAAAA	9.80, 36, pAUC=11.3
#1	k=9	KSM= ATGAATTGA 	1.11, 49, kAUC=29.1	PWM= AAGAATTGA	6.90, 58, pAUC=12.6
#2	k=9	KSM= AATTCTTGG 	1.11, 24, kAUC=17.8	PWM= ATTCTTG	7.50, 39, pAUC=8.6
#3	k=9	KSM= ATATAATTT 	1.11, 19, kAUC=15.8	PWM= ATATAATTT	8.99, 19, pAUC=10.1
#4	k=9	KSM= AATGCAGTT 	1.32, 3, kAUC=6.5	PWM= AATGCAGTT	9.39, 5, pAUC=3.2



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=74
k=10+1, gapped kmer=0, base kmer=0
k=10+2, gapped kmer=0, base kmer=0
k=10+3, gapped kmer=0, base kmer=0
k=10+4, gapped kmer=0, base kmer=0

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 74

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAAAAT/ATTTTTTTTT	6	6	1	-2.1    	0	14.3	8.0	114.3	16
AATTATGAAA/TTTCATAATT	4	4	0	-2.1    	2	8.2	3.0	24.7	3
TAGCCGACGA/TCGTCGGCTA	4	3	0	-1.8    	30	3.0	6.0	18.0	3
TTAATATTAA/TTAATATTAA	2	2	0	-1.5    	73	5.3	3.0	15.9	2
ATTGAACTTA/TAAGTTCAAT	3	3	0	-1.8    	17	3.9	4.0	15.5	2

Align and cluster 43 k-mers with seed AAAAAAAAAT,   	#0
Align and cluster 38 k-mers with seed AATTATGAAA,   	#1
Align and cluster 9 k-mers with seed TAGCCGACGA,   	#2
Align and cluster 30 k-mers with seed TTAATATTAA,   	#3
Align and cluster 31 k-mers with seed ATTGAACTTA,   	#4

Merging redundant motifs ...
39ms: Remove motif #2.

------------------------- k = 10 ----------------------------

#0	k=10	KSM= AATTATGAAA 	1.21, 25, kAUC=20.3	PWM= AAAAATGAAA	9.44, 40, pAUC=11.3
#1	k=10	KSM= TTAATATTAA 	1.11, 20, kAUC=15.8	PWM= TTAAAAATAA	8.35, 33, pAUC=7.4
#2	k=10	KSM= AAAAAAAAAT 	1.11, 19, kAUC=15.3	PWM= AAAAAAAAAT	10.04, 31, pAUC=9.2
#3	k=10	KSM= TAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TAGCCGACGA	10.89, 6, pAUC=8.5
#4	k=10	KSM= ATTGAACTTA 	1.32, 3, kAUC=6.5	PWM= ATTGAACCTA	8.83, 13, pAUC=8.3

After merging:
#0	k=10	KSM= AATTATGAAA 	1.21, 25, kAUC=20.3	PWM= AAAAATGAAA	9.44, 40, pAUC=11.3
#1	k=10	KSM= TTAATATTAA 	1.11, 20, kAUC=15.8	PWM= TTAAAAATAA	8.35, 33, pAUC=7.4
#3	k=10	KSM= TAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TAGCCGACGA	10.89, 6, pAUC=8.5
#4	k=10	KSM= ATTGAACTTA 	1.32, 3, kAUC=6.5	PWM= ATTGAACCTA	8.83, 13, pAUC=8.3



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=51
k=11+1, gapped kmer=0, base kmer=0
k=11+2, gapped kmer=0, base kmer=0
k=11+3, gapped kmer=0, base kmer=0
k=11+4, gapped kmer=0, base kmer=0

------------------------- k = 11 ----------------------------

Total number of k-mers: n = 51

Top 5 cluster center k-mers:
# k-mer/r.c.           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
GAAAAAAAAAA/TTTTTTTTTTC	3	3	0	-1.8    	8	7.3	9.0	66.1	9
ACAAATATCTT/AAGATATTTGT	3	3	0	-1.8    	2	3.9	5.0	19.4	2
ACTTTAGCCGA/TCGGCTAAAGT	4	3	0	-1.8    	3	3.0	5.0	15.0	3
ATATGAATTGA/TCAATTCATAT	4	3	0	-1.8    	5	3.0	4.0	12.0	2
TTAGCCGACGA/TCGTCGGCTAA	4	3	0	-1.8    	13	3.0	4.0	12.0	3
AAAAAAAAAAT/ATTTTTTTTTT	3	3	0	-1.8

Align and cluster 20 k-mers with seed GAAAAAAAAAA,   	#0
Align and cluster 16 k-mers with seed ACAAATATCTT,   	#1
Align and cluster 9 k-mers with seed ACTTTAGCCGA,   	#2
Align and cluster 8 k-mers with seed ATATGAATTGA,   	#3
Align and cluster 7 k-mers with seed TTAGCCGACGA,   	#4

Merging redundant motifs ...
33ms: Remove motif #3.

------------------------- k = 11 ----------------------------

#0	k=11	KSM= GAAAAAAAAAA 	1.00, 16, kAUC=12.7	PWM= GAAAAAAAAAA	9.39, 39, pAUC=13.4
#1	k=11	KSM= ATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= ATATGAATTGA	9.76, 18, pAUC=9.1
#2	k=11	KSM= TTAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TTAGCCGACGA	11.26, 5, pAUC=8.9
#3	k=11	KSM= ACTTTAGCCGA 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGA	15.55, 4, pAUC=7.3
#4	k=11	KSM= ACAAATATCTT 	1.21, 3, kAUC=6.5	PWM= ACAGATATCTT	10.82, 10, pAUC=7.8

After merging:
#0	k=11	KSM= GAAAAAAAAAA 	1.00, 16, kAUC=12.7	PWM= GAAAAAAAAAA	9.39, 39, pAUC=13.4
#1	k=11	KSM= ATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= ATATGAATTGA	9.76, 18, pAUC=9.1
#2	k=11	KSM= TTAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TTAGCCGACGA	11.26, 5, pAUC=8.9
#4	k=11	KSM= ACAAATATCTT 	1.21, 3, kAUC=6.5	PWM= ACAGATATCTT	10.82, 10, pAUC=7.8



----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=39
k=12+1, gapped kmer=0, base kmer=0
k=12+2, gapped kmer=0, base kmer=0
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=0, base kmer=0

------------------------- k = 12 ----------------------------

Total number of k-mers: n = 39

Top 5 cluster center k-mers:
# k-mer/r.c.             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
ACTTTAGCCGAC/GTCGGCTAAAGT	4	3	0	-1.8    	0	3.0	10.0	30.0	6
TATATGAATTGA/TCAATTCATATA	4	3	0	-1.8    	2	3.0	8.0	24.0	1
AAACCTAGGTTT/AAACCTAGGTTT	2	2	0	-1.5    	4	2.0	7.0	14.0	1
AAAGAACTTTAG/CTAAAGTTCTTT	3	2	0	-1.5    	5	2.0	6.0	12.0	5
AATATAAAGAAC/GTTCTTTATATT	3	2	0	-1.5    	8	2.0	6.0	12.0	5

Align and cluster 13 k-mers with seed ACTTTAGCCGAC,   	#0
Align and cluster 6 k-mers with seed TATATGAATTGA,   	#1
Align and cluster 3 k-mers with seed AAACCTAGGTTT,   	#2
Align and cluster 10 k-mers with seed AAAGAACTTTAG,   	#3
Align and cluster 11 k-mers with seed AATATAAAGAAC,   	#4

Merging redundant motifs ...
52ms: Remove motif #3.

------------------------- k = 12 ----------------------------

#0	k=12	KSM= AATATAAAGAAC 	1.00, 6, kAUC=9.1	PWM= AAGAGAAAGAAC	9.55, 31, pAUC=11.3
#1	k=12	KSM= TATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= TATATGAATTGA	10.23, 21, pAUC=12.6
#2	k=12	KSM= ACTTTAGCCGAC 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGAC	15.06, 4, pAUC=6.4
#3	k=12	KSM= AAAGAACTTTAG 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTGG	11.68, 5, pAUC=7.5
#4	k=12	KSM= AAACCTAGGTTT 	1.11, 2, kAUC=6.0	PWM= AAACCTCGGTTT	10.08, 6, pAUC=5.4

After merging:
#0	k=12	KSM= AATATAAAGAAC 	1.00, 6, kAUC=9.1	PWM= AAGAGAAAGAAC	9.55, 31, pAUC=11.3
#1	k=12	KSM= TATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= TATATGAATTGA	10.23, 21, pAUC=12.6
#2	k=12	KSM= ACTTTAGCCGAC 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGAC	15.06, 4, pAUC=6.4
#4	k=12	KSM= AAACCTAGGTTT 	1.11, 2, kAUC=6.0	PWM= AAACCTCGGTTT	10.08, 6, pAUC=5.4



----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=36
k=13+1, gapped kmer=0, base kmer=0
k=13+2, gapped kmer=0, base kmer=0
k=13+3, gapped kmer=0, base kmer=0
k=13+4, gapped kmer=0, base kmer=0

------------------------- k = 13 ----------------------------

Total number of k-mers: n = 36

Top 5 cluster center k-mers:
# k-mer/r.c.               	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
ACTTTAGCCGACG/CGTCGGCTAAAGT	4	3	0	-1.8    	0	3.0	11.0	33.0	5
AATATAAAGAACT/AGTTCTTTATATT	3	2	0	-1.5    	5	2.0	7.0	14.0	5
ACGAAAATATAAT/ATTATATTTTCGT	3	2	0	-1.5    	6	2.0	7.0	14.0	3
AAAGAACTTTAGC/GCTAAAGTTCTTT	3	2	0	-1.5    	2	2.0	6.0	12.0	6
ATCTGCGAATATA/TATATTCGCAGAT	3	2	0	-1.5    	12	2.0	4.0	8.0	4

Align and cluster 10 k-mers with seed ACTTTAGCCGACG,   	#0
Align and cluster 9 k-mers with seed AATATAAAGAACT,   	#1
Align and cluster 9 k-mers with seed ACGAAAATATAAT,   	#2
Align and cluster 8 k-mers with seed AAAGAACTTTAGC,   	#3
Align and cluster 12 k-mers with seed ATCTGCGAATATA,   	#4

Merging redundant motifs ...
42ms: Remove motif #3.

------------------------- k = 13 ----------------------------

#0	k=13	KSM= ACTTTAGCCGACG 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACG	17.08, 4, pAUC=7.6
#1	k=13	KSM= AATATAAAGAACT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACC	11.35, 19, pAUC=14.4
#2	k=13	KSM= ATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= ATCTGCGAATATA	9.52, 6, pAUC=9.4
#3	k=13	KSM= AAAGAACTTTAGC 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGC	12.64, 10, pAUC=8.8
#4	k=13	KSM= ACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= ACGAAAATATAAT	10.65, 4, pAUC=5.6

After merging:
#0	k=13	KSM= ACTTTAGCCGACG 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACG	17.08, 4, pAUC=7.6
#1	k=13	KSM= AATATAAAGAACT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACC	11.35, 19, pAUC=14.4
#2	k=13	KSM= ATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= ATCTGCGAATATA	9.52, 6, pAUC=9.4
#4	k=13	KSM= ACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= ACGAAAATATAAT	10.65, 4, pAUC=5.6



----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=35
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

------------------------- k = 14 ----------------------------

Total number of k-mers: n = 35

Top 5 cluster center k-mers:
# k-mer/r.c.                 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
ACTTTAGCCGACGA/TCGTCGGCTAAAGT	4	3	0	-1.8    	0	3.0	10.0	30.0	5
AGATCTGCGAATAT/ATATTCGCAGATCT	3	2	0	-1.5    	6	2.0	7.0	14.0	5
AAAGAACTTTAGCC/GGCTAAAGTTCTTT	3	2	0	-1.5    	1	2.0	6.0	12.0	5
AATATAAAGAACTT/AAGTTCTTTATATT	3	2	0	-1.5    	4	2.0	6.0	12.0	5
AGCCGACGAAAATA/TATTTTCGTCGGCT	3	2	0	-1.5    	7	2.0	6.0	12.0	5

Align and cluster 11 k-mers with seed ACTTTAGCCGACGA,   	#0
Align and cluster 11 k-mers with seed AGATCTGCGAATAT,   	#1
Align and cluster 10 k-mers with seed AAAGAACTTTAGCC,   	#2
Align and cluster 10 k-mers with seed AATATAAAGAACTT,   	#3
Align and cluster 13 k-mers with seed AGCCGACGAAAATA,   	#4

Merging redundant motifs ...
50ms: Remove motif #1.
74ms: Remove motif #3.

------------------------- k = 14 ----------------------------

#0	k=14	KSM= ACTTTAGCCGACGA 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACGA	23.71, 3, pAUC=6.5
#1	k=14	KSM= AGCCGACGAAAATA 	1.00, 3, kAUC=7.0	PWM= AGCCGACGAAAATA	16.41, 4, pAUC=5.8
#2	k=14	KSM= AATATAAAGAACTT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACTT	10.35, 6, pAUC=9.6
#3	k=14	KSM= AAAGAACTTTAGCC 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCC	16.60, 4, pAUC=6.5
#4	k=14	KSM= AGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATAT	21.47, 2, pAUC=4.6

After merging:
#0	k=14	KSM= ACTTTAGCCGACGA 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACGA	23.71, 3, pAUC=6.5
#2	k=14	KSM= AATATAAAGAACTT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACTT	10.35, 6, pAUC=9.6
#4	k=14	KSM= AGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATAT	21.47, 2, pAUC=4.6



----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=34
k=15+1, gapped kmer=0, base kmer=0
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

------------------------- k = 15 ----------------------------

Total number of k-mers: n = 34

Top 5 cluster center k-mers:
# k-mer/r.c.                   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAACTTTAGCCG/CGGCTAAAGTTCTTT	3	2	0	-1.5    	0	2.0	13.0	26.0	5
AACTTTAGCCGACGA/TCGTCGGCTAAAGTT	3	2	0	-1.5    	1	2.0	8.0	16.0	5
AGATCTGCGAATATA/TATATTCGCAGATCT	3	2	0	-1.5    	6	2.0	8.0	16.0	5
AGCCGACGAAAATAT/ATATTTTCGTCGGCT	3	2	0	-1.5    	7	2.0	8.0	16.0	5
CTGTCCAGATCTGCG/CGCAGATCTGGACAG	3	2	0	-1.5    	17	2.0	7.0	14.0	3

Align and cluster 10 k-mers with seed AAAGAACTTTAGCCG,   	#0
Align and cluster 9 k-mers with seed AACTTTAGCCGACGA,   	#1
Align and cluster 8 k-mers with seed AGATCTGCGAATATA,   	#2
Align and cluster 9 k-mers with seed AGCCGACGAAAATAT,   	#3
Align and cluster 5 k-mers with seed CTGTCCAGATCTGCG,   	#4

Merging redundant motifs ...
11ms: PWM AACTTTAGCCGACGA has been removed.
14ms: PWM CTGTCCAGATCTGCG has been removed.

------------------------- k = 15 ----------------------------

#0	k=15	KSM= AGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATAT	18.83, 4, pAUC=6.7
#1	k=15	KSM= AAAGAACTTTAGCCG 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCG	23.34, 2, pAUC=5.8
#2	k=15	KSM= AGATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATA	22.82, 2, pAUC=5.0
#3	k=15	KSM= AACTTTAGCCGACGA 	1.21, 2, kAUC=6.5	PWM= AACTTTAGCCGACGA	23.95, 3, pAUC=4.7
#4	k=15	KSM= CTGTCCAGATCTGCG 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCG	24.92, 2, pAUC=4.4

After merging:
#0	k=15	KSM= AGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATAT	18.83, 4, pAUC=6.7
#1	k=15	KSM= AAAGAACTTTAGCCG 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCG	23.34, 2, pAUC=5.8
#2	k=15	KSM= AGATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATA	22.82, 2, pAUC=5.0



----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=33
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

------------------------- k = 16 ----------------------------

Total number of k-mers: n = 33

Top 5 cluster center k-mers:
# k-mer/r.c.                     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAACTTTAGCCGA/TCGGCTAAAGTTCTTT	3	2	0	-1.5    	0	2.0	12.0	24.0	5
AGATCTGCGAATATAA/TTATATTCGCAGATCT	3	2	0	-1.5    	6	2.0	9.0	18.0	5
AACTTTAGCCGACGAA/TTCGTCGGCTAAAGTT	3	2	0	-1.5    	1	2.0	8.0	16.0	5
AATATAAAGAACTTTA/TAAAGTTCTTTATATT	3	2	0	-1.5    	3	2.0	8.0	16.0	5
AGCCGACGAAAATATA/TATATTTTCGTCGGCT	3	2	0	-1.5    	7	2.0	8.0	16.0	5

Align and cluster 12 k-mers with seed AAAGAACTTTAGCCGA,   	#0
Align and cluster 11 k-mers with seed AGATCTGCGAATATAA,   	#1
Align and cluster 10 k-mers with seed AACTTTAGCCGACGAA,   	#2
Align and cluster 11 k-mers with seed AATATAAAGAACTTTA,   	#3
Align and cluster 10 k-mers with seed AGCCGACGAAAATATA,   	#4

Merging redundant motifs ...
11ms: PWM AAAGAACTTTAGCCGA has been removed.
14ms: PWM AACTTTAGCCGACGAA has been removed.

------------------------- k = 16 ----------------------------

#0	k=16	KSM= AATATAAAGAACTTTA 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTA	20.91, 3, pAUC=10.1
#1	k=16	KSM= AGCCGACGAAAATATA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATA	19.18, 3, pAUC=5.7
#2	k=16	KSM= AAAGAACTTTAGCCGA 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCGA	24.69, 2, pAUC=4.3
#3	k=16	KSM= AGATCTGCGAATATAA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAA	24.16, 2, pAUC=4.2
#4	k=16	KSM= AACTTTAGCCGACGAA 	1.21, 2, kAUC=6.5	PWM= AACTTTAGCCGACGAA	24.19, 3, pAUC=4.2

After merging:
#0	k=16	KSM= AATATAAAGAACTTTA 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTA	20.91, 3, pAUC=10.1
#1	k=16	KSM= AGCCGACGAAAATATA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATA	19.18, 3, pAUC=5.7
#3	k=16	KSM= AGATCTGCGAATATAA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAA	24.16, 2, pAUC=4.2



----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=32
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

------------------------- k = 17 ----------------------------

Total number of k-mers: n = 32

Top 5 cluster center k-mers:
# k-mer/r.c.                       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAACTTTAGCCGAC/GTCGGCTAAAGTTCTTT	3	2	0	-1.5    	0	2.0	13.0	26.0	5
AATATAAAGAACTTTAG/CTAAAGTTCTTTATATT	3	2	0	-1.5    	3	2.0	10.0	20.0	5
AGATCTGCGAATATAAA/TTTATATTCGCAGATCT	3	2	0	-1.5    	6	2.0	9.0	18.0	5
AGCCGACGAAAATATAA/TTATATTTTCGTCGGCT	3	2	0	-1.5    	7	2.0	9.0	18.0	4
CTGTCCAGATCTGCGAA/TTCGCAGATCTGGACAG	3	2	0	-1.5    	15	2.0	8.0	16.0	3

Align and cluster 11 k-mers with seed AAAGAACTTTAGCCGAC,   	#0
Align and cluster 9 k-mers with seed AATATAAAGAACTTTAG,   	#1
Align and cluster 10 k-mers with seed AGATCTGCGAATATAAA,   	#2
Align and cluster 7 k-mers with seed AGCCGACGAAAATATAA,   	#3
Align and cluster 5 k-mers with seed CTGTCCAGATCTGCGAA,   	#4

Merging redundant motifs ...
22ms: PWM AAAGAACTTTAGCCGAC has been removed.
29ms: PWM AGATCTGCGAATATAAA has been removed.

------------------------- k = 17 ----------------------------

#0	k=17	KSM= AGCCGACGAAAATATAA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAA	26.04, 2, pAUC=6.0
#1	k=17	KSM= AATATAAAGAACTTTAG 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAG	24.46, 2, pAUC=6.0
#2	k=17	KSM= AAAGAACTTTAGCCGAC 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCGAC	26.56, 2, pAUC=5.8
#3	k=17	KSM= CTGTCCAGATCTGCGAA 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAA	27.61, 2, pAUC=4.8
#4	k=17	KSM= AGATCTGCGAATATAAA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAAA	25.51, 2, pAUC=4.5

After merging:
#0	k=17	KSM= AGCCGACGAAAATATAA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAA	26.04, 2, pAUC=6.0
#1	k=17	KSM= AATATAAAGAACTTTAG 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAG	24.46, 2, pAUC=6.0
#3	k=17	KSM= CTGTCCAGATCTGCGAA 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAA	27.61, 2, pAUC=4.8



----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=31
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

------------------------- k = 18 ----------------------------

Total number of k-mers: n = 31

Top 5 cluster center k-mers:
# k-mer/r.c.                         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAACTTTAGCCGACG/CGTCGGCTAAAGTTCTTT	3	2	0	-1.5    	0	2.0	14.0	28.0	7
AATATAAAGAACTTTAGC/GCTAAAGTTCTTTATATT	3	2	0	-1.5    	3	2.0	10.0	20.0	7
AGATCTGCGAATATAAAG/CTTTATATTCGCAGATCT	3	2	0	-1.5    	6	2.0	10.0	20.0	7
AGCCGACGAAAATATAAT/ATTATATTTTCGTCGGCT	3	2	0	-1.5    	7	2.0	9.0	18.0	4
CTGTCCAGATCTGCGAAT/ATTCGCAGATCTGGACAG	3	2	0	-1.5    	15	2.0	8.0	16.0	4

Align and cluster 11 k-mers with seed AAAGAACTTTAGCCGACG,   	#0
Align and cluster 9 k-mers with seed AATATAAAGAACTTTAGC,   	#1
Align and cluster 10 k-mers with seed AGATCTGCGAATATAAAG,   	#2
Align and cluster 7 k-mers with seed AGCCGACGAAAATATAAT,   	#3
Align and cluster 6 k-mers with seed CTGTCCAGATCTGCGAAT,   	#4

Merging redundant motifs ...
12ms: PWM AGATCTGCGAATATAAAG has been removed.
41ms: Remove motif #4.

------------------------- k = 18 ----------------------------

#0	k=18	KSM= AGCCGACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAAT	27.38, 2, pAUC=7.3
#1	k=18	KSM= AATATAAAGAACTTTAGC 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAGC	26.33, 2, pAUC=6.2
#2	k=18	KSM= AGATCTGCGAATATAAAG 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAAAG	27.38, 2, pAUC=5.6
#3	k=18	KSM= CTGTCCAGATCTGCGAAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAAT	28.96, 2, pAUC=5.1
#4	k=18	KSM= AAAGAACTTTAGCCGACG 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCGACG	22.42, 3, pAUC=4.7

After merging:
#0	k=18	KSM= AGCCGACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAAT	27.38, 2, pAUC=7.3
#1	k=18	KSM= AATATAAAGAACTTTAGC 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAGC	26.33, 2, pAUC=6.2
#3	k=18	KSM= CTGTCCAGATCTGCGAAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAAT	28.96, 2, pAUC=5.1



----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=30
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

------------------------- k = 19 ----------------------------

Total number of k-mers: n = 30

Top 5 cluster center k-mers:
# k-mer/r.c.                           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAACTTTAGCCGACGA/TCGTCGGCTAAAGTTCTTT	3	2	0	-1.5    	0	2.0	14.0	28.0	7
AATATAAAGAACTTTAGCC/GGCTAAAGTTCTTTATATT	3	2	0	-1.5    	3	2.0	10.0	20.0	7
AGATCTGCGAATATAAAGA/TCTTTATATTCGCAGATCT	3	2	0	-1.5    	6	2.0	10.0	20.0	7
CTGTCCAGATCTGCGAATA/TATTCGCAGATCTGGACAG	3	2	0	-1.5    	14	2.0	8.0	16.0	4
ACTTTAGCCGACGAAAATA/TATTTTCGTCGGCTAAAGT	3	2	0	-1.5    	4	2.0	2.0	4.0	6

Align and cluster 11 k-mers with seed AAAGAACTTTAGCCGACGA,   	#0
Align and cluster 9 k-mers with seed AATATAAAGAACTTTAGCC,   	#1
Align and cluster 10 k-mers with seed AGATCTGCGAATATAAAGA,   	#2
Align and cluster 5 k-mers with seed CTGTCCAGATCTGCGAATA,   	#3
Align and cluster 9 k-mers with seed ACTTTAGCCGACGAAAATA,   	#4

Merging redundant motifs ...
23ms: PWM AATATAAAGAACTTTAGCC has been removed.
30ms: PWM CTGTCCAGATCTGCGAATA has been removed.
37ms: PWM AAAGAACTTTAGCCGACGA has been removed.

------------------------- k = 19 ----------------------------

#0	k=19	KSM= AGATCTGCGAATATAAAGA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAAAGA	28.73, 2, pAUC=5.6
#1	k=19	KSM= ACTTTAGCCGACGAAAATA 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATA	24.52, 3, pAUC=5.6
#2	k=19	KSM= AAAGAACTTTAGCCGACGA 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCGACGA	23.85, 3, pAUC=5.4
#3	k=19	KSM= AATATAAAGAACTTTAGCC 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAGCC	28.20, 2, pAUC=5.2
#4	k=19	KSM= CTGTCCAGATCTGCGAATA 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAATA	30.30, 2, pAUC=4.6

After merging:
#0	k=19	KSM= AGATCTGCGAATATAAAGA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAAAGA	28.73, 2, pAUC=5.6
#1	k=19	KSM= ACTTTAGCCGACGAAAATA 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATA	24.52, 3, pAUC=5.6



----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=29
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

------------------------- k = 20 ----------------------------

Total number of k-mers: n = 29

Top 5 cluster center k-mers:
# k-mer/r.c.                             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAGAACTTTAGCCGACGAA/TTCGTCGGCTAAAGTTCTTT	3	2	0	-1.5    	0	2.0	14.0	28.0	7
AGATCTGCGAATATAAAGAA/TTCTTTATATTCGCAGATCT	3	2	0	-1.5    	6	2.0	11.0	22.0	7
AATATAAAGAACTTTAGCCG/CGGCTAAAGTTCTTTATATT	3	2	0	-1.5    	3	2.0	10.0	20.0	7
CTGTCCAGATCTGCGAATAT/ATATTCGCAGATCTGGACAG	3	2	0	-1.5    	14	2.0	8.0	16.0	4
ACTTTAGCCGACGAAAATAT/ATATTTTCGTCGGCTAAAGT	3	2	0	-1.5    	4	2.0	2.0	4.0	6

Align and cluster 13 k-mers with seed AAAGAACTTTAGCCGACGAA,   	#0
Align and cluster 12 k-mers with seed AGATCTGCGAATATAAAGAA,   	#1
Align and cluster 11 k-mers with seed AATATAAAGAACTTTAGCCG,   	#2
Align and cluster 8 k-mers with seed CTGTCCAGATCTGCGAATAT,   	#3
Align and cluster 9 k-mers with seed ACTTTAGCCGACGAAAATAT,   	#4

Merging redundant motifs ...
25ms: PWM AATATAAAGAACTTTAGCCG has been removed.
32ms: PWM AAAGAACTTTAGCCGACGAA has been removed.
39ms: PWM AGATCTGCGAATATAAAGAA has been removed.

------------------------- k = 20 ----------------------------

#0	k=20	KSM= ACTTTAGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATAT	25.95, 3, pAUC=7.7
#1	k=20	KSM= CTGTCCAGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAATAT	31.65, 2, pAUC=5.9
#2	k=20	KSM= AATATAAAGAACTTTAGCCG 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAGCCG	30.07, 2, pAUC=5.5
#3	k=20	KSM= AAAGAACTTTAGCCGACGAA 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCGACGAA	31.12, 2, pAUC=5.3
#4	k=20	KSM= AGATCTGCGAATATAAAGAA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAAAGAA	30.07, 2, pAUC=5.1

After merging:
#0	k=20	KSM= ACTTTAGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATAT	25.95, 3, pAUC=7.7
#1	k=20	KSM= CTGTCCAGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAATAT	31.65, 2, pAUC=5.9



------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= ATTTTTTT 	1.32, 90, kAUC=67.9	PWM= TTTTTT	5.92, 121, pAUC=5.1
#1	k=8	KSM= TTTCAATT 	1.54, 30, kAUC=51.6	PWM= ATCAAA	5.43, 112, pAUC=8.1
#2	k=7	KSM= AAAGAAA 	1.49, 49, kAUC=40.3	PWM= AAAGAAA	6.82, 79, pAUC=8.1
#3	k=7	KSM= TTATTCA 	1.32, 56, kAUC=38.6	PWM= TTATTCA	7.51, 41, pAUC=11.3
#4	k=6	KSM= CAAATA 	1.21, 75, kAUC=37.8	PWM= CAAATA	5.72, 71, pAUC=12.4
#5	k=7	KSM= ATCAAAT 	1.39, 46, kAUC=37.6	PWM= ATCAAAT	7.13, 67, pAUC=11.1
#6	k=8	KSM= GATTCATT 	1.31, 31, kAUC=37.4	PWM= ATTCAT	6.88, 72, pAUC=13.2
#7	k=6	KSM= AAAGAA 	1.54, 65, kAUC=36.3	PWM= AAAGAA	5.98, 148, pAUC=7.4
#8	k=6	KSM= TTTATC 	1.21, 94, kAUC=33.3	PWM= TTTATC	5.93, 114, pAUC=10.1
#9	k=9	KSM= AATAAAAAA 	1.32, 30, kAUC=30.4	PWM= AATAAAAAA	9.80, 36, pAUC=11.3
#10	k=9	KSM= ATGAATTGA 	1.11, 49, kAUC=29.1	PWM= AAGAATTGA	6.90, 58, pAUC=12.6
#11	k=7	KSM= AAAATTG 	1.32, 36, kAUC=27.2	PWM= AAATT	5.96, 144, pAUC=5.0
#12	k=10	KSM= AATTATGAAA 	1.21, 25, kAUC=20.3	PWM= AAAAATGAAA	9.44, 40, pAUC=11.3
#13	k=9	KSM= AATTCTTGG 	1.11, 24, kAUC=17.8	PWM= ATTCTTG	7.50, 39, pAUC=8.6
#14	k=9	KSM= ATATAATTT 	1.11, 19, kAUC=15.8	PWM= ATATAATTT	8.99, 19, pAUC=10.1
#15	k=10	KSM= TTAATATTAA 	1.11, 20, kAUC=15.8	PWM= TTAAAAATAA	8.35, 33, pAUC=7.4
#16	k=11	KSM= GAAAAAAAAAA 	1.00, 16, kAUC=12.7	PWM= GAAAAAAAAAA	9.39, 39, pAUC=13.4
#17	k=7	KSM= AGATCTA 	1.00, 19, kAUC=12.3	PWM= AGATCTA	6.94, 23, pAUC=10.8
#18	k=12	KSM= AATATAAAGAAC 	1.00, 6, kAUC=9.1	PWM= AAGAGAAAGAAC	9.55, 31, pAUC=11.3
#19	k=12	KSM= TATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= TATATGAATTGA	10.23, 21, pAUC=12.6
#20	k=11	KSM= ATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= ATATGAATTGA	9.76, 18, pAUC=9.1
#21	k=11	KSM= TTAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TTAGCCGACGA	11.26, 5, pAUC=8.9
#22	k=10	KSM= TAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TAGCCGACGA	10.89, 6, pAUC=8.5
#23	k=13	KSM= ACTTTAGCCGACG 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACG	17.08, 4, pAUC=7.6
#24	k=14	KSM= ACTTTAGCCGACGA 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACGA	23.71, 3, pAUC=6.5
#25	k=12	KSM= ACTTTAGCCGAC 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGAC	15.06, 4, pAUC=6.4
#26	k=13	KSM= AATATAAAGAACT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACC	11.35, 19, pAUC=14.4
#27	k=16	KSM= AATATAAAGAACTTTA 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTA	20.91, 3, pAUC=10.1
#28	k=14	KSM= AATATAAAGAACTT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACTT	10.35, 6, pAUC=9.6
#29	k=13	KSM= ATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= ATCTGCGAATATA	9.52, 6, pAUC=9.4
#30	k=10	KSM= ATTGAACTTA 	1.32, 3, kAUC=6.5	PWM= ATTGAACCTA	8.83, 13, pAUC=8.3
#31	k=11	KSM= ACAAATATCTT 	1.21, 3, kAUC=6.5	PWM= ACAGATATCTT	10.82, 10, pAUC=7.8
#32	k=20	KSM= ACTTTAGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATAT	25.95, 3, pAUC=7.7
#33	k=18	KSM= AGCCGACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAAT	27.38, 2, pAUC=7.3
#34	k=15	KSM= AGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATAT	18.83, 4, pAUC=6.7
#35	k=18	KSM= AATATAAAGAACTTTAGC 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAGC	26.33, 2, pAUC=6.2
#36	k=17	KSM= AGCCGACGAAAATATAA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAA	26.04, 2, pAUC=6.0
#37	k=17	KSM= AATATAAAGAACTTTAG 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAG	24.46, 2, pAUC=6.0
#38	k=20	KSM= CTGTCCAGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAATAT	31.65, 2, pAUC=5.9
#39	k=15	KSM= AAAGAACTTTAGCCG 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCG	23.34, 2, pAUC=5.8
#40	k=16	KSM= AGCCGACGAAAATATA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATA	19.18, 3, pAUC=5.7
#41	k=19	KSM= AGATCTGCGAATATAAAGA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAAAGA	28.73, 2, pAUC=5.6
#42	k=19	KSM= ACTTTAGCCGACGAAAATA 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATA	24.52, 3, pAUC=5.6
#43	k=13	KSM= ACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= ACGAAAATATAAT	10.65, 4, pAUC=5.6
#44	k=18	KSM= CTGTCCAGATCTGCGAAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAAT	28.96, 2, pAUC=5.1
#45	k=15	KSM= AGATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATA	22.82, 2, pAUC=5.0
#46	k=17	KSM= CTGTCCAGATCTGCGAA 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAA	27.61, 2, pAUC=4.8
#47	k=14	KSM= AGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATAT	21.47, 2, pAUC=4.6
#48	k=16	KSM= AGATCTGCGAATATAA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAA	24.16, 2, pAUC=4.2
#49	k=9	KSM= AATGCAGTT 	1.32, 3, kAUC=6.5	PWM= AATGCAGTT	9.39, 5, pAUC=3.2
#50	k=12	KSM= AAACCTAGGTTT 	1.11, 2, kAUC=6.0	PWM= AAACCTCGGTTT	10.08, 6, pAUC=5.4

86ms: Remove motif #2.
109ms: Remove motif #7.
131ms: Remove motif #9.
155ms: Remove motif #11.
203ms: Remove motif #12.
225ms: Remove motif #15.
247ms: Remove motif #16.
268ms: Remove motif #33.
289ms: Remove motif #34.
310ms: Remove motif #36.
373ms: Remove motif #40.
403ms: Remove motif #43.
454ms: Remove motif #29.
483ms: Remove motif #27.
487ms: PWM GCTAAAGTTCTTTATATT has been removed.
491ms: PWM CTAAAGTTCTTTATATT has been removed.
494ms: PWM CTGTCCAGATCTGCGAATAT has been removed.
498ms: PWM CGGCTAAAGTTCTTT has been removed.
502ms: PWM CTGTCCAGATCTGCGAAT has been removed.
505ms: PWM CTGTCCAGATCTGCGAA has been removed.
547ms: Remove motif #26.
579ms: Remove motif #28.
583ms: PWM AGATCTGCGAATATAAAGA has been removed.
613ms: Remove motif #22.
642ms: Remove motif #23.
646ms: PWM ACTTTAGCCGACGA has been removed.
675ms: Remove motif #25.
680ms: PWM ACTTTAGCCGACGAAAATAT has been removed.
684ms: PWM ACTTTAGCCGACGAAAATA has been removed.
689ms: PWM AGATCTGCGAATAT has been removed.
693ms: PWM AGATCTGCGAATATAA has been removed.

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=8	KSM= ATTTTTTT 	1.32, 90, kAUC=67.9	PWM= TTTTTT	5.92, 121, pAUC=5.1
#1	k=8	KSM= TTTCAATT 	1.54, 30, kAUC=51.6	PWM= ATCAAA	5.43, 112, pAUC=8.1
#2	k=7	KSM= AAAGAAA 	1.49, 49, kAUC=40.3	PWM= AAAGAAA	6.82, 79, pAUC=8.1
#3	k=7	KSM= TTATTCA 	1.32, 56, kAUC=38.6	PWM= TTATTCA	7.51, 41, pAUC=11.3
#4	k=6	KSM= CAAATA 	1.21, 75, kAUC=37.8	PWM= CAAATA	5.72, 71, pAUC=12.4
#5	k=7	KSM= ATCAAAT 	1.39, 46, kAUC=37.6	PWM= ATCAAAT	7.13, 67, pAUC=11.1
#6	k=8	KSM= GATTCATT 	1.31, 31, kAUC=37.4	PWM= ATTCAT	6.88, 72, pAUC=13.2
#7	k=6	KSM= AAAGAA 	1.54, 65, kAUC=36.3	PWM= AAAGAA	5.98, 148, pAUC=7.4
#8	k=6	KSM= TTTATC 	1.21, 94, kAUC=33.3	PWM= TTTATC	5.93, 114, pAUC=10.1
#9	k=9	KSM= AATAAAAAA 	1.32, 30, kAUC=30.4	PWM= AATAAAAAA	9.80, 36, pAUC=11.3
#10	k=9	KSM= ATGAATTGA 	1.11, 49, kAUC=29.1	PWM= AAGAATTGA	6.90, 58, pAUC=12.6
#11	k=7	KSM= AAAATTG 	1.32, 36, kAUC=27.2	PWM= AAATT	5.96, 144, pAUC=5.0
#12	k=10	KSM= AATTATGAAA 	1.21, 25, kAUC=20.3	PWM= AAAAATGAAA	9.44, 40, pAUC=11.3
#13	k=9	KSM= AATTCTTGG 	1.11, 24, kAUC=17.8	PWM= ATTCTTG	7.50, 39, pAUC=8.6
#14	k=9	KSM= ATATAATTT 	1.11, 19, kAUC=15.8	PWM= ATATAATTT	8.99, 19, pAUC=10.1
#15	k=10	KSM= TTAATATTAA 	1.11, 20, kAUC=15.8	PWM= TTAAAAATAA	8.35, 33, pAUC=7.4
#16	k=11	KSM= GAAAAAAAAAA 	1.00, 16, kAUC=12.7	PWM= GAAAAAAAAAA	9.39, 39, pAUC=13.4
#17	k=7	KSM= AGATCTA 	1.00, 19, kAUC=12.3	PWM= AGATCTA	6.94, 23, pAUC=10.8
#18	k=12	KSM= AATATAAAGAAC 	1.00, 6, kAUC=9.1	PWM= AAGAGAAAGAAC	9.55, 31, pAUC=11.3
#19	k=12	KSM= TATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= TATATGAATTGA	10.23, 21, pAUC=12.6
#20	k=11	KSM= ATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= ATATGAATTGA	9.76, 18, pAUC=9.1
#21	k=11	KSM= TTAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TTAGCCGACGA	11.26, 5, pAUC=8.9
#22	k=10	KSM= TAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TAGCCGACGA	10.89, 6, pAUC=8.5
#23	k=13	KSM= ACTTTAGCCGACG 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACG	17.08, 4, pAUC=7.6
#24	k=14	KSM= ACTTTAGCCGACGA 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGACGA	23.71, 3, pAUC=6.5
#25	k=12	KSM= ACTTTAGCCGAC 	1.00, 3, kAUC=7.0	PWM= ACTTTAGCCGAC	15.06, 4, pAUC=6.4
#26	k=13	KSM= AATATAAAGAACT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACC	11.35, 19, pAUC=14.4
#27	k=16	KSM= AATATAAAGAACTTTA 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTA	20.91, 3, pAUC=10.1
#28	k=14	KSM= AATATAAAGAACTT 	1.21, 2, kAUC=6.5	PWM= AATAGAAAGAACTT	10.35, 6, pAUC=9.6
#29	k=13	KSM= ATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= ATCTGCGAATATA	9.52, 6, pAUC=9.4
#30	k=10	KSM= ATTGAACTTA 	1.32, 3, kAUC=6.5	PWM= ATTGAACCTA	8.83, 13, pAUC=8.3
#31	k=11	KSM= ACAAATATCTT 	1.21, 3, kAUC=6.5	PWM= ACAGATATCTT	10.82, 10, pAUC=7.8
#32	k=20	KSM= ACTTTAGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATAT	25.95, 3, pAUC=7.7
#33	k=18	KSM= AGCCGACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAAT	27.38, 2, pAUC=7.3
#34	k=15	KSM= AGCCGACGAAAATAT 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATAT	18.83, 4, pAUC=6.7
#35	k=18	KSM= AATATAAAGAACTTTAGC 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAGC	26.33, 2, pAUC=6.2
#36	k=17	KSM= AGCCGACGAAAATATAA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATAA	26.04, 2, pAUC=6.0
#37	k=17	KSM= AATATAAAGAACTTTAG 	1.21, 2, kAUC=6.5	PWM= AATATAAAGAACTTTAG	24.46, 2, pAUC=6.0
#38	k=20	KSM= CTGTCCAGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAATAT	31.65, 2, pAUC=5.9
#39	k=15	KSM= AAAGAACTTTAGCCG 	1.21, 2, kAUC=6.5	PWM= AAAGAACTTTAGCCG	23.34, 2, pAUC=5.8
#40	k=16	KSM= AGCCGACGAAAATATA 	1.21, 2, kAUC=6.5	PWM= AGCCGACGAAAATATA	19.18, 3, pAUC=5.7
#41	k=19	KSM= AGATCTGCGAATATAAAGA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAAAGA	28.73, 2, pAUC=5.6
#42	k=19	KSM= ACTTTAGCCGACGAAAATA 	1.21, 2, kAUC=6.5	PWM= ACTTTAGCCGACGAAAATA	24.52, 3, pAUC=5.6
#43	k=13	KSM= ACGAAAATATAAT 	1.21, 2, kAUC=6.5	PWM= ACGAAAATATAAT	10.65, 4, pAUC=5.6
#44	k=18	KSM= CTGTCCAGATCTGCGAAT 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAAT	28.96, 2, pAUC=5.1
#45	k=15	KSM= AGATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATA	22.82, 2, pAUC=5.0
#46	k=17	KSM= CTGTCCAGATCTGCGAA 	1.21, 2, kAUC=6.5	PWM= CTGTCCAGATCTGCGAA	27.61, 2, pAUC=4.8
#47	k=14	KSM= AGATCTGCGAATAT 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATAT	21.47, 2, pAUC=4.6
#48	k=16	KSM= AGATCTGCGAATATAA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATAA	24.16, 2, pAUC=4.2
#49	k=9	KSM= AATGCAGTT 	1.32, 3, kAUC=6.5	PWM= AATGCAGTT	9.39, 5, pAUC=3.2
#50	k=12	KSM= AAACCTAGGTTT 	1.11, 2, kAUC=6.0	PWM= AAACCTCGGTTT	10.08, 6, pAUC=5.4

------------------------- GEMout_1, k = ALL (after merging) -------------------------
#0	k=8	KSM= ATTTTTTT 	1.32, 93, kAUC=72.0	PWM= TTTTTT	5.92, 108, pAUC=5.1
#1	k=8	KSM= TTTCAATT 	1.54, 30, kAUC=51.6	PWM= ATCAAA	5.43, 112, pAUC=8.1
#3	k=7	KSM= TTATTCA 	1.32, 77, kAUC=47.7	PWM= TATTCA	5.99, 103, pAUC=9.2
#4	k=6	KSM= CAAATA 	1.21, 75, kAUC=37.8	PWM= CAAATA	5.72, 71, pAUC=12.4
#5	k=7	KSM= ATCAAAT 	1.39, 46, kAUC=37.6	PWM= ATCAAAT	7.13, 67, pAUC=11.1
#6	k=8	KSM= GATTCATT 	1.31, 31, kAUC=37.4	PWM= ATTCAT	6.88, 72, pAUC=13.2
#8	k=6	KSM= TTTATC 	1.21, 94, kAUC=33.3	PWM= TTTATC	5.93, 114, pAUC=10.1
#10	k=9	KSM= ATGAATTGA 	1.11, 49, kAUC=29.1	PWM= AAGAATTGA	6.90, 58, pAUC=12.6
#13	k=9	KSM= AATTCTTGG 	1.11, 24, kAUC=17.8	PWM= ATTCTTG	7.50, 39, pAUC=8.6
#14	k=9	KSM= ATATAATTT 	1.11, 19, kAUC=15.8	PWM= ATATAATTT	8.99, 19, pAUC=10.1
#17	k=7	KSM= AGATCTA 	1.00, 19, kAUC=12.3	PWM= AGATCTA	6.94, 23, pAUC=10.8
#18	k=12	KSM= AATATAAAGAAC 	1.00, 6, kAUC=9.1	PWM= AAGAGAAAGAAC	9.55, 31, pAUC=11.3
#19	k=12	KSM= TATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= TATATGAATTGA	10.23, 21, pAUC=12.6
#20	k=11	KSM= ATATGAATTGA 	1.32, 3, kAUC=7.0	PWM= ATATGAATTGA	9.76, 18, pAUC=9.1
#21	k=11	KSM= TTAGCCGACGA 	1.11, 3, kAUC=7.0	PWM= TTAGCCGACGA	11.26, 5, pAUC=8.9
#30	k=10	KSM= ATTGAACTTA 	1.32, 3, kAUC=6.5	PWM= ATTGAACCTA	8.83, 13, pAUC=8.3
#31	k=11	KSM= ACAAATATCTT 	1.21, 3, kAUC=6.5	PWM= ACAGATATCTT	10.82, 10, pAUC=7.8
#45	k=15	KSM= AGATCTGCGAATATA 	1.21, 2, kAUC=6.5	PWM= AGATCTGCGAATATA	22.82, 2, pAUC=5.0
#49	k=9	KSM= AATGCAGTT 	1.32, 3, kAUC=6.5	PWM= AATGCAGTT	9.39, 5, pAUC=3.2
#50	k=12	KSM= AAACCTAGGTTT 	1.11, 2, kAUC=6.0	PWM= AAACCTCGGTTT	10.08, 6, pAUC=5.4

836ms: Finalizing 20 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_1 final motifs ----------------------------


GEMout_1 motif #0

KSM top k-mer: ATTTTTTT/AAAAAAAT, total 120 k-mers
KSM threshold: 1.32, 		hit=93+/0-, kAUC=72.0

    |
TTTTTT
      
      
      
PWM threshold: 5.92/8.87, 	hit=108+/109-, pAUC=5.1
--------------------------------------------------------------

GEMout_1 motif #1

KSM top k-mer: TTTCAATT/AATTGAAA, total 87 k-mers
KSM threshold: 1.54, 		hit=30+/0-, kAUC=51.6

 |
ATCAAA
T  C T
      
      
PWM threshold: 5.43/8.71, 	hit=112+/107-, pAUC=8.1
--------------------------------------------------------------

GEMout_1 motif #2

KSM top k-mer: TTATTCA/TGAATAA, total 45 k-mers
KSM threshold: 1.32, 		hit=77+/3-, kAUC=47.7

      |
TATTCA
      
      
      
PWM threshold: 5.99/8.60, 	hit=103+/91-, pAUC=9.2
--------------------------------------------------------------

GEMout_1 motif #3

KSM top k-mer: CAAATA/TATTTG, total 40 k-mers
KSM threshold: 1.21, 		hit=75+/6-, kAUC=37.8

   |
CAAATA
      
      
      
PWM threshold: 5.72/9.81, 	hit=71+/39-, pAUC=12.4
--------------------------------------------------------------

GEMout_1 motif #4

KSM top k-mer: ATCAAAT/ATTTGAT, total 28 k-mers
KSM threshold: 1.39, 		hit=46+/1-, kAUC=37.6

      |
ATCAAAT
  G   A
       
       
PWM threshold: 7.13/10.73, 	hit=67+/43-, pAUC=11.1
--------------------------------------------------------------

GEMout_1 motif #5

KSM top k-mer: GATTCATT/AATGAATC, total 41 k-mers
KSM threshold: 1.31, 		hit=31+/0-, kAUC=37.4

ATTCAT
PWM threshold: 6.88/10.06, 	hit=72+/46-, pAUC=13.2
--------------------------------------------------------------

GEMout_1 motif #6

KSM top k-mer: TTTATC/GATAAA, total 41 k-mers
KSM threshold: 1.21, 		hit=94+/15-, kAUC=33.3

    |
TTTATC
   T  
      
      
PWM threshold: 5.93/8.71, 	hit=114+/104-, pAUC=10.1
--------------------------------------------------------------

GEMout_1 motif #7

KSM top k-mer: ATGAATTGA/TCAATTCAT, total 21 k-mers
KSM threshold: 1.11, 		hit=49+/0-, kAUC=29.1

        |
AAGAATTGA
     AAA 
         
         
PWM threshold: 6.90/9.88, 	hit=58+/40-, pAUC=12.6
--------------------------------------------------------------

GEMout_1 motif #8

KSM top k-mer: AATTCTTGG/CCAAGAATT, total 10 k-mers
KSM threshold: 1.11, 		hit=24+/0-, kAUC=17.8

 |
ATTCTTG
      T
       
       
PWM threshold: 7.50/10.75, 	hit=39+/28-, pAUC=8.6
--------------------------------------------------------------

GEMout_1 motif #9

KSM top k-mer: ATATAATTT/AAATTATAT, total 11 k-mers
KSM threshold: 1.11, 		hit=19+/0-, kAUC=15.8

ATATAATTT
PWM threshold: 8.99/13.77, 	hit=19+/6-, pAUC=10.1
--------------------------------------------------------------

GEMout_1 motif #10

KSM top k-mer: AGATCTA/TAGATCT, total 6 k-mers
KSM threshold: 1.00, 		hit=19+/2-, kAUC=12.3

          |
AGATCTA
  T    
       
       
PWM threshold: 6.94/11.51, 	hit=23+/10-, pAUC=10.8
--------------------------------------------------------------

GEMout_1 motif #11

KSM top k-mer: AATATAAAGAAC/GTTCTTTATATT, total 11 k-mers
KSM threshold: 1.00, 		hit=6+/0-, kAUC=9.1

               |
AAGAGAAAGAAC
  T T    G A
            
            
PWM threshold: 9.55/15.04, 	hit=31+/14-, pAUC=11.3
--------------------------------------------------------------

GEMout_1 motif #12

KSM top k-mer: TATATGAATTGA/TCAATTCATATA, total 1 k-mers
KSM threshold: 1.32, 		hit=3+/0-, kAUC=7.0

         |
TATATGAATTGA
            
            
            
PWM threshold: 10.23/17.87, 	hit=21+/2-, pAUC=12.6
--------------------------------------------------------------

GEMout_1 motif #13

KSM top k-mer: ATATGAATTGA/TCAATTCATAT, total 2 k-mers
KSM threshold: 1.32, 		hit=3+/0-, kAUC=7.0

        |
ATATGAATTGA
        GA 
           
           
PWM threshold: 9.76/16.68, 	hit=18+/8-, pAUC=9.1
--------------------------------------------------------------

GEMout_1 motif #14

KSM top k-mer: ATTGAACTTA/TAAGTTCAAT, total 1 k-mers
KSM threshold: 1.32, 		hit=3+/0-, kAUC=6.5

ATTGAACCTA
C      T
PWM threshold: 8.83/14.96, 	hit=13+/4-, pAUC=8.3
--------------------------------------------------------------

GEMout_1 motif #15

KSM top k-mer: ACAAATATCTT/AAGATATTTGT, total 1 k-mers
KSM threshold: 1.21, 		hit=3+/0-, kAUC=6.5

ACAGATATCTT
 TCA
PWM threshold: 10.82/16.13, 	hit=10+/3-, pAUC=7.8
--------------------------------------------------------------



============================ Round 2 ============================
Running EM with motif positional prior ...

Running with 10 threads ...

100	/221604	21.2s
1000	/221604	39.2s
10000	/221604	55.2s
100000	/221604	3.3m
200000	/221604	5.9m
221604	/221604	6.8m
---------------------------
Events discovered 
Significant:	479
Insignificant:	144
Filtered:	7

Total 173 homotypic events (within 501bp of other events).

Finish binding event prediction: 6.9m

Percentage of significant events with a motif match, divided in 10 bins (~48 events per bin):
85 81 95 93 93 97 97 97 100 97 

Percentage of filtered events with a motif match, divided in 7 bins (~1 events per bin):
0 0 0 0 0 0 0 

Percentage of insignificant events with a motif match, divided in 10 bins (~15 events per bin):
100 93 100 93 100 86 86 80 86 100 

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.30. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=168
k=6+1, gapped kmer=374, base kmer=448
k=6+2, gapped kmer=2216, base kmer=1026
k=6+3, gapped kmer=4438, base kmer=1138
k=6+4, gapped kmer=4888, base kmer=950

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 168

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAA/TTTTTT	203	205	150	-3.7    	0	344.2	5.0	1721.1	37
TTCATA/TATGAA	78	77	57	-1.6    	20	203.0	2.0	405.9	22
ACCATT/AATGGT	83	77	28	-7.0    	17	179.6	2.0	359.2	21
TAAATG/CATTTA	80	76	43	-3.3    	21	177.0	2.0	353.9	17
ATGTTA/TAACAT	79	71	46	-2.3    	23	163.0	2.0	325.9	9

Align and cluster 209 k-mers with seed AAAAAA,   	#0
Align and cluster 157 k-mers with seed TTCATA,   	#1
Align and cluster 118 k-mers with seed ACCATT,   	#2
Align and cluster 141 k-mers with seed TAAATG,   	#3
Align and cluster 158 k-mers with seed ATGTTA,   	#4

Merging redundant motifs ...
82ms: Remove motif #3.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= ACCATT 	1.56, 112, kAUC=26.7	PWM= ACCATT	6.16, 234, pAUC=12.7
#1	k=6	KSM= TTCATA 	1.80, 135, kAUC=26.1	PWM= TTCAT	5.18, 425, pAUC=6.0
#2	k=6	KSM= ATGTTA 	2.65, 160, kAUC=25.3	PWM= ATGTTA	5.57, 340, pAUC=8.4
#3	k=6	KSM= TAAATG 	1.89, 72, kAUC=25.0	PWM= AAATG	5.57, 392, pAUC=6.6
#4	k=6	KSM= AAAAAA 	4.74, 84, kAUC=14.6	PWM= AAAAAA	6.38, 381, pAUC=6.8

After merging:
#0	k=6	KSM= ACCATT 	1.55, 88, kAUC=27.0	PWM= ACCATT	5.37, 408, pAUC=12.7
#1	k=6	KSM= TTCATA 	1.80, 135, kAUC=26.1	PWM= TTCAT	5.18, 425, pAUC=6.0
#2	k=6	KSM= ATGTTA 	2.65, 160, kAUC=25.3	PWM= ATGTTA	5.57, 340, pAUC=8.4
#4	k=6	KSM= AAAAAA 	4.74, 84, kAUC=14.6	PWM= AAAAAA	6.38, 381, pAUC=6.8



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=602
k=7+1, gapped kmer=650, base kmer=730
k=7+2, gapped kmer=2287, base kmer=1085
k=7+3, gapped kmer=2982, base kmer=923
k=7+4, gapped kmer=3278, base kmer=790

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 602

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAT/ATTTTTT	115	119	75	-3.7    	0	241.5	6.0	1448.8	37
TTTCATA/TATGAAA	49	47	22	-3.3    	26	141.3	2.0	282.7	30
TGATTTC/GAAATCA	40	37	11	-4.7    	49	105.9	2.0	211.8	17
TCTACAT/ATGTAGA	38	34	9	-4.8    	58	103.8	2.0	207.6	28
TTAATGG/CCATTAA	40	36	5	-7.5    	53	92.4	2.0	184.7	23

Align and cluster 599 k-mers with seed AAAAAAT,   	#0
Align and cluster 315 k-mers with seed TTTCATA,   	#1
Align and cluster 234 k-mers with seed TGATTTC,   	#2
Align and cluster 246 k-mers with seed TCTACAT,   	#3
Align and cluster 184 k-mers with seed TTAATGG,   	#4

Merging redundant motifs ...

------------------------- k = 7 ----------------------------

#0	k=7	KSM= TTTCATA 	3.54, 122, kAUC=33.9	PWM= TTTCATA	8.03, 142, pAUC=8.8
#1	k=7	KSM= AAAAAAT 	3.15, 124, kAUC=33.8	PWM= AAAAAAT	7.58, 305, pAUC=7.7
#2	k=7	KSM= TCTACAT 	2.31, 64, kAUC=32.7	PWM= TCTACAT	7.20, 201, pAUC=13.4
#3	k=7	KSM= TTAATGG 	1.69, 74, kAUC=26.4	PWM= TTAATGG	7.84, 122, pAUC=15.3
#4	k=7	KSM= TGATTTC 	2.46, 47, kAUC=23.3	PWM= TGATTTC	7.11, 193, pAUC=9.1

After merging:
#0	k=7	KSM= TTTCATA 	3.54, 122, kAUC=33.9	PWM= TTTCATA	8.03, 142, pAUC=8.8
#1	k=7	KSM= AAAAAAT 	3.15, 124, kAUC=33.8	PWM= AAAAAAT	7.58, 305, pAUC=7.7
#2	k=7	KSM= TCTACAT 	2.31, 64, kAUC=32.7	PWM= TCTACAT	7.20, 201, pAUC=13.4
#3	k=7	KSM= TTAATGG 	1.69, 74, kAUC=26.4	PWM= TTAATGG	7.84, 122, pAUC=15.3
#4	k=7	KSM= TGATTTC 	2.46, 47, kAUC=23.3	PWM= TGATTTC	7.11, 193, pAUC=9.1



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=1041
k=8+1, gapped kmer=571, base kmer=709
k=8+2, gapped kmer=1101, base kmer=761
k=8+3, gapped kmer=1401, base kmer=704
k=8+4, gapped kmer=1693, base kmer=666

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 1041

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAAT/ATTTTTTT	79	82	30	-7.4    	0	182.2	7.0	1275.7	50
TTTTCATA/TATGAAAA	39	36	6	-6.8    	24	102.9	2.0	205.8	35
TTGATTTC/GAAATCAA	33	30	7	-4.8    	44	84.9	2.0	169.7	30
TCTTATGT/ACATAAGA	42	37	3	-9.4    	21	80.5	2.0	160.9	24
CTACCATT/AATGGTAG	56	50	2	-14.4    	4	74.5	2.0	149.0	18

Align and cluster 1243 k-mers with seed AAAAAAAT,   	#0
Align and cluster 709 k-mers with seed TTTTCATA,   	#1
Align and cluster 594 k-mers with seed TTGATTTC,   	#2
Align and cluster 551 k-mers with seed TCTTATGT,   	#3
Align and cluster 408 k-mers with seed CTACCATT,   	#4

Merging redundant motifs ...

------------------------- k = 8 ----------------------------

#0	k=8	KSM= AAAAAAAT 	2.20, 94, kAUC=50.4	PWM= AAAAAAAT	8.45, 257, pAUC=10.7
#1	k=8	KSM= TTTTCATA 	2.03, 86, kAUC=39.7	PWM= TTTTCATT	7.39, 248, pAUC=8.2
#2	k=8	KSM= CTACCATT 	1.56, 75, kAUC=30.7	PWM= CTACCATT	7.41, 127, pAUC=17.3
#3	k=8	KSM= TTGATTTC 	1.50, 93, kAUC=29.2	PWM= TTGATTTC	8.64, 111, pAUC=11.3
#4	k=8	KSM= TCTTATGT 	2.70, 52, kAUC=25.7	PWM= TCTTATGT	7.98, 119, pAUC=14.4

After merging:
#0	k=8	KSM= AAAAAAAT 	2.20, 94, kAUC=50.4	PWM= AAAAAAAT	8.45, 257, pAUC=10.7
#1	k=8	KSM= TTTTCATA 	2.03, 86, kAUC=39.7	PWM= TTTTCATT	7.39, 248, pAUC=8.2
#2	k=8	KSM= CTACCATT 	1.56, 75, kAUC=30.7	PWM= CTACCATT	7.41, 127, pAUC=17.3
#3	k=8	KSM= TTGATTTC 	1.50, 93, kAUC=29.2	PWM= TTGATTTC	8.64, 111, pAUC=11.3
#4	k=8	KSM= TCTTATGT 	2.70, 52, kAUC=25.7	PWM= TCTTATGT	7.98, 119, pAUC=14.4



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=1146
k=9+1, gapped kmer=307, base kmer=423
k=9+2, gapped kmer=467, base kmer=467
k=9+3, gapped kmer=616, base kmer=487
k=9+4, gapped kmer=762, base kmer=526

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 1146

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAAAT/ATTTTTTTT	40	42	16	-4.0    	1	144.6	8.0	1157.0	245
TTATGTTAA/TTAACATAA	45	39	1	-12.3    	3	109.6	3.0	328.8	83
ACACAACTA/TAGTTGTGT	20	18	2	-4.8    	72	63.6	3.0	190.9	49
TTCCAGAAA/TTTCTGGAA	22	19	0	-6.7    	68	61.0	3.0	183.1	50
TAAGACACA/TGTGTCTTA	33	29	0	-9.8    	25	82.9	2.0	165.8	47
TTTTTTCAT/ATGAAAAAA	52	47	5	-10.5

Align and cluster 1041 k-mers with seed AAAAAAAAT,   	#0
Align and cluster 389 k-mers with seed TTATGTTAA,   	#1
Align and cluster 306 k-mers with seed ACACAACTA,   	#2
Align and cluster 240 k-mers with seed TTCCAGAAA,   	#3
Align and cluster 262 k-mers with seed TAAGACACA,   	#4

Merging redundant motifs ...
353ms: Remove motif #1.
459ms: Remove motif #2.
614ms: Remove motif #3.

------------------------- k = 9 ----------------------------

#0	k=9	KSM= AAAAAAAAT 	1.74, 177, kAUC=57.1	PWM= AAAAAAAAA	8.44, 148, pAUC=8.6
#1	k=9	KSM= TTATGTTAA 	2.13, 64, kAUC=32.5	PWM= TTATGTTAA	8.80, 156, pAUC=10.7
#2	k=9	KSM= ACACAACTA 	2.46, 53, kAUC=31.8	PWM= ACAAAACAA	8.13, 241, pAUC=10.1
#3	k=9	KSM= TAAGACACA 	2.13, 42, kAUC=28.4	PWM= TAAGACACA	9.39, 146, pAUC=14.7
#4	k=9	KSM= TTCCAGAAA 	1.20, 57, kAUC=16.0	PWM= TTCCAGAAA	9.27, 56, pAUC=10.7

After merging:
#0	k=9	KSM= AAAAAAAAT 	1.82, 144, kAUC=62.6	PWM= AAAAAAA	7.13, 309, pAUC=7.1
#4	k=9	KSM= TTCCAGAAA 	1.20, 57, kAUC=16.0	PWM= TTCCAGAAA	9.27, 56, pAUC=10.7



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=951
k=10+1, gapped kmer=172, base kmer=259
k=10+2, gapped kmer=203, base kmer=264
k=10+3, gapped kmer=237, base kmer=292
k=10+4, gapped kmer=318, base kmer=375

------------------------- k = 10 ----------------------------

Total number of k-mers: n = 951

Top 5 cluster center k-mers:
# k-mer/r.c.         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAAAAT/ATTTTTTTTT	24	25	10	-2.6    	27	93.3	9.0	839.5	173
TCTTATGTTA/TAACATAAGA	42	37	0	-12.3    	1	69.9	4.0	279.5	33
TTTTTCATAA/TTATGAAAAA	35	31	1	-9.8    	8	75.7	2.0	151.5	52
AAATGTCTAC/GTAGACATTT	24	21	0	-7.3    	35	42.0	3.0	126.0	19
TATGTTAATG/CATTAACATA	36	32	0	-10.8    	3	62.2	2.0	124.5	26
CTTATGTTAA/TTAACATAAG	44	39	0	-13.0

Align and cluster 888 k-mers with seed AAAAAAAAAT,   	#0
Align and cluster 481 k-mers with seed TCTTATGTTA,   	#1
Align and cluster 536 k-mers with seed TTTTTCATAA,   	#2
Align and cluster 317 k-mers with seed AAATGTCTAC,   	#3
Align and cluster 425 k-mers with seed TATGTTAATG,   	#4

Merging redundant motifs ...
129ms: Remove motif #1.
335ms: Remove motif #2.

------------------------- k = 10 ----------------------------

#0	k=10	KSM= TTTTTCATAA 	1.41, 145, kAUC=43.7	PWM= TTTTTTTTTA	8.47, 192, pAUC=9.5
#1	k=10	KSM= AAAAAAAAAT 	1.35, 133, kAUC=41.2	PWM= AAAAAAAAAA	10.08, 115, pAUC=9.6
#2	k=10	KSM= TCTTATGTTA 	2.04, 56, kAUC=35.7	PWM= TCTTTTGTTA	9.90, 143, pAUC=14.7
#3	k=10	KSM= TATGTTAATG 	1.00, 66, kAUC=16.3	PWM= TATGTTAATG	12.98, 39, pAUC=11.1
#4	k=10	KSM= AAATGTCTAC 	1.00, 53, kAUC=13.7	PWM= AAATGTCTAC	9.20, 48, pAUC=15.9

After merging:
#0	k=10	KSM= TTTTTCATAA 	1.20, 216, kAUC=45.4	PWM= TTTTTCTT	7.96, 285, pAUC=12.6
#3	k=10	KSM= TATGTTAATG 	1.00, 66, kAUC=16.3	PWM= TATGTTAATG	12.98, 39, pAUC=11.1
#4	k=10	KSM= AAATGTCTAC 	1.00, 53, kAUC=13.7	PWM= AAATGTCTAC	9.20, 48, pAUC=15.9



----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=798
k=11+1, gapped kmer=126, base kmer=203
k=11+2, gapped kmer=96, base kmer=171
k=11+3, gapped kmer=88, base kmer=162
k=11+4, gapped kmer=145, base kmer=235

------------------------- k = 11 ----------------------------

Total number of k-mers: n = 798

Top 5 cluster center k-mers:
# k-mer/r.c.           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AAAAAAAAAAT/ATTTTTTTTTT	16	16	4	-2.9    	54	57.8	9.0	520.4	102
TTATGTTAATG/CATTAACATAA	36	32	0	-10.8    	1	50.0	4.0	199.8	15
TCTACATTTAA/TTAAATGTAGA	30	26	0	-8.9    	20	41.4	4.0	165.7	11
TGTCTACCATT/AATGGTAGACA	32	28	0	-9.5    	10	38.2	4.0	152.6	10
TTCCAGAAAGT/ACTTTCTGGAA	21	18	0	-6.4    	45	25.8	4.0	103.2	13
TCTTATGTTAA/TTAACATAAGA	41	36	0	-12.0

Align and cluster 513 k-mers with seed AAAAAAAAAAT,   	#0
Align and cluster 240 k-mers with seed TTATGTTAATG,   	#1
Align and cluster 209 k-mers with seed TCTACATTTAA,   	#2
Align and cluster 122 k-mers with seed TGTCTACCATT,   	#3
Align and cluster 135 k-mers with seed TTCCAGAAAGT,   	#4

Merging redundant motifs ...
139ms: Remove motif #1.
256ms: Remove motif #3.

------------------------- k = 11 ----------------------------

#0	k=11	KSM= AAAAAAAAAAT 	1.10, 194, kAUC=37.1	PWM= AAATGAAAAAA	9.52, 166, pAUC=13.4
#1	k=11	KSM= TTATGTTAATG 	1.20, 125, kAUC=27.1	PWM= TTTTGTTTATT	9.69, 127, pAUC=13.9
#2	k=11	KSM= TCTACATTTAA 	1.00, 56, kAUC=15.0	PWM= TCTACCATTAA	9.02, 72, pAUC=15.9
#3	k=11	KSM= TGTCTACCATT 	1.00, 54, kAUC=14.7	PWM= TGTCTACCATT	10.08, 64, pAUC=17.1
#4	k=11	KSM= TTCCAGAAAGT 	1.00, 38, kAUC=11.7	PWM= TTCCAGAAAGT	9.31, 59, pAUC=9.6

After merging:
#0	k=11	KSM= AAAAAAAAAAT 	1.10, 194, kAUC=37.1	PWM= AAATGAAAAAA	9.52, 166, pAUC=13.4
#2	k=11	KSM= TCTACATTTAA 	1.00, 60, kAUC=15.6	PWM= TCTACCATTAA	8.99, 80, pAUC=17.1
#4	k=11	KSM= TTCCAGAAAGT 	1.00, 38, kAUC=11.7	PWM= TTCCAGAAAGT	9.31, 59, pAUC=9.6



----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=718
k=12+1, gapped kmer=108, base kmer=184
k=12+2, gapped kmer=65, base kmer=135
k=12+3, gapped kmer=45, base kmer=101
k=12+4, gapped kmer=76, base kmer=147

------------------------- k = 12 ----------------------------

Total number of k-mers: n = 718

Top 5 cluster center k-mers:
# k-mer/r.c.             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
ACTTTTTTCATA/TATGAAAAAAGT	25	22	0	-7.6    	18	53.1	10.0	530.7	58
TTATGTTAATGG/CCATTAACATAA	35	31	0	-10.4    	1	51.0	6.0	306.2	26
AACTACTTTTTT/AAAAAAGTAGTT	18	16	0	-5.8    	45	43.6	5.0	218.2	48
CATAAGACACAA/TTGTGTCTTATG	31	27	0	-9.2    	12	45.3	4.0	181.2	23
TAACATAAGACA/TGTCTTATGTTA	33	29	0	-9.8    	5	42.1	4.0	168.2	18
TATGTTAATGGT/ACCATTAACATA	35	31	0	-10.4

Align and cluster 389 k-mers with seed ACTTTTTTCATA,   	#0
Align and cluster 260 k-mers with seed TTATGTTAATGG,   	#1
Align and cluster 290 k-mers with seed AACTACTTTTTT,   	#2
Align and cluster 230 k-mers with seed CATAAGACACAA,   	#3
Align and cluster 255 k-mers with seed TAACATAAGACA,   	#4

Merging redundant motifs ...
120ms: Remove motif #1.
199ms: Remove motif #2.

------------------------- k = 12 ----------------------------

#0	k=12	KSM= AACTACTTTTTT 	1.00, 126, kAUC=25.6	PWM= AACTACTTTTTT	10.22, 145, pAUC=15.9
#1	k=12	KSM= ACTTTTTTCATA 	1.20, 100, kAUC=24.3	PWM= CTTTTTTCAT	9.32, 165, pAUC=15.3
#2	k=12	KSM= TAACATAAGACA 	1.10, 104, kAUC=23.6	PWM= AAACATAAGAAA	10.40, 120, pAUC=17.3
#3	k=12	KSM= CATAAGACACAA 	1.10, 89, kAUC=20.8	PWM= CATAAGAAACAA	10.75, 111, pAUC=15.6
#4	k=12	KSM= TTATGTTAATGG 	1.10, 49, kAUC=14.4	PWM= TTATGTTAATGG	11.15, 50, pAUC=12.8

After merging:
#0	k=12	KSM= AACTACTTTTTT 	1.09, 136, kAUC=28.4	PWM= AACTATTTTTTT	10.16, 159, pAUC=16.7
#3	k=12	KSM= CATAAGACACAA 	1.10, 89, kAUC=20.8	PWM= CATAAGAAACAA	10.75, 111, pAUC=15.6
#4	k=12	KSM= TTATGTTAATGG 	1.10, 49, kAUC=14.4	PWM= TTATGTTAATGG	11.15, 50, pAUC=12.8



----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=670
k=13+1, gapped kmer=100, base kmer=174
k=13+2, gapped kmer=54, base kmer=123
k=13+3, gapped kmer=34, base kmer=79
k=13+4, gapped kmer=46, base kmer=103

------------------------- k = 13 ----------------------------

Total number of k-mers: n = 670

Top 5 cluster center k-mers:
# k-mer/r.c.               	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TTATGTTAATGGT/ACCATTAACATAA	35	31	0	-10.4    	0	45.2	10.0	452.3	19
AAAGTCTACTTTT/AAAAGTAGACTTT	24	21	0	-7.3    	17	34.3	6.0	205.8	14
TTTCATAAGAAAT/ATTTCTTATGAAA	24	21	0	-7.3    	24	35.8	5.0	179.0	18
TTTTGATTTCCAG/CTGGAAATCAAAA	16	14	0	-5.2    	55	24.8	6.0	148.9	21
AATCAAAATTGTT/AACAATTTTGATT	14	12	0	-4.6    	60	23.5	5.0	117.5	18

Align and cluster 163 k-mers with seed TTATGTTAATGGT,   	#0
Align and cluster 118 k-mers with seed AAAGTCTACTTTT,   	#1
Align and cluster 138 k-mers with seed TTTCATAAGAAAT,   	#2
Align and cluster 138 k-mers with seed TTTTGATTTCCAG,   	#3
Align and cluster 184 k-mers with seed AATCAAAATTGTT,   	#4

Merging redundant motifs ...
99ms: Remove motif #4.

------------------------- k = 13 ----------------------------

#0	k=13	KSM= TTATGTTAATGGT 	1.00, 51, kAUC=14.0	PWM= TTATGTTAATGGT	16.15, 43, pAUC=13.4
#1	k=13	KSM= TTTCATAAGAAAT 	1.00, 30, kAUC=10.4	PWM= TTTCATAAGAAAT	12.00, 50, pAUC=11.6
#2	k=13	KSM= TTTTGATTTCCAG 	1.00, 25, kAUC=9.6	PWM= TTTTGATTTCCAG	15.15, 23, pAUC=9.9
#3	k=13	KSM= AAAGTCTACTTTT 	1.31, 21, kAUC=8.8	PWM= AAAGTCTACTTTT	18.76, 24, pAUC=11.9
#4	k=13	KSM= AATCAAAATTGTT 	1.00, 15, kAUC=7.7	PWM= AATCAAAATTGTT	11.26, 27, pAUC=8.8

After merging:
#0	k=13	KSM= TTATGTTAATGGT 	1.00, 51, kAUC=14.0	PWM= TTATGTTAATGGT	16.15, 43, pAUC=13.4
#1	k=13	KSM= TTTCATAAGAAAT 	1.00, 30, kAUC=10.4	PWM= TTTCATAAGAAAT	12.00, 50, pAUC=11.6
#2	k=13	KSM= TTTTGATTTCCAG 	1.00, 28, kAUC=10.1	PWM= TTTTGATTTCCAG	12.84, 32, pAUC=10.3
#3	k=13	KSM= AAAGTCTACTTTT 	1.31, 21, kAUC=8.8	PWM= AAAGTCTACTTTT	18.76, 24, pAUC=11.9



----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=645
k=14+1, gapped kmer=98, base kmer=172
k=14+2, gapped kmer=50, base kmer=116
k=14+3, gapped kmer=29, base kmer=66
k=14+4, gapped kmer=33, base kmer=80

------------------------- k = 14 ----------------------------

Total number of k-mers: n = 645

Top 5 cluster center k-mers:
# k-mer/r.c.                 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTTATGTTAATGGT/ACCATTAACATAAG	34	30	0	-10.1    	0	41.4	11.0	454.9	15
AAATGTCTACCATT/AATGGTAGACATTT	24	21	0	-7.3    	14	33.7	7.0	235.7	16
ACAACTACTTTTTT/AAAAAAGTAGTTGT	17	15	0	-5.5    	40	29.2	6.0	175.4	16
AACATAAGACACAA/TTGTGTCTTATGTT	31	27	0	-9.2    	4	34.5	5.0	172.3	12
TTTTGATTTCCAGA/TCTGGAAATCAAAA	16	14	0	-5.2    	50	20.8	7.0	145.8	15

Align and cluster 182 k-mers with seed CTTATGTTAATGGT,   	#0
Align and cluster 143 k-mers with seed AAATGTCTACCATT,   	#1
Align and cluster 190 k-mers with seed ACAACTACTTTTTT,   	#2
Align and cluster 176 k-mers with seed AACATAAGACACAA,   	#3
Align and cluster 173 k-mers with seed TTTTGATTTCCAGA,   	#4

Merging redundant motifs ...
138ms: Remove motif #1.
268ms: Remove motif #2.

------------------------- k = 14 ----------------------------

#0	k=14	KSM= CTTATGTTAATGGT 	1.00, 53, kAUC=14.7	PWM= CTTATGTTAATGGT	12.54, 44, pAUC=13.6
#1	k=14	KSM= AACATAAGACACAA 	1.20, 37, kAUC=13.2	PWM= AACATAAGACACAA	11.28, 53, pAUC=14.9
#2	k=14	KSM= AAATGTCTACCATT 	1.00, 42, kAUC=12.6	PWM= AAATGTCTACCATT	14.63, 46, pAUC=14.3
#3	k=14	KSM= ACAACTACTTTTTT 	1.10, 34, kAUC=11.0	PWM= ACAACTACTTTTTT	15.12, 34, pAUC=15.1
#4	k=14	KSM= TTTTGATTTCCAGA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGA	17.23, 25, pAUC=10.6

After merging:
#0	k=14	KSM= CTTATGTTAATGGT 	1.00, 56, kAUC=15.1	PWM= CTTATGTTAATGGT	11.06, 52, pAUC=15.0
#3	k=14	KSM= ACAACTACTTTTTT 	1.10, 34, kAUC=11.0	PWM= ACAACTACTTTTTT	15.12, 34, pAUC=15.1
#4	k=14	KSM= TTTTGATTTCCAGA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGA	17.23, 25, pAUC=10.6



----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=625
k=15+1, gapped kmer=96, base kmer=168
k=15+2, gapped kmer=48, base kmer=111
k=15+3, gapped kmer=23, base kmer=53
k=15+4, gapped kmer=32, base kmer=77

------------------------- k = 15 ----------------------------

Total number of k-mers: n = 625

Top 5 cluster center k-mers:
# k-mer/r.c.                   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTTATGTTAATGGTA/TACCATTAACATAAG	33	29	0	-9.8    	0	39.6	12.0	474.9	19
TAACATAAGACACAA/TTGTGTCTTATGTTA	31	27	0	-9.2    	3	35.6	6.0	213.7	14
GAAATGTCTACCATT/AATGGTAGACATTTC	19	17	0	-6.1    	22	30.3	7.0	212.1	18
CACAACTACTTTTTT/AAAAAAGTAGTTGTG	17	15	0	-5.5    	38	29.2	7.0	204.7	21
TTTTGATTTCCAGAA/TTCTGGAAATCAAAA	16	14	0	-5.2    	45	21.2	7.0	148.2	24

Align and cluster 114 k-mers with seed CTTATGTTAATGGTA,   	#0
Align and cluster 115 k-mers with seed TAACATAAGACACAA,   	#1
Align and cluster 96 k-mers with seed GAAATGTCTACCATT,   	#2
Align and cluster 110 k-mers with seed CACAACTACTTTTTT,   	#3
Align and cluster 116 k-mers with seed TTTTGATTTCCAGAA,   	#4

Merging redundant motifs ...
32ms: PWM TTGTGTCTTATGTTA has been removed.

------------------------- k = 15 ----------------------------

#0	k=15	KSM= CTTATGTTAATGGTA 	1.00, 48, kAUC=13.7	PWM= CTTATGTTAATGGTA	14.19, 44, pAUC=14.2
#1	k=15	KSM= TAACATAAGACACAA 	1.00, 37, kAUC=11.6	PWM= TAACATAAGACACAA	22.48, 31, pAUC=12.8
#2	k=15	KSM= GAAATGTCTACCATT 	1.00, 34, kAUC=11.2	PWM= GAAATGTCTACCATT	17.84, 32, pAUC=11.8
#3	k=15	KSM= CACAACTACTTTTTT 	1.00, 26, kAUC=9.8	PWM= CACAACTACTTTTTT	14.03, 32, pAUC=11.6
#4	k=15	KSM= TTTTGATTTCCAGAA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGAA	17.06, 25, pAUC=9.9

After merging:
#0	k=15	KSM= CTTATGTTAATGGTA 	1.00, 48, kAUC=13.7	PWM= CTTATGTTAATGGTA	14.19, 44, pAUC=14.2
#2	k=15	KSM= GAAATGTCTACCATT 	1.00, 34, kAUC=11.2	PWM= GAAATGTCTACCATT	17.84, 32, pAUC=11.8
#3	k=15	KSM= CACAACTACTTTTTT 	1.00, 26, kAUC=9.8	PWM= CACAACTACTTTTTT	14.03, 32, pAUC=11.6
#4	k=15	KSM= TTTTGATTTCCAGAA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGAA	17.06, 25, pAUC=9.9



----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=612
k=16+1, gapped kmer=95, base kmer=166
k=16+2, gapped kmer=46, base kmer=109
k=16+3, gapped kmer=19, base kmer=47
k=16+4, gapped kmer=23, base kmer=58

------------------------- k = 16 ----------------------------

Total number of k-mers: n = 612

Top 5 cluster center k-mers:
# k-mer/r.c.                     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTACATTTAACATAAG/CTTATGTTAAATGTAG	28	25	0	-8.6    	7	36.7	13.0	477.7	16
AGAAATGTCTACCATT/AATGGTAGACATTTCT	19	17	0	-6.1    	17	23.0	7.0	160.7	15
ATTTAACATAAGACAC/GTGTCTTATGTTAAAT	29	26	0	-8.9    	2	33.8	4.0	135.3	9
TTTGATTTCCAGAAAG/CTTTCTGGAAATCAAA	15	13	0	-4.9    	42	19.1	6.0	114.5	21
ACATAAGACACAACTA/TAGTTGTGTCTTATGT	19	17	0	-6.1    	16	25.1	4.0	100.3	16
CTTATGTTAATGGTAG/CTACCATTAACATAAG	32	28	0	-9.5

Align and cluster 174 k-mers with seed CTACATTTAACATAAG,   	#0
Align and cluster 134 k-mers with seed AGAAATGTCTACCATT,   	#1
Align and cluster 127 k-mers with seed ATTTAACATAAGACAC,   	#2
Align and cluster 156 k-mers with seed TTTGATTTCCAGAAAG,   	#3
Align and cluster 147 k-mers with seed ACATAAGACACAACTA,   	#4

Merging redundant motifs ...
219ms: Remove motif #2.
229ms: PWM AGAAATGTCTACCATT has been removed.

------------------------- k = 16 ----------------------------

#0	k=16	KSM= ACATAAGACACAACTA 	1.00, 51, kAUC=14.2	PWM= ACATAAGACACAACTA	12.32, 45, pAUC=15.5
#1	k=16	KSM= CTACATTTAACATAAG 	1.00, 47, kAUC=13.5	PWM= CTACATTTAACATAAG	13.23, 45, pAUC=14.7
#2	k=16	KSM= ATTTAACATAAGACAC 	1.10, 37, kAUC=11.6	PWM= ATTTAACATAAGACAC	12.52, 32, pAUC=13.1
#3	k=16	KSM= AGAAATGTCTACCATT 	1.00, 36, kAUC=11.5	PWM= AGAAATGTCTACCATT	19.92, 31, pAUC=10.5
#4	k=16	KSM= TTTGATTTCCAGAAAG 	1.00, 22, kAUC=9.0	PWM= TTTGATTTCCAGAAAG	20.85, 24, pAUC=8.8

After merging:
#0	k=16	KSM= ACATAAGACACAACTA 	1.00, 54, kAUC=14.8	PWM= ACATAAGACACAACAA	12.02, 65, pAUC=16.0
#1	k=16	KSM= CTACATTTAACATAAG 	1.00, 47, kAUC=13.5	PWM= CTACATTTAACATAAG	13.23, 45, pAUC=14.7
#4	k=16	KSM= TTTGATTTCCAGAAAG 	1.00, 22, kAUC=9.0	PWM= TTTGATTTCCAGAAAG	20.85, 24, pAUC=8.8



----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=598
k=17+1, gapped kmer=93, base kmer=162
k=17+2, gapped kmer=44, base kmer=106
k=17+3, gapped kmer=18, base kmer=47
k=17+4, gapped kmer=18, base kmer=47

------------------------- k = 17 ----------------------------

Total number of k-mers: n = 598

Top 5 cluster center k-mers:
# k-mer/r.c.                       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
CTACATTTAACATAAGA/TCTTATGTTAAATGTAG	28	25	0	-8.6    	5	34.3	13.0	445.6	15
CCAGAAAGTCTACTTTT/AAAAGTAGACTTTCTGG	19	16	0	-5.8    	15	18.3	8.0	146.6	11
ATTTTGATTTCCAGAAA/TTTCTGGAAATCAAAAT	16	14	0	-5.2    	33	18.7	7.0	131.0	18
AACATAAGACACAACTA/TAGTTGTGTCTTATGTT	19	17	0	-6.1    	11	24.4	4.0	97.6	14
ACAACTACTTTTTTCAT/ATGAAAAAAGTAGTTGT	17	15	0	-5.5    	21	22.2	4.0	89.0	17
ACATTTAACATAAGACA/TGTCTTATGTTAAATGT	30	26	0	-8.9

Align and cluster 120 k-mers with seed CTACATTTAACATAAGA,   	#0
Align and cluster 63 k-mers with seed CCAGAAAGTCTACTTTT,   	#1
Align and cluster 114 k-mers with seed ATTTTGATTTCCAGAAA,   	#2
Align and cluster 109 k-mers with seed AACATAAGACACAACTA,   	#3
Align and cluster 134 k-mers with seed ACAACTACTTTTTTCAT,   	#4

Merging redundant motifs ...
36ms: PWM AACATAAGACACAACTA has been removed.

------------------------- k = 17 ----------------------------

#0	k=17	KSM= CTACATTTAACATAAGA 	1.00, 46, kAUC=13.2	PWM= CTACATTTAACATAAGA	16.38, 40, pAUC=13.5
#1	k=17	KSM= AACATAAGACACAACTA 	1.00, 30, kAUC=10.4	PWM= AACATAAGACACAACTA	23.23, 31, pAUC=12.7
#2	k=17	KSM= ACAACTACTTTTTTCAT 	1.00, 28, kAUC=10.1	PWM= ACAACTACTTTTTTCAT	23.98, 31, pAUC=11.1
#3	k=17	KSM= CCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= CCAGAAAGTCTACTTTT	26.88, 24, pAUC=9.7
#4	k=17	KSM= ATTTTGATTTCCAGAAA 	1.00, 22, kAUC=9.0	PWM= ATTTTGATTTCCAGAAA	20.96, 24, pAUC=8.9

After merging:
#0	k=17	KSM= CTACATTTAACATAAGA 	1.00, 46, kAUC=13.2	PWM= CTACATTTAACATAAGA	16.38, 40, pAUC=13.5
#2	k=17	KSM= ACAACTACTTTTTTCAT 	1.00, 28, kAUC=10.1	PWM= ACAACTACTTTTTTCAT	23.98, 31, pAUC=11.1
#3	k=17	KSM= CCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= CCAGAAAGTCTACTTTT	26.88, 24, pAUC=9.7
#4	k=17	KSM= ATTTTGATTTCCAGAAA 	1.00, 22, kAUC=9.0	PWM= ATTTTGATTTCCAGAAA	20.96, 24, pAUC=8.9



----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=586
k=18+1, gapped kmer=90, base kmer=156
k=18+2, gapped kmer=44, base kmer=107
k=18+3, gapped kmer=16, base kmer=43
k=18+4, gapped kmer=15, base kmer=40

------------------------- k = 18 ----------------------------

Total number of k-mers: n = 586

Top 5 cluster center k-mers:
# k-mer/r.c.                         	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TCTACATTTAACATAAGA/TCTTATGTTAAATGTAGA	28	25	0	-8.6    	5	36.4	14.0	509.6	23
ATTTAACATAAGACACAA/TTGTGTCTTATGTTAAAT	29	26	0	-8.9    	0	33.8	8.0	270.6	16
TCCAGAAAGTCTACTTTT/AAAAGTAGACTTTCTGGA	19	16	0	-5.8    	11	20.0	8.0	160.0	16
ACTTTTTTCATAAGAAAT/ATTTCTTATGAAAAAAGT	23	20	0	-7.0    	7	30.3	5.0	151.7	22
CATAAGAAATGTCTACCA/TGGTAGACATTTCTTATG	13	12	0	-4.6    	38	20.5	6.0	123.0	17

Align and cluster 157 k-mers with seed TCTACATTTAACATAAGA,   	#0
Align and cluster 133 k-mers with seed ATTTAACATAAGACACAA,   	#1
Align and cluster 98 k-mers with seed TCCAGAAAGTCTACTTTT,   	#2
Align and cluster 180 k-mers with seed ACTTTTTTCATAAGAAAT,   	#3
Align and cluster 113 k-mers with seed CATAAGAAATGTCTACCA,   	#4

Merging redundant motifs ...
218ms: Remove motif #1.
317ms: Remove motif #3.

------------------------- k = 18 ----------------------------

#0	k=18	KSM= TCTACATTTAACATAAGA 	1.00, 45, kAUC=13.1	PWM= TCTACATTTAACATAAGA	14.57, 41, pAUC=14.1
#1	k=18	KSM= ATTTAACATAAGACACAA 	1.00, 37, kAUC=11.6	PWM= ATTTAACATAAGACACAA	14.98, 32, pAUC=12.7
#2	k=18	KSM= CATAAGAAATGTCTACCA 	1.00, 32, kAUC=10.7	PWM= CATAAGAAATGTCTACCA	22.38, 31, pAUC=11.5
#3	k=18	KSM= ACTTTTTTCATAAGAAAT 	1.00, 31, kAUC=10.5	PWM= ACTTTTTTCATAAGAAAT	18.41, 33, pAUC=10.6
#4	k=18	KSM= TCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= TCCAGAAAGTCTACTTTT	28.59, 24, pAUC=10.3

After merging:
#0	k=18	KSM= TCTACATTTAACATAAGA 	1.00, 45, kAUC=13.1	PWM= TCTACATTTAACATAAGA	14.57, 41, pAUC=14.1
#2	k=18	KSM= CATAAGAAATGTCTACCA 	1.00, 32, kAUC=10.7	PWM= CATAAGAAATGTCTACCA	22.38, 31, pAUC=11.5
#4	k=18	KSM= TCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= TCCAGAAAGTCTACTTTT	28.59, 24, pAUC=10.3



----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=573
k=19+1, gapped kmer=87, base kmer=150
k=19+2, gapped kmer=44, base kmer=107
k=19+3, gapped kmer=14, base kmer=40
k=19+4, gapped kmer=13, base kmer=36

------------------------- k = 19 ----------------------------

Total number of k-mers: n = 573

Top 5 cluster center k-mers:
# k-mer/r.c.                           	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
ACATTTAACATAAGACACA/TGTGTCTTATGTTAAATGT	28	25	0	-8.6    	0	33.2	15.0	497.5	15
TAACATAAGACACAACTAC/GTAGTTGTGTCTTATGTTA	18	16	0	-5.8    	7	24.3	6.0	146.0	20
TTCCAGAAAGTCTACTTTT/AAAAGTAGACTTTCTGGAA	18	16	0	-5.8    	9	17.9	8.0	143.1	15
TGTCTACCATTAACATAAG/CTTATGTTAATGGTAGACA	13	12	0	-4.6    	38	23.5	6.0	141.0	19
ATTTTGATTTCCAGAAAGT/ACTTTCTGGAAATCAAAAT	15	13	0	-4.9    	29	16.9	8.0	135.3	19

Align and cluster 89 k-mers with seed ACATTTAACATAAGACACA,   	#0
Align and cluster 100 k-mers with seed TAACATAAGACACAACTAC,   	#1
Align and cluster 77 k-mers with seed TTCCAGAAAGTCTACTTTT,   	#2
Align and cluster 75 k-mers with seed TGTCTACCATTAACATAAG,   	#3
Align and cluster 113 k-mers with seed ATTTTGATTTCCAGAAAGT,   	#4

Merging redundant motifs ...
40ms: PWM ACATTTAACATAAGACACA has been removed.
50ms: PWM ATTTTGATTTCCAGAAAGT has been removed.

------------------------- k = 19 ----------------------------

#0	k=19	KSM= TGTCTACCATTAACATAAG 	1.00, 45, kAUC=13.1	PWM= TGTCTACCATTAACATAAG	20.32, 51, pAUC=14.1
#1	k=19	KSM= TAACATAAGACACAACTAC 	1.00, 30, kAUC=10.4	PWM= TAACATAAGACACAACTAC	23.27, 31, pAUC=11.8
#2	k=19	KSM= ACATTTAACATAAGACACA 	1.00, 28, kAUC=10.1	PWM= ACATTTAACATAAGACACA	29.26, 30, pAUC=12.7
#3	k=19	KSM= TTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTCCAGAAAGTCTACTTTT	30.24, 24, pAUC=9.6
#4	k=19	KSM= ATTTTGATTTCCAGAAAGT 	1.00, 22, kAUC=9.0	PWM= ATTTTGATTTCCAGAAAGT	24.85, 24, pAUC=7.8

After merging:
#0	k=19	KSM= TGTCTACCATTAACATAAG 	1.00, 45, kAUC=13.1	PWM= TGTCTACCATTAACATAAG	20.32, 51, pAUC=14.1
#1	k=19	KSM= TAACATAAGACACAACTAC 	1.00, 30, kAUC=10.4	PWM= TAACATAAGACACAACTAC	23.27, 31, pAUC=11.8
#3	k=19	KSM= TTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTCCAGAAAGTCTACTTTT	30.24, 24, pAUC=9.6



----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=560
k=20+1, gapped kmer=84, base kmer=144
k=20+2, gapped kmer=44, base kmer=107
k=20+3, gapped kmer=12, base kmer=36
k=20+4, gapped kmer=12, base kmer=36

------------------------- k = 20 ----------------------------

Total number of k-mers: n = 560

Top 5 cluster center k-mers:
# k-mer/r.c.                             	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
ACATTTAACATAAGACACAA/TTGTGTCTTATGTTAAATGT	28	25	0	-8.6    	0	32.8	16.0	524.6	15
ACACAACTACTTTTTTCATA/TATGAAAAAAGTAGTTGTGT	17	15	0	-5.5    	10	21.6	10.0	215.6	19
ATGTCTACCATTAACATAAG/CTTATGTTAATGGTAGACAT	13	12	0	-4.6    	31	21.9	8.0	175.3	18
TTTCCAGAAAGTCTACTTTT/AAAAGTAGACTTTCTGGAAA	18	16	0	-5.8    	7	17.4	9.0	156.9	17
ATTTTGATTTCCAGAAAGTC/GACTTTCTGGAAATCAAAAT	14	12	0	-4.6    	32	16.2	9.0	146.2	19

Align and cluster 125 k-mers with seed ACATTTAACATAAGACACAA,   	#0
Align and cluster 149 k-mers with seed ACACAACTACTTTTTTCATA,   	#1
Align and cluster 91 k-mers with seed ATGTCTACCATTAACATAAG,   	#2
Align and cluster 112 k-mers with seed TTTCCAGAAAGTCTACTTTT,   	#3
Align and cluster 121 k-mers with seed ATTTTGATTTCCAGAAAGTC,   	#4

Merging redundant motifs ...
38ms: PWM ATTTTGATTTCCAGAAAGTC has been removed.

------------------------- k = 20 ----------------------------

#0	k=20	KSM= ATGTCTACCATTAACATAAG 	1.00, 42, kAUC=12.6	PWM= ATGTCTACCATTAACATAAG	19.87, 38, pAUC=13.0
#1	k=20	KSM= ACATTTAACATAAGACACAA 	1.00, 30, kAUC=10.4	PWM= ACATTTAACATAAGACACAA	31.51, 30, pAUC=12.1
#2	k=20	KSM= ACACAACTACTTTTTTCATA 	1.00, 28, kAUC=10.1	PWM= ACACAACTACTTTTTTCATA	27.93, 31, pAUC=10.4
#3	k=20	KSM= TTTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTTCCAGAAAGTCTACTTTT	31.95, 24, pAUC=9.3
#4	k=20	KSM= ATTTTGATTTCCAGAAAGTC 	1.00, 22, kAUC=9.0	PWM= ATTTTGATTTCCAGAAAGTC	27.02, 24, pAUC=8.4

After merging:
#0	k=20	KSM= ATGTCTACCATTAACATAAG 	1.00, 42, kAUC=12.6	PWM= ATGTCTACCATTAACATAAG	19.87, 38, pAUC=13.0
#1	k=20	KSM= ACATTTAACATAAGACACAA 	1.00, 30, kAUC=10.4	PWM= ACATTTAACATAAGACACAA	31.51, 30, pAUC=12.1
#2	k=20	KSM= ACACAACTACTTTTTTCATA 	1.00, 28, kAUC=10.1	PWM= ACACAACTACTTTTTTCATA	27.93, 31, pAUC=10.4
#3	k=20	KSM= TTTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTTCCAGAAAGTCTACTTTT	31.95, 24, pAUC=9.3



------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=9	KSM= AAAAAAAAT 	1.82, 144, kAUC=62.6	PWM= AAAAAAA	7.13, 309, pAUC=7.1
#1	k=8	KSM= AAAAAAAT 	2.20, 94, kAUC=50.4	PWM= AAAAAAAT	8.45, 257, pAUC=10.7
#2	k=10	KSM= TTTTTCATAA 	1.20, 216, kAUC=45.4	PWM= TTTTTCTT	7.96, 285, pAUC=12.6
#3	k=8	KSM= TTTTCATA 	2.03, 86, kAUC=39.7	PWM= TTTTCATT	7.39, 248, pAUC=8.2
#4	k=11	KSM= AAAAAAAAAAT 	1.10, 194, kAUC=37.1	PWM= AAATGAAAAAA	9.52, 166, pAUC=13.4
#5	k=7	KSM= TTTCATA 	3.54, 122, kAUC=33.9	PWM= TTTCATA	8.03, 142, pAUC=8.8
#6	k=7	KSM= AAAAAAT 	3.15, 124, kAUC=33.8	PWM= AAAAAAT	7.58, 305, pAUC=7.7
#7	k=7	KSM= TCTACAT 	2.31, 64, kAUC=32.7	PWM= TCTACAT	7.20, 201, pAUC=13.4
#8	k=8	KSM= CTACCATT 	1.56, 75, kAUC=30.7	PWM= CTACCATT	7.41, 127, pAUC=17.3
#9	k=8	KSM= TTGATTTC 	1.50, 93, kAUC=29.2	PWM= TTGATTTC	8.64, 111, pAUC=11.3
#10	k=12	KSM= AACTACTTTTTT 	1.09, 136, kAUC=28.4	PWM= AACTATTTTTTT	10.16, 159, pAUC=16.7
#11	k=6	KSM= ACCATT 	1.55, 88, kAUC=27.0	PWM= ACCATT	5.37, 408, pAUC=12.7
#12	k=7	KSM= TTAATGG 	1.69, 74, kAUC=26.4	PWM= TTAATGG	7.84, 122, pAUC=15.3
#13	k=6	KSM= TTCATA 	1.80, 135, kAUC=26.1	PWM= TTCAT	5.18, 425, pAUC=6.0
#14	k=8	KSM= TCTTATGT 	2.70, 52, kAUC=25.7	PWM= TCTTATGT	7.98, 119, pAUC=14.4
#15	k=6	KSM= ATGTTA 	2.65, 160, kAUC=25.3	PWM= ATGTTA	5.57, 340, pAUC=8.4
#16	k=7	KSM= TGATTTC 	2.46, 47, kAUC=23.3	PWM= TGATTTC	7.11, 193, pAUC=9.1
#17	k=12	KSM= CATAAGACACAA 	1.10, 89, kAUC=20.8	PWM= CATAAGAAACAA	10.75, 111, pAUC=15.6
#18	k=10	KSM= TATGTTAATG 	1.00, 66, kAUC=16.3	PWM= TATGTTAATG	12.98, 39, pAUC=11.1
#19	k=9	KSM= TTCCAGAAA 	1.20, 57, kAUC=16.0	PWM= TTCCAGAAA	9.27, 56, pAUC=10.7
#20	k=11	KSM= TCTACATTTAA 	1.00, 60, kAUC=15.6	PWM= TCTACCATTAA	8.99, 80, pAUC=17.1
#21	k=14	KSM= CTTATGTTAATGGT 	1.00, 56, kAUC=15.1	PWM= CTTATGTTAATGGT	11.06, 52, pAUC=15.0
#22	k=16	KSM= ACATAAGACACAACTA 	1.00, 54, kAUC=14.8	PWM= ACATAAGACACAACAA	12.02, 65, pAUC=16.0
#23	k=6	KSM= AAAAAA 	4.74, 84, kAUC=14.6	PWM= AAAAAA	6.38, 381, pAUC=6.8
#24	k=12	KSM= TTATGTTAATGG 	1.10, 49, kAUC=14.4	PWM= TTATGTTAATGG	11.15, 50, pAUC=12.8
#25	k=13	KSM= TTATGTTAATGGT 	1.00, 51, kAUC=14.0	PWM= TTATGTTAATGGT	16.15, 43, pAUC=13.4
#26	k=15	KSM= CTTATGTTAATGGTA 	1.00, 48, kAUC=13.7	PWM= CTTATGTTAATGGTA	14.19, 44, pAUC=14.2
#27	k=10	KSM= AAATGTCTAC 	1.00, 53, kAUC=13.7	PWM= AAATGTCTAC	9.20, 48, pAUC=15.9
#28	k=16	KSM= CTACATTTAACATAAG 	1.00, 47, kAUC=13.5	PWM= CTACATTTAACATAAG	13.23, 45, pAUC=14.7
#29	k=17	KSM= CTACATTTAACATAAGA 	1.00, 46, kAUC=13.2	PWM= CTACATTTAACATAAGA	16.38, 40, pAUC=13.5
#30	k=18	KSM= TCTACATTTAACATAAGA 	1.00, 45, kAUC=13.1	PWM= TCTACATTTAACATAAGA	14.57, 41, pAUC=14.1
#31	k=19	KSM= TGTCTACCATTAACATAAG 	1.00, 45, kAUC=13.1	PWM= TGTCTACCATTAACATAAG	20.32, 51, pAUC=14.1
#32	k=20	KSM= ATGTCTACCATTAACATAAG 	1.00, 42, kAUC=12.6	PWM= ATGTCTACCATTAACATAAG	19.87, 38, pAUC=13.0
#33	k=11	KSM= TTCCAGAAAGT 	1.00, 38, kAUC=11.7	PWM= TTCCAGAAAGT	9.31, 59, pAUC=9.6
#34	k=15	KSM= GAAATGTCTACCATT 	1.00, 34, kAUC=11.2	PWM= GAAATGTCTACCATT	17.84, 32, pAUC=11.8
#35	k=14	KSM= ACAACTACTTTTTT 	1.10, 34, kAUC=11.0	PWM= ACAACTACTTTTTT	15.12, 34, pAUC=15.1
#36	k=18	KSM= CATAAGAAATGTCTACCA 	1.00, 32, kAUC=10.7	PWM= CATAAGAAATGTCTACCA	22.38, 31, pAUC=11.5
#37	k=13	KSM= TTTCATAAGAAAT 	1.00, 30, kAUC=10.4	PWM= TTTCATAAGAAAT	12.00, 50, pAUC=11.6
#38	k=20	KSM= ACATTTAACATAAGACACAA 	1.00, 30, kAUC=10.4	PWM= ACATTTAACATAAGACACAA	31.51, 30, pAUC=12.1
#39	k=19	KSM= TAACATAAGACACAACTAC 	1.00, 30, kAUC=10.4	PWM= TAACATAAGACACAACTAC	23.27, 31, pAUC=11.8
#40	k=17	KSM= ACAACTACTTTTTTCAT 	1.00, 28, kAUC=10.1	PWM= ACAACTACTTTTTTCAT	23.98, 31, pAUC=11.1
#41	k=20	KSM= ACACAACTACTTTTTTCATA 	1.00, 28, kAUC=10.1	PWM= ACACAACTACTTTTTTCATA	27.93, 31, pAUC=10.4
#42	k=13	KSM= TTTTGATTTCCAG 	1.00, 28, kAUC=10.1	PWM= TTTTGATTTCCAG	12.84, 32, pAUC=10.3
#43	k=15	KSM= CACAACTACTTTTTT 	1.00, 26, kAUC=9.8	PWM= CACAACTACTTTTTT	14.03, 32, pAUC=11.6
#44	k=18	KSM= TCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= TCCAGAAAGTCTACTTTT	28.59, 24, pAUC=10.3
#45	k=17	KSM= CCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= CCAGAAAGTCTACTTTT	26.88, 24, pAUC=9.7
#46	k=19	KSM= TTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTCCAGAAAGTCTACTTTT	30.24, 24, pAUC=9.6
#47	k=20	KSM= TTTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTTCCAGAAAGTCTACTTTT	31.95, 24, pAUC=9.3
#48	k=14	KSM= TTTTGATTTCCAGA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGA	17.23, 25, pAUC=10.6
#49	k=15	KSM= TTTTGATTTCCAGAA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGAA	17.06, 25, pAUC=9.9
#50	k=17	KSM= ATTTTGATTTCCAGAAA 	1.00, 22, kAUC=9.0	PWM= ATTTTGATTTCCAGAAA	20.96, 24, pAUC=8.9
#51	k=16	KSM= TTTGATTTCCAGAAAG 	1.00, 22, kAUC=9.0	PWM= TTTGATTTCCAGAAAG	20.85, 24, pAUC=8.8
#52	k=13	KSM= AAAGTCTACTTTT 	1.31, 21, kAUC=8.8	PWM= AAAGTCTACTTTT	18.76, 24, pAUC=11.9

374ms: Remove motif #1.
480ms: Remove motif #2.
696ms: Remove motif #4.
803ms: Remove motif #6.
1.0s: Remove motif #10.
1.2s: Remove motif #17.
1.4s: Remove motif #23.
1.6s: Remove motif #35.
1.7s: Remove motif #40.
1.9s: Remove motif #41.
2.2s: Remove motif #5.
2.4s: Remove motif #37.
2.5s: Remove motif #20.
2.5s: PWM TCCAGAAAGTCTACTTTT has been removed.
2.5s: PWM CCAGAAAGTCTACTTTT has been removed.
2.5s: PWM AAAGTCTACTTTT has been removed.
2.5s: Remove motif #11.
2.6s: Remove motif #12.
2.7s: Remove motif #31.
2.7s: PWM ATGTCTACCATTAACATAAG has been removed.
2.8s: Remove motif #16.
2.9s: Remove motif #42.
3.0s: Remove motif #48.
3.1s: Remove motif #49.
3.1s: PWM ATTTTGATTTCCAGAAA has been removed.
3.1s: PWM TTTGATTTCCAGAAAG has been removed.
3.2s: Remove motif #15.
3.4s: Remove motif #38.
3.5s: Remove motif #21.
3.6s: Remove motif #24.
3.6s: PWM TTATGTTAATGGT has been removed.
3.7s: Remove motif #26.
3.8s: Remove motif #28.
3.9s: Remove motif #33.
3.9s: PWM TAACATAAGACACAACTAC has been removed.
3.9s: PWM CACAACTACTTTTTT has been removed.
3.9s: PWM GAAATGTCTACCATT has been removed.
3.9s: PWM CATAAGAAATGTCTACCA has been removed.
4.1s: Remove motif #30.
4.1s: PWM TTTCCAGAAAGTCTACTTTT has been removed.
4.4s: Remove motif #18.
4.5s: Remove motif #29.

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=9	KSM= AAAAAAAAT 	1.82, 144, kAUC=62.6	PWM= AAAAAAA	7.13, 309, pAUC=7.1
#1	k=8	KSM= AAAAAAAT 	2.20, 94, kAUC=50.4	PWM= AAAAAAAT	8.45, 257, pAUC=10.7
#2	k=10	KSM= TTTTTCATAA 	1.20, 216, kAUC=45.4	PWM= TTTTTCTT	7.96, 285, pAUC=12.6
#3	k=8	KSM= TTTTCATA 	2.03, 86, kAUC=39.7	PWM= TTTTCATT	7.39, 248, pAUC=8.2
#4	k=11	KSM= AAAAAAAAAAT 	1.10, 194, kAUC=37.1	PWM= AAATGAAAAAA	9.52, 166, pAUC=13.4
#5	k=7	KSM= TTTCATA 	3.54, 122, kAUC=33.9	PWM= TTTCATA	8.03, 142, pAUC=8.8
#6	k=7	KSM= AAAAAAT 	3.15, 124, kAUC=33.8	PWM= AAAAAAT	7.58, 305, pAUC=7.7
#7	k=7	KSM= TCTACAT 	2.31, 64, kAUC=32.7	PWM= TCTACAT	7.20, 201, pAUC=13.4
#8	k=8	KSM= CTACCATT 	1.56, 75, kAUC=30.7	PWM= CTACCATT	7.41, 127, pAUC=17.3
#9	k=8	KSM= TTGATTTC 	1.50, 93, kAUC=29.2	PWM= TTGATTTC	8.64, 111, pAUC=11.3
#10	k=12	KSM= AACTACTTTTTT 	1.09, 136, kAUC=28.4	PWM= AACTATTTTTTT	10.16, 159, pAUC=16.7
#11	k=6	KSM= ACCATT 	1.55, 88, kAUC=27.0	PWM= ACCATT	5.37, 408, pAUC=12.7
#12	k=7	KSM= TTAATGG 	1.69, 74, kAUC=26.4	PWM= TTAATGG	7.84, 122, pAUC=15.3
#13	k=6	KSM= TTCATA 	1.80, 135, kAUC=26.1	PWM= TTCAT	5.18, 425, pAUC=6.0
#14	k=8	KSM= TCTTATGT 	2.70, 52, kAUC=25.7	PWM= TCTTATGT	7.98, 119, pAUC=14.4
#15	k=6	KSM= ATGTTA 	2.65, 160, kAUC=25.3	PWM= ATGTTA	5.57, 340, pAUC=8.4
#16	k=7	KSM= TGATTTC 	2.46, 47, kAUC=23.3	PWM= TGATTTC	7.11, 193, pAUC=9.1
#17	k=12	KSM= CATAAGACACAA 	1.10, 89, kAUC=20.8	PWM= CATAAGAAACAA	10.75, 111, pAUC=15.6
#18	k=10	KSM= TATGTTAATG 	1.00, 66, kAUC=16.3	PWM= TATGTTAATG	12.98, 39, pAUC=11.1
#19	k=9	KSM= TTCCAGAAA 	1.20, 57, kAUC=16.0	PWM= TTCCAGAAA	9.27, 56, pAUC=10.7
#20	k=11	KSM= TCTACATTTAA 	1.00, 60, kAUC=15.6	PWM= TCTACCATTAA	8.99, 80, pAUC=17.1
#21	k=14	KSM= CTTATGTTAATGGT 	1.00, 56, kAUC=15.1	PWM= CTTATGTTAATGGT	11.06, 52, pAUC=15.0
#22	k=16	KSM= ACATAAGACACAACTA 	1.00, 54, kAUC=14.8	PWM= ACATAAGACACAACAA	12.02, 65, pAUC=16.0
#23	k=6	KSM= AAAAAA 	4.74, 84, kAUC=14.6	PWM= AAAAAA	6.38, 381, pAUC=6.8
#24	k=12	KSM= TTATGTTAATGG 	1.10, 49, kAUC=14.4	PWM= TTATGTTAATGG	11.15, 50, pAUC=12.8
#25	k=13	KSM= TTATGTTAATGGT 	1.00, 51, kAUC=14.0	PWM= TTATGTTAATGGT	16.15, 43, pAUC=13.4
#26	k=15	KSM= CTTATGTTAATGGTA 	1.00, 48, kAUC=13.7	PWM= CTTATGTTAATGGTA	14.19, 44, pAUC=14.2
#27	k=10	KSM= AAATGTCTAC 	1.00, 53, kAUC=13.7	PWM= AAATGTCTAC	9.20, 48, pAUC=15.9
#28	k=16	KSM= CTACATTTAACATAAG 	1.00, 47, kAUC=13.5	PWM= CTACATTTAACATAAG	13.23, 45, pAUC=14.7
#29	k=17	KSM= CTACATTTAACATAAGA 	1.00, 46, kAUC=13.2	PWM= CTACATTTAACATAAGA	16.38, 40, pAUC=13.5
#30	k=18	KSM= TCTACATTTAACATAAGA 	1.00, 45, kAUC=13.1	PWM= TCTACATTTAACATAAGA	14.57, 41, pAUC=14.1
#31	k=19	KSM= TGTCTACCATTAACATAAG 	1.00, 45, kAUC=13.1	PWM= TGTCTACCATTAACATAAG	20.32, 51, pAUC=14.1
#32	k=20	KSM= ATGTCTACCATTAACATAAG 	1.00, 42, kAUC=12.6	PWM= ATGTCTACCATTAACATAAG	19.87, 38, pAUC=13.0
#33	k=11	KSM= TTCCAGAAAGT 	1.00, 38, kAUC=11.7	PWM= TTCCAGAAAGT	9.31, 59, pAUC=9.6
#34	k=15	KSM= GAAATGTCTACCATT 	1.00, 34, kAUC=11.2	PWM= GAAATGTCTACCATT	17.84, 32, pAUC=11.8
#35	k=14	KSM= ACAACTACTTTTTT 	1.10, 34, kAUC=11.0	PWM= ACAACTACTTTTTT	15.12, 34, pAUC=15.1
#36	k=18	KSM= CATAAGAAATGTCTACCA 	1.00, 32, kAUC=10.7	PWM= CATAAGAAATGTCTACCA	22.38, 31, pAUC=11.5
#37	k=13	KSM= TTTCATAAGAAAT 	1.00, 30, kAUC=10.4	PWM= TTTCATAAGAAAT	12.00, 50, pAUC=11.6
#38	k=20	KSM= ACATTTAACATAAGACACAA 	1.00, 30, kAUC=10.4	PWM= ACATTTAACATAAGACACAA	31.51, 30, pAUC=12.1
#39	k=19	KSM= TAACATAAGACACAACTAC 	1.00, 30, kAUC=10.4	PWM= TAACATAAGACACAACTAC	23.27, 31, pAUC=11.8
#40	k=17	KSM= ACAACTACTTTTTTCAT 	1.00, 28, kAUC=10.1	PWM= ACAACTACTTTTTTCAT	23.98, 31, pAUC=11.1
#41	k=20	KSM= ACACAACTACTTTTTTCATA 	1.00, 28, kAUC=10.1	PWM= ACACAACTACTTTTTTCATA	27.93, 31, pAUC=10.4
#42	k=13	KSM= TTTTGATTTCCAG 	1.00, 28, kAUC=10.1	PWM= TTTTGATTTCCAG	12.84, 32, pAUC=10.3
#43	k=15	KSM= CACAACTACTTTTTT 	1.00, 26, kAUC=9.8	PWM= CACAACTACTTTTTT	14.03, 32, pAUC=11.6
#44	k=18	KSM= TCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= TCCAGAAAGTCTACTTTT	28.59, 24, pAUC=10.3
#45	k=17	KSM= CCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.4	PWM= CCAGAAAGTCTACTTTT	26.88, 24, pAUC=9.7
#46	k=19	KSM= TTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTCCAGAAAGTCTACTTTT	30.24, 24, pAUC=9.6
#47	k=20	KSM= TTTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTTCCAGAAAGTCTACTTTT	31.95, 24, pAUC=9.3
#48	k=14	KSM= TTTTGATTTCCAGA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGA	17.23, 25, pAUC=10.6
#49	k=15	KSM= TTTTGATTTCCAGAA 	1.00, 23, kAUC=9.1	PWM= TTTTGATTTCCAGAA	17.06, 25, pAUC=9.9
#50	k=17	KSM= ATTTTGATTTCCAGAAA 	1.00, 22, kAUC=9.0	PWM= ATTTTGATTTCCAGAAA	20.96, 24, pAUC=8.9
#51	k=16	KSM= TTTGATTTCCAGAAAG 	1.00, 22, kAUC=9.0	PWM= TTTGATTTCCAGAAAG	20.85, 24, pAUC=8.8
#52	k=13	KSM= AAAGTCTACTTTT 	1.31, 21, kAUC=8.8	PWM= AAAGTCTACTTTT	18.76, 24, pAUC=11.9

------------------------- GEMout_2, k = ALL (after merging) -------------------------
#0	k=9	KSM= AAAAAAAAT 	1.38, 240, kAUC=63.0	PWM= AAAAAAA	7.15, 337, pAUC=7.1
#8	k=8	KSM= CTACCATT 	1.64, 106, kAUC=40.4	PWM= TTCCAT	5.47, 419, pAUC=5.6
#14	k=8	KSM= TCTTATGT 	1.30, 248, kAUC=40.4	PWM= TTTTGTT	7.01, 275, pAUC=4.9
#3	k=8	KSM= TTTTCATA 	2.03, 86, kAUC=39.7	PWM= TTTTCATT	7.39, 248, pAUC=8.2
#9	k=8	KSM= TTGATTTC 	1.31, 172, kAUC=33.6	PWM= TTGATTTC	8.58, 171, pAUC=9.8
#7	k=7	KSM= TCTACAT 	2.31, 64, kAUC=32.7	PWM= TCTACAT	7.20, 201, pAUC=13.4
#13	k=6	KSM= TTCATA 	1.80, 135, kAUC=26.1	PWM= TTCAT	5.18, 425, pAUC=6.0
#19	k=9	KSM= TTCCAGAAA 	1.20, 75, kAUC=20.1	PWM= TTCCAGAAA	8.28, 102, pAUC=11.0
#22	k=16	KSM= ACATAAGACACAACTA 	1.00, 54, kAUC=14.8	PWM= ACATAAGACACAACAA	12.02, 65, pAUC=16.0
#27	k=10	KSM= AAATGTCTAC 	1.00, 53, kAUC=13.7	PWM= AAATGTCTAC	9.20, 48, pAUC=15.9
#46	k=19	KSM= TTCCAGAAAGTCTACTTTT 	1.00, 24, kAUC=9.3	PWM= TTCCAGAAAGTCTACTTTT	30.24, 24, pAUC=9.6

4.6s: Finalizing 11 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_2 final motifs ----------------------------


GEMout_2 motif #0

KSM top k-mer: AAAAAAAAT/ATTTTTTTT, total 590 k-mers
KSM threshold: 1.38, 		hit=240+/3-, kAUC=63.0

   |
AAAAAAA
      G
       
       
PWM threshold: 7.15/10.60, 	hit=337+/225-, pAUC=7.1
--------------------------------------------------------------

GEMout_2 motif #1

KSM top k-mer: CTACCATT/AATGGTAG, total 165 k-mers
KSM threshold: 1.64, 		hit=106+/2-, kAUC=40.4

  |
TTCCAT
 AT   
      
      
PWM threshold: 5.47/8.34, 	hit=419+/379-, pAUC=5.6
--------------------------------------------------------------

GEMout_2 motif #2

KSM top k-mer: TCTTATGT/ACATAAGA, total 159 k-mers
KSM threshold: 1.30, 		hit=248+/21-, kAUC=40.4

       |
TTTTGTT
AAA    
       
       
PWM threshold: 7.01/10.10, 	hit=275+/200-, pAUC=4.9
--------------------------------------------------------------

GEMout_2 motif #3

KSM top k-mer: TTTTCATA/TATGAAAA, total 202 k-mers
KSM threshold: 2.03, 		hit=86+/0-, kAUC=39.7

  |
TTTTCATT
    TC A
        
        
PWM threshold: 7.39/10.57, 	hit=248+/224-, pAUC=8.2
--------------------------------------------------------------

GEMout_2 motif #4

KSM top k-mer: TTGATTTC/GAAATCAA, total 140 k-mers
KSM threshold: 1.31, 		hit=172+/15-, kAUC=33.6

        |
TTGATTTC
  T    T
        
        
PWM threshold: 8.58/12.29, 	hit=171+/129-, pAUC=9.8
--------------------------------------------------------------

GEMout_2 motif #5

KSM top k-mer: TCTACAT/ATGTAGA, total 68 k-mers
KSM threshold: 2.31, 		hit=64+/2-, kAUC=32.7

               |
TCTACAT
    AT 
       
       
PWM threshold: 7.20/10.95, 	hit=201+/141-, pAUC=13.4
--------------------------------------------------------------

GEMout_2 motif #6

KSM top k-mer: TTCATA/TATGAA, total 59 k-mers
KSM threshold: 1.80, 		hit=135+/18-, kAUC=26.1

     |
TTCAT
     
     
     
PWM threshold: 5.18/8.37, 	hit=425+/421-, pAUC=6.0
--------------------------------------------------------------

GEMout_2 motif #7

KSM top k-mer: TTCCAGAAA/TTTCTGGAA, total 56 k-mers
KSM threshold: 1.20, 		hit=75+/1-, kAUC=20.1

                  |
TTCCAGAAA
  T      
         
         
PWM threshold: 8.28/14.22, 	hit=102+/57-, pAUC=11.0
--------------------------------------------------------------

GEMout_2 motif #8

KSM top k-mer: ACATAAGACACAACTA/TAGTTGTGTCTTATGT, total 73 k-mers
KSM threshold: 1.00, 		hit=54+/0-, kAUC=14.8

                     |
ACATAAGACACAACAA
        A A  AT 
                
                
PWM threshold: 12.02/23.20, 	hit=65+/10-, pAUC=16.0
--------------------------------------------------------------

GEMout_2 motif #9

KSM top k-mer: AAATGTCTAC/GTAGACATTT, total 30 k-mers
KSM threshold: 1.00, 		hit=53+/0-, kAUC=13.7

AAATGTCTAC
     G
PWM threshold: 9.20/17.67, 	hit=48+/5-, pAUC=15.9
--------------------------------------------------------------



Finished! GEM analysis results are printed to:
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYBb/vescaAlpina/GEMout/GEMout.GEM_events.txt
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYBb/vescaAlpina/GEMout/GEMout.results.htm
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS144-MYBb/vescaAlpina/GEMout/GEMout_outputs (folder with all other files)

----------------------------------

End time: 2021/12/04 22:32:59

Total running time: 18.5m

pigz 2.6
Analysis complete for INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1.fq.gz
Analysis complete for INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2.fq.gz
Archive:  INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc.zip
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/warning.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/error.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Icons/tick.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/summary.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_tile_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/per_base_n_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/duplication_levels.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/Images/adapter_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_report.html  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/fastqc_data.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_1_val_1_fastqc/fastqc.fo  
Archive:  INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc.zip
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/
   creating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/fastqc_icon.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/warning.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/error.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Icons/tick.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/summary.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_tile_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_quality.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_sequence_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_sequence_gc_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/per_base_n_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/sequence_length_distribution.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/duplication_levels.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/Images/adapter_content.png  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_report.html  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/fastqc_data.txt  
  inflating: INDEX49_BKDL210038288-1A-49_HG7TTCCX2_L8_2_val_2_fastqc/fastqc.fo  

GEM (version 3.4)!

Please cite: 
Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. 
High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. 
doi:10.1371/journal.pcbi.1002638

Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).

----------------------------------

Start time: 2021/12/04 23:30:05

Loading data...
    Loading reads from: CS149-HAT9bvescaAlpinasorted.bam ... Loaded
    Loading reads from: InputvescaAlpinasorted.bam ... Loaded

Options:
--d /home/joaquin/projects/methylation/programs/gem/Read_Distribution_default.txt --g /home/joaquin/projects/methylation/data/commonData/fragariaVesca/genome.index.txt --genome /home/joaquin/projects/methylation/data/commonData/fragariaVesca/FVechrs/ --s 150000000 --expt CS149-HAT9bvescaAlpinasorted.bam --ctrl /home/joaquin/projects/methylation/data/fragaria_vesca_data/Input/vescaAlpina/InputvescaAlpinasorted.bam --out /home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9b/vescaAlpina/GEMout --f SAM --outNP --excluded_fraction 0 --range 200 --smooth 0 --mrc 1 --fold 2 --q 1.301029996 --k_min 6 --k_max 20 --k_seqs 600 --k_neg_dinu_shuffle --pp_nmotifs 1 --t 10 

Getting 5' positions of all reads...

Original read count stats:
_IP  	Bases: 21523559	HitCounts: 3.9209356E7
_CTRL	Bases: 14558969	HitCounts: 1.8658693E7

Filter duplicate reads.
_IP  	Bases: 21523559	HitCounts: 3.537534E7
_CTRL	Bases: 14558969	HitCounts: 1.5832687E7

At Poisson p-value 1.0e-03, in a 501bp window, expect 153 reads.


	IP: 35375340	Ctrl: 15832687	 IP/Ctrl: 2.23

Sorting reads and selecting enriched regions ...
For condition , IP/Control = 0.21

The genome is segmented into 204058 regions for analysis.

============================ Round 0 ============================

Running with 10 threads ...

100	/204058	34.3s
1000	/204058	56.3s
10000	/204058	4.1m
100000	/204058	5.9m
200000	/204058	5.9m
204058	/204058	5.9m

Scaling condition , IP/Control = 1.09

---------------------------
Events discovered 
Significant:	33104
Insignificant:	372
Filtered:	2123

Total 20258 homotypic events (within 501bp of other events).

Finish binding event prediction: 5.9m

Refine read distribution from 2000 binding events. 


============================ Round 1 ============================

Running with 10 threads ...

100	/204058	2.3s
1000	/204058	4.4s
10000	/204058	20.4s
100000	/204058	3.0m
200000	/204058	6.0m
204058	/204058	6.4m
---------------------------
Events discovered 
Significant:	47766
Insignificant:	89
Filtered:	35

Total 11346 homotypic events (within 545bp of other events).

Finish binding event prediction: 6.5m

Refine read distribution from 2000 binding events. 

Loading genome sequences ...
Done, 2.4s
GC content=0.41

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.32. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=8
k=6+1, gapped kmer=18, base kmer=31
k=6+2, gapped kmer=29, base kmer=42
k=6+3, gapped kmer=22, base kmer=37
k=6+4, gapped kmer=0, base kmer=0

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 8

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATCAT/ATGATT	221	227	99	-16.9    	0	332.8	3.0	998.5	7
TAATTA/TAATTA	88	88	41	-5.5    	6	183.3	1.0	183.3	1

Align and cluster 34 k-mers with seed AATCAT,   	#0
Align and cluster 24 k-mers with seed TAATTA,   	#1

Merging redundant motifs ...
163ms: Remove motif #1.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= AATCAT 	3.02, 169, kAUC=24.5	PWM= AATCAT	5.60, 464, pAUC=11.2
#1	k=6	KSM= TAATTA 	2.76, 156, kAUC=21.4	PWM= TAATCA	6.09, 396, pAUC=12.6

After merging:
#0	k=6	KSM= AATCAT 	3.02, 169, kAUC=24.5	PWM= AATCAT	5.60, 464, pAUC=11.2



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=16
k=7+1, gapped kmer=25, base kmer=36
k=7+2, gapped kmer=19, base kmer=35
k=7+3, gapped kmer=0, base kmer=0
k=7+4, gapped kmer=0, base kmer=0

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 16

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TAATCAT/ATGATTA	87	90	21	-12.6    	2	192.3	3.0	577.0	12
AATCATT/AATGATT	106	110	33	-12.3

Align and cluster 44 k-mers with seed TAATCAT,   	#0
Align and cluster 35 k-mers with seed AATCATT,   	#1

Merging redundant motifs ...
96ms: Remove motif #1.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= TAATCAT 	3.08, 69, kAUC=32.9	PWM= CAATCAT	7.39, 400, pAUC=18.1
#1	k=7	KSM= AATCATT 	3.39, 64, kAUC=27.5	PWM= AATCATT	7.34, 342, pAUC=14.4

After merging:
#0	k=7	KSM= TAATCAT 	2.97, 65, kAUC=33.1	PWM= CAATCAT	7.46, 426, pAUC=18.1



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=10
k=8+1, gapped kmer=15, base kmer=25
k=8+2, gapped kmer=0, base kmer=0
k=8+3, gapped kmer=0, base kmer=0
k=8+4, gapped kmer=0, base kmer=0

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 10

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATGATTG/CAATCATT	42	44	3	-11.4    	0	83.9	3.0	251.7	7

Align and cluster 22 k-mers with seed AATGATTG,   	#0

------------------------- k = 8 ----------------------------

#0	k=8	KSM= AATGATTG 	2.70, 93, kAUC=29.2	PWM= AATGATTG	9.49, 180, pAUC=22.1



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=3
k=9+1, gapped kmer=1, base kmer=2
k=9+2, gapped kmer=0, base kmer=0
k=9+3, gapped kmer=0, base kmer=0
k=9+4, gapped kmer=0, base kmer=0

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 3

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATGATTGA/TCAATCATT	21	22	0	-7.6    	0	29.3	2.0	58.6	3

Align and cluster 4 k-mers with seed AATGATTGA,   	#0

------------------------- k = 9 ----------------------------

#0	k=9	KSM= AATGATTGA 	1.41, 46, kAUC=11.8	PWM= AATGATTGA	8.88, 177, pAUC=19.1



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=0
k=10+1, gapped kmer=0, base kmer=0
k=10+2, gapped kmer=0, base kmer=0
k=10+3, gapped kmer=0, base kmer=0
k=10+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=0
k=11+1, gapped kmer=0, base kmer=0
k=11+2, gapped kmer=0, base kmer=0
k=11+3, gapped kmer=0, base kmer=0
k=11+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=0
k=12+1, gapped kmer=0, base kmer=0
k=12+2, gapped kmer=0, base kmer=0
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=0
k=13+1, gapped kmer=0, base kmer=0
k=13+2, gapped kmer=0, base kmer=0
k=13+3, gapped kmer=0, base kmer=0
k=13+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=0
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=0, base kmer=0
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=7	KSM= TAATCAT 	2.97, 65, kAUC=33.1	PWM= CAATCAT	7.46, 426, pAUC=18.1
#1	k=8	KSM= AATGATTG 	2.70, 93, kAUC=29.2	PWM= AATGATTG	9.49, 180, pAUC=22.1
#2	k=6	KSM= AATCAT 	3.02, 169, kAUC=24.5	PWM= AATCAT	5.60, 464, pAUC=11.2
#3	k=9	KSM= AATGATTGA 	1.41, 46, kAUC=11.8	PWM= AATGATTGA	8.88, 177, pAUC=19.1

24ms: PWM CAATCATT has been removed.
102ms: Remove motif #2.
107ms: PWM TCAATCATT has been removed.

------------------------- GEMout_1, k = ALL (before merging) -------------------------
#0	k=7	KSM= TAATCAT 	2.97, 65, kAUC=33.1	PWM= CAATCAT	7.46, 426, pAUC=18.1
#1	k=8	KSM= AATGATTG 	2.70, 93, kAUC=29.2	PWM= AATGATTG	9.49, 180, pAUC=22.1
#2	k=6	KSM= AATCAT 	3.02, 169, kAUC=24.5	PWM= AATCAT	5.60, 464, pAUC=11.2
#3	k=9	KSM= AATGATTGA 	1.41, 46, kAUC=11.8	PWM= AATGATTGA	8.88, 177, pAUC=19.1

------------------------- GEMout_1, k = ALL (after merging) -------------------------
#0	k=7	KSM= TAATCAT 	2.87, 75, kAUC=35.1	PWM= CAATCAT	7.39, 424, pAUC=18.1

182ms: Finalizing 1 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_1 final motifs ----------------------------


GEMout_1 motif #0

KSM top k-mer: TAATCAT/ATGATTA, total 36 k-mers
KSM threshold: 2.87, 		hit=75+/0-, kAUC=35.1

    |
CAATCAT
T      
       
       
PWM threshold: 7.39/11.06, 	hit=424+/199-, pAUC=18.1
--------------------------------------------------------------



============================ Round 2 ============================
Running EM with motif positional prior ...

Running with 10 threads ...

100	/204058	18.2s
1000	/204058	35.2s
10000	/204058	56.2s
100000	/204058	4.4m
200000	/204058	8.3m
204058	/204058	8.8m
---------------------------
Events discovered 
Significant:	52390
Insignificant:	77
Filtered:	9

Total 20729 homotypic events (within 545bp of other events).

Finish binding event prediction: 8.8m

Refine read distribution from 2000 binding events. 

Percentage of significant events with a motif match, divided in 10 bins (~5239 events per bin):
89 87 87 85 85 84 83 83 84 84 

Percentage of filtered events with a motif match, divided in 9 bins (~1 events per bin):
0 0 0 0 0 0 0 0 0 

Percentage of insignificant events with a motif match, divided in 10 bins (~8 events per bin):
87 75 87 75 50 37 50 25 62 80 

Use di-nucleotide shuffled sequences as negative sequences.

Estimated GC content is 0.31. Set [--gc -1] to use the estimated GC content.
Provided  GC content is 0.41.

KMAC motif discovery with 600+/600- sequences.


Motif discovery from k_min=6 to k_max=20 ...

----------------------------------------------------------
Running k=6 ...

k=6, exact kmers=12
k=6+1, gapped kmer=29, base kmer=46
k=6+2, gapped kmer=44, base kmer=75
k=6+3, gapped kmer=47, base kmer=72
k=6+4, gapped kmer=10, base kmer=16

------------------------- k = 6 ----------------------------

Total number of k-mers: n = 12

Top 5 cluster center k-mers:
# k-mer/r.c. 	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATCAT/ATGATT	342	344	92	-54.1    	0	443.2	4.0	1772.8	6
CTAATC/GATTAG	69	69	16	-9.8    	10	178.2	2.0	356.3	3
TTATTG/CAATAA	141	140	65	-8.7    	6	275.7	1.0	275.7	3

Align and cluster 47 k-mers with seed AATCAT,   	#0
Align and cluster 26 k-mers with seed CTAATC,   	#1
Align and cluster 40 k-mers with seed TTATTG,   	#2

Merging redundant motifs ...
139ms: Remove motif #1.
374ms: Remove motif #2.

------------------------- k = 6 ----------------------------

#0	k=6	KSM= TTATTG 	2.44, 365, kAUC=37.1	PWM= TGATTG	6.04, 516, pAUC=18.0
#1	k=6	KSM= AATCAT 	2.04, 373, kAUC=35.0	PWM= AATCAT	6.52, 526, pAUC=18.6
#2	k=6	KSM= CTAATC 	3.77, 235, kAUC=27.8	PWM= TAAT	4.31, 590, pAUC=5.8

After merging:
#0	k=6	KSM= TTATTG 	2.44, 365, kAUC=37.1	PWM= TGATTG	6.04, 516, pAUC=18.0



----------------------------------------------------------
Running k=7 ...

k=7, exact kmers=21
k=7+1, gapped kmer=47, base kmer=69
k=7+2, gapped kmer=46, base kmer=74
k=7+3, gapped kmer=13, base kmer=17
k=7+4, gapped kmer=0, base kmer=0

------------------------- k = 7 ----------------------------

Total number of k-mers: n = 21

Top 5 cluster center k-mers:
# k-mer/r.c.   	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATCATT/AATGATT	173	174	30	-30.8    	0	282.3	4.0	1129.2	9
TCAATCA/TGATTGA	82	83	22	-10.5    	5	173.6	2.0	347.2	6

Align and cluster 54 k-mers with seed AATCATT,   	#0
Align and cluster 51 k-mers with seed TCAATCA,   	#1

Merging redundant motifs ...
177ms: Remove motif #1.

------------------------- k = 7 ----------------------------

#0	k=7	KSM= TCAATCA 	4.32, 264, kAUC=47.7	PWM= TCAATCA	7.82, 360, pAUC=18.6
#1	k=7	KSM= AATCATT 	2.43, 395, kAUC=40.5	PWM= AATCATT	7.80, 456, pAUC=25.9

After merging:
#0	k=7	KSM= TCAATCA 	4.32, 264, kAUC=47.7	PWM= TCAATCA	7.82, 360, pAUC=18.6



----------------------------------------------------------
Running k=8 ...

k=8, exact kmers=29
k=8+1, gapped kmer=42, base kmer=62
k=8+2, gapped kmer=12, base kmer=17
k=8+3, gapped kmer=0, base kmer=0
k=8+4, gapped kmer=0, base kmer=0

------------------------- k = 8 ----------------------------

Total number of k-mers: n = 29

Top 5 cluster center k-mers:
# k-mer/r.c.     	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
AATGATTG/CAATCATT	65	66	3	-18.2    	0	164.3	4.0	657.2	17
TGATTGAT/ATCAATCA	39	40	7	-7.3    	13	67.5	2.0	135.1	3
GTAATTAT/ATAATTAC	23	23	1	-7.3    	26	64.0	1.0	64.0	1

Align and cluster 72 k-mers with seed AATGATTG,   	#0
Align and cluster 53 k-mers with seed TGATTGAT,   	#1
Align and cluster 56 k-mers with seed GTAATTAT,   	#2

Merging redundant motifs ...
110ms: Remove motif #2.

------------------------- k = 8 ----------------------------

#0	k=8	KSM= AATGATTG 	5.23, 87, kAUC=46.6	PWM= AATGATTG	9.51, 299, pAUC=31.9
#1	k=8	KSM= GTAATTAT 	4.77, 58, kAUC=44.2	PWM= TAATCA	5.78, 410, pAUC=16.0
#2	k=8	KSM= TGATTGAT 	2.13, 210, kAUC=43.8	PWM= TGATTGAT	8.57, 221, pAUC=19.7

After merging:
#0	k=8	KSM= AATGATTG 	5.23, 84, kAUC=47.6	PWM= AATGATTG	8.96, 300, pAUC=29.9
#1	k=8	KSM= GTAATTAT 	4.77, 58, kAUC=44.2	PWM= TAATCA	5.78, 410, pAUC=16.0



----------------------------------------------------------
Running k=9 ...

k=9, exact kmers=14
k=9+1, gapped kmer=7, base kmer=10
k=9+2, gapped kmer=0, base kmer=0
k=9+3, gapped kmer=0, base kmer=0
k=9+4, gapped kmer=0, base kmer=0

------------------------- k = 9 ----------------------------

Total number of k-mers: n = 14

Top 5 cluster center k-mers:
# k-mer/r.c.       	PosCt	wPosCt	NegCt	Hgp_lg10	Id	Density	Delta	Gamma	Cluster_size
TCAATCATT/AATGATTGA	27	28	2	-7.7    	0	65.7	4.0	262.7	12

Align and cluster 20 k-mers with seed TCAATCATT,   	#0

------------------------- k = 9 ----------------------------

#0	k=9	KSM= TCAATCATT 	2.49, 67, kAUC=26.2	PWM= CAATCAT	8.87, 227, pAUC=24.6



----------------------------------------------------------
Running k=10 ...

k=10, exact kmers=0
k=10+1, gapped kmer=0, base kmer=0
k=10+2, gapped kmer=0, base kmer=0
k=10+3, gapped kmer=0, base kmer=0
k=10+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=11 ...

k=11, exact kmers=0
k=11+1, gapped kmer=0, base kmer=0
k=11+2, gapped kmer=0, base kmer=0
k=11+3, gapped kmer=0, base kmer=0
k=11+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=12 ...

k=12, exact kmers=0
k=12+1, gapped kmer=0, base kmer=0
k=12+2, gapped kmer=0, base kmer=0
k=12+3, gapped kmer=0, base kmer=0
k=12+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=13 ...

k=13, exact kmers=0
k=13+1, gapped kmer=0, base kmer=0
k=13+2, gapped kmer=0, base kmer=0
k=13+3, gapped kmer=0, base kmer=0
k=13+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=14 ...

k=14, exact kmers=0
k=14+1, gapped kmer=0, base kmer=0
k=14+2, gapped kmer=0, base kmer=0
k=14+3, gapped kmer=0, base kmer=0
k=14+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=15 ...

k=15, exact kmers=0
k=15+1, gapped kmer=0, base kmer=0
k=15+2, gapped kmer=0, base kmer=0
k=15+3, gapped kmer=0, base kmer=0
k=15+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=16 ...

k=16, exact kmers=0
k=16+1, gapped kmer=0, base kmer=0
k=16+2, gapped kmer=0, base kmer=0
k=16+3, gapped kmer=0, base kmer=0
k=16+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=17 ...

k=17, exact kmers=0
k=17+1, gapped kmer=0, base kmer=0
k=17+2, gapped kmer=0, base kmer=0
k=17+3, gapped kmer=0, base kmer=0
k=17+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=18 ...

k=18, exact kmers=0
k=18+1, gapped kmer=0, base kmer=0
k=18+2, gapped kmer=0, base kmer=0
k=18+3, gapped kmer=0, base kmer=0
k=18+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=19 ...

k=19, exact kmers=0
k=19+1, gapped kmer=0, base kmer=0
k=19+2, gapped kmer=0, base kmer=0
k=19+3, gapped kmer=0, base kmer=0
k=19+4, gapped kmer=0, base kmer=0

No enriched k-mer!

----------------------------------------------------------
Running k=20 ...

k=20, exact kmers=0
k=20+1, gapped kmer=0, base kmer=0
k=20+2, gapped kmer=0, base kmer=0
k=20+3, gapped kmer=0, base kmer=0
k=20+4, gapped kmer=0, base kmer=0

No enriched k-mer!

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=7	KSM= TCAATCA 	4.32, 264, kAUC=47.7	PWM= TCAATCA	7.82, 360, pAUC=18.6
#1	k=8	KSM= AATGATTG 	5.23, 84, kAUC=47.6	PWM= AATGATTG	8.96, 300, pAUC=29.9
#2	k=8	KSM= GTAATTAT 	4.77, 58, kAUC=44.2	PWM= TAATCA	5.78, 410, pAUC=16.0
#3	k=6	KSM= TTATTG 	2.44, 365, kAUC=37.1	PWM= TGATTG	6.04, 516, pAUC=18.0
#4	k=9	KSM= TCAATCATT 	2.49, 67, kAUC=26.2	PWM= CAATCAT	8.87, 227, pAUC=24.6

95ms: Remove motif #1.
218ms: Remove motif #2.
428ms: Remove motif #3.
513ms: Remove motif #4.

------------------------- GEMout_2, k = ALL (before merging) -------------------------
#0	k=7	KSM= TCAATCA 	4.32, 264, kAUC=47.7	PWM= TCAATCA	7.82, 360, pAUC=18.6
#1	k=8	KSM= AATGATTG 	5.23, 84, kAUC=47.6	PWM= AATGATTG	8.96, 300, pAUC=29.9
#2	k=8	KSM= GTAATTAT 	4.77, 58, kAUC=44.2	PWM= TAATCA	5.78, 410, pAUC=16.0
#3	k=6	KSM= TTATTG 	2.44, 365, kAUC=37.1	PWM= TGATTG	6.04, 516, pAUC=18.0
#4	k=9	KSM= TCAATCATT 	2.49, 67, kAUC=26.2	PWM= CAATCAT	8.87, 227, pAUC=24.6

------------------------- GEMout_2, k = ALL (after merging) -------------------------
#0	k=7	KSM= TCAATCA 	4.32, 264, kAUC=47.7	PWM= TCAATCA	7.82, 360, pAUC=18.6

545ms: Finalizing 1 motifs ...


Compute motif distance distribution ...

------------------------- GEMout_2 final motifs ----------------------------


GEMout_2 motif #0

KSM top k-mer: TCAATCA/TGATTGA, total 44 k-mers
KSM threshold: 4.32, 		hit=264+/17-, kAUC=47.7

     |
TCAATCA
C      
       
       
PWM threshold: 7.82/11.86, 	hit=360+/108-, pAUC=18.6
--------------------------------------------------------------



Finished! GEM analysis results are printed to:
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9b/vescaAlpina/GEMout/GEMout.GEM_events.txt
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9b/vescaAlpina/GEMout/GEMout.results.htm
/home/joaquin/projects/methylation/data/fragaria_vesca_data/CS149-HAT9b/vescaAlpina/GEMout/GEMout_outputs (folder with all other files)

----------------------------------

End time: 2021/12/04 23:55:40

Total running time: 25.6m

Checking MD5 from JM11-INDEX42
Checking Fastaq lenghts from JM11-INDEX42
Doing Trim galore in CS142-CDFb/vescaAlpina from JM11-INDEX42
Trim galore finished,checking results...
Doing Bowtie2 in CS142-CDFb/vescaAlpina from JM11-INDEX42
Bowtie2 summary: b''
Bowtie2 errors if any: b'Time loading reference: 00:00:00\nTime loading forward index: 00:00:00\nTime loading mirror index: 00:00:00\nMultiseed full-index search: 00:23:57\n28744913 reads; of these:\n  28744913 (100.00%) were paired; of these:\n    3442465 (11.98%) aligned concordantly 0 times\n    18270039 (63.56%) aligned concordantly exactly 1 time\n    7032409 (24.46%) aligned concordantly >1 times\n    ----\n    3442465 pairs aligned concordantly 0 times; of these:\n      472687 (13.73%) aligned discordantly 1 time\n    ----\n    2969778 pairs aligned 0 times concordantly or discordantly; of these:\n      5939556 mates make up the pairs; of these:\n        4464034 (75.16%) aligned 0 times\n        788931 (13.28%) aligned exactly 1 time\n        686591 (11.56%) aligned >1 times\n92.24% overall alignment rate\nTime searching: 00:23:57\nOverall time: 00:23:57\n'
Checking MD5 from JM11-INDEX44
Checking Fastaq lenghts from JM11-INDEX44
Doing Trim galore in CS144-MYBb/vescaAlpina from JM11-INDEX44
Trim galore finished,checking results...
Doing Bowtie2 in CS144-MYBb/vescaAlpina from JM11-INDEX44
Bowtie2 summary: b''
Bowtie2 errors if any: b'Time loading reference: 00:00:00\nTime loading forward index: 00:00:00\nTime loading mirror index: 00:00:00\nMultiseed full-index search: 00:22:25\n31197576 reads; of these:\n  31197576 (100.00%) were paired; of these:\n    5446315 (17.46%) aligned concordantly 0 times\n    18322695 (58.73%) aligned concordantly exactly 1 time\n    7428566 (23.81%) aligned concordantly >1 times\n    ----\n    5446315 pairs aligned concordantly 0 times; of these:\n      173005 (3.18%) aligned discordantly 1 time\n    ----\n    5273310 pairs aligned 0 times concordantly or discordantly; of these:\n      10546620 mates make up the pairs; of these:\n        9686682 (91.85%) aligned 0 times\n        518867 (4.92%) aligned exactly 1 time\n        341071 (3.23%) aligned >1 times\n84.48% overall alignment rate\nTime searching: 00:22:25\nOverall time: 00:22:25\n'
Checking MD5 from JM11-INDEX49
Checking Fastaq lenghts from JM11-INDEX49
Doing Trim galore in CS149-HAT9b/vescaAlpina from JM11-INDEX49
Trim galore finished,checking results...
Doing Bowtie2 in CS149-HAT9b/vescaAlpina from JM11-INDEX49
Bowtie2 summary: b''
Bowtie2 errors if any: b'Time loading reference: 00:00:00\nTime loading forward index: 00:00:00\nTime loading mirror index: 00:00:00\nMultiseed full-index search: 00:19:09\n20525632 reads; of these:\n  20525632 (100.00%) were paired; of these:\n    1803672 (8.79%) aligned concordantly 0 times\n    12422009 (60.52%) aligned concordantly exactly 1 time\n    6299951 (30.69%) aligned concordantly >1 times\n    ----\n    1803672 pairs aligned concordantly 0 times; of these:\n      348944 (19.35%) aligned discordantly 1 time\n    ----\n    1454728 pairs aligned 0 times concordantly or discordantly; of these:\n      2909456 mates make up the pairs; of these:\n        1841908 (63.31%) aligned 0 times\n        444419 (15.27%) aligned exactly 1 time\n        623129 (21.42%) aligned >1 times\n95.51% overall alignment rate\nTime searching: 00:19:09\nOverall time: 00:19:09\n'
